
## Protein–protein interaction database

Database | Web link | References
---------|----------|-----------
IntAct | https://www.ebi.ac.uk/intact/ | [51]
Database of Interacting Proteins | https://dip.doe-mbi.ucla.edu/dip/Main.cgi | [52]
The Human Protein Reference Database | http://www.hprd.org/ | [53]
The Molecular INTeraction database (MINT) | https://mint.bio.uniroma2.it/ | [54]
Munich MPact/MIPS database | https://mips.helmholtz-muenchen.de/proj/ppi/ | [55]
BIOGRID | https://thebiogrid.org/ | [56]
HINT | (High-quality INTeractomes) http://hint.yulab.org/ | [57]
STITCH | http://stitch.embl.de/ | [58]
STRING | https://string-db.org/ | [59]
PIBASE database|  https://modbase.compbio.ucsf.edu/pibase/queries.html | [60]
3did database | https://3did.irbbarcelona.org/ | [61]

## Tools

Tools | Web link | References
------|----------|-----------
Cytoscape | https://cytoscape.org/ | [62]
PathVisio | https://pathvisio.github.io/ | [63]
Systrip | https://tulip.labri.fr/TulipDrupal/?q=systrip | [64]
MAPPIS | http://bioinfo3d.cs.tau.ac.il/MAPPIS/ | [64]
PSOPIA | https://mizuguchilab.org/PSOPIA/ | [65]
Agile Protein Interaction DataAnalyzer | http://cicblade.dep.usal.es:8080/APID/init.action | [66]
iFrag | http://sbi.imim.es/web/index.php/research/servers/iFrag [67]
Interactome3D | https://interactome3d.irbbarcelona.org/ | [68]
PCRPi-W | http://www.bioinsilico.org/PCRPi/ | [69]
PPCheck | http://caps.ncbs.res.in/ppcheck/ | [69]

## References

https://link.springer.com/protocol/10.1007/7653_2020_62

51. Keshava Prasad TS, Goel R, Kandasamy K et al (2009) Human protein reference database - 2009 update. Nucleic Acids Res 37:767–772.  https://doi.org/10.1093/nar/gkn892
52. Zanzoni A, Montecchi-Palazzi L, Quondam M et al (2002) MINT: a molecular INTeraction database. FEBS Lett 513:135–140
53. Mewes H, Frishman D et al (2002) MIPS: a database for genomes and protein sequences. Nucleic Acids Res 30:31
54. Stark C, Breitkreutz B, Reguly T et al (2006) BioGRID: a general repository for interaction datasets. Nucleic Acids Res 34:D35
55. Das J, Yu H (2012) HINT: high-quality protein interactomes and their applications in understanding human disease. BMC Syst Biol 6:92.  https://doi.org/10.1186/1752-0509-6-92
56. Kuhn M, von Mering C, Campillos M et al (2008) STITCH: interaction networks of chemicals and proteins. Nucleic Acids Res 36:D684–D688.  https://doi.org/10.1093/nar/gkm795
57. Szklarczyk D, Morris J, Cook H et al (2017) The STRING database in 2017: quality-controlled protein–protein association networks, made broadly accessible. Nucleic Acids Res 45:D362
58. Davis F, Sali A (2005) PIBASE: a comprehensive database of structurally defined protein interfaces. Bioinformatics 21:1901
59. Stein A, Russell R, Aloy P (2005) 3did: interacting protein domains of known three-dimensional structure. Nucleic Acids Res 33:D413
60. Shannon P, Markiel A, Ozier O et al (2003) Cytoscape: a software environment for integrated models of biomolecular interaction networks. Genome Res 13:2498–2504.  https://doi.org/10.1101/gr.1239303
61. Kutmon M, van Iersel MP, Bohler A et al (2015) PathVisio 3: an extendable pathway analysis toolbox. PLoS Comput Biol 11:e1004085.  https://doi.org/10.1371/journal.pcbi.1004085
62. Dubois J, Cottret L, Ghozlane A et al (2012) Systrip: a visual environment for the investigation of time-series data in the context of metabolic networks. IEEE, Washington, DC
63. Shulman-Peleg A, Shatsky M, Nussinov R, Wolfson HJ (2005) MAPPIS: multiple 3D alignment of protein-protein interfaces. In: Lecture notes in computer science (including subseries Lecture notes in artificial intelligence and Lecture notes in bioinformatics). Springer, Berlin, pp 91–103
64. Murakami Y, Mizuguchi K (2018) PSOPIA: toward more reliable protein-protein interaction prediction from sequence information. In: ICIIBMS 2017 - 2nd International Conference on Intelligent Informatics and Biomedical Sciences. Institute of Electrical and Electronics Engineers Inc, Washington, DC, pp 255–261
65. Prieto C, de Las RJ (2006) APID: agile protein interaction DataAnalyzer. Nucleic Acids Res 34:W298–W302.  https://doi.org/10.1093/nar/gkl128
66. Garcia-Garcia J, Valls-Comamala V, Guney E et al (2017) iFrag: a protein–protein interface prediction server based on sequence fragments. J Mol Biol 429:382–389.  https://doi.org/10.1016/j.jmb.2016.11.034
67. Mosca R, Céol A, Aloy P (2013) Interactome3D: adding structural details to protein networks. Nat Methods 10:47
68. Segura Mora S, Assi SA, Fernandez-Fuentes N (2010) Presaging critical residues in protein interfaces-web server (PCRPi-W): a web server to chart Hot spots in protein interfaces. PLoS One 5:e12352.  https://doi.org/10.1371/journal.pone.0012352
69. Sukhwal A, Sowdhamini R (2015) PPcheck: a webserver for the quantitative analysis of protein-protein interfaces and prediction of residue hotspots. Bioinformatics Biol Insights 9:141–151.  https://doi.org/10.4137/BBI.S25928
