% 29-06-00 as new chapter by upgrade report
% pending for further literature review 11/12/00
% 21/9/2001, 16/10/2001, 28/11/2001 revised

\chapter{Marker characteristics and fine mapping}

\section{Introduction}

The importance of marker characteristics in both linkage and association
analysis is now well recognised.  For example, when a parent passes a gamete
to an offspring, a recombination may occur in the parent only if the parent is
heterozygous at two joint loci.  Ott (1992) examined the sample size necessary
to characterise a marker and the effects of wrongly assuming equal allele
frequencies to linkage analysis.  Terwilliger et al.  (1992) focused on the
relative importance of marker heterozygosity and intermarker distance and
showed that high marker density may be a good substitute for low marker
heterozygosity.  Kruglyak (1997) investigated how many polymorphic and densely
spaced biallelic markers are needed for extraction of most of the mode of
inheritance information as compared to microsatellite markers.  Kruglyak (1999)
used coalescent theory and computer simulation to examine the extent to which
LD can extend in the context of genome screen and using SNPs.

Of particular interets is the advantage and disadvantage of SNPs and dense maps
over traditional markers in fine mapping.  Ott \& Rabinowitz (1997), Chapman
\& Wijsman (1998) both considered the effect of marker polymorphism on
mutation detection assuming single ancestral mutation.  Ott \& Rabinowitz
(1997) found that greater marker heterozygosity results in increased power
to detect linkage disequilibrium. A similar conclusion was reached by Chapman
\& Wijsman (1998), that multiallelic markers always have more power to detect
LD than biallelic markers.  It is desirable, however, multiple mutations can be
incorporated.

\subsection*{Chapter aims}

This chapter uses a simple deterministic model to investigate the effect of
marker polymorphism on the power of fine mapping.  The simple population
genetics model incorporates multiple ancestral mutations.  Under this model,
the effect of marker polymorphism on the power to detect linkage disequilibrium
due to multiple mutations is examined for several scenarios.  Some results of
this investigation are reported in Sham, Zhao \& Curtis (2000).


\section{Methods}

We begin with the simplest assumption that current disease chromosomes are due
to multiple ancestral mutations at a single marker locus at generation $g$.  We
assume the disease locus is recombination rate $\theta$ from the marker.  We
also assume that the marker locus have $n$ alleles, with frequencies $\pi_j,
j=1, \ldots, n$ in the population, so that there are at most $m$ mutations at
an allele and $n^m$ possible patterns of mutations in total.  Let $d_i=1,
\dots, m$ be the specific pattern of each of these $n^m$ enumerations, with a
non-zero value indicating a mutation at that allele and their counts
accumulating to $c_j$, so that marker allele frequency at the disease-carrying
chromosome at generation 0 is $f_j=c_j/m$, $j=1, \ldots, n$.  Under random
mating and nonoverlapping generations, the expected frequencies of the marker
alleles among chromosomes containing a disease mutation at generation $g$ are
obtained via the following basic relation
\begin{eqnarray}
f_j^{(g)}=f_j(1-\theta)^g+\left[1-(1-\theta)^g\right]\pi_j\label{eqn1}
\end{eqnarray}
i.e., the expected frequencies for the marker chromosome at generation $g$
is due to chromosomes that remain intact with the founder mutations and those
experienced at least one recombinations.  The allele frequencies discrepancy
between the disease chromosome and chromosomes in the population can be
measured by Pearson $\chi^2$ and likelihood ratio test statistics as follows.
\begin{eqnarray}
\chi_i^2 &=& N\sum_j\frac{(\pi_j-f_j^{(g)})^2}{\pi_j+f_j^{(g)}}\label{eqn2}
\end{eqnarray}
and
\begin{eqnarray}
l_i &=& 2N \sum_j\left[\pi_j\ln \pi_j+f_j^{(g)}\ln
f_j^{(g)}-(\pi_j+f_j^{(g)})\ln\frac{\pi_j+f_j^{(g)}}{2}\right]\label{eqn3}
\end{eqnarray}
for $i=1, \ldots, n^m$ and $N$ is the appropriate sample size. Asymptotically
$\chi_i^2$ and $l_i$ would have noncentral $\chi^2$ distribution with degree
of freedom $n-1$, and noncentrality parameters $\chi_i^2$ and $l_i$
respectively.

The appropriate area $s_i$ is calculated as the power estimate from these
two distributions that exceeds the critical value set by the global
$\alpha$ value, the type I error under the null distribution of $\chi^2$.
Denote $w_i=\prod_j \pi_j^{c_j}$,$i=1,\ldots,n^m; j=1, \dots, n$ the
multinomial probability associated with the $i-th$ of the $n^m$
enumerations, we can get an overall estimate of the power by $\sum_i
w_is_i$, $i=1,\ldots,n^m$. The estimate of $N$ can be iteratively refined
to achieve predefined power. In practice to enumerate all the $n^m$
possibilities is quite slow so we can set a number of values that are
interesting to us and get their overall estimates of power. The minimum
and maximum of $\chi_i^2$ and $l_i$ can also be recorded for each
mutation. The $n^m$ possibilities are enumerated using a recursive algorithm.

Two sets of allele frequencies are examined for markers with alleles varying
from 2 to 15, which represents fairly common polymorphisms.  In the first
scheme each allele has frequency $\pi_j=1/n$ for $j=1, \ldots, n$, so the
marker heterozygosity is easily calculated as $1-\sum \pi_j^2=1-1/n$.  In the
second scheme different alleles follow a geometric relationship,
$p_1=(1-r)/(1-r^n), \pi_j=\pi_{j-1}r, j=2, \ldots, n$, the marker
heterozygosity is $1-(1+r^n)/(1+r)p_1=[2r/(1+r)][(1-r^{(n-1)})/(1-r^n)]$.  By
setting $r=0.5$ here allele 1 has the highest frequency 0.67 for biallelic
marker and 0.5 for marker with 15 alleles, while the marker heterozygosity
increases from 0.44 to 0.67.  A proportion of $r=9/11\approx 0.82$ is used so
that allele 1 has the highest frequency 0.55 for biallelic marker and 0.19 for
15-allele marker while the marker heterozygosity increases from 0.50 to 0.89.

The following combinations are examined for 1 to 5 mutations:  the generation
$g=20$, 50 and 80, the recombination fractions $\theta=0.005$ and 0.05.  It is
expected that combinations such as $g=50$, $\theta=0.005$ and $g=50$,
$\theta=0.05$ will give comparable results to $g=500,\theta=0.0005$ and $g=500,
\theta=0.005$.

The entire calculation is done by a C program calling up a number of
subroutines:  subroutine {\em digit} for the recursive enumeration, Fortran
routines to calculate central and noncentral $\chi^2$ distribution functions
and noncentrality parameters, unidimensional optimisation procedures {\em
fibonacci} and {\em fmin} to determine the required sample size for a given
type I error rate and power.  Subroutines {\em digit}, {\em fibonacci}, {\em
fmin} are listed in Appendix B.  Fortran routines are based on algorithms
published in Applied Statistics (AS91, AS170, AS245, AS275, available from
http://lib.stat.cmu.edu).


\section{Results}

Table~\ref{table4_1} shows required the sample size based on the exact
calculation.  In general, the likelihood ratio test statistic exhibits more
power (data not shown) than Pearson $\chi^2$.  As the number of mutations
increases, the effect of number of alleles is more pronounced.

\begin{table}[h]
\caption{Required sample sizes for $\alpha=0.001$ and $1-\beta=0.9$ over
$g=20, 50, 80$ and $\theta=0.005, 0.05$\label{table4_1}}
\end{table}
%\oddsidemargin=-0.5cm
%\evensidemargin=-1.5cm
\begin{alltt}
               20                    50                         80
       0.005      0.05       0.005        0.05         0.005         0.05

mutation=1
2    41   61   315   521    59   90   7000  12404    83  130   153301  273655
3    30   49   207   396    43   71   4166   9138    59  102    88030  200080
4    27   47   170   372    38   68   3132   8435    51   96    64423  184002
5    26   48   152   372    36   69   2594   8373    48   97    51989  182552
6    26   49   142   380    35   71   2262   8557    46  100    44229  185929
7    25   51   135   393    34   73   2037   8813    45  103    38891  191671
8    25   53   131   407    34   76   1874   9124    45  107    34973  198183
9    26   54   128   421    34   78   1752   9427    45  111    31966  204930
10   26   56   127   435    34   81   1656   9734    45  114    29579  211906
11   26   58   125   448    35   83   1579  10033    45  118    27635  218092
12   26   60   125   461    35   86   1516  10321    46  121    26017  224348
13   27   61   124   473    35   88   1464  10598    46  125    24659  230376
14   27   63   124   485    36   90   1421  10865    46  128    23476  236202
15   27   64   124   497    36   93   1386  11123    47  131    24560  241776

mutation=2
2     a  348     a   500     a  463      a  44655     a  625        a  962221
3    70  105   597   806   103  151  14000  18118   149  213   313316  394666
4    47   84   364   649    68  121   8157  14590    96  171   177318  317607
5    39   80   284   614    56  115   6018  13761    78  162   128958  299360
6    36   80   243   615    50  115   4901  13744    70  162   103561  298950
7    34   82   219   629    47  118   4208  14042    65  166    87774  305250
8    33   85   203   647    45  121   3735  14469    62  171    76781  314506
9    32   87   191   669    44  125   3387  14954    59  177    68822  325265
10   32   90   183   691    43  130   3122  15440    58  183    62571  335538
11   31   93   177   713    43  134   2914  15926    57  188    57733  346085
12   31   96   172   733    42  137   2745  16397    56  194    53700  356640
13   31   98   168   754    42  141   2606  16857    56  199    50318  366325
14   31  101   165   774    42  145   2488  17303    56  204    47487  375945
15   31  103   163   793    42  149   2388  17724    56  209    45054  385233


mutation=3
2   416    b  2704     b   566    b  58884      b   768    b  1278369       b
3     c  159     c  1287     c  229      c  31032     c  325        c  688633
4    90  114   792   908   133  164  19342  21294   193  233   426784  469190
5    60  104   490   821    88  150  11371  19050   125  212   250193  418463
6    50  103   382   806    72  148   8917  18605   101  209   185013  408269
7    45  104   325   817    64  150   6968  18815    89  212   149953  412510
8    42  107   289   838    59  154   5991  19298    81  218   127695  422995
9    40  111   265   864    56  159   5312  19893    76  225   112179  436012
10   38  114   248   892    53  164   4811  20541    73  232   100678  449925
11   37  118   235   919    52  169   4424  21166    70  239    91740  463938
12   37  121   225   946    51  174   4116  21796    69  246    84601  478114
13   36  124   217   973    50  179   3865  22406    67  252    78743  491271
14   36  127   210   998    49  183   3658  23011    66  259    73836  504429
15   36  130   205  1022    49  188   3481  23578    65  265    69670  516839

mutation=4
2     d 1147     d  7608     d 1564      d 168285     d 2131        d 3666858
3   402  349  2549  2066   538  461  51458  43946   721  612  1111038  949769
4   145  149  1126  1140   209  213  25214  25979   296  300   600000  567967
5    94  128   812   987   138  184  19767  22497   199  259   438705  491914
6    69  124   561   953   100  178  13159  21686   143  250   300000  474056
7    58  125   453   958    84  179  10500  21796   118  252   200000  476267
8    52  128   390   980    75  183   8948  22282   105  257   185964  486771
9    48  131   349  1008    69  188   7465  22913    96  265   160620  500750
10   46  135   320  1038    65  194   6668  23618    90  273   142447  516039
11   44  139   299  1069    62  199   6066  24324    85  281   128703  531745
12   43  143   283  1100    60  205   5594  25035    82  289   117840  547029
13   42  147   269  1130    58  210   5213  25723    80  297   109060  562118
14   41  151   259  1158    57  216   4900  26377    78  304   101786  576605
15   40  154   250  1186    56  221   4634  27019    76  311    95649  590692


mutation=5
2  1113 1841  7137 11455  1507 2476 155053 245369  2038 3328  3365250 5310142
3   482  532  3303  3408   663  720  73968  73848   910  975  1616146 1601564
4   370  192  2178  1428   492  305  44962  32560   655  380   964316  713259
5   120  156  1929  1176   173  272  20801  26795   244  311   452149  586413
6    91  147   726  1115   132  222  16362  25343   188  295   355886  554258
7    74  147   500  1112   107  210  14000  25253   153  294   300000  552080
8    64  150   499  1133    93  214  11191  25723   131  300   254320  562118
9    58  154   440  1163    84  219  10500  26395   118  308   200000  577296
10   54  158   398  1197    77  225   8986  27181   108  317   184955  593975
11   51  163   367  1233    73  232   7765  27980   102  326   166406  611709
12   49  167   344  1269    70  239   7121  28783    97  335   151890  629200
13   48  172   325  1303    67  245   6605  29570    93  344   140077  646404
14   46  176   310  1336    65  251   6180  30339    90  353   130371  663337
15   45  180   298  1368    63  257   5824  31072    87  362   122251  679534

The maximum attainable powers for these cases are (a) 0.500, (b) 0.778, (c)
0.556 and (d) 0.625. See text for more details.
\end{alltt}

Several observations can be made.  Firstly, for the three generations
considered here the decay of linkage disequilibrium is not so remarkable for
$\theta=0.005$ compared to $\theta=0.050$.  Secondly, power usually increases
with the number of alleles, so the sample size required for biallelic marker
may be ten times or more for marker with fifteen alleles.  The power seems to
attain its maximum at about 10 alleles.  Finally, a larger sample size is
required for more mutations.  For $g=20$ generations this increase is very slow
but for $g=50$ and $\theta=0.05$, the increase is roughly proportional to
number of mutations.  Moreover, the log-likelihood ratio statistic provide
slightly larger power than Pearson $\chi^2$.

There are four special cases; the mutation-allele combinations 2-2, 3-3 and 4-2
under equal allele frequencies when there is one mutation in each allele, and
3-2 under unequal allele frequencies when there are two mutations occurring in
the more frequent allele and one mutation occurring in the less frequent
allele.  In such cases, the noncentrality parameters reduce to zero, so it is
impossible to achieve more power than certain magnitude by the statistics
proposed here.  Suppose the number of instances with zero noncentrality
parameter is $n_0$, the maximum attainable power is predicted as follows,
\begin{eqnarray}
1-n_0(1-\alpha)w_i\label{eqn4}
\end{eqnarray}
where $\alpha$ is the type I error rate, $w_i$ is the probability weight.

For equifrequent marker, according to equation (\ref{eqn1}), the
three combinations each result in the same expected allele frequencies as
observed in $n_0$=2, 6 and 6 instances, respectively. Since the weight
$w_i$ is $1/n^m$, the optimal powers are therefore 0.500, 0.778 and 0.625.
Under unequal allele frequencies the weight is $4/27$ and there are
$n_0=3$ such instances so the optimal power is 0.556.


\section{Discussion}

Fine mapping of disease mutation depends on both mutation-marker distance and
age of mutation.  We have in addition quantified the effect of number of
mutations and number of alleles.  When $\theta\ll 0.5$, equation (\ref{eqn1})
is roughly a linear function of $g\theta$, so that for old mutations to be
detected, the marker has to be fairly close.  Even in population isolates,
recent mutations are less interesting than those brought in by founders.  This
has important implications with respect to the recent attention to SNPs (Guo \&
Lange 2000).

We only examined two simple scenarios, one assuming equal allele frequencies
and one with a geometric decrease between alleles.  Results for 20 and 80
generations not reported in Sham, Zhao \& Curtis (2000) are also included.
Equal allele frequency tend to have greater power, as noted by Ott (1992) in
the context of linkage analysis.  The optimal number of alleles won't
necessarily be large for detecting specific number of mutations although the
reality may be more complicated.  We do not consider actual mutation mechanism
such as single and multiple step mutations (Farrall \& Weeks 1997).  Many
factors such as selection or population growth rate would affect the result.
An obvious note from Chapman \& Wijsman (1998) is that when choosing between
markers with the same heterozygosity and different number of alleles, the one
with fewer alleles will be more powerful.

The current model is simple to interpret and easy to apply to multiple
mutations.  Ott \& Rabinowitz (1997), Chapman \& Wijsman (1998) used simple
deterministic models and assumed that a single founder of the population was
responsible for a mutation at the disease locus at $g$ generations ago, and a
marker locus with recombination rate $\theta$ with the disease mutation was
used, which had $n$ alleles each with frequencies $\pi_1, \ldots, \pi_n$.  Ott
\& Rabinowitz (1997) assumed that a priori any member of the population was
equally likely to be the founder so the probability that mutation coupled with
$i$-th allele was $\pi_i$.  After $g$ generations a proportion
$\rho=1-(1-\theta)^{(g-1)}$ had undergone at least one recombination between
the disease locus and the marker locus.  Also among the proportion of
chromosomes in which the disease mutation undergone a recombination, the
distribution of marker alleles was not different from that of general
population, which implied there were random mixing and negligible genetic
drift.  Let $O_i$ denote the number of chromosomes in the sample with the
$i$-th allele, and $E_i=N\pi_i$ the expectation under the null hypothesis of no
disequilibrium, a Pearson $\chi^2$ statistic $\sum_{i=1}^n{(O_i-E_i)^2}/{E_i}$
can be constructed and referred to $\chi^2$ distribution with $n-1$ degrees of
freedom.  If the original mutation was in coupling with a fixed but arbitrary
allele $i^\star$, so that the prevalence of the $i^\star$-th allele in the
population of chromosomes containing the original mutation was
$(1-\rho)+\rho\pi_{i^\star}$, the noncentrality parameter became
$N(1-\rho)^2(\pi_{i^\star}^{-1}-1)$, or $N(1-\rho)^2(n-1) \equiv
N(n-1)(1-\theta)^{2(g-1)}$ with equal allele frequencies, where $N$ is the
sample size.  Setting $N=100$ and assume equal allele frequencies, they
conducted 3,600 simulations for $g=20, 50, 80$ and $n=2, \ldots, 10$.  In
Chapman \& Wijsman (1998), a mutation introduced into the population had a very
small frequency $r$, then the marker allele frequencies in the disease
mutation-carrying chromosomes and the normal chromosomes were
$$x_j^{(g)}=\cases{[1-(1-\theta)^g]\pi_j & $j=1,\ldots,i-1,i+1,\ldots,n$\cr
(1-\theta)^g+[1-(1-\theta)^g]\pi_j & $j=i$} $$ and $y_j^{(g)}\approx\pi_j,
~j=1,\ldots,n$, where $i$ was the index for initial mutation on a background of
allele $i$.  They examined several disease models.  In the recessive case, the
observed marker allele frequencies in cases and controls were roughly
$x_j^{(g)}$ and $y_j^{(g)}, ~j=1,\ldots,n$; in the dominant case,
$0.5(x_j^{(g)}+\pi_j)$ and $y_j^{(g)}, ~j=1,\ldots,n$; for mixture model with
any other fractions.  The same $\chi^2$ is constructed but Chapman \& Wijsman
(1998) assumed marker polymorphism proceed the disease mutation and an
expected $\chi^2$ statistic was obtained from the weighted average of the
$\chi^2$ for each allele.  Their considerations were mainly for biallelic and
multiallelic markers with equifrequent alleles.  Equation (\ref{eqn1}) is
comparable to Ott \& Rabinowitz (1997) for one mutant allele.  Since
$(1-\theta)^g\approx 1-\theta g$, we have $\rho\approx\theta g$, their basic
relation becomes $1-\theta g + \theta g \pi_{i^\star}$.  By equation
(\ref{eqn1}), it would be $(1-\theta g+\theta g\pi_j)$, and we evaluate $w_j$
by enumeration instead of by simulation.  Chapman \& Wijsman (1998) also had
similar forms.


\section{Bibliographic notes}

Consider a specific haplotype $(A_i,B_j)$ formed by loci $A$ and $B$ whose
alleles are indexed by $i$ and $j$.  Suppose that start from generation 0, all
individuals are produced by random mating.  Then the probability of haplotype
$(A_i,B_j)$ in generation 1 is $h_1(A_i,B_j)=(1-\theta)h_0(A_i,B_j)+\theta
P(A_i)P(B_j)$, where $\theta$ is the recombination rate between $A$ and $B$,
and $h_0(A_i,B_j)$ is the probability of $(A_i,B_j)$ in generation 0.  The
disequilibrium in generation 1 is
$\Delta_1=h_1(A_i,B_j)-P(A_i)P(B_j)=(1-\theta)\Delta_0$, where $\Delta_0$ is
the disequilibrium in generation 0.  After $g$ generations, the disequilibrium
is $\Delta_g=(1-\theta)^g\Delta_0$.  For loci that are unlinked ($1-\theta$ =
1/2), the equilibrium is reached quickly; for example, $(1/2)^{10}=1/1024$.  In
contrast, for tightly linked loci the disequilibrium decays very slowly; for
example $(1-\theta)^{10}>1/3$ for $\theta=0.1$.  In the case of tightly linked
loci $(1-\theta)^g\approx 1-g\theta$, and $\theta g$ dominates the decay of
disequilibrium.  Similar result can be obtained for multiple loci (Geiringer
1944; Bennett 1954).  For simplicity subscripts of loci A, B and C are dropped
and the expressions become $h_{g+1}(AB)=(1-\theta)h_g(AB)+\theta P(A)P(B)$,
$\Delta_g(AB)=h_g(AB)-P(A)P(B)$,
$\Delta_g(AB)=\theta\Delta_{g-1}(AB)=(1-\theta)^g\Delta_0(AB)$.  For three loci
A, B and C, we have
$h_{g+1}(ABC)=(1-\theta_{AB})(1-\theta_{BC})(1-\theta_{AC})h_g(ABC)
+\theta_{AB}(1-\theta_{BC})P(A)h_g(BC)+\theta_{AB}(1-\theta_{AC})P(B)h_g(AC)
+\theta_{AC}(1-\theta_{AB})P(C)h_g(AB)$, and if we define
$\Delta_g(ABC)=h_g(ABC)-P(A)\Delta_g(BC)-P(B)h_g(AC)-P(C)\Delta_g(AB)
-P(A)P(B)P(C)$, then
$\Delta_g(ABC)=(1-\theta_{AB})(1-\theta_{BC})(1-\theta_{AC})\Delta_{g-1}(ABC)
=[(1-\theta_{AB})(1-\theta_{BC})(1-\theta_{AC})]^g\Delta_0(ABC)$.

H\"{a}stbacka et al.  (1992) noted that the best chance of success with allelic
association mapping is when most of the disease chromosomes in the population
descend from a single ancestral mutation and the mutation is old enough to
allow recombination to break up the ancestral haplotype, but not so old that
the maintained neighbourhood around the disease locus is too small to be easily
detected.  These conditions often hold for isolated founder populations that
are not very old.  If one can identify an isolated human population in which a
significant fraction of affected individuals have inherited the predisposing
allele from a common ancestor, one can exploit the tremendous power of LD
mapping (allelic association mapping).  The resolution of recombinational
mapping in a limited family series can be greatly improved, even in the case of
relatively recently expanded genes in isolated founder populations, for disease
chromosomes that descend from the same ancestral mutation should share a common
haplotype in a neighborhood of the disease locus, reflecting the haplotype on
the ancestral chromosome on which the mutation occurred.  De la Chapelle (1993,
1998) gave many practical examples.

Several methods were proposed to formulate expected haplotype frequencies
assuming affected individuals are descended from a common founder, especially
for Mendelian disorders in population isolates such as Finland (Terwilliger
1995; Jorde 1995; Kaplan 1995; Devlin et al.  1996; Xiong \& Guo 1997; Guo
1997; Rannala \& Slatkin 1998; Lazzeroni 1998).  These developments rely on
simplicity of the underlying population genetics model.  The methods of both
Kaplan et al.  (1995) and Devlin et al.  (1996) would only be effective when a
single haplotype currently has a high frequency in disease chromosomes.
Terwilliger's method combines the likelihood ratio statistics from separate LD
analyses of each marker and does not use all the information in a haplotype.
Service et al.  (1999) evaluates a LD-mapping method for genome screening in
founder populations which make direct use of haplotype information assuming
affected individuals are related to a common founder.  Rannala \& Slatkin
(1998) showed this method will produce maximum likelihood estimate when
genealogy of affected individuals is of ``star'' type, which implies
individuals are related only through one common ancestor.  Other works include
Thompson \& Neel (1997), Lazzeroni (1998), Graham \& Thompson (1998, 2000),
McPeek \& Strahs (1999), MacLean et al.  (2000), Morris et al.  (2000), Wright
et al.  (2000).  Clayton (2000) \& Lazzeroni (2001) gave comprehensive reviews
of these works. A coalescent approach to study LD between SNPs was also
conducted by Z\"{o}llner \& von Haeseler (2000). A Bayesian approach has been
adopted by Liu et al (2001).

A procedure developed by Waller et al.  (1995) for evaluating power is
described here.  Consider the distribution function of $Z=\sum_{k=1}^na_kY_k$,
where $a_k, k=1,\ldots, n$ are the weights taking real values, and $Y_k,
k=1,\ldots,n$ are independent non-central chi-squared random variables with
$n_k$ degrees of freedom and noncentral parameters $\delta_k$, for
$k=1,\ldots,n$.  The distribution functions of $Y_k$ and $Z$ do not have closed
form, but their characteristic functions do.  From above the characteristic
function for $Z$ is $\phi_Z(t)=\prod_{k=1}^n\phi_{Y_{k}}(a_kt)$.  Also
$E(Z)=\mu=\sum_{k=1}^na_k(n_k+\delta_k)$ and
$V(Z)=\sigma^2=\sum_{k=1}^na_k^2(n_k+2\delta_k)$.  We scale $Z$ to obtain
$Z'=(Z-\mu)/\sigma$, whose characteristic function is given by the formula
$\phi_{Z'}(t)=\exp(-it\mu)\phi_Z(t/\sigma)$, the following inversion formula
can be utilized to obtain the cumulative distribution function.
$$F_Z(z)=\frac{1}{2}+\frac{\mu
z}{2\pi}-\sum\limits_{n=1-H}^{H-1}\frac{\phi_Z(\eta n)}{2\pi in}e^{-i\eta nz},
\quad n\neq 0$$ where $\eta$ is chosen to ensure the range of $F_Z(z)$ is fully
represented, i.e., the values of $F_Z(z)$ include both 0 and 1.  $H$ is chosen
to define the $(2H-1)$ points where the cumulative distribution function is
estimated.  The values for $z$ may be chosen as the Fourier frequencies, that
is $z_j=(2\pi(j-H)/2\eta(H-1)))$ for $j=1,\ldots,2H-1$.  Thus the sum can be
efficiently calculated using FFT.
