---
title: "An Overview of pQTLtools"
author: "Jing Hua Zhao"
output:
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 2
    check_title: false
bibliography: refs.bib
vignette: >
  %\VignetteIndexEntry{An Overview of pQTLtools}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
set.seed(0)
knitr::opts_chunk$set(
  out.extra = 'style="display:block; margin: auto"',
  fig.align = "center",
  fig.path = "pQTLtools/",
  collapse = TRUE,
  comment = "#>",
  dev = "png")
```

## cis/trans classification
```{r cistrans, fig.height=7, fig.width=7}
options(width=200)
require(gap)
INF <- Sys.getenv("INF")
f <- file.path(INF,"work","INF1.merge")
clumped <- read.delim(f,as.is=TRUE)
hits <- merge(clumped[c("CHR","POS","MarkerName","prot","log10p")],
              inf1[c("prot","uniprot")],by="prot")
names(hits) <- c("prot","Chr","bp","SNP","log10p","uniprot")
cistrans <- cis.vs.trans.classification(hits,inf1,"uniprot")
cis.vs.trans <- with(cistrans,data)
knitr::kable(with(cistrans,table),caption="Table 1. cis/trans classification")
with(cistrans,total)
circos.cis.vs.trans.plot(hits=f,inf1,"uniprot")
```
This is used to generate 2d-Manhattan plot
```{r mht2d, fig.height=8}
INF <- Sys.getenv("INF")
d <- read.csv(file.path(INF,"work","INF1.merge.cis.vs.trans"),as.is=TRUE)
t2d <- mhtplot2d(d)
```
and ideogram (biomaRt is not always on so we include the plot below)
```r
INF <- Sys.getenv("INF")
f <- file.path(INF,"work","INF1.merge")
INF1_merge <- read.delim(f)[c("Chrom","Start","End","prot","MarkerName")]
singletons <- with(INF1_merge,End-Start==1)
INF1_merge[singletons,"Start"] <- INF1_merge[singletons,"Start"] - 1e6
INF1_merge[singletons,"End"] <- INF1_merge[singletons,"End"] + 1e6
small <- with(INF1_merge, Start<0)
INF1_merge[small,"Start"] <- 0
cvt <- read.csv(file.path(INF,"work","/INF1.merge.cis.vs.trans"),as.is=TRUE)
INF1_merge_cvt <- merge(INF1_merge,cvt,by.x=c("prot","MarkerName"),by.y=c("prot","SNP"))
ord <- with(INF1_merge_cvt,order(Chr,bp))
INF1_merge_cvt <- INF1_merge_cvt[ord,]

library(httr)
set_config(config(ssl_verifypeer = 0L))
library(biomaRt)
library(regioneR)
mart <- useMart(biomart = "ensembl", dataset = "hsapiens_gene_ensembl")
attrs <- c("hgnc_symbol", "chromosome_name", "start_position", "end_position", "band")
bm <- with(INF1_merge_cvt, getBM(attributes = attrs,
           filters="chromosomal_region", values=paste0(Chr,":",bp,":",bp), mart=mart))
genes <- with(bm,toGRanges(chromosome_name,start_position,end_position,labels=hgnc_symbol))

library(karyoploteR)
attach(INF1_merge_cvt)
sentinels <- toGRanges(Chr,bp-1,bp,labels=p.gene)
cis.regions <- toGRanges(Chr,cis.start,cis.end)
loci <- toGRanges(Chr,Start,End)
panel <- toGRanges(p.chr,p.start,p.end,labels=p.gene)
colors <- c("red","blue")
seqlevelsStyle(sentinels) <- "UCSC"
kp <- plotKaryotype(genome="hg19",chromosomes=levels(seqnames(sentinels)))
kpAddBaseNumbers(kp)
kpPlotRegions(kp,data=loci,r0=0.05,r1=0.15,border="black")
kpPlotMarkers(kp, data=sentinels, labels=p.gene, text.orientation="vertical",
              cex=0.45, y=0.3*seq_along(p.gene)/length(p.gene), srt=30,
              ignore.chromosome.ends=TRUE,
              adjust.label.position=TRUE, label.color=colors[2-cis], label.dist=0.002,
              cex.axis=3, cex.lab=3)
legend("bottomright", bty="n", pch=c(19,19), col=colors, pt.cex=0.4,
       legend=c("cis", "trans"), text.col=colors, cex=0.8, horiz=FALSE)
# kpPlotLinks(kp, data=loci, data2=panel, col=colors[2-cis])
detach(INF1_merge_cvt)
```
```{r karyoplot, echo=FALSE, fig.height=12, fig.width=8}
knitr::include_graphics('karyoplot.png',dpi=300)
```

## Genomic regions enrichment analysis

It is now considerably easier with Genomic Regions Enrichment of Annotations Tool (GREAT).

```{r GREAT, fig.height=6,fig.width=8, results='hide', warning=FALSE}
post <- function(regions)
{
  library(rGREAT)
  job <- invisible(submitGreatJob(get(regions), species="hg19", version="3.0.0"))
  et <- invisible(getEnrichmentTables(job,download_by = 'tsv'))
  tb <- do.call('rbind',et)
  write.table(tb,file=paste0(regions,".tsv"),quote=FALSE,row.names=FALSE,sep="\t")
  invisible(list(job=job,tb=tb))
}

all_regions <- function()
{
  cistrans.post <- post("cistrans")
  job <- with(cistrans.post,job)
  plotRegionGeneAssociationGraphs(job,type=c(1,3))
  availableOntologies(job)
# plot of the top term
  par(mfcol=c(3,1))
  plotRegionGeneAssociationGraphs(job, ontology="GO Molecular Function",
                                  termID="GO:0005126", type=c(1,3))
  plotRegionGeneAssociationGraphs(job, ontology="GO Biological Process",
                                  termID="GO:0009611", type=c(1,3))
  plotRegionGeneAssociationGraphs(job, ontology="GO Cellular Component",
                                  termID="GO:0005615", type=c(1,3))
}

three_regions <- function()
{
  tb_all <- data.frame()
  for (r in c("IL12B","KITLG","TNFSF10"))
  {
    r.post <- post(r)
    tb_all <- rbind(tb_all,data.frame(gene=r,with(r.post,tb)))
  }
  write.table(tb_all,file="IL12B-KITLG-TNFSF10.tsv",quote=FALSE,row.names=FALSE,sep="\t")
}

library(dplyr)
M <- 1e+6
INF1_merge <- read.delim(file.path(INF,"work","INF1.merge")) %>%
              mutate(chr=Chrom, start=POS-M, end=POS+M) %>%
              mutate(start=if_else(start<0,0,start)) %>%
              select(prot,MarkerName,chr,start,end)
cistrans <- INF1_merge %>% select(chr,start,end) %>% arrange(chr,start,end) %>% distinct()
all_regions()

# Specific regions
IL12B <- filter(INF1_merge,prot=="IL.12B") %>% select(-c(prot,MarkerName))
KITLG <- filter(INF1_merge,prot=="SCF") %>% select(-c(prot,MarkerName))
TNFSF10 <- filter(INF1_merge,prot=="TRAIL") %>% select(-c(prot,MarkerName))

three_regions()
```
The top terms could be extracted as follows,
```{r, echo=FALSE}
options(width=100)
great3 <- read.delim("IL12B-KITLG-TNFSF10.tsv") %>%
          mutate(flag=if_else(BinomP<1e-4,"x","")) %>% arrange(desc(flag))
knitr::kable(subset(great3,flag=="x"),caption="Table 2. GREAT results (IL12B-KITLG-TNFSF10)")
great <- read.delim("cistrans.tsv") %>%
         mutate(flag=if_else(BinomP<1e-4,"x","")) %>% arrange(desc(flag))
knitr::kable(subset(great,flag=="x"),caption="Table 2. GREAT results (All regions)")
unlink("IL12B-KITLG-TNFSF10.tsv")
unlink("IL12B.tsv")
unlink("KITLG.tsv")
unlink("TNFSF10.tsv")
unlink("cistrans.tsv")
```

## eQTL Catalogue for colocalization analysis

See example associated with `import_eQTLCatalogue`. A related function is `import_OpenGWAS` used to fetch data from OpenGWAS. The cis-pQTLs and 1e+6 
flanking regions were considered and data are actually fetched from files stored locally. For illustration, only the first sentinel was used.

```{r, results='hide', warning=FALSE}
library(pQTLtools)
f <- file.path(system.file(package="pQTLtools"),"eQTL-Catalogue","hg19ToHg38.over.chain")
chain <- rtracklayer::import.chain(f)
pkgs <- c("dplyr", "ggplot2", "readr", "coloc", "GenomicRanges","seqminer")
invisible(lapply(pkgs, require, character.only = TRUE))
HPC_WORK <- Sys.getenv("HPC_WORK")
gwasvcf::set_bcftools(file.path(HPC_WORK,"bin","bcftools"))
f <- file.path(system.file(package="pQTLtools"),"eQTL-Catalogue","tabix_ftp_paths.tsv")
tabix_paths <- read.delim(f, sep = "\t", header = TRUE, stringsAsFactors = FALSE) %>%
               dplyr::as_tibble()
HOME <- Sys.getenv("HOME")
f <- file.path(system.file(package="pQTLtools"),"eQTL-Catalogue",
               "tabix_ftp_paths_imported.tsv")
imported_tabix_paths <- within(read.delim(f, stringsAsFactors = FALSE) %>%
                        dplyr::as_tibble(),
      {ftp_path <- gsub("ftp://ftp.ebi.ac.uk/pub/databases/spot/eQTL/csv/GTEx_V8/ge",
                        paste0(HOME,"/rds/public_databases/GTEx/csv"),ftp_path)})
INF <- Sys.getenv("INF")
M <- 1e6
sentinels <- subset(read.csv(file.path(INF,"work","INF1.merge.cis.vs.trans")),cis)

liftRegion <- function(x,chain,flanking=1e6)
{
  require(GenomicRanges)
  gr <- with(x,GenomicRanges::GRanges(seqnames=chr,IRanges::IRanges(start,end))+flanking)
  seqlevelsStyle(gr) <- "UCSC"
  gr38 <- rtracklayer::liftOver(gr, chain)
  chr <- colnames(table(seqnames(gr38)))
  chr <- gsub("chr","",chr)
  start <- min(unlist(start(gr38)))
  end <- max(unlist(end(gr38)))
  invisible(list(chr=chr,start=start,end=end,region=paste0(chr,":",start,"-",end)))
}

sumstats <- function(prot,chr,region37)
{
  cat("GWAS sumstats\n")
  f <- file.path(INF,"METAL/gwas2vcf",paste0(prot,".vcf.gz"))
  gwas_stats <- gwasvcf::query_gwas(f, chrompos = region37)
  gwas_stats <- gwasvcf::vcf_to_granges(gwas_stats) %>%
                keepSeqlevels(chr) %>%
                renameSeqlevels(paste0("chr",chr))
  gwas_stats_hg38 <- rtracklayer::liftOver(gwas_stats, chain) %>%
                     unlist() %>%
                     renameSeqlevels(chr) %>%
                     dplyr::as_tibble() %>%
                     dplyr::transmute(chromosome = seqnames,
                                      position = start, AF, ES, SE, LP, SS) %>%
                     dplyr::mutate(id = paste(chromosome, position, sep = ":")) %>%
                     dplyr::mutate(MAF = pmin(AF, 1-AF)) %>%
                     dplyr::group_by(id) %>%
                     dplyr::mutate(row_count = n()) %>%
                     dplyr::ungroup() %>%
                     dplyr::filter(row_count == 1)
}

coloc <- function(gwas_stats_hg38,ensGene,region38)
{
  cat("c. GTEx_v8 imported eQTL datasets\n")
  rnaseq_df <- dplyr::filter(imported_tabix_paths, quant_method == "ge") %>%
               dplyr::mutate(qtl_id = paste(study, qtl_group, sep = "_"))
  ftp_path_list <- setNames(as.list(rnaseq_df$ftp_path), rnaseq_df$qtl_id)
  hdr <- file.path(system.file(package="pQTLtools"),"eQTL-Catalogue","column_names.GTEx")
  column_names <- names(read.delim(hdr))
  safe_import <- purrr::safely(import_eQTLCatalogue)
  summary_list <- purrr::map(ftp_path_list, ~safe_import(., region38,
                             selected_gene_id = ensGene, column_names))
  result_list <- purrr::map(summary_list, ~.$result)
  result_list <- result_list[!unlist(purrr::map(result_list, is.null))]
  result_filtered <- purrr::map(result_list[lapply(result_list,nrow)!=0],
                                ~dplyr::filter(., !is.na(se)))
  purrr::map_df(result_filtered, ~run_coloc(., gwas_stats_hg38), .id = "qtl_id")
}

sentinel <- sentinels[1,]
isnpid <- within(gap::inv_chr_pos_a1_a2(sentinel[["SNP"]]),
{
    chr <- gsub("chr","",chr)
    pos <- as.integer(pos)
    start <- pos-M
    if (start<0) start <- 0
    end <- pos+M
})
chr <- with(isnpid,chr)
region37 <- with(isnpid, paste0(chr,":",start,"-",end))
ensRegion37 <- with(subset(inf1,prot==sentinel[["prot"]]),paste0(chr,":",start,"-",end))
region38 <- with(liftRegion(isnpid,chain),region)
ensGene <- subset(inf1,prot==sentinel[["prot"]])[["ensembl_gene_id"]]
ensRegion38 <- with(liftRegion(subset(inf1,prot==sentinel[["prot"]]),chain),region)
f <- with(sentinel,paste0(prot,"-",SNP,".RDS"))
gwas_stats_hg38 <- sumstats(sentinel[["prot"]],chr,region37)
coloc_df_imported <- coloc(gwas_stats_hg38,ensGene,region38)
saveRDS(coloc_df_imported,file=f)
```

```{r coloc, fig.height=7, fig.width=8, echo=FALSE}
ggplot(gwas_stats_hg38, aes(x = position, y = LP)) +
geom_point() +
ggtitle(with(sentinel,paste0(prot,"-",SNP," association plot")))
rds <- readRDS(f)
ktitle <- with(sentinel,paste0("Table 4. Colocalisation results for ",prot,"-",SNP))
knitr::kable(rds, caption=ktitle)
unlink(f)
```

The function `sumstats()` obtained meta-analysis summary statistics (in build 37 and therefore lifted over to build 38) to be used in colocalisation 
analysis.  The output are saved in the `.RDS` files.

## pQTL-based Mendelian Randomization (MR)

The function `pqtlMR` is derived from @zheng20. It has an attractive feature that multiple pQTLs
can be used together for conducting MR with a list of outcomes from MR-Base. For generic applications
the `run_TwoSampleMR()` function can be used.

```{r, warning=FALSE}
library(TwoSampleMR)
f <- file.path(system.file(package="pQTLtools"),"tests","Ins.csv")
ivs <- format_data(read.csv(f))
caption2 <- "Table 3. ABO/LIFR variants and CHD/FEV1"
knitr::kable(ivs, caption=paste(caption2,"(instruments)"))
ids <- c("ieu-a-7","ebi-a-GCST007432")
pqtlMR(ivs, ids)
result <- read.delim("pQTL-combined-result.txt",header=TRUE)
knitr::kable(result,caption=paste(caption2, "(result)"))
single <- read.delim("pQTL-combined-single.txt",header=TRUE)
knitr::kable(subset(single,!grepl("All",SNP)), caption=paste(caption2, "(single)"))
invisible(sapply(c("harmonise","result","single"),
                 function(x) unlink(paste0("pQTL-combined-",x,".txt"))))
```

## run_TwoSampleMR

The function has similiarity to @dt18. The documentation example is quoted here,

```{r run_TwoSampleMR, fig.height=7, fig.width=7, results='hide', warning=FALSE}
library(TwoSampleMR)
outcomes <- "ebi-a-GCST007432"
prot <- "MMP.10"
type <- "cis"
f <- paste0(prot,"-",type,".mrx")
d <- read.table(file.path(system.file(package="pQTLtools"),"tests",f),
                header=TRUE)
exposure <- format_data(within(d,{P=10^logP}), phenotype_col="prot", snp_col="rsid",
                        chr_col="Chromosome", pos_col="Posistion",
                        effect_allele_col="Allele1", other_allele_col="Allele2",
                        eaf_col="Freq1", beta_col="Effect", se_col="StdErr",
                        pval_col="P", log_pval=FALSE,
                        samplesize_col="N")
clump <- clump_data(exposure)
outcome <- extract_outcome_data(snps=exposure$SNP,outcomes=outcomes)
harmonise <- harmonise_data(clump,outcome)
prefix <- paste(outcomes,prot,type,sep="-")
run_TwoSampleMR(harmonise, mr_plot="New", prefix=prefix)
```

The output is contained in individual `.txt` files, together with the scatter, forest, funnel and leave-one-out plots.

```{r, echo=FALSE}
caption4 <- "Table 5. MMP.10 variants and FEV1"
knitr::kable(read.delim(paste0(prefix,"-result.txt"),header=TRUE),caption=paste(caption4, "(result)"))
knitr::kable(read.delim(paste0(prefix,"-heterogeneity.txt"),header=TRUE),caption=paste(caption4,"(heterogeneity)"))
knitr::kable(read.delim(paste0(prefix,"-pleiotropy.txt"),header=TRUE),caption=paste(caption4,"(pleiotropy)"))
knitr::kable(read.delim(paste0(prefix,"-single.txt"),header=TRUE),caption=paste(caption4,"(single)"))
knitr::kable(read.delim(paste0(prefix,"-loo.txt"),header=TRUE),caption=paste(caption4,"(loo)"))
for (x in c("result","heterogeneity","pleiotropy","single","loo")) unlink(paste0(prefix,"-",x,".txt"))
```

## Literature on pQTLs

Those as identified by @sun18 and @suhre20 are included as EndNote libraries.

## UniProt IDs

The function `uniprot2ids` converts UniProt IDs to others.

## References
