<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a></li>
    <li><a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a></li>
    <li><a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a></li>
    <li><a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a></li>
    <li><a class="dropdown-item" href="../articles/spectrum.html">Spectrum data analysis</a></li>
    <li><a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../reference/pQTLtools.html">Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/"><span class="fa fab fa-r-project"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/"><span class="fa fas fa-fast-backward"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-urls" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">URLs</button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-urls"><li><a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a></li>
    <li><a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a></li>
    <li><a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a></li>
    <li><a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a></li>
    <li><a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.histo.fyi/">histo.fyi</a></li>
    <li><a class="external-link dropdown-item" href="https://metascape.org/">metascape</a></li>
    <li><a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a></li>
    <li><a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a></li>
    <li><a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a></li>
    <li><a class="external-link dropdown-item" href="https://string-db.org/">STRING</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a></li>
    <li><a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a></li>
  </ul></li>
<li class="nav-item"><a class="external-link nav-link" href="https://www.uniprot.org/"><span class="fa fas fa-fast-forward"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Changelog</h1>
      <small>Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.4" id="pqtltools-04">pQTLtools 0.4<a class="anchor" aria-label="anchor" href="#pqtltools-04"></a></h2>
<p>(2024/09/25)</p>
<ul><li>Add csq(), protein_altering_variants (so data/), histo.fyi</li>
<li>Adopt new hg38 LD blocks for turboman().</li>
<li>Add spectrum.Rmd.</li>
<li>Renew call to ldops in novelty_check() in accordance with ieugwasr.</li>
<li>Replace tests/ with most recent PROC results.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.3" id="pqtltools-03">pQTLtools 0.3<a class="anchor" aria-label="anchor" href="#pqtltools-03"></a></h2>
<p>(2024/05/26)</p>
<ul><li>Activate package logo for the GitHub repository.</li>
<li>Dedicated .R files for (blocks of) functions.</li>
<li>Use of scope operator for clarification, e.g., ensembldb::genes().</li>
<li>Use of OPENGWAS_JWT from <a href="https://api.opengwas.io/profile/" class="external-link uri">https://api.opengwas.io/profile/</a>.</li>
<li>Suggest IlluminaHumanMethylation450kmanifest, OUTRIDER.</li>
<li>Fix URL in DESCRIPTION/snakemake.Rmd.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.2" id="pqtltools-02">pQTLtools 0.2<a class="anchor" aria-label="anchor" href="#pqtltools-02"></a></h2>
<p>(2024/04/24)</p>
<div class="section level3">
<h3 id="milestones-0-2">Milestones<a class="anchor" aria-label="anchor" href="#milestones-0-2"></a></h3>
<ul><li>
<strong><em>2023.12</em></strong>. Add LocusZoom.js article.</li>
<li>
<strong><em>2023.05</em></strong>. Test data are fully available with medRxiv post of the SCALLOP paper.</li>
<li>
<strong><em>2022.12</em></strong>. A new package <a href="https://github.com/jinghuazhao/pQTLdata" class="external-link">pQTLdata</a> is created to hold panel and meta data.</li>
<li>
<strong><em>2022.06</em></strong>. It passes <a href="https://cran.r-project.org/" class="external-link">CRAN</a> checks with no warning.</li>
<li>
<strong><em>2021.02</em></strong>. A web-driven documentation is now available, <a href="https://jinghuazhao.github.io/pQTLtools/" class="uri">https://jinghuazhao.github.io/pQTLtools/</a>.</li>
</ul></div>
<div class="section level3">
<h3 id="package-0-2">Package<a class="anchor" aria-label="anchor" href="#package-0-2"></a></h3>
<p>The information here mirrors the package DESCRIPTION,</p>
<ul><li>Depends R (&gt;=3.5.0), pQTLdata.</li>
<li>Import dplyr, gap, ggplot2, Rdpack.</li>
<li>importFrom utils read.table tail.</li>
<li>Import from lmm as template, use save(compress=‘xz’).</li>
<li>Replace ChangeLog with NEWS.md.</li>
<li>LICENSE.md and README.md.</li>
<li>Reduce size by <code>sed -i '/ISSN/d' REFERENCES.bib</code>.</li>
<li>Suggest Biobase, BioStrings, GenomeInfoDb, GenomicRanges, IRanges, VariantAnnotation.</li>
<li>Suggest Roxygen2.</li>
<li>Suggest biomaRt, bookdown, circlize, cowplot.</li>
<li>Suggest gap.datasets, gwasvcf, htmlwidgets, httr, ieugwasr.</li>
<li>Suggest knitr, mclust, meta.</li>
<li>Suggest openxlsx, plotly, plyr.</li>
<li>Suggest regione, rgl, rmarkdown, rtracklayer.</li>
<li>Suggest scatterplot3d, seqminer, stringr.</li>
<li>GitHub action</li>
<li>inst/Bioconductor/.</li>
<li>inst/STRING/change_STRING_colors.py.</li>
<li>inst/UniProt|PPI/README.md.</li>
<li>inst/snakemake</li>
<li>pQTLtools.Rmd, bioconductor.Rmd, es.Rmd, LocusZoom.js.Rmd, snakemake.Rmd and SCALLOP-INF.Rmd articles.</li>
<li>List publications on pQTLs by Sun et al. (2018) and Suhre et al. (2020).</li>
</ul></div>
<div class="section level3">
<h3 id="functions-0-2">Functions<a class="anchor" aria-label="anchor" href="#functions-0-2"></a></h3>
<p>The list is in no particular order,</p>
<ul><li>peptideMapping(), peptideAssociationPlot().</li>
<li>make_ExpressionSet(), novelty_check(), qtl_lookup(), turboman(), turboqq().</li>
<li>Reflow turboman.r/[partial]turboqq.r by formatR::tidy_source().</li>
<li>pqtlMR(), run_TwoSampleMR().</li>
<li>run_coloc().</li>
<li>import_OpenGWAS()</li>
<li>import_eQTLCatalogue().</li>
<li>genequries(), regionqueries(), snpqueries().</li>
<li>uniprot2ids().</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.1" id="pqtltools-01">pQTLtools 0.1<a class="anchor" aria-label="anchor" href="#pqtltools-01"></a></h2>
<ul><li>First release</li>
</ul></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer></div>





  </body></html>

