<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Bioconductor notes â€¢ pQTLtools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Bioconductor notes">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span></a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a></li>
    <li><a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a></li>
    <li><a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a></li>
    <li><a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a></li>
    <li><a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/pQTLtools.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/"><span class="fa fab fa-r-project"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/"><span class="fa fas fa-fast-backward"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-urls" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">URLs</button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-urls">
<li><a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a></li>
    <li><a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a></li>
    <li><a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a></li>
    <li><a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a></li>
    <li><a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://metascape.org/">metascape</a></li>
    <li><a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a></li>
    <li><a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a></li>
    <li><a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a></li>
    <li><a class="external-link dropdown-item" href="https://string-db.org/">STRING</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a></li>
    <li><a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a></li>
  </ul>
</li>
<li class="nav-item"><a class="external-link nav-link" href="https://www.uniprot.org/"><span class="fa fas fa-fast-forward"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Bioconductor notes</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/vignettes/bioconductor.Rmd" class="external-link"><code>vignettes/bioconductor.Rmd</code></a></small>
      <div class="d-none name"><code>bioconductor.Rmd</code></div>
    </div>

    
    
<p>This article collects notes on Bioconductor packages, made available here to faciliate their use and extensions.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AnnotationDbi"</span>, <span class="st">"AnnotationFilter"</span>, <span class="st">"Biostrings"</span>, <span class="st">"ComplexHeatmap"</span>, <span class="st">"DESeq2"</span>, <span class="st">"EnsDb.Hsapiens.v86"</span>,</span>
<span>          <span class="st">"FlowSorted.DLPFC.450k"</span>, <span class="st">"GeneNet"</span>, <span class="st">"GenomicFeatures"</span>, <span class="st">"IlluminaHumanMethylation450kmanifest"</span>,</span>
<span>          <span class="st">"OUTRIDER"</span>,<span class="st">"RColorBrewer"</span>, <span class="st">"RMariaDB"</span>, <span class="st">"Rgraphviz"</span>, <span class="st">"S4Vectors"</span>, <span class="st">"SummarizedExperiment"</span>,</span>
<span>          <span class="st">"TxDb.Hsapiens.UCSC.hg38.knownGene"</span>, <span class="st">"bladderbatch"</span>, <span class="st">"clusterProfiler"</span>,</span>
<span>          <span class="st">"corpcor"</span>, <span class="st">"doParallel"</span>, <span class="st">"ensembldb"</span>, <span class="st">"fdrtool"</span>, <span class="st">"graph"</span>, <span class="st">"graphite"</span>, <span class="st">"heatmaply"</span>,</span>
<span>          <span class="st">"minfi"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"plyr"</span>, <span class="st">"quantro"</span>, <span class="st">"recount3"</span>, <span class="st">"sva"</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">p</span> <span class="kw">in</span> <span class="va">pkgs</span><span class="op">)</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"^package:"</span>, <span class="va">p</span>, <span class="st">"$"</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/search.html" class="external-link">search</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/warning.html" class="external-link">warning</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"This vignette needs package `"</span>, <span class="va">p</span>, <span class="st">"'; please install"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">pkgs</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2" number="1">
<h2 id="liftover">
<span class="header-section-number">1</span> liftover<a class="anchor" aria-label="anchor" href="#liftover"></a>
</h2>
<p>See <code>inst/turboman</code>in the source, <a href="https://github.com/jinghuazhao/pQTLtools/tree/master/inst/turboman" class="external-link uri">https://github.com/jinghuazhao/pQTLtools/tree/master/inst/turboman</a>, or <code>turboman/</code> directory in the installed package.</p>
</div>
<div class="section level2" number="2">
<h2 id="normalisation">
<span class="header-section-number">2</span> Normalisation<a class="anchor" aria-label="anchor" href="#normalisation"></a>
</h2>
<div class="section level3" number="2.1">
<h3 id="combat">
<span class="header-section-number">2.1</span> ComBat<a class="anchor" aria-label="anchor" href="#combat"></a>
</h3>
<p>This is the documentation example, based on Bioconductor 3.14.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">bladderdata</span>, package<span class="op">=</span><span class="st">"bladderbatch"</span><span class="op">)</span></span>
<span><span class="va">edat</span> <span class="op">&lt;-</span> <span class="va">bladderEset</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span></span>
<span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">edat</span><span class="op">)</span></span>
<span><span class="va">batch</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">$</span><span class="va">batch</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">batch</span><span class="op">)</span></span>
<span><span class="co">#&gt; batch</span></span>
<span><span class="co">#&gt;  1  2  3  4  5 </span></span>
<span><span class="co">#&gt; 11 18  4  5 19</span></span>
<span><span class="fu">quantro</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/quantro/man/matboxplot.html" class="external-link">matboxplot</a></span><span class="op">(</span><span class="va">edat</span>,<span class="va">batch</span>,cex.axis<span class="op">=</span><span class="fl">0.6</span>,notch<span class="op">=</span><span class="cn">TRUE</span>,pch<span class="op">=</span><span class="fl">19</span>,ylab<span class="op">=</span><span class="st">"Expression"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:ComBat-1"></span>
<img src="bioconductor/ComBat-1.png" alt="ComBat example" width="672"><p class="caption">
Figure 2.1: ComBat example
</p>
</div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">quantro</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/quantro/man/matdensity.html" class="external-link">matdensity</a></span><span class="op">(</span><span class="va">edat</span>,<span class="va">batch</span>,xlab<span class="op">=</span><span class="st">" "</span>,ylab<span class="op">=</span><span class="st">"density"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topleft"</span>,legend<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,col<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,lty<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:ComBat-2"></span>
<img src="bioconductor/ComBat-2.png" alt="ComBat example" width="672"><p class="caption">
Figure 2.2: ComBat example
</p>
</div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># 1. parametric adjustment</span></span>
<span><span class="va">combat_edata1</span> <span class="op">&lt;-</span> <span class="fu">sva</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sva/man/ComBat.html" class="external-link">ComBat</a></span><span class="op">(</span>dat<span class="op">=</span><span class="va">edat</span>, batch<span class="op">=</span><span class="va">batch</span>, par.prior<span class="op">=</span><span class="cn">TRUE</span>, prior.plots<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Found5batches</span></span>
<span><span class="co">#&gt; Adjusting for0covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; Standardizing Data across genes</span></span>
<span><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span><span class="co">#&gt; Finding parametric adjustments</span></span>
<span><span class="co">#&gt; Adjusting the Data</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:ComBat-3"></span>
<img src="bioconductor/ComBat-3.png" alt="ComBat example" width="672"><p class="caption">
Figure 2.3: ComBat example
</p>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># 2. non-parametric adjustment, mean-only version</span></span>
<span><span class="va">combat_edata2</span> <span class="op">&lt;-</span> <span class="fu">sva</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sva/man/ComBat.html" class="external-link">ComBat</a></span><span class="op">(</span>dat<span class="op">=</span><span class="va">edat</span>, batch<span class="op">=</span><span class="va">batch</span>, par.prior<span class="op">=</span><span class="cn">FALSE</span>, mean.only<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using the 'mean only' version of ComBat</span></span>
<span><span class="co">#&gt; Found5batches</span></span>
<span><span class="co">#&gt; Adjusting for0covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; Standardizing Data across genes</span></span>
<span><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span><span class="co">#&gt; Finding nonparametric adjustments</span></span>
<span><span class="co">#&gt; Adjusting the Data</span></span>
<span></span>
<span><span class="co"># 3. reference-batch version, with covariates</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">cancer</span><span class="op">)</span>, data<span class="op">=</span><span class="va">pheno</span><span class="op">)</span></span>
<span><span class="va">combat_edata3</span> <span class="op">&lt;-</span> <span class="fu">sva</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sva/man/ComBat.html" class="external-link">ComBat</a></span><span class="op">(</span>dat<span class="op">=</span><span class="va">edat</span>, batch<span class="op">=</span><span class="va">batch</span>, mod<span class="op">=</span><span class="va">mod</span>, par.prior<span class="op">=</span><span class="cn">TRUE</span>, ref.batch<span class="op">=</span><span class="fl">3</span>, prior.plots<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using batch =3as a reference batch (this batch won't change)</span></span>
<span><span class="co">#&gt; Found5batches</span></span>
<span><span class="co">#&gt; Adjusting for2covariate(s) or covariate level(s)</span></span>
<span><span class="co">#&gt; Standardizing Data across genes</span></span>
<span><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span><span class="co">#&gt; Finding parametric adjustments</span></span>
<span><span class="co">#&gt; Adjusting the Data</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:ComBat-4"></span>
<img src="bioconductor/ComBat-4.png" alt="ComBat example" width="672"><p class="caption">
Figure 2.4: ComBat example
</p>
</div>
</div>
<div class="section level3" number="2.2">
<h3 id="quantro">
<span class="header-section-number">2.2</span> quantro<a class="anchor" aria-label="anchor" href="#quantro"></a>
</h3>
<p>This is also adapted from the package vignette but with <code>FlowSorted.DLPFC.450k</code> in place of <code>FlowSorted</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">FlowSorted.DLPFC.450k</span>,package<span class="op">=</span><span class="st">"FlowSorted.DLPFC.450k"</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">getBeta</span><span class="op">(</span><span class="va">FlowSorted.DLPFC.450k</span>,offset<span class="op">=</span><span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">pd</span> <span class="op">&lt;-</span> <span class="fu">Biobase</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">FlowSorted.DLPFC.450k</span><span class="op">)</span></span>
<span><span class="fu">quantro</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/quantro/man/matboxplot.html" class="external-link">matboxplot</a></span><span class="op">(</span><span class="va">p</span>, groupFactor <span class="op">=</span> <span class="va">pd</span><span class="op">$</span><span class="va">CellType</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"Beta Values"</span>, pch<span class="op">=</span><span class="fl">19</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:quantro-1"></span>
<img src="bioconductor/quantro-1.png" alt="quantro example" width="672"><p class="caption">
Figure 2.5: quantro example
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">quantro</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/quantro/man/matdensity.html" class="external-link">matdensity</a></span><span class="op">(</span><span class="va">p</span>, groupFactor <span class="op">=</span> <span class="va">pd</span><span class="op">$</span><span class="va">CellType</span>, xlab <span class="op">=</span> <span class="st">" "</span>, ylab <span class="op">=</span> <span class="st">"density"</span>,</span>
<span>                    main <span class="op">=</span> <span class="st">"Beta Values"</span>, brewer.n <span class="op">=</span> <span class="fl">8</span>, brewer.name <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">'top'</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"NeuN_neg"</span>, <span class="st">"NeuN_pos"</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, lty <span class="op">=</span> <span class="fl">1</span>, lwd <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:quantro-2"></span>
<img src="bioconductor/quantro-2.png" alt="quantro example" width="672"><p class="caption">
Figure 2.6: quantro example
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtest</span> <span class="op">&lt;-</span> <span class="fu">quantro</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/quantro/man/quantro.html" class="external-link">quantro</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">p</span>, groupFactor <span class="op">=</span> <span class="va">pd</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span></span>
<span><span class="co">#&gt; [quantro] Average medians of the distributions are </span></span>
<span><span class="co">#&gt;                         not equal across groups.</span></span>
<span><span class="co">#&gt; [quantro] Calculating the quantro test statistic.</span></span>
<span><span class="co">#&gt; [quantro] No permutation testing performed. </span></span>
<span><span class="co">#&gt;                          Use B &gt; 0 for permutation testing.</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="fu">doParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/doParallel/man/registerDoParallel.html" class="external-link">registerDoParallel</a></span><span class="op">(</span>cores<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span>  <span class="va">qtestPerm</span> <span class="op">&lt;-</span> <span class="fu">quantro</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/quantro/man/quantro.html" class="external-link">quantro</a></span><span class="op">(</span><span class="va">p</span>, groupFactor <span class="op">=</span> <span class="va">pd</span><span class="op">$</span><span class="va">CellType</span>, B <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span>  <span class="fu">quantro</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/quantro/man/quantroPlot.html" class="external-link">quantroPlot</a></span><span class="op">(</span><span class="va">qtestPerm</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="3">
<h2 id="outlier-detection-in-rna-seq">
<span class="header-section-number">3</span> Outlier detection in RNA-Seq<a class="anchor" aria-label="anchor" href="#outlier-detection-in-rna-seq"></a>
</h2>
<p>The following is adapted from package OUTRIDER,</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ctsFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'KremerNBaderSmall.tsv'</span>, package<span class="op">=</span><span class="st">'OUTRIDER'</span><span class="op">)</span></span>
<span><span class="va">ctsTable</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">ctsFile</span>, check.names<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">ods</span> <span class="op">&lt;-</span> <span class="fu">OUTRIDER</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/OUTRIDER/man/OutriderDataSet-class.html" class="external-link">OutriderDataSet</a></span><span class="op">(</span>countData<span class="op">=</span><span class="va">ctsTable</span><span class="op">)</span></span>
<span><span class="va">ods</span> <span class="op">&lt;-</span> <span class="fu">OUTRIDER</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/OUTRIDER/man/filterExpression.html" class="external-link">filterExpression</a></span><span class="op">(</span><span class="va">ods</span>, minCounts<span class="op">=</span><span class="cn">TRUE</span>, filterGenes<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; 229 genes did not pass the filter due to zero counts. This is 22.9% of the genes.</span></span>
<span><span class="va">ods</span> <span class="op">&lt;-</span> <span class="fu">OUTRIDER</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/OUTRIDER/man/OUTRIDER.html" class="external-link">OUTRIDER</a></span><span class="op">(</span><span class="va">ods</span><span class="op">)</span></span>
<span><span class="co">#&gt; Mon Jul 29 15:19:24 2024: SizeFactor estimation ...</span></span>
<span><span class="co">#&gt; Mon Jul 29 15:19:24 2024: Controlling for confounders ...</span></span>
<span><span class="co">#&gt; Using estimated q with: 23</span></span>
<span><span class="co">#&gt; Mon Jul 29 15:19:24 2024: Using the autoencoder implementation for controlling.</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:30 2024: Initial PCA loss: 4.73997327486604"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:37 2024: Iteration: 1 loss: 4.19416269506454"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:40 2024: Iteration: 2 loss: 4.17550752619036"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:43 2024: Iteration: 3 loss: 4.16639365666912"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:45 2024: Iteration: 4 loss: 4.16142359470334"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:47 2024: Iteration: 5 loss: 4.15785341106832"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:49 2024: Iteration: 6 loss: 4.15533343090454"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:50 2024: Iteration: 7 loss: 4.15339892434562"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:52 2024: Iteration: 8 loss: 4.15175378925737"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:53 2024: Iteration: 9 loss: 4.15069201289976"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:54 2024: Iteration: 10 loss: 4.1501222741797"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:56 2024: Iteration: 11 loss: 4.14904801948777"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:58 2024: Iteration: 12 loss: 4.14805452270911"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:19:59 2024: Iteration: 13 loss: 4.14796461892655"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:20:00 2024: Iteration: 14 loss: 4.14722109314569"</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:20:01 2024: Iteration: 15 loss: 4.14696284053289"</span></span>
<span><span class="co">#&gt; Time difference of 28.68971 secs</span></span>
<span><span class="co">#&gt; [1] "Mon Jul 29 15:20:01 2024: 15 Final nb-AE loss: 4.14696284053289"</span></span>
<span><span class="co">#&gt; Mon Jul 29 15:20:01 2024: Used the autoencoder implementation for controlling.</span></span>
<span><span class="co">#&gt; Mon Jul 29 15:20:01 2024: P-value calculation ...</span></span>
<span><span class="co">#&gt; Mon Jul 29 15:20:08 2024: Zscore calculation ...</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">OUTRIDER</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/OUTRIDER/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">ods</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">res</span>,caption<span class="op">=</span><span class="st">"A check list of outliers"</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>
<span id="tab:ods">Table 3.1: </span>A check list of outliers</caption>
<colgroup>
<col width="8%">
<col width="5%">
<col width="5%">
<col width="6%">
<col width="4%">
<col width="3%">
<col width="6%">
<col width="8%">
<col width="6%">
<col width="8%">
<col width="4%">
<col width="5%">
<col width="10%">
<col width="9%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left">geneID</th>
<th align="left">sampleID</th>
<th align="right">pValue</th>
<th align="right">padjust</th>
<th align="right">zScore</th>
<th align="right">l2fc</th>
<th align="right">rawcounts</th>
<th align="right">meanRawcounts</th>
<th align="right">normcounts</th>
<th align="right">meanCorrected</th>
<th align="right">theta</th>
<th align="left">aberrant</th>
<th align="right">AberrantBySample</th>
<th align="right">AberrantByGene</th>
<th align="right">padj_rank</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ATAD3C</td>
<td align="left">MUC1360</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000001</td>
<td align="right">5.28</td>
<td align="right">1.88</td>
<td align="right">948</td>
<td align="right">82.29</td>
<td align="right">248.14</td>
<td align="right">67.33</td>
<td align="right">16.48</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">NBPF15</td>
<td align="left">MUC1351</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000035</td>
<td align="right">5.80</td>
<td align="right">0.77</td>
<td align="right">7591</td>
<td align="right">4224.88</td>
<td align="right">7028.35</td>
<td align="right">4121.14</td>
<td align="right">112.78</td>
<td align="left">TRUE</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">MSTO1</td>
<td align="left">MUC1367</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000177</td>
<td align="right">-6.26</td>
<td align="right">-0.81</td>
<td align="right">761</td>
<td align="right">1327.87</td>
<td align="right">727.74</td>
<td align="right">1276.02</td>
<td align="right">153.26</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">HDAC1</td>
<td align="left">MUC1350</td>
<td align="right">0.0e+00</td>
<td align="right">0.0001133</td>
<td align="right">-5.88</td>
<td align="right">-0.78</td>
<td align="right">2215</td>
<td align="right">3805.56</td>
<td align="right">2128.19</td>
<td align="right">3649.04</td>
<td align="right">135.60</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">DCAF6</td>
<td align="left">MUC1374</td>
<td align="right">1.0e-07</td>
<td align="right">0.0004182</td>
<td align="right">-5.67</td>
<td align="right">-0.61</td>
<td align="right">2348</td>
<td align="right">4869.53</td>
<td align="right">3085.40</td>
<td align="right">4724.25</td>
<td align="right">196.31</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">NBPF16</td>
<td align="left">MUC1351</td>
<td align="right">2.0e-07</td>
<td align="right">0.0006791</td>
<td align="right">4.83</td>
<td align="right">0.67</td>
<td align="right">4014</td>
<td align="right">2459.90</td>
<td align="right">3822.18</td>
<td align="right">2402.36</td>
<td align="right">107.85</td>
<td align="left">TRUE</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">FAM102B</td>
<td align="left">MUC1363</td>
<td align="right">1.2e-06</td>
<td align="right">0.0067682</td>
<td align="right">-5.37</td>
<td align="right">-1.29</td>
<td align="right">455</td>
<td align="right">1138.75</td>
<td align="right">440.62</td>
<td align="right">1076.81</td>
<td align="right">41.98</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">LOC100288142</td>
<td align="left">MUC1361</td>
<td align="right">3.0e-06</td>
<td align="right">0.0167927</td>
<td align="right">4.25</td>
<td align="right">0.85</td>
<td align="right">637</td>
<td align="right">356.12</td>
<td align="right">622.40</td>
<td align="right">345.76</td>
<td align="right">57.44</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">TARDBP</td>
<td align="left">MUC0486</td>
<td align="right">7.3e-06</td>
<td align="right">0.0407384</td>
<td align="right">-4.56</td>
<td align="right">-0.32</td>
<td align="right">5911</td>
<td align="right">5780.34</td>
<td align="right">4449.25</td>
<td align="right">5565.44</td>
<td align="right">464.81</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">ZMPSTE24</td>
<td align="left">MUC1370</td>
<td align="right">7.7e-06</td>
<td align="right">0.0426990</td>
<td align="right">4.28</td>
<td align="right">0.38</td>
<td align="right">6180</td>
<td align="right">4026.77</td>
<td align="right">5088.19</td>
<td align="right">3905.38</td>
<td align="right">273.05</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">OUTRIDER</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/OUTRIDER/man/plotFunctions.html" class="external-link">plotQQ</a></span><span class="op">(</span><span class="va">ods</span>, <span class="va">res</span><span class="op">[</span><span class="st">"geneID"</span><span class="op">]</span>,global<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:ods"></span>
<img src="bioconductor/ods-1.png" alt="Q-Q plot for Kremer data" width="672"><p class="caption">
Figure 3.1: Q-Q plot for Kremer data
</p>
</div>
</div>
<div class="section level2" number="4">
<h2 id="differential-expression">
<span class="header-section-number">4</span> Differential expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/makeExampleDESeqDataSet.html" class="external-link">makeExampleDESeqDataSet</a></span><span class="op">(</span>m<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">ex</span><span class="op">)</span></span>
<span><span class="co">#&gt; estimating size factors</span></span>
<span><span class="co">#&gt; estimating dispersions</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; fitting model and testing</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span>, contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>,<span class="st">"B"</span>,<span class="st">"A"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rld</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/rlog.html" class="external-link">rlogTransformation</a></span><span class="op">(</span><span class="va">ex</span>, blind<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/plotPCA.html" class="external-link">plotPCA</a></span><span class="op">(</span><span class="va">rld</span>, intgroup<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span><span class="op">)</span>,returnData<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; using ntop=500 top features by variance</span></span>
<span><span class="va">percentVar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">100</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">dat</span>,<span class="st">"percentVar"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">dat</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">PC1</span>, <span class="va">PC2</span>, color<span class="op">=</span><span class="va">condition</span>, shape<span class="op">=</span><span class="va">condition</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC1:"</span>,<span class="va">percentVar</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,<span class="st">"% variance"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"PC2:"</span>,<span class="va">percentVar</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,<span class="st">"% variance"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:DESeq2"></span>
<img src="bioconductor/DESeq2-1.png" alt="DESeq2 example" width="672"><p class="caption">
Figure 4.1: DESeq2 example
</p>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex</span><span class="op">$</span><span class="va">condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">ex</span><span class="op">$</span><span class="va">condition</span>, ref<span class="op">=</span><span class="st">"B"</span><span class="op">)</span></span>
<span><span class="va">dds2</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="co">#&gt; using pre-existing size factors</span></span>
<span><span class="co">#&gt; estimating dispersions</span></span>
<span><span class="co">#&gt; found already estimated dispersions, replacing these</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt; fitting model and testing</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds2</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="8%">
<col width="13%">
<col width="20%">
<col width="13%">
<col width="15%">
<col width="13%">
<col width="13%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">stat</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">gene1</td>
<td align="right">277.46471</td>
<td align="right">0.0164732</td>
<td align="right">0.4477271</td>
<td align="right">0.0367929</td>
<td align="right">0.9706501</td>
<td align="right">0.9998477</td>
</tr>
<tr class="even">
<td align="left">gene2</td>
<td align="right">10.51017</td>
<td align="right">0.9062864</td>
<td align="right">1.2748781</td>
<td align="right">0.7108808</td>
<td align="right">0.4771581</td>
<td align="right">0.9998477</td>
</tr>
<tr class="odd">
<td align="left">gene3</td>
<td align="right">37.01996</td>
<td align="right">0.4181164</td>
<td align="right">0.9271126</td>
<td align="right">0.4509877</td>
<td align="right">0.6519984</td>
<td align="right">0.9998477</td>
</tr>
<tr class="even">
<td align="left">gene4</td>
<td align="right">78.76033</td>
<td align="right">-0.9719377</td>
<td align="right">0.6074553</td>
<td align="right">-1.6000153</td>
<td align="right">0.1095952</td>
<td align="right">0.9998477</td>
</tr>
<tr class="odd">
<td align="left">gene5</td>
<td align="right">14.43047</td>
<td align="right">-0.5318057</td>
<td align="right">1.4614592</td>
<td align="right">-0.3638868</td>
<td align="right">0.7159426</td>
<td align="right">0.9998477</td>
</tr>
<tr class="even">
<td align="left">gene6</td>
<td align="right">41.86626</td>
<td align="right">1.2435933</td>
<td align="right">0.7644820</td>
<td align="right">1.6267136</td>
<td align="right">0.1037979</td>
<td align="right">0.9998477</td>
</tr>
</tbody>
</table>
<p>See the package in action from a snakemake workflow<span class="citation"><sup><a href="#ref-koster21" role="doc-biblioref">1</a></sup></span>.</p>
</div>
<div class="section level2" number="5">
<h2 id="gene-co-expression-and-network-analysis">
<span class="header-section-number">5</span> Gene co-expression and network analysis<a class="anchor" aria-label="anchor" href="#gene-co-expression-and-network-analysis"></a>
</h2>
<p>A simple network is furnished with the <code>GeneNet</code> documentation example,</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## A random network with 40 nodes </span></span>
<span><span class="co"># it contains 780=40*39/2 edges of which 5 percent (=39) are non-zero</span></span>
<span><span class="va">true.pcor</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/ggm.simulate.pcor.html" class="external-link">ggm.simulate.pcor</a></span><span class="op">(</span><span class="fl">40</span><span class="op">)</span></span>
<span>  </span>
<span><span class="co"># A data set with 40 observations</span></span>
<span><span class="va">m.sim</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/ggm.simulate.data.html" class="external-link">ggm.simulate.data</a></span><span class="op">(</span><span class="fl">40</span>, <span class="va">true.pcor</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># A simple estimate of partial correlations</span></span>
<span><span class="va">estimated.pcor</span> <span class="op">&lt;-</span> <span class="fu">corpcor</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/corpcor/man/cor2pcor.html" class="external-link">cor2pcor</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">m.sim</span><span class="op">)</span> <span class="op">)</span></span>
<span></span>
<span><span class="co"># A comparison of estimated and true values</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">true.pcor</span><span class="op">-</span><span class="va">estimated.pcor</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 593.2851</span></span>
<span></span>
<span><span class="co"># A slightly better estimate ...</span></span>
<span><span class="va">estimated.pcor.2</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/ggm.estimate.pcor.html" class="external-link">ggm.estimate.pcor</a></span><span class="op">(</span><span class="va">m.sim</span><span class="op">)</span></span>
<span><span class="co">#&gt; Estimating optimal shrinkage intensity lambda (correlation matrix): 0.3689</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="va">true.pcor</span><span class="op">-</span><span class="va">estimated.pcor.2</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 10.85778</span></span>
<span></span>
<span><span class="co">## ecoli data </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ecoli</span>, package<span class="op">=</span><span class="st">"GeneNet"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># partial correlation matrix </span></span>
<span><span class="va">inferred.pcor</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/ggm.estimate.pcor.html" class="external-link">ggm.estimate.pcor</a></span><span class="op">(</span><span class="va">ecoli</span><span class="op">)</span></span>
<span><span class="co">#&gt; Estimating optimal shrinkage intensity lambda (correlation matrix): 0.1804</span></span>
<span></span>
<span><span class="co"># p-values, q-values and posterior probabilities for each potential edge </span></span>
<span><span class="va">test.results</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/network.test.edges.html" class="external-link">network.test.edges</a></span><span class="op">(</span><span class="va">inferred.pcor</span><span class="op">)</span></span>
<span><span class="co">#&gt; Estimate (local) false discovery rates (partial correlations):</span></span>
<span><span class="co">#&gt; Step 1... determine cutoff point</span></span>
<span><span class="co">#&gt; Step 2... estimate parameters of null distribution and eta0</span></span>
<span><span class="co">#&gt; Step 3... compute p-values and estimate empirical PDF/CDF</span></span>
<span><span class="co">#&gt; Step 4... compute q-values and local fdr</span></span>
<span><span class="co">#&gt; Step 5... prepare for plotting</span></span>
<span></span>
<span><span class="co"># best 20 edges (strongest correlation)</span></span>
<span><span class="va">test.results</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span></span>
<span><span class="co"># network containing edges with prob &gt; 0.9 (i.e. local fdr &lt; 0.1)</span></span>
<span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/network.test.edges.html" class="external-link">extract.network</a></span><span class="op">(</span><span class="va">test.results</span>, cutoff.ggm<span class="op">=</span><span class="fl">0.9</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Significant edges:  65 </span></span>
<span><span class="co">#&gt;     Corresponding to  1.26 %  of possible edges</span></span>
<span><span class="va">net</span></span>
<span><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span><span class="co">#&gt; 21  0.09576091    89    95 1.443542e-05 3.354777e-03 0.9891317</span></span>
<span><span class="co">#&gt; 22  0.09473210     7    51 1.784126e-05 3.864825e-03 0.9891317</span></span>
<span><span class="co">#&gt; 23 -0.09386896    53    58 2.127622e-05 4.313590e-03 0.9891317</span></span>
<span><span class="co">#&gt; 24 -0.09366615    29    83 2.217013e-05 4.421099e-03 0.9891317</span></span>
<span><span class="co">#&gt; 25 -0.09341148    21    89 2.334321e-05 4.556947e-03 0.9810727</span></span>
<span><span class="co">#&gt; 26 -0.09156391    49    93 3.380043e-05 5.955972e-03 0.9810727</span></span>
<span><span class="co">#&gt; 27 -0.09150710    80    90 3.418363e-05 6.002083e-03 0.9810727</span></span>
<span><span class="co">#&gt; 28  0.09101505     7    53 3.767966e-05 6.408102e-03 0.9810727</span></span>
<span><span class="co">#&gt; 29  0.09050688    21    84 4.164471e-05 6.838782e-03 0.9810727</span></span>
<span><span class="co">#&gt; 30  0.08965490    72    73 4.919365e-05 7.581866e-03 0.9810727</span></span>
<span><span class="co">#&gt; 31 -0.08934025    29    99 5.229604e-05 7.861416e-03 0.9810727</span></span>
<span><span class="co">#&gt; 32 -0.08906819     9    95 5.512708e-05 8.104759e-03 0.9810727</span></span>
<span><span class="co">#&gt; 33  0.08888345     2    49 5.713144e-05 8.270673e-03 0.9810727</span></span>
<span><span class="co">#&gt; 34  0.08850681    86    90 6.143363e-05 8.610161e-03 0.9810727</span></span>
<span><span class="co">#&gt; 35  0.08805868    17    53 6.695170e-05 9.015175e-03 0.9810727</span></span>
<span><span class="co">#&gt; 36  0.08790809    28    48 6.890884e-05 9.151291e-03 0.9810727</span></span>
<span><span class="co">#&gt; 37  0.08783471    33    58 6.988211e-05 9.217597e-03 0.9682377</span></span>
<span><span class="co">#&gt; 38 -0.08705796     7    49 8.101244e-05 1.021362e-02 0.9682377</span></span>
<span><span class="co">#&gt; 39  0.08645033    20    46 9.086547e-05 1.102466e-02 0.9682377</span></span>
<span><span class="co">#&gt; 40  0.08609950    48    86 9.705862e-05 1.150392e-02 0.9682377</span></span>
<span><span class="co">#&gt; 41  0.08598769    21    52 9.911458e-05 1.165816e-02 0.9682377</span></span>
<span><span class="co">#&gt; 42  0.08555275    32    95 1.075099e-04 1.226435e-02 0.9682377</span></span>
<span><span class="co">#&gt; 43  0.08548231    17    51 1.089311e-04 1.236337e-02 0.9424721</span></span>
<span><span class="co">#&gt; 44  0.08470370    80    83 1.258659e-04 1.382356e-02 0.9424721</span></span>
<span><span class="co">#&gt; 45  0.08442510    80    82 1.325062e-04 1.437068e-02 0.9174573</span></span>
<span><span class="co">#&gt; 46  0.08271606    80    93 1.810275e-04 1.845632e-02 0.9174573</span></span>
<span><span class="co">#&gt; 47  0.08235175    46    91 1.933329e-04 1.941579e-02 0.9174573</span></span>
<span><span class="co">#&gt; 48  0.08217787    25    95 1.994788e-04 1.988432e-02 0.9174573</span></span>
<span><span class="co">#&gt; 49 -0.08170331    29    87 2.171999e-04 2.119715e-02 0.9174573</span></span>
<span><span class="co">#&gt; 50  0.08123632    19    29 2.360716e-04 2.253606e-02 0.9174573</span></span>
<span><span class="co">#&gt; 51  0.08101702    51    84 2.454547e-04 2.318024e-02 0.9174573</span></span>
<span><span class="co">#&gt; 52  0.08030748    16    93 2.782643e-04 2.532796e-02 0.9174573</span></span>
<span><span class="co">#&gt; 53  0.08006503    28    52 2.903870e-04 2.608271e-02 0.9174573</span></span>
<span><span class="co">#&gt; 54 -0.07941656    41    80 3.252833e-04 2.814824e-02 0.9174573</span></span>
<span><span class="co">#&gt; 55  0.07941410    54    89 3.254229e-04 2.815620e-02 0.9174573</span></span>
<span><span class="co">#&gt; 56 -0.07934653    28    80 3.292784e-04 2.837511e-02 0.9174573</span></span>
<span><span class="co">#&gt; 57  0.07916783    29    92 3.396802e-04 2.895702e-02 0.9174573</span></span>
<span><span class="co">#&gt; 58 -0.07866905    17    86 3.703635e-04 3.060293e-02 0.9174573</span></span>
<span><span class="co">#&gt; 59  0.07827749    16    29 3.962446e-04 3.191462e-02 0.9174573</span></span>
<span><span class="co">#&gt; 60 -0.07808262    73    89 4.097452e-04 3.257290e-02 0.9174573</span></span>
<span><span class="co">#&gt; 61  0.07766261    52    67 4.403165e-04 3.400207e-02 0.9174573</span></span>
<span><span class="co">#&gt; 62  0.07762917    25    87 4.428396e-04 3.411637e-02 0.9174573</span></span>
<span><span class="co">#&gt; 63 -0.07739378     9    93 4.609872e-04 3.492295e-02 0.9174573</span></span>
<span><span class="co">#&gt; 64  0.07738885    31    80 4.613747e-04 3.493988e-02 0.9174573</span></span>
<span><span class="co">#&gt; 65 -0.07718681    80    94 4.775136e-04 3.563444e-02 0.9174573</span></span>
<span></span>
<span><span class="co"># significant based on FDR cutoff Q=0.05?</span></span>
<span><span class="va">num.significant.1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">test.results</span><span class="op">$</span><span class="va">qval</span> <span class="op">&lt;=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">test.results</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">num.significant.1</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span><span class="co">#&gt; 21  0.09576091    89    95 1.443542e-05 3.354777e-03 0.9891317</span></span>
<span><span class="co">#&gt; 22  0.09473210     7    51 1.784126e-05 3.864825e-03 0.9891317</span></span>
<span><span class="co">#&gt; 23 -0.09386896    53    58 2.127622e-05 4.313590e-03 0.9891317</span></span>
<span><span class="co">#&gt; 24 -0.09366615    29    83 2.217013e-05 4.421099e-03 0.9891317</span></span>
<span><span class="co">#&gt; 25 -0.09341148    21    89 2.334321e-05 4.556947e-03 0.9810727</span></span>
<span><span class="co">#&gt; 26 -0.09156391    49    93 3.380043e-05 5.955972e-03 0.9810727</span></span>
<span><span class="co">#&gt; 27 -0.09150710    80    90 3.418363e-05 6.002083e-03 0.9810727</span></span>
<span><span class="co">#&gt; 28  0.09101505     7    53 3.767966e-05 6.408102e-03 0.9810727</span></span>
<span><span class="co">#&gt; 29  0.09050688    21    84 4.164471e-05 6.838782e-03 0.9810727</span></span>
<span><span class="co">#&gt; 30  0.08965490    72    73 4.919365e-05 7.581866e-03 0.9810727</span></span>
<span><span class="co">#&gt; 31 -0.08934025    29    99 5.229604e-05 7.861416e-03 0.9810727</span></span>
<span><span class="co">#&gt; 32 -0.08906819     9    95 5.512708e-05 8.104759e-03 0.9810727</span></span>
<span><span class="co">#&gt; 33  0.08888345     2    49 5.713144e-05 8.270673e-03 0.9810727</span></span>
<span><span class="co">#&gt; 34  0.08850681    86    90 6.143363e-05 8.610161e-03 0.9810727</span></span>
<span><span class="co">#&gt; 35  0.08805868    17    53 6.695170e-05 9.015175e-03 0.9810727</span></span>
<span><span class="co">#&gt; 36  0.08790809    28    48 6.890884e-05 9.151291e-03 0.9810727</span></span>
<span><span class="co">#&gt; 37  0.08783471    33    58 6.988211e-05 9.217597e-03 0.9682377</span></span>
<span><span class="co">#&gt; 38 -0.08705796     7    49 8.101244e-05 1.021362e-02 0.9682377</span></span>
<span><span class="co">#&gt; 39  0.08645033    20    46 9.086547e-05 1.102466e-02 0.9682377</span></span>
<span><span class="co">#&gt; 40  0.08609950    48    86 9.705862e-05 1.150392e-02 0.9682377</span></span>
<span><span class="co">#&gt; 41  0.08598769    21    52 9.911458e-05 1.165816e-02 0.9682377</span></span>
<span><span class="co">#&gt; 42  0.08555275    32    95 1.075099e-04 1.226435e-02 0.9682377</span></span>
<span><span class="co">#&gt; 43  0.08548231    17    51 1.089311e-04 1.236337e-02 0.9424721</span></span>
<span><span class="co">#&gt; 44  0.08470370    80    83 1.258659e-04 1.382356e-02 0.9424721</span></span>
<span><span class="co">#&gt; 45  0.08442510    80    82 1.325062e-04 1.437068e-02 0.9174573</span></span>
<span><span class="co">#&gt; 46  0.08271606    80    93 1.810275e-04 1.845632e-02 0.9174573</span></span>
<span><span class="co">#&gt; 47  0.08235175    46    91 1.933329e-04 1.941579e-02 0.9174573</span></span>
<span><span class="co">#&gt; 48  0.08217787    25    95 1.994788e-04 1.988432e-02 0.9174573</span></span>
<span><span class="co">#&gt; 49 -0.08170331    29    87 2.171999e-04 2.119715e-02 0.9174573</span></span>
<span><span class="co">#&gt; 50  0.08123632    19    29 2.360716e-04 2.253606e-02 0.9174573</span></span>
<span><span class="co">#&gt; 51  0.08101702    51    84 2.454547e-04 2.318024e-02 0.9174573</span></span>
<span><span class="co">#&gt; 52  0.08030748    16    93 2.782643e-04 2.532796e-02 0.9174573</span></span>
<span><span class="co">#&gt; 53  0.08006503    28    52 2.903870e-04 2.608271e-02 0.9174573</span></span>
<span><span class="co">#&gt; 54 -0.07941656    41    80 3.252833e-04 2.814824e-02 0.9174573</span></span>
<span><span class="co">#&gt; 55  0.07941410    54    89 3.254229e-04 2.815620e-02 0.9174573</span></span>
<span><span class="co">#&gt; 56 -0.07934653    28    80 3.292784e-04 2.837511e-02 0.9174573</span></span>
<span><span class="co">#&gt; 57  0.07916783    29    92 3.396802e-04 2.895702e-02 0.9174573</span></span>
<span><span class="co">#&gt; 58 -0.07866905    17    86 3.703635e-04 3.060293e-02 0.9174573</span></span>
<span><span class="co">#&gt; 59  0.07827749    16    29 3.962446e-04 3.191462e-02 0.9174573</span></span>
<span><span class="co">#&gt; 60 -0.07808262    73    89 4.097452e-04 3.257290e-02 0.9174573</span></span>
<span><span class="co">#&gt; 61  0.07766261    52    67 4.403165e-04 3.400207e-02 0.9174573</span></span>
<span><span class="co">#&gt; 62  0.07762917    25    87 4.428396e-04 3.411637e-02 0.9174573</span></span>
<span><span class="co">#&gt; 63 -0.07739378     9    93 4.609872e-04 3.492295e-02 0.9174573</span></span>
<span><span class="co">#&gt; 64  0.07738885    31    80 4.613747e-04 3.493988e-02 0.9174573</span></span>
<span><span class="co">#&gt; 65 -0.07718681    80    94 4.775136e-04 3.563444e-02 0.9174573</span></span>
<span><span class="co">#&gt; 66  0.07706275    27    58 4.876831e-04 3.606179e-02 0.8297811</span></span>
<span><span class="co">#&gt; 67 -0.07610709    16    83 5.730532e-04 4.085920e-02 0.8297811</span></span>
<span><span class="co">#&gt; 68  0.07550557    53    84 6.337143e-04 4.406472e-02 0.8297811</span></span>
<span></span>
<span><span class="co"># significant based on "local fdr" cutoff (prob &gt; 0.9)?</span></span>
<span><span class="va">num.significant.2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">test.results</span><span class="op">$</span><span class="va">prob</span> <span class="op">&gt;</span> <span class="fl">0.9</span><span class="op">)</span></span>
<span><span class="va">test.results</span><span class="op">[</span><span class="va">test.results</span><span class="op">$</span><span class="va">prob</span> <span class="op">&gt;</span> <span class="fl">0.9</span>,<span class="op">]</span></span>
<span><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span><span class="co">#&gt; 21  0.09576091    89    95 1.443542e-05 3.354777e-03 0.9891317</span></span>
<span><span class="co">#&gt; 22  0.09473210     7    51 1.784126e-05 3.864825e-03 0.9891317</span></span>
<span><span class="co">#&gt; 23 -0.09386896    53    58 2.127622e-05 4.313590e-03 0.9891317</span></span>
<span><span class="co">#&gt; 24 -0.09366615    29    83 2.217013e-05 4.421099e-03 0.9891317</span></span>
<span><span class="co">#&gt; 25 -0.09341148    21    89 2.334321e-05 4.556947e-03 0.9810727</span></span>
<span><span class="co">#&gt; 26 -0.09156391    49    93 3.380043e-05 5.955972e-03 0.9810727</span></span>
<span><span class="co">#&gt; 27 -0.09150710    80    90 3.418363e-05 6.002083e-03 0.9810727</span></span>
<span><span class="co">#&gt; 28  0.09101505     7    53 3.767966e-05 6.408102e-03 0.9810727</span></span>
<span><span class="co">#&gt; 29  0.09050688    21    84 4.164471e-05 6.838782e-03 0.9810727</span></span>
<span><span class="co">#&gt; 30  0.08965490    72    73 4.919365e-05 7.581866e-03 0.9810727</span></span>
<span><span class="co">#&gt; 31 -0.08934025    29    99 5.229604e-05 7.861416e-03 0.9810727</span></span>
<span><span class="co">#&gt; 32 -0.08906819     9    95 5.512708e-05 8.104759e-03 0.9810727</span></span>
<span><span class="co">#&gt; 33  0.08888345     2    49 5.713144e-05 8.270673e-03 0.9810727</span></span>
<span><span class="co">#&gt; 34  0.08850681    86    90 6.143363e-05 8.610161e-03 0.9810727</span></span>
<span><span class="co">#&gt; 35  0.08805868    17    53 6.695170e-05 9.015175e-03 0.9810727</span></span>
<span><span class="co">#&gt; 36  0.08790809    28    48 6.890884e-05 9.151291e-03 0.9810727</span></span>
<span><span class="co">#&gt; 37  0.08783471    33    58 6.988211e-05 9.217597e-03 0.9682377</span></span>
<span><span class="co">#&gt; 38 -0.08705796     7    49 8.101244e-05 1.021362e-02 0.9682377</span></span>
<span><span class="co">#&gt; 39  0.08645033    20    46 9.086547e-05 1.102466e-02 0.9682377</span></span>
<span><span class="co">#&gt; 40  0.08609950    48    86 9.705862e-05 1.150392e-02 0.9682377</span></span>
<span><span class="co">#&gt; 41  0.08598769    21    52 9.911458e-05 1.165816e-02 0.9682377</span></span>
<span><span class="co">#&gt; 42  0.08555275    32    95 1.075099e-04 1.226435e-02 0.9682377</span></span>
<span><span class="co">#&gt; 43  0.08548231    17    51 1.089311e-04 1.236337e-02 0.9424721</span></span>
<span><span class="co">#&gt; 44  0.08470370    80    83 1.258659e-04 1.382356e-02 0.9424721</span></span>
<span><span class="co">#&gt; 45  0.08442510    80    82 1.325062e-04 1.437068e-02 0.9174573</span></span>
<span><span class="co">#&gt; 46  0.08271606    80    93 1.810275e-04 1.845632e-02 0.9174573</span></span>
<span><span class="co">#&gt; 47  0.08235175    46    91 1.933329e-04 1.941579e-02 0.9174573</span></span>
<span><span class="co">#&gt; 48  0.08217787    25    95 1.994788e-04 1.988432e-02 0.9174573</span></span>
<span><span class="co">#&gt; 49 -0.08170331    29    87 2.171999e-04 2.119715e-02 0.9174573</span></span>
<span><span class="co">#&gt; 50  0.08123632    19    29 2.360716e-04 2.253606e-02 0.9174573</span></span>
<span><span class="co">#&gt; 51  0.08101702    51    84 2.454547e-04 2.318024e-02 0.9174573</span></span>
<span><span class="co">#&gt; 52  0.08030748    16    93 2.782643e-04 2.532796e-02 0.9174573</span></span>
<span><span class="co">#&gt; 53  0.08006503    28    52 2.903870e-04 2.608271e-02 0.9174573</span></span>
<span><span class="co">#&gt; 54 -0.07941656    41    80 3.252833e-04 2.814824e-02 0.9174573</span></span>
<span><span class="co">#&gt; 55  0.07941410    54    89 3.254229e-04 2.815620e-02 0.9174573</span></span>
<span><span class="co">#&gt; 56 -0.07934653    28    80 3.292784e-04 2.837511e-02 0.9174573</span></span>
<span><span class="co">#&gt; 57  0.07916783    29    92 3.396802e-04 2.895702e-02 0.9174573</span></span>
<span><span class="co">#&gt; 58 -0.07866905    17    86 3.703635e-04 3.060293e-02 0.9174573</span></span>
<span><span class="co">#&gt; 59  0.07827749    16    29 3.962446e-04 3.191462e-02 0.9174573</span></span>
<span><span class="co">#&gt; 60 -0.07808262    73    89 4.097452e-04 3.257290e-02 0.9174573</span></span>
<span><span class="co">#&gt; 61  0.07766261    52    67 4.403165e-04 3.400207e-02 0.9174573</span></span>
<span><span class="co">#&gt; 62  0.07762917    25    87 4.428396e-04 3.411637e-02 0.9174573</span></span>
<span><span class="co">#&gt; 63 -0.07739378     9    93 4.609872e-04 3.492295e-02 0.9174573</span></span>
<span><span class="co">#&gt; 64  0.07738885    31    80 4.613747e-04 3.493988e-02 0.9174573</span></span>
<span><span class="co">#&gt; 65 -0.07718681    80    94 4.775136e-04 3.563444e-02 0.9174573</span></span>
<span></span>
<span><span class="co"># parameters of the mixture distribution used to compute p-values etc.</span></span>
<span><span class="va">c</span> <span class="op">&lt;-</span> <span class="fu">fdrtool</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/fdrtool/man/fdrtool.html" class="external-link">fdrtool</a></span><span class="op">(</span><span class="fu">corpcor</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/corpcor/man/smtools.html" class="external-link">sm2vec</a></span><span class="op">(</span><span class="va">inferred.pcor</span><span class="op">)</span>, statistic<span class="op">=</span><span class="st">"correlation"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Step 1... determine cutoff point</span></span>
<span><span class="co">#&gt; Step 2... estimate parameters of null distribution and eta0</span></span>
<span><span class="co">#&gt; Step 3... compute p-values and estimate empirical PDF/CDF</span></span>
<span><span class="co">#&gt; Step 4... compute q-values and local fdr</span></span>
<span><span class="co">#&gt; Step 5... prepare for plotting</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:GeneNet-1"></span>
<img src="bioconductor/GeneNet-1.png" alt="GeneNet example" width="672"><p class="caption">
Figure 5.1: GeneNet example
</p>
</div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">c</span><span class="op">$</span><span class="va">param</span></span>
<span><span class="co">#&gt;          cutoff N.cens      eta0     eta0.SE    kappa kappa.SE</span></span>
<span><span class="co">#&gt; [1,] 0.03553068   4352 0.9474623 0.005656465 2043.377 94.72267</span></span>
<span></span>
<span><span class="co">## A random network with 20 nodes and 10 percent (=19) edges</span></span>
<span><span class="va">true.pcor</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/ggm.simulate.pcor.html" class="external-link">ggm.simulate.pcor</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">0.1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># convert to edge list</span></span>
<span><span class="va">test.results</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/GeneNet-internal.html" class="external-link">ggm.list.edges</a></span><span class="op">(</span><span class="va">true.pcor</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">19</span>,<span class="op">]</span></span>
<span><span class="va">nlab</span> <span class="op">&lt;-</span> <span class="va">LETTERS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># graphviz</span></span>
<span><span class="co"># network.make.dot(filename="test.dot", test.results, nlab, main = "A graph")</span></span>
<span><span class="co"># system("fdp -T svg -o test.svg test.dot")</span></span>
<span></span>
<span><span class="co"># Rgraphviz</span></span>
<span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/network.make.graph.html" class="external-link">network.make.graph</a></span><span class="op">(</span> <span class="va">test.results</span>, <span class="va">nlab</span><span class="op">)</span></span>
<span><span class="va">gr</span></span>
<span><span class="co">#&gt; A graphNEL graph with directed edges</span></span>
<span><span class="co">#&gt; Number of Nodes = 20 </span></span>
<span><span class="co">#&gt; Number of Edges = 38</span></span>
<span><span class="fu">num.nodes</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 20</span></span>
<span><span class="fu">edge.info</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span></span>
<span><span class="co">#&gt; $weight</span></span>
<span><span class="co">#&gt;      A~T      A~Q      A~J      A~L      B~N      B~J      C~R      D~N </span></span>
<span><span class="co">#&gt; -0.06270  0.19879  0.40924  0.62571  0.19421  0.52032 -0.69851  0.38015 </span></span>
<span><span class="co">#&gt;      D~H      F~Q      F~M      G~K      H~I      J~K      K~M      M~T </span></span>
<span><span class="co">#&gt;  0.61005 -0.23242  0.52489  0.48304  0.67365  0.25697  0.44416  0.18103 </span></span>
<span><span class="co">#&gt;      M~S      N~Q      R~T </span></span>
<span><span class="co">#&gt;  0.31499 -0.43478 -0.59535 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dir</span></span>
<span><span class="co">#&gt;    A~T    A~Q    A~J    A~L    B~N    B~J    C~R    D~N    D~H    F~Q    F~M </span></span>
<span><span class="co">#&gt; "none" "none" "none" "none" "none" "none" "none" "none" "none" "none" "none" </span></span>
<span><span class="co">#&gt;    G~K    H~I    J~K    K~M    M~T    M~S    N~Q    R~T </span></span>
<span><span class="co">#&gt; "none" "none" "none" "none" "none" "none" "none" "none"</span></span>
<span><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/network.make.graph.html" class="external-link">network.make.graph</a></span><span class="op">(</span> <span class="va">test.results</span>, <span class="va">nlab</span>, drop.singles<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">gr2</span></span>
<span><span class="co">#&gt; A graphNEL graph with directed edges</span></span>
<span><span class="co">#&gt; Number of Nodes = 17 </span></span>
<span><span class="co">#&gt; Number of Edges = 38</span></span>
<span><span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/network.make.graph.html" class="external-link">num.nodes</a></span><span class="op">(</span><span class="va">gr2</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 17</span></span>
<span><span class="fu">GeneNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GeneNet/man/network.make.graph.html" class="external-link">edge.info</a></span><span class="op">(</span><span class="va">gr2</span><span class="op">)</span></span>
<span><span class="co">#&gt; $weight</span></span>
<span><span class="co">#&gt;      A~T      A~Q      A~J      A~L      B~N      B~J      C~R      D~N </span></span>
<span><span class="co">#&gt; -0.06270  0.19879  0.40924  0.62571  0.19421  0.52032 -0.69851  0.38015 </span></span>
<span><span class="co">#&gt;      D~H      F~Q      F~M      G~K      H~I      J~K      K~M      M~T </span></span>
<span><span class="co">#&gt;  0.61005 -0.23242  0.52489  0.48304  0.67365  0.25697  0.44416  0.18103 </span></span>
<span><span class="co">#&gt;      M~S      N~Q      R~T </span></span>
<span><span class="co">#&gt;  0.31499 -0.43478 -0.59535 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $dir</span></span>
<span><span class="co">#&gt;    A~T    A~Q    A~J    A~L    B~N    B~J    C~R    D~N    D~H    F~Q    F~M </span></span>
<span><span class="co">#&gt; "none" "none" "none" "none" "none" "none" "none" "none" "none" "none" "none" </span></span>
<span><span class="co">#&gt;    G~K    H~I    J~K    K~M    M~T    M~S    N~Q    R~T </span></span>
<span><span class="co">#&gt; "none" "none" "none" "none" "none" "none" "none" "none"</span></span>
<span></span>
<span><span class="co"># plot network</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gr</span>, <span class="st">"fdp"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:GeneNet-2"></span>
<img src="bioconductor/GeneNet-2.png" alt="GeneNet example" width="672"><p class="caption">
Figure 5.2: GeneNet example
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gr2</span>, <span class="st">"fdp"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:GeneNet-3"></span>
<img src="bioconductor/GeneNet-3.png" alt="GeneNet example" width="672"><p class="caption">
Figure 5.3: GeneNet example
</p>
</div>
<p>A side-by-side heatmaps</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123454321</span><span class="op">)</span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">2500</span><span class="op">)</span>,<span class="fl">50</span><span class="op">)</span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">r</span><span class="op">&gt;=</span><span class="fl">0.7</span><span class="op">)</span><span class="op">+</span><span class="fl">0</span></span>
<span><span class="va">f1</span> <span class="op">&lt;-</span> <span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span><span class="va">f2</span> <span class="op">&lt;-</span> <span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="va">f1</span><span class="op">+</span><span class="va">f2</span></span>
<span><span class="fu">ComplexHeatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:heatmap"></span>
<img src="bioconductor/heatmap-1.png" alt="Heatmaps" width="672"><p class="caption">
Figure 5.4: Heatmaps
</p>
</div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">heatmaply</span><span class="fu">::</span><span class="fu"><a href="https://talgalili.github.io/heatmaply/reference/normalize.html" class="external-link">normalize</a></span><span class="op">(</span><span class="va">mtcars</span><span class="op">)</span></span>
<span><span class="va">hm</span> <span class="op">&lt;-</span> <span class="fu">heatmaply</span><span class="fu">::</span><span class="fu"><a href="https://talgalili.github.io/heatmaply/reference/heatmaply.html" class="external-link">heatmaply</a></span><span class="op">(</span><span class="va">df</span>,k_col<span class="op">=</span><span class="fl">5</span>,k_row<span class="op">=</span><span class="fl">5</span>,</span>
<span>                           colors <span class="op">=</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">3</span>, <span class="st">"RdBu"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">256</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">htmlwidgets</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/htmlwidgets/man/saveWidget.html" class="external-link">saveWidget</a></span><span class="op">(</span><span class="va">hm</span>,file<span class="op">=</span><span class="st">"heatmaply.html"</span><span class="op">)</span></span>
<span><span class="fu">htmltools</span><span class="fu">::</span><span class="va"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">tags</a></span><span class="op">$</span><span class="fu">iframe</span><span class="op">(</span>src <span class="op">=</span> <span class="st">"heatmaply.html"</span>, width <span class="op">=</span> <span class="st">"100%"</span>, height <span class="op">=</span> <span class="st">"550px"</span><span class="op">)</span></span></code></pre></div>
<iframe src="heatmaply.html" width="100%" height="550px"></iframe>
<p>so we have <a href="heatmaply.html">heatmaply.html</a> and a module analysis with WGCNA,</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pwr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>from<span class="op">=</span><span class="fl">12</span>, to<span class="op">=</span><span class="fl">30</span>, by<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sft</span> <span class="op">&lt;-</span> <span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/pickSoftThreshold.html" class="external-link">pickSoftThreshold</a></span><span class="op">(</span><span class="va">dat</span>, powerVector<span class="op">=</span><span class="va">pwr</span>, verbose<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">ADJ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">dat</span>, method<span class="op">=</span><span class="st">"pearson"</span>, use<span class="op">=</span><span class="st">"pairwise.complete.obs"</span><span class="op">)</span><span class="op">)</span><span class="op">^</span><span class="fl">6</span></span>
<span><span class="va">dissADJ</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">-</span><span class="va">ADJ</span></span>
<span><span class="va">dissTOM</span> <span class="op">&lt;-</span> <span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/TOMsimilarity.html" class="external-link">TOMdist</a></span><span class="op">(</span><span class="va">ADJ</span><span class="op">)</span></span>
<span><span class="va">TOM</span> <span class="op">&lt;-</span> <span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/TOMsimilarityFromExpr.html" class="external-link">TOMsimilarityFromExpr</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span>
<span><span class="va">Tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="va">TOM</span><span class="op">)</span>, method<span class="op">=</span><span class="st">"average"</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">pwr</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">pam_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pam"</span>,<span class="va">j</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/assign.html" class="external-link">assign</a></span><span class="op">(</span><span class="va">pam_name</span>, <span class="fu">cluster</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/cluster/man/pam.html" class="external-link">pam</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="va">dissADJ</span><span class="op">)</span>,<span class="va">j</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">pamTOM_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pamTOM"</span>,<span class="va">j</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/assign.html" class="external-link">assign</a></span><span class="op">(</span><span class="va">pamTOM_name</span>,<span class="fu">cluster</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/cluster/man/pam.html" class="external-link">pam</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="va">dissTOM</span><span class="op">)</span>,<span class="va">j</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">tc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="va">pam_name</span><span class="op">)</span><span class="op">$</span><span class="va">clustering</span>,<span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="va">pamTOM_name</span><span class="op">)</span><span class="op">$</span><span class="va">clustering</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">tc</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">tc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">colorStaticTOM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/cutreeStaticColor.html" class="external-link">cutreeStaticColor</a></span><span class="op">(</span><span class="va">Tree</span>,cutHeight<span class="op">=</span><span class="fl">.99</span>,minSize<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">colorDynamicTOM</span> <span class="op">&lt;-</span> <span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/labels2colors.html" class="external-link">labels2colors</a></span><span class="op">(</span><span class="fu">cutreeDynamic</span><span class="op">(</span><span class="va">Tree</span>,method<span class="op">=</span><span class="st">"tree"</span>,minClusterSize<span class="op">=</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">Colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">pamTOM6</span><span class="op">$</span><span class="va">clustering</span>,<span class="va">colorStaticTOM</span>,<span class="va">colorDynamicTOM</span><span class="op">)</span></span>
<span><span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/plotDendroAndColors.html" class="external-link">plotDendroAndColors</a></span><span class="op">(</span><span class="va">Tree</span>, <span class="va">Colors</span>, dendroLabels<span class="op">=</span><span class="cn">FALSE</span>, hang<span class="op">=</span><span class="fl">0.03</span>, addGuide<span class="op">=</span><span class="cn">TRUE</span>, guideHang<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">meg</span> <span class="op">&lt;-</span> <span class="fu">WGCNA</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/WGCNA/man/moduleEigengenes.html" class="external-link">moduleEigengenes</a></span><span class="op">(</span><span class="va">dat</span>, color<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span>, softPower<span class="op">=</span><span class="fl">6</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="6">
<h2 id="meta-data">
<span class="header-section-number">6</span> Meta-data<a class="anchor" aria-label="anchor" href="#meta-data"></a>
</h2>
<p>This section is based on package <code>recount3</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu">recount3</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount3/man/available_projects.html" class="external-link">available_projects</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">hs</span>,<span class="va">file_source</span><span class="op">==</span><span class="st">"gtex"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">recount3</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount3/man/annotation_options.html" class="external-link">annotation_options</a></span><span class="op">(</span><span class="st">"human"</span><span class="op">)</span></span>
<span><span class="va">blood_rse</span> <span class="op">&lt;-</span> <span class="fu">recount3</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount3/man/create_rse.html" class="external-link">create_rse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">hs</span>,<span class="va">project</span><span class="op">==</span><span class="st">"BLOOD"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">blood_rse</span><span class="op">)</span></span>
<span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">blood_rse</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">blood_rse</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></span>
<span><span class="fu">recount3</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount3/man/expand_sra_attributes.html" class="external-link">expand_sra_attributes</a></span><span class="op">(</span><span class="va">blood_rse</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="7">
<h2 id="pathway-and-enrichment-analysis">
<span class="header-section-number">7</span> Pathway and enrichment analysis<a class="anchor" aria-label="anchor" href="#pathway-and-enrichment-analysis"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reactome</span> <span class="op">&lt;-</span> <span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathways.html" class="external-link">pathways</a></span><span class="op">(</span><span class="st">"hsapiens"</span>, <span class="st">"reactome"</span><span class="op">)</span></span>
<span><span class="va">kegg</span> <span class="op">&lt;-</span> <span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathways.html" class="external-link">pathways</a></span><span class="op">(</span><span class="st">"hsapiens"</span>,<span class="st">"kegg"</span><span class="op">)</span></span>
<span><span class="va">pharmgkb</span> <span class="op">&lt;-</span> <span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathways.html" class="external-link">pathways</a></span><span class="op">(</span><span class="st">"hsapiens"</span>,<span class="st">"pharmgkb"</span><span class="op">)</span></span>
<span><span class="fu">nodes</span><span class="op">(</span><span class="va">kegg</span><span class="op">[[</span><span class="fl">21</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "ENTREZID:102724560" "ENTREZID:10993"     "ENTREZID:113675"   </span></span>
<span><span class="co">#&gt;  [4] "ENTREZID:132158"    "ENTREZID:1610"      "ENTREZID:1738"     </span></span>
<span><span class="co">#&gt;  [7] "ENTREZID:1757"      "ENTREZID:189"       "ENTREZID:211"      </span></span>
<span><span class="co">#&gt; [10] "ENTREZID:212"       "ENTREZID:23464"     "ENTREZID:2593"     </span></span>
<span><span class="co">#&gt; [13] "ENTREZID:26227"     "ENTREZID:2628"      "ENTREZID:27232"    </span></span>
<span><span class="co">#&gt; [16] "ENTREZID:2731"      "ENTREZID:275"       "ENTREZID:29958"    </span></span>
<span><span class="co">#&gt; [19] "ENTREZID:29968"     "ENTREZID:441531"    "ENTREZID:501"      </span></span>
<span><span class="co">#&gt; [22] "ENTREZID:51268"     "ENTREZID:5223"      "ENTREZID:5224"     </span></span>
<span><span class="co">#&gt; [25] "ENTREZID:55349"     "ENTREZID:5723"      "ENTREZID:635"      </span></span>
<span><span class="co">#&gt; [28] "ENTREZID:63826"     "ENTREZID:6470"      "ENTREZID:6472"     </span></span>
<span><span class="co">#&gt; [31] "ENTREZID:64902"     "ENTREZID:669"       "ENTREZID:875"      </span></span>
<span><span class="co">#&gt; [34] "ENTREZID:9380"      "ENTREZID:1491"</span></span>
<span><span class="va">kegg_t2g</span> <span class="op">&lt;-</span> <span class="fu">ldply</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">kegg</span>, <span class="va">nodes</span><span class="op">)</span>, <span class="va">data.frame</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">kegg_t2g</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gs_name"</span>, <span class="st">"gene_symbol"</span><span class="op">)</span></span>
<span><span class="va">VEGF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">kegg_t2g</span>,<span class="va">gs_name</span><span class="op">==</span><span class="st">"VEGF signaling pathway"</span><span class="op">)</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">eKEGG</span> <span class="op">&lt;-</span> <span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/clusterProfiler/man/enricher.html" class="external-link">enricher</a></span><span class="op">(</span>gene<span class="op">=</span><span class="va">VEGF</span>, TERM2GENE <span class="op">=</span> <span class="va">kegg_t2g</span>,</span>
<span>                                   universe<span class="op">=</span>,</span>
<span>                                   pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                                   pvalueCutoff <span class="op">=</span> <span class="fl">0.1</span>, qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                   minGSSize <span class="op">=</span> <span class="fl">10</span>, maxGSSize <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="8">
<h2 id="peptide-sequence">
<span class="header-section-number">8</span> Peptide sequence<a class="anchor" aria-label="anchor" href="#peptide-sequence"></a>
</h2>
<p>Here is an example for PROC_HUMAN, which is handled by the <code>Biostrings</code> package,</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fasta_file_path</span> <span class="op">&lt;-</span> <span class="st">'https://rest.uniprot.org/uniprotkb/P04070.fasta'</span></span>
<span><span class="va">fasta_sequences</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">fasta_file_path</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span></span>
<span><span class="va">AA_sequence</span> <span class="op">&lt;-</span> <span class="va">fasta_sequences</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Sequence:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/toString.html" class="external-link">toString</a></span><span class="op">(</span><span class="va">AA_sequence</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="va">iso_442688365</span> <span class="op">&lt;-</span> <span class="st">'TDGEGALSEPSATVTIEELAAPPPPVLMHHGESSQVLHPGNK'</span></span>
<span><span class="va">match_position</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">regexpr</a></span><span class="op">(</span><span class="va">iso_442688365</span>, <span class="va">AA_sequence</span><span class="op">)</span></span>
<span><span class="va">match_position</span></span>
<span><span class="va">mp</span> <span class="op">&lt;-</span> <span class="fu">matchPattern</span><span class="op">(</span><span class="va">iso_442688365</span>,<span class="va">AA_sequence</span><span class="op">)</span></span>
<span><span class="va">mp</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"~/pQTLtools/tests/PROC.rda"</span><span class="op">)</span></span>
<span><span class="fu">pQTLtools</span><span class="fu">::</span><span class="fu"><a href="../reference/peptideAssociationPlot.html">peptideAssociationPlot</a></span><span class="op">(</span><span class="va">protein</span>,<span class="va">cistrans</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(Modified.Peptide.Sequence)`</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:a1bg"></span>
<img src="bioconductor/a1bg-1.png" alt="peptide association plot" width="960"><p class="caption">
Figure 8.1: peptide association plot
</p>
</div>
</div>
<div class="section level2" number="9">
<h2 id="transcript-databases">
<span class="header-section-number">9</span> Transcript databases<a class="anchor" aria-label="anchor" href="#transcript-databases"></a>
</h2>
<p>An overview of annotation is available<span class="citation"><sup><a href="#ref-carlson16" role="doc-biblioref">2</a></sup></span>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># columns(org.Hs.eg.db)</span></span>
<span><span class="co"># keyref &lt;- keys(org.Hs.eg.db, keytype="ENTREZID")</span></span>
<span><span class="co"># symbol_uniprot &lt;- select(org.Hs.eg.db,keys=keyref,columns = c("SYMBOL","UNIPROT"))</span></span>
<span><span class="co"># subset(symbol_uniprot,SYMBOL=="MC4R")</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">EnsDb.Hsapiens.v86</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">listColumns</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"protein"</span>, skip.keys<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "tx_id"            "protein_id"       "protein_sequence"</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">listGenebiotypes</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "protein_coding"                     "unitary_pseudogene"                 "unprocessed_pseudogene"             "processed_pseudogene"               "processed_transcript"              </span></span>
<span><span class="co">#&gt;  [6] "transcribed_unprocessed_pseudogene" "antisense"                          "transcribed_unitary_pseudogene"     "polymorphic_pseudogene"             "lincRNA"                           </span></span>
<span><span class="co">#&gt; [11] "sense_intronic"                     "transcribed_processed_pseudogene"   "sense_overlapping"                  "IG_V_pseudogene"                    "pseudogene"                        </span></span>
<span><span class="co">#&gt; [16] "TR_V_gene"                          "3prime_overlapping_ncRNA"           "IG_V_gene"                          "bidirectional_promoter_lncRNA"      "snRNA"                             </span></span>
<span><span class="co">#&gt; [21] "miRNA"                              "misc_RNA"                           "snoRNA"                             "rRNA"                               "Mt_tRNA"                           </span></span>
<span><span class="co">#&gt; [26] "Mt_rRNA"                            "IG_C_gene"                          "IG_J_gene"                          "TR_J_gene"                          "TR_C_gene"                         </span></span>
<span><span class="co">#&gt; [31] "TR_V_pseudogene"                    "TR_J_pseudogene"                    "IG_D_gene"                          "ribozyme"                           "IG_C_pseudogene"                   </span></span>
<span><span class="co">#&gt; [36] "TR_D_gene"                          "TEC"                                "IG_J_pseudogene"                    "scRNA"                              "scaRNA"                            </span></span>
<span><span class="co">#&gt; [41] "vaultRNA"                           "sRNA"                               "macro_lncRNA"                       "non_coding"                         "IG_pseudogene"                     </span></span>
<span><span class="co">#&gt; [46] "LRG_gene"</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">listTxbiotypes</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "protein_coding"                     "processed_transcript"               "nonsense_mediated_decay"            "retained_intron"                    "unitary_pseudogene"                </span></span>
<span><span class="co">#&gt;  [6] "TEC"                                "miRNA"                              "misc_RNA"                           "non_stop_decay"                     "unprocessed_pseudogene"            </span></span>
<span><span class="co">#&gt; [11] "processed_pseudogene"               "transcribed_unprocessed_pseudogene" "lincRNA"                            "antisense"                          "transcribed_unitary_pseudogene"    </span></span>
<span><span class="co">#&gt; [16] "polymorphic_pseudogene"             "sense_intronic"                     "transcribed_processed_pseudogene"   "sense_overlapping"                  "IG_V_pseudogene"                   </span></span>
<span><span class="co">#&gt; [21] "pseudogene"                         "TR_V_gene"                          "3prime_overlapping_ncRNA"           "IG_V_gene"                          "bidirectional_promoter_lncRNA"     </span></span>
<span><span class="co">#&gt; [26] "snRNA"                              "snoRNA"                             "rRNA"                               "Mt_tRNA"                            "Mt_rRNA"                           </span></span>
<span><span class="co">#&gt; [31] "IG_C_gene"                          "IG_J_gene"                          "TR_J_gene"                          "TR_C_gene"                          "TR_V_pseudogene"                   </span></span>
<span><span class="co">#&gt; [36] "TR_J_pseudogene"                    "IG_D_gene"                          "ribozyme"                           "IG_C_pseudogene"                    "TR_D_gene"                         </span></span>
<span><span class="co">#&gt; [41] "IG_J_pseudogene"                    "scRNA"                              "scaRNA"                             "vaultRNA"                           "sRNA"                              </span></span>
<span><span class="co">#&gt; [46] "macro_lncRNA"                       "non_coding"                         "IG_pseudogene"                      "LRG_gene"</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">listTables</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; $gene</span></span>
<span><span class="co">#&gt; [1] "gene_id"          "gene_name"        "gene_biotype"     "gene_seq_start"   "gene_seq_end"     "seq_name"         "seq_strand"       "seq_coord_system" "symbol"          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $tx</span></span>
<span><span class="co">#&gt; [1] "tx_id"            "tx_biotype"       "tx_seq_start"     "tx_seq_end"       "tx_cds_seq_start" "tx_cds_seq_end"   "gene_id"          "tx_name"         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $tx2exon</span></span>
<span><span class="co">#&gt; [1] "tx_id"    "exon_id"  "exon_idx"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $exon</span></span>
<span><span class="co">#&gt; [1] "exon_id"        "exon_seq_start" "exon_seq_end"  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $chromosome</span></span>
<span><span class="co">#&gt; [1] "seq_name"    "seq_length"  "is_circular"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $protein</span></span>
<span><span class="co">#&gt; [1] "tx_id"            "protein_id"       "protein_sequence"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $uniprot</span></span>
<span><span class="co">#&gt; [1] "protein_id"           "uniprot_id"           "uniprot_db"           "uniprot_mapping_type"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $protein_domain</span></span>
<span><span class="co">#&gt; [1] "protein_id"            "protein_domain_id"     "protein_domain_source" "interpro_accession"    "prot_dom_start"        "prot_dom_end"         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $entrezgene</span></span>
<span><span class="co">#&gt; [1] "gene_id"  "entrezid"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $metadata</span></span>
<span><span class="co">#&gt; [1] "name"  "value"</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  name                               value</span></span>
<span><span class="co">#&gt; 1             Db type                               EnsDb</span></span>
<span><span class="co">#&gt; 2     Type of Gene ID                     Ensembl Gene ID</span></span>
<span><span class="co">#&gt; 3  Supporting package                           ensembldb</span></span>
<span><span class="co">#&gt; 4       Db created by ensembldb package from Bioconductor</span></span>
<span><span class="co">#&gt; 5      script_version                               0.3.0</span></span>
<span><span class="co">#&gt; 6       Creation time            Thu May 18 16:32:27 2017</span></span>
<span><span class="co">#&gt; 7     ensembl_version                                  86</span></span>
<span><span class="co">#&gt; 8        ensembl_host                           localhost</span></span>
<span><span class="co">#&gt; 9            Organism                        homo_sapiens</span></span>
<span><span class="co">#&gt; 10        taxonomy_id                                9606</span></span>
<span><span class="co">#&gt; 11       genome_build                              GRCh38</span></span>
<span><span class="co">#&gt; 12    DBSCHEMAVERSION                                 2.0</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/organism_species.html" class="external-link">organism</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Homo sapiens"</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">returnFilterColumns</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqinfo</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; Seqinfo object with 357 sequences (1 circular) from GRCh38 genome:</span></span>
<span><span class="co">#&gt;   seqnames seqlengths isCircular genome</span></span>
<span><span class="co">#&gt;   X         156040895      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   20         64444167      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   1         248956422      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   6         170805979      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   3         198295559      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   ...             ...        ...    ...</span></span>
<span><span class="co">#&gt;   LRG_239      114904      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   LRG_311      115492      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   LRG_721       33396      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   LRG_741      231167      FALSE GRCh38</span></span>
<span><span class="co">#&gt;   LRG_93        22459      FALSE GRCh38</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt;   [1] "1"                                      "10"                                     "11"                                     "12"                                    </span></span>
<span><span class="co">#&gt;   [5] "13"                                     "14"                                     "15"                                     "16"                                    </span></span>
<span><span class="co">#&gt;   [9] "17"                                     "18"                                     "19"                                     "2"                                     </span></span>
<span><span class="co">#&gt;  [13] "20"                                     "21"                                     "22"                                     "3"                                     </span></span>
<span><span class="co">#&gt;  [17] "4"                                      "5"                                      "6"                                      "7"                                     </span></span>
<span><span class="co">#&gt;  [21] "8"                                      "9"                                      "CHR_HG107_PATCH"                        "CHR_HG126_PATCH"                       </span></span>
<span><span class="co">#&gt;  [25] "CHR_HG1311_PATCH"                       "CHR_HG1342_HG2282_PATCH"                "CHR_HG1362_PATCH"                       "CHR_HG142_HG150_NOVEL_TEST"            </span></span>
<span><span class="co">#&gt;  [29] "CHR_HG151_NOVEL_TEST"                   "CHR_HG1651_PATCH"                       "CHR_HG1832_PATCH"                       "CHR_HG2021_PATCH"                      </span></span>
<span><span class="co">#&gt;  [33] "CHR_HG2022_PATCH"                       "CHR_HG2023_PATCH"                       "CHR_HG2030_PATCH"                       "CHR_HG2058_PATCH"                      </span></span>
<span><span class="co">#&gt;  [37] "CHR_HG2062_PATCH"                       "CHR_HG2063_PATCH"                       "CHR_HG2066_PATCH"                       "CHR_HG2072_PATCH"                      </span></span>
<span><span class="co">#&gt;  [41] "CHR_HG2095_PATCH"                       "CHR_HG2104_PATCH"                       "CHR_HG2116_PATCH"                       "CHR_HG2128_PATCH"                      </span></span>
<span><span class="co">#&gt;  [45] "CHR_HG2191_PATCH"                       "CHR_HG2213_PATCH"                       "CHR_HG2217_PATCH"                       "CHR_HG2232_PATCH"                      </span></span>
<span><span class="co">#&gt;  [49] "CHR_HG2233_PATCH"                       "CHR_HG2235_PATCH"                       "CHR_HG2239_PATCH"                       "CHR_HG2247_PATCH"                      </span></span>
<span><span class="co">#&gt;  [53] "CHR_HG2249_PATCH"                       "CHR_HG2288_HG2289_PATCH"                "CHR_HG2290_PATCH"                       "CHR_HG2291_PATCH"                      </span></span>
<span><span class="co">#&gt;  [57] "CHR_HG2334_PATCH"                       "CHR_HG26_PATCH"                         "CHR_HG986_PATCH"                        "CHR_HSCHR10_1_CTG1"                    </span></span>
<span><span class="co">#&gt;  [61] "CHR_HSCHR10_1_CTG2"                     "CHR_HSCHR10_1_CTG3"                     "CHR_HSCHR10_1_CTG4"                     "CHR_HSCHR10_1_CTG6"                    </span></span>
<span><span class="co">#&gt;  [65] "CHR_HSCHR11_1_CTG1_2"                   "CHR_HSCHR11_1_CTG5"                     "CHR_HSCHR11_1_CTG6"                     "CHR_HSCHR11_1_CTG7"                    </span></span>
<span><span class="co">#&gt;  [69] "CHR_HSCHR11_1_CTG8"                     "CHR_HSCHR11_2_CTG1"                     "CHR_HSCHR11_2_CTG1_1"                   "CHR_HSCHR11_3_CTG1"                    </span></span>
<span><span class="co">#&gt;  [73] "CHR_HSCHR12_1_CTG1"                     "CHR_HSCHR12_1_CTG2_1"                   "CHR_HSCHR12_2_CTG1"                     "CHR_HSCHR12_2_CTG2"                    </span></span>
<span><span class="co">#&gt;  [77] "CHR_HSCHR12_2_CTG2_1"                   "CHR_HSCHR12_3_CTG2"                     "CHR_HSCHR12_3_CTG2_1"                   "CHR_HSCHR12_4_CTG2"                    </span></span>
<span><span class="co">#&gt;  [81] "CHR_HSCHR12_4_CTG2_1"                   "CHR_HSCHR12_5_CTG2"                     "CHR_HSCHR12_5_CTG2_1"                   "CHR_HSCHR12_6_CTG2_1"                  </span></span>
<span><span class="co">#&gt;  [85] "CHR_HSCHR13_1_CTG1"                     "CHR_HSCHR13_1_CTG3"                     "CHR_HSCHR13_1_CTG5"                     "CHR_HSCHR13_1_CTG8"                    </span></span>
<span><span class="co">#&gt;  [89] "CHR_HSCHR14_1_CTG1"                     "CHR_HSCHR14_2_CTG1"                     "CHR_HSCHR14_3_CTG1"                     "CHR_HSCHR14_7_CTG1"                    </span></span>
<span><span class="co">#&gt;  [93] "CHR_HSCHR15_1_CTG1"                     "CHR_HSCHR15_1_CTG3"                     "CHR_HSCHR15_1_CTG8"                     "CHR_HSCHR15_2_CTG3"                    </span></span>
<span><span class="co">#&gt;  [97] "CHR_HSCHR15_2_CTG8"                     "CHR_HSCHR15_3_CTG3"                     "CHR_HSCHR15_3_CTG8"                     "CHR_HSCHR15_4_CTG8"                    </span></span>
<span><span class="co">#&gt; [101] "CHR_HSCHR15_5_CTG8"                     "CHR_HSCHR15_6_CTG8"                     "CHR_HSCHR16_1_CTG1"                     "CHR_HSCHR16_1_CTG3_1"                  </span></span>
<span><span class="co">#&gt; [105] "CHR_HSCHR16_2_CTG3_1"                   "CHR_HSCHR16_3_CTG1"                     "CHR_HSCHR16_4_CTG1"                     "CHR_HSCHR16_4_CTG3_1"                  </span></span>
<span><span class="co">#&gt; [109] "CHR_HSCHR16_5_CTG1"                     "CHR_HSCHR16_CTG2"                       "CHR_HSCHR17_10_CTG4"                    "CHR_HSCHR17_1_CTG1"                    </span></span>
<span><span class="co">#&gt; [113] "CHR_HSCHR17_1_CTG2"                     "CHR_HSCHR17_1_CTG4"                     "CHR_HSCHR17_1_CTG5"                     "CHR_HSCHR17_1_CTG9"                    </span></span>
<span><span class="co">#&gt; [117] "CHR_HSCHR17_2_CTG1"                     "CHR_HSCHR17_2_CTG2"                     "CHR_HSCHR17_2_CTG4"                     "CHR_HSCHR17_2_CTG5"                    </span></span>
<span><span class="co">#&gt; [121] "CHR_HSCHR17_3_CTG2"                     "CHR_HSCHR17_3_CTG4"                     "CHR_HSCHR17_4_CTG4"                     "CHR_HSCHR17_5_CTG4"                    </span></span>
<span><span class="co">#&gt; [125] "CHR_HSCHR17_6_CTG4"                     "CHR_HSCHR17_7_CTG4"                     "CHR_HSCHR17_8_CTG4"                     "CHR_HSCHR17_9_CTG4"                    </span></span>
<span><span class="co">#&gt; [129] "CHR_HSCHR18_1_CTG1_1"                   "CHR_HSCHR18_1_CTG2_1"                   "CHR_HSCHR18_2_CTG1_1"                   "CHR_HSCHR18_2_CTG2"                    </span></span>
<span><span class="co">#&gt; [133] "CHR_HSCHR18_2_CTG2_1"                   "CHR_HSCHR18_3_CTG2_1"                   "CHR_HSCHR18_5_CTG1_1"                   "CHR_HSCHR18_ALT21_CTG2_1"              </span></span>
<span><span class="co">#&gt; [137] "CHR_HSCHR18_ALT2_CTG2_1"                "CHR_HSCHR19KIR_ABC08_A1_HAP_CTG3_1"     "CHR_HSCHR19KIR_ABC08_AB_HAP_C_P_CTG3_1" "CHR_HSCHR19KIR_ABC08_AB_HAP_T_P_CTG3_1"</span></span>
<span><span class="co">#&gt; [141] "CHR_HSCHR19KIR_FH05_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_FH05_B_HAP_CTG3_1"       "CHR_HSCHR19KIR_FH06_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_FH06_BA1_HAP_CTG3_1"    </span></span>
<span><span class="co">#&gt; [145] "CHR_HSCHR19KIR_FH08_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_FH08_BAX_HAP_CTG3_1"     "CHR_HSCHR19KIR_FH13_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_FH13_BA2_HAP_CTG3_1"    </span></span>
<span><span class="co">#&gt; [149] "CHR_HSCHR19KIR_FH15_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_FH15_B_HAP_CTG3_1"       "CHR_HSCHR19KIR_G085_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_G085_BA1_HAP_CTG3_1"    </span></span>
<span><span class="co">#&gt; [153] "CHR_HSCHR19KIR_G248_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_G248_BA2_HAP_CTG3_1"     "CHR_HSCHR19KIR_GRC212_AB_HAP_CTG3_1"    "CHR_HSCHR19KIR_GRC212_BA1_HAP_CTG3_1"  </span></span>
<span><span class="co">#&gt; [157] "CHR_HSCHR19KIR_LUCE_A_HAP_CTG3_1"       "CHR_HSCHR19KIR_LUCE_BDEL_HAP_CTG3_1"    "CHR_HSCHR19KIR_RP5_B_HAP_CTG3_1"        "CHR_HSCHR19KIR_RSH_A_HAP_CTG3_1"       </span></span>
<span><span class="co">#&gt; [161] "CHR_HSCHR19KIR_RSH_BA2_HAP_CTG3_1"      "CHR_HSCHR19KIR_T7526_A_HAP_CTG3_1"      "CHR_HSCHR19KIR_T7526_BDEL_HAP_CTG3_1"   "CHR_HSCHR19LRC_COX1_CTG3_1"            </span></span>
<span><span class="co">#&gt; [165] "CHR_HSCHR19LRC_COX2_CTG3_1"             "CHR_HSCHR19LRC_LRC_I_CTG3_1"            "CHR_HSCHR19LRC_LRC_J_CTG3_1"            "CHR_HSCHR19LRC_LRC_S_CTG3_1"           </span></span>
<span><span class="co">#&gt; [169] "CHR_HSCHR19LRC_LRC_T_CTG3_1"            "CHR_HSCHR19LRC_PGF1_CTG3_1"             "CHR_HSCHR19LRC_PGF2_CTG3_1"             "CHR_HSCHR19_1_CTG2"                    </span></span>
<span><span class="co">#&gt; [173] "CHR_HSCHR19_1_CTG3_1"                   "CHR_HSCHR19_2_CTG2"                     "CHR_HSCHR19_2_CTG3_1"                   "CHR_HSCHR19_3_CTG2"                    </span></span>
<span><span class="co">#&gt; [177] "CHR_HSCHR19_3_CTG3_1"                   "CHR_HSCHR19_4_CTG2"                     "CHR_HSCHR19_4_CTG3_1"                   "CHR_HSCHR19_5_CTG2"                    </span></span>
<span><span class="co">#&gt; [181] "CHR_HSCHR1_1_CTG11"                     "CHR_HSCHR1_1_CTG3"                      "CHR_HSCHR1_1_CTG31"                     "CHR_HSCHR1_1_CTG32_1"                  </span></span>
<span><span class="co">#&gt; [185] "CHR_HSCHR1_2_CTG3"                      "CHR_HSCHR1_2_CTG31"                     "CHR_HSCHR1_2_CTG32_1"                   "CHR_HSCHR1_3_CTG3"                     </span></span>
<span><span class="co">#&gt; [189] "CHR_HSCHR1_3_CTG31"                     "CHR_HSCHR1_3_CTG32_1"                   "CHR_HSCHR1_4_CTG3"                      "CHR_HSCHR1_4_CTG31"                    </span></span>
<span><span class="co">#&gt; [193] "CHR_HSCHR1_5_CTG3"                      "CHR_HSCHR1_5_CTG32_1"                   "CHR_HSCHR1_ALT2_1_CTG32_1"              "CHR_HSCHR20_1_CTG1"                    </span></span>
<span><span class="co">#&gt; [197] "CHR_HSCHR20_1_CTG2"                     "CHR_HSCHR20_1_CTG3"                     "CHR_HSCHR20_1_CTG4"                     "CHR_HSCHR21_2_CTG1_1"                  </span></span>
<span><span class="co">#&gt; [201] "CHR_HSCHR21_3_CTG1_1"                   "CHR_HSCHR21_4_CTG1_1"                   "CHR_HSCHR21_5_CTG2"                     "CHR_HSCHR21_6_CTG1_1"                  </span></span>
<span><span class="co">#&gt; [205] "CHR_HSCHR21_8_CTG1_1"                   "CHR_HSCHR22_1_CTG1"                     "CHR_HSCHR22_1_CTG2"                     "CHR_HSCHR22_1_CTG3"                    </span></span>
<span><span class="co">#&gt; [209] "CHR_HSCHR22_1_CTG4"                     "CHR_HSCHR22_1_CTG5"                     "CHR_HSCHR22_1_CTG6"                     "CHR_HSCHR22_1_CTG7"                    </span></span>
<span><span class="co">#&gt; [213] "CHR_HSCHR22_2_CTG1"                     "CHR_HSCHR22_3_CTG1"                     "CHR_HSCHR22_4_CTG1"                     "CHR_HSCHR22_5_CTG1"                    </span></span>
<span><span class="co">#&gt; [217] "CHR_HSCHR22_6_CTG1"                     "CHR_HSCHR22_7_CTG1"                     "CHR_HSCHR22_8_CTG1"                     "CHR_HSCHR2_1_CTG1"                     </span></span>
<span><span class="co">#&gt; [221] "CHR_HSCHR2_1_CTG15"                     "CHR_HSCHR2_1_CTG5"                      "CHR_HSCHR2_1_CTG7"                      "CHR_HSCHR2_1_CTG7_2"                   </span></span>
<span><span class="co">#&gt; [225] "CHR_HSCHR2_2_CTG1"                      "CHR_HSCHR2_2_CTG15"                     "CHR_HSCHR2_2_CTG7"                      "CHR_HSCHR2_2_CTG7_2"                   </span></span>
<span><span class="co">#&gt; [229] "CHR_HSCHR2_3_CTG1"                      "CHR_HSCHR2_3_CTG15"                     "CHR_HSCHR2_3_CTG7_2"                    "CHR_HSCHR2_4_CTG1"                     </span></span>
<span><span class="co">#&gt; [233] "CHR_HSCHR2_6_CTG7_2"                    "CHR_HSCHR3_1_CTG1"                      "CHR_HSCHR3_1_CTG2_1"                    "CHR_HSCHR3_1_CTG3"                     </span></span>
<span><span class="co">#&gt; [237] "CHR_HSCHR3_2_CTG2_1"                    "CHR_HSCHR3_2_CTG3"                      "CHR_HSCHR3_3_CTG1"                      "CHR_HSCHR3_3_CTG3"                     </span></span>
<span><span class="co">#&gt; [241] "CHR_HSCHR3_4_CTG2_1"                    "CHR_HSCHR3_4_CTG3"                      "CHR_HSCHR3_5_CTG2_1"                    "CHR_HSCHR3_5_CTG3"                     </span></span>
<span><span class="co">#&gt; [245] "CHR_HSCHR3_6_CTG3"                      "CHR_HSCHR3_7_CTG3"                      "CHR_HSCHR3_8_CTG3"                      "CHR_HSCHR3_9_CTG3"                     </span></span>
<span><span class="co">#&gt; [249] "CHR_HSCHR4_11_CTG12"                    "CHR_HSCHR4_1_CTG12"                     "CHR_HSCHR4_1_CTG4"                      "CHR_HSCHR4_1_CTG6"                     </span></span>
<span><span class="co">#&gt; [253] "CHR_HSCHR4_1_CTG9"                      "CHR_HSCHR4_2_CTG12"                     "CHR_HSCHR4_2_CTG4"                      "CHR_HSCHR4_3_CTG12"                    </span></span>
<span><span class="co">#&gt; [257] "CHR_HSCHR4_4_CTG12"                     "CHR_HSCHR4_5_CTG12"                     "CHR_HSCHR4_6_CTG12"                     "CHR_HSCHR4_7_CTG12"                    </span></span>
<span><span class="co">#&gt; [261] "CHR_HSCHR4_8_CTG12"                     "CHR_HSCHR4_9_CTG12"                     "CHR_HSCHR5_1_CTG1"                      "CHR_HSCHR5_1_CTG1_1"                   </span></span>
<span><span class="co">#&gt; [265] "CHR_HSCHR5_1_CTG5"                      "CHR_HSCHR5_2_CTG1"                      "CHR_HSCHR5_2_CTG1_1"                    "CHR_HSCHR5_2_CTG5"                     </span></span>
<span><span class="co">#&gt; [269] "CHR_HSCHR5_3_CTG1"                      "CHR_HSCHR5_3_CTG5"                      "CHR_HSCHR5_4_CTG1"                      "CHR_HSCHR5_4_CTG1_1"                   </span></span>
<span><span class="co">#&gt; [273] "CHR_HSCHR5_5_CTG1"                      "CHR_HSCHR5_6_CTG1"                      "CHR_HSCHR5_7_CTG1"                      "CHR_HSCHR6_1_CTG10"                    </span></span>
<span><span class="co">#&gt; [277] "CHR_HSCHR6_1_CTG2"                      "CHR_HSCHR6_1_CTG3"                      "CHR_HSCHR6_1_CTG4"                      "CHR_HSCHR6_1_CTG5"                     </span></span>
<span><span class="co">#&gt; [281] "CHR_HSCHR6_1_CTG6"                      "CHR_HSCHR6_1_CTG7"                      "CHR_HSCHR6_1_CTG8"                      "CHR_HSCHR6_1_CTG9"                     </span></span>
<span><span class="co">#&gt; [285] "CHR_HSCHR6_8_CTG1"                      "CHR_HSCHR6_MHC_APD_CTG1"                "CHR_HSCHR6_MHC_COX_CTG1"                "CHR_HSCHR6_MHC_DBB_CTG1"               </span></span>
<span><span class="co">#&gt; [289] "CHR_HSCHR6_MHC_MANN_CTG1"               "CHR_HSCHR6_MHC_MCF_CTG1"                "CHR_HSCHR6_MHC_QBL_CTG1"                "CHR_HSCHR6_MHC_SSTO_CTG1"              </span></span>
<span><span class="co">#&gt; [293] "CHR_HSCHR7_1_CTG1"                      "CHR_HSCHR7_1_CTG4_4"                    "CHR_HSCHR7_1_CTG6"                      "CHR_HSCHR7_1_CTG7"                     </span></span>
<span><span class="co">#&gt; [297] "CHR_HSCHR7_2_CTG1"                      "CHR_HSCHR7_2_CTG4_4"                    "CHR_HSCHR7_2_CTG6"                      "CHR_HSCHR7_2_CTG7"                     </span></span>
<span><span class="co">#&gt; [301] "CHR_HSCHR7_3_CTG6"                      "CHR_HSCHR8_1_CTG1"                      "CHR_HSCHR8_1_CTG6"                      "CHR_HSCHR8_1_CTG7"                     </span></span>
<span><span class="co">#&gt; [305] "CHR_HSCHR8_2_CTG1"                      "CHR_HSCHR8_2_CTG7"                      "CHR_HSCHR8_3_CTG1"                      "CHR_HSCHR8_3_CTG7"                     </span></span>
<span><span class="co">#&gt; [309] "CHR_HSCHR8_4_CTG1"                      "CHR_HSCHR8_4_CTG7"                      "CHR_HSCHR8_5_CTG1"                      "CHR_HSCHR8_5_CTG7"                     </span></span>
<span><span class="co">#&gt; [313] "CHR_HSCHR8_6_CTG1"                      "CHR_HSCHR8_7_CTG1"                      "CHR_HSCHR8_8_CTG1"                      "CHR_HSCHR8_9_CTG1"                     </span></span>
<span><span class="co">#&gt; [317] "CHR_HSCHR9_1_CTG1"                      "CHR_HSCHR9_1_CTG2"                      "CHR_HSCHR9_1_CTG3"                      "CHR_HSCHR9_1_CTG4"                     </span></span>
<span><span class="co">#&gt; [321] "CHR_HSCHR9_1_CTG5"                      "CHR_HSCHR9_1_CTG6"                      "CHR_HSCHRX_1_CTG3"                      "CHR_HSCHRX_2_CTG12"                    </span></span>
<span><span class="co">#&gt; [325] "CHR_HSCHRX_2_CTG3"                      "GL000009.2"                             "GL000194.1"                             "GL000195.1"                            </span></span>
<span><span class="co">#&gt; [329] "GL000205.2"                             "GL000213.1"                             "GL000216.2"                             "GL000218.1"                            </span></span>
<span><span class="co">#&gt; [333] "GL000219.1"                             "GL000220.1"                             "GL000225.1"                             "KI270442.1"                            </span></span>
<span><span class="co">#&gt; [337] "KI270711.1"                             "KI270713.1"                             "KI270721.1"                             "KI270726.1"                            </span></span>
<span><span class="co">#&gt; [341] "KI270727.1"                             "KI270728.1"                             "KI270731.1"                             "KI270733.1"                            </span></span>
<span><span class="co">#&gt; [345] "KI270734.1"                             "KI270744.1"                             "KI270750.1"                             "LRG_183"                               </span></span>
<span><span class="co">#&gt; [349] "LRG_187"                                "LRG_239"                                "LRG_311"                                "LRG_721"                               </span></span>
<span><span class="co">#&gt; [353] "LRG_741"                                "LRG_93"                                 "MT"                                     "X"                                     </span></span>
<span><span class="co">#&gt; [357] "Y"</span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">updateEnsDb</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="co">#&gt; EnsDb for Ensembl:</span></span>
<span><span class="co">#&gt; |Backend: SQLite</span></span>
<span><span class="co">#&gt; |Db type: EnsDb</span></span>
<span><span class="co">#&gt; |Type of Gene ID: Ensembl Gene ID</span></span>
<span><span class="co">#&gt; |Supporting package: ensembldb</span></span>
<span><span class="co">#&gt; |Db created by: ensembldb package from Bioconductor</span></span>
<span><span class="co">#&gt; |script_version: 0.3.0</span></span>
<span><span class="co">#&gt; |Creation time: Thu May 18 16:32:27 2017</span></span>
<span><span class="co">#&gt; |ensembl_version: 86</span></span>
<span><span class="co">#&gt; |ensembl_host: localhost</span></span>
<span><span class="co">#&gt; |Organism: homo_sapiens</span></span>
<span><span class="co">#&gt; |taxonomy_id: 9606</span></span>
<span><span class="co">#&gt; |genome_build: GRCh38</span></span>
<span><span class="co">#&gt; |DBSCHEMAVERSION: 2.0</span></span>
<span><span class="co">#&gt; | No. of genes: 63970.</span></span>
<span><span class="co">#&gt; | No. of transcripts: 216741.</span></span>
<span><span class="co">#&gt; |Protein data available.</span></span>
<span></span>
<span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">x</span>, columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene_name"</span><span class="op">)</span>,</span>
<span>             filter<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu">SeqNameFilter</span><span class="op">(</span><span class="st">"X"</span><span class="op">)</span>, <span class="fu">AnnotationFilter</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationFilter/man/AnnotationFilter.html" class="external-link">GeneBiotypeFilter</a></span><span class="op">(</span><span class="st">"protein_coding"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 841 ranges and 3 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames              ranges strand |   gene_name         gene_id   gene_biotype</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000182378        X       276322-303356      + |      PLCXD1 ENSG00000182378 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000178605        X       304529-318819      - |      GTPBP6 ENSG00000178605 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000167393        X       333963-386955      - |     PPP2R3B ENSG00000167393 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000185960        X       624344-659411      + |        SHOX ENSG00000185960 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000205755        X     1187549-1212750      - |       CRLF2 ENSG00000205755 protein_coding</span></span>
<span><span class="co">#&gt;               ...      ...                 ...    ... .         ...             ...            ...</span></span>
<span><span class="co">#&gt;   ENSG00000277745        X 155459415-155460005      - |      H2AFB3 ENSG00000277745 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000185973        X 155490115-155669944      - |       TMLHE ENSG00000185973 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000168939        X 155767812-155782459      + |       SPRY3 ENSG00000168939 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000124333        X 155881293-155943769      + |       VAMP7 ENSG00000124333 protein_coding</span></span>
<span><span class="co">#&gt;   ENSG00000124334        X 155997581-156010817      + |        IL9R ENSG00000124334 protein_coding</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from GRCh38 genome</span></span>
<span><span class="fu">ensembldb</span> <span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">x</span>, columns<span class="op">=</span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">listColumns</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"tx"</span><span class="op">)</span>,</span>
<span>                        filter <span class="op">=</span> <span class="fu">AnnotationFilter</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationFilter/man/AnnotationFilterList.html" class="external-link">AnnotationFilterList</a></span><span class="op">(</span><span class="op">)</span>, order.type <span class="op">=</span> <span class="st">"asc"</span>, return.type <span class="op">=</span> <span class="st">"GRanges"</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 216741 ranges and 6 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames            ranges strand |           tx_id             tx_biotype tx_cds_seq_start tx_cds_seq_end         gene_id         tx_name</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;            &lt;character&gt;        &lt;integer&gt;      &lt;integer&gt;     &lt;character&gt;     &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENST00000456328        1       11869-14409      + | ENST00000456328   processed_transcript             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000456328</span></span>
<span><span class="co">#&gt;   ENST00000450305        1       12010-13670      + | ENST00000450305 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000450305</span></span>
<span><span class="co">#&gt;   ENST00000488147        1       14404-29570      - | ENST00000488147 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000227232 ENST00000488147</span></span>
<span><span class="co">#&gt;   ENST00000619216        1       17369-17436      - | ENST00000619216                  miRNA             &lt;NA&gt;           &lt;NA&gt; ENSG00000278267 ENST00000619216</span></span>
<span><span class="co">#&gt;   ENST00000473358        1       29554-31097      + | ENST00000473358                lincRNA             &lt;NA&gt;           &lt;NA&gt; ENSG00000243485 ENST00000473358</span></span>
<span><span class="co">#&gt;               ...      ...               ...    ... .             ...                    ...              ...            ...             ...             ...</span></span>
<span><span class="co">#&gt;   ENST00000420810        Y 26549425-26549743      + | ENST00000420810   processed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000224240 ENST00000420810</span></span>
<span><span class="co">#&gt;   ENST00000456738        Y 26586642-26591601      - | ENST00000456738 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000227629 ENST00000456738</span></span>
<span><span class="co">#&gt;   ENST00000435945        Y 26594851-26634652      - | ENST00000435945 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000237917 ENST00000435945</span></span>
<span><span class="co">#&gt;   ENST00000435741        Y 26626520-26627159      - | ENST00000435741   processed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000231514 ENST00000435741</span></span>
<span><span class="co">#&gt;   ENST00000431853        Y 56855244-56855488      + | ENST00000431853   processed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000235857 ENST00000431853</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 357 sequences (1 circular) from GRCh38 genome</span></span>
<span></span>
<span><span class="va">txdbEnsemblGRCh38</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromEnsembl.html" class="external-link">makeTxDbFromEnsembl</a></span><span class="op">(</span>organism<span class="op">=</span><span class="st">"Homo sapiens"</span>, release<span class="op">=</span><span class="fl">98</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in call_fun_in_txdbmaker("makeTxDbFromEnsembl", ...): makeTxDbFromEnsembl() has moved to the txdbmaker package. Please call txdbmaker::makeTxDbFromEnsembl() to get rid of this warning.</span></span>
<span><span class="co">#&gt; Fetch transcripts and genes from Ensembl ... OK</span></span>
<span><span class="co">#&gt;   (fetched 250194 transcripts from 67946 genes)</span></span>
<span><span class="co">#&gt; Fetch exons and CDS from Ensembl ... OK</span></span>
<span><span class="co">#&gt; Fetch chromosome names and lengths from Ensembl ...OK</span></span>
<span><span class="co">#&gt; Gather the metadata ... OK</span></span>
<span><span class="co">#&gt; Make the TxDb object ... OK</span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">txdbEnsemblGRCh38</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">txdb</span>,<span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $transcripts</span></span>
<span><span class="co">#&gt;   tx_id         tx_name        tx_type        tx_chrom tx_strand tx_start  tx_end</span></span>
<span><span class="co">#&gt; 1     1 ENST00000636745         lncRNA CHR_HG107_PATCH         +  1049876 1055745</span></span>
<span><span class="co">#&gt; 2     2 ENST00000636387         lncRNA CHR_HG107_PATCH         +  1052607 1055745</span></span>
<span><span class="co">#&gt; 3     3 ENST00000643422 protein_coding CHR_HG107_PATCH         +  1075018 1112365</span></span>
<span><span class="co">#&gt; 4     4 ENST00000645631 protein_coding CHR_HG107_PATCH         +  1075018 1112365</span></span>
<span><span class="co">#&gt; 5     5 ENST00000636567 protein_coding CHR_HG107_PATCH         +  1159911 1203106</span></span>
<span><span class="co">#&gt; 6     6 ENST00000636545 protein_coding CHR_HG107_PATCH         -  1012823 1036718</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $splicings</span></span>
<span><span class="co">#&gt;   tx_id exon_rank exon_id       exon_name      exon_chrom exon_strand exon_start exon_end cds_id        cds_name cds_start cds_end</span></span>
<span><span class="co">#&gt; 1     1         1       1 ENSE00003797146 CHR_HG107_PATCH           +    1049876  1049958     NA            &lt;NA&gt;        NA      NA</span></span>
<span><span class="co">#&gt; 2     1         2       2 ENSE00003795151 CHR_HG107_PATCH           +    1051619  1051839     NA            &lt;NA&gt;        NA      NA</span></span>
<span><span class="co">#&gt; 3     1         3       4 ENSE00003793692 CHR_HG107_PATCH           +    1054235  1054388     NA            &lt;NA&gt;        NA      NA</span></span>
<span><span class="co">#&gt; 4     1         4       5 ENSE00003797325 CHR_HG107_PATCH           +    1055110  1055745     NA            &lt;NA&gt;        NA      NA</span></span>
<span><span class="co">#&gt; 5     2         1       3 ENSE00003798310 CHR_HG107_PATCH           +    1052607  1055745     NA            &lt;NA&gt;        NA      NA</span></span>
<span><span class="co">#&gt; 6     3         1       6 ENSE00003815958 CHR_HG107_PATCH           +    1075018  1075093      1 ENSP00000494473   1075018 1075093</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $genes</span></span>
<span><span class="co">#&gt;   tx_id         gene_id</span></span>
<span><span class="co">#&gt; 1     1 ENSG00000283640</span></span>
<span><span class="co">#&gt; 2     2 ENSG00000283640</span></span>
<span><span class="co">#&gt; 3     3 ENSG00000284971</span></span>
<span><span class="co">#&gt; 4     4 ENSG00000284971</span></span>
<span><span class="co">#&gt; 5     5 ENSG00000283158</span></span>
<span><span class="co">#&gt; 6     6 ENSG00000283350</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $chrominfo</span></span>
<span><span class="co">#&gt;              chrom    length is_circular</span></span>
<span><span class="co">#&gt; 1  CHR_HG107_PATCH 135088590          NA</span></span>
<span><span class="co">#&gt; 2  CHR_HG109_PATCH  58617934          NA</span></span>
<span><span class="co">#&gt; 3  CHR_HG126_PATCH 198295908          NA</span></span>
<span><span class="co">#&gt; 4 CHR_HG1277_PATCH 133754853          NA</span></span>
<span><span class="co">#&gt; 5 CHR_HG1296_PATCH 190208697          NA</span></span>
<span><span class="co">#&gt; 6 CHR_HG1298_PATCH 190196285          NA</span></span>
<span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span></span>
<span></span>
<span><span class="co"># liverExprs &lt;- quantifyExpressionsFromBWs(txdb = txdb,BWfiles=,experimentalDesign=)</span></span></code></pre></div>
</div>
<div class="section level2" number="10">
<h2 id="bionconductor-forum">
<span class="header-section-number">10</span> Bionconductor forum<a class="anchor" aria-label="anchor" href="#bionconductor-forum"></a>
</h2>
<p>Web: <a href="https://support.bioconductor.org/" class="external-link">https://support.bioconductor.org/</a></p>
</div>
<div class="section level2" number="11">
<h2 id="bioconductorcran-packages">
<span class="header-section-number">11</span> Bioconductor/CRAN packages<a class="anchor" aria-label="anchor" href="#bioconductorcran-packages"></a>
</h2>
<table class="table">
<colgroup>
<col width="40%">
<col width="60%">
</colgroup>
<thead><tr class="header">
<th>Package</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Bioconductor</strong></td>
<td></td>
</tr>
<tr class="even">
<td>AnnotationDbi</td>
<td>AnnotationDb objects and their progeny, methods etc.</td>
</tr>
<tr class="odd">
<td>Biobase</td>
<td>Base functions for Bioconductor</td>
</tr>
<tr class="even">
<td>Biostrings</td>
<td>Efficient manipulation of biological strings</td>
</tr>
<tr class="odd">
<td>clusterProfiler</td>
<td>Functional profiles for genes and gene clusters</td>
</tr>
<tr class="even">
<td>ComplexHeatmap</td>
<td>Make complex heatmaps</td>
</tr>
<tr class="odd">
<td>DESSeq2</td>
<td>Differential gene expression analysis based on the negative binomial distribution</td>
</tr>
<tr class="even">
<td>edgeR</td>
<td>Empirical analysis of digital gene expression</td>
</tr>
<tr class="odd">
<td>EnsDb.Hsapiens.v86</td>
<td>Exposes an annotation databases generated from Ensembl</td>
</tr>
<tr class="even">
<td>ensembldb</td>
<td>Retrieve annotation data from an Ensembl based package</td>
</tr>
<tr class="odd">
<td>FlowSorted.DLPFC.450k</td>
<td>Illumina HumanMethylation data on sorted frontal cortex cell populations</td>
</tr>
<tr class="even">
<td>graphite</td>
<td>GRAPH Interaction from pathway topological environment</td>
</tr>
<tr class="odd">
<td>IlluminaHumanMethylation450kmanifest</td>
<td>Annotation for Illuminaâ€™s 450k methylation arrays</td>
</tr>
<tr class="even">
<td>INSPEcT</td>
<td>Quantification of the intronic and exonic gene features and the post-transcriptional regulation analysis</td>
</tr>
<tr class="odd">
<td>org.Hs.eg.db</td>
<td>Conversion of Entrez ID â€“ gene symbols</td>
</tr>
<tr class="even">
<td>OUTRIDER</td>
<td>OUTlier in RNA-Seq fInDER</td>
</tr>
<tr class="odd">
<td>Pi</td>
<td>Priority index, leveraging genetic evidence to prioritise drug targets at the gene and pathway level</td>
</tr>
<tr class="even">
<td>quantro</td>
<td>A test for when to use quantile normalisation</td>
</tr>
<tr class="odd">
<td>recount3</td>
<td>Interface to uniformly processed RNA-seq data</td>
</tr>
<tr class="even">
<td>Rgraphiz</td>
<td>Interfaces R with the AT&amp;T graphviz library for plotting R graph objects from the graph package</td>
</tr>
<tr class="odd">
<td>sva</td>
<td>Surrogate Variable Analysis</td>
</tr>
<tr class="even">
<td>TxDb.Hsapiens.UCSC.hg38.knownGene</td>
<td>Annotation of the human genome</td>
</tr>
<tr class="odd">
<td><strong>CRAN</strong></td>
<td></td>
</tr>
<tr class="even">
<td>doParallel</td>
<td>Foreach Parallel Adaptor for the â€˜parallelâ€™ Package</td>
</tr>
<tr class="odd">
<td>GeneNet</td>
<td>Modeling and Inferring Gene Networks</td>
</tr>
<tr class="even">
<td>ggplot2</td>
<td>Data Visualisations Using the grammar of graphics</td>
</tr>
<tr class="odd">
<td>heatmaply</td>
<td>Interactive Cluster Heat Maps Using <code>plotly</code> and <code>ggplot2</code>
</td>
</tr>
<tr class="even">
<td>pheatmap</td>
<td>results visualisation</td>
</tr>
<tr class="odd">
<td>plyr</td>
<td>Splitting, applying and combining data</td>
</tr>
<tr class="even">
<td>RColorBrewer</td>
<td>ColorBrewer Palettes</td>
</tr>
<tr class="odd">
<td>WGCNA</td>
<td>Weighted correlation network analysis</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-koster21" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">KÃ¶ster, J., Forster, J., Schmeier, S. &amp; Salazar, V. <span class="nocase">snakemake-workflows/rna-seq-star-deseq2: Version 1.2.0</span>. (2021) doi:<a href="https://doi.org/10.5281/zenodo.5245549" class="external-link">10.5281/zenodo.5245549</a>.</div>
</div>
<div id="ref-carlson16" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Carlson, M. R. J., PagÃ¨s, H., Arora, S., Obenchain, V. &amp; Morgan, M. Genomic annotation resources in <span>R/Bioconductor</span>. in <em>Statistical genomics: Methods and protocols</em> (eds. MathÃ©, E. &amp; Davis, S.) 67â€“69 (Springer New York, New York, NY, 2016). doi:<a href="https://doi.org/10.1007/978-1-4939-3578-9_4" class="external-link">10.1007/978-1-4939-3578-9_4</a>.</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>





  </body>
</html>
