<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="pQTLtools"><title>Bioconductor notes â€¢ pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Bioconductor notes"><meta property="og:description" content="pQTLtools"><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Vignettes:</h6>
    <a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Articles:</h6>
    <a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a>
    <a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a>
    <a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a>
    <a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a>
    <a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/pQTLtools.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://jinghuazhao.github.io/R/">
    <span class="fab fa fab fa-r-project"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/">
    <span class="fas fa fas fa-fast-backward"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-urls">URLs</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-urls">
    <a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a>
    <a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a>
    <a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a>
    <a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a>
    <a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a>
    <a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a>
    <a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a>
    <a class="external-link dropdown-item" href="https://metascape.org/">metascape</a>
    <a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a>
    <a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a>
    <a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a>
    <a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a>
    <a class="external-link dropdown-item" href="https://string-db.org/">STRING</a>
    <a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a>
    <a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.uniprot.org/">
    <span class="fas fa fas fa-fast-forward"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Bioconductor notes</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/vignettes/bioconductor.Rmd" class="external-link"><code>vignettes/bioconductor.Rmd</code></a></small>
      <div class="d-none name"><code>bioconductor.Rmd</code></div>
    </div>

    
    
<p>This article collects notes on Bioconductor packages, made available here to faciliate their use and extensions.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;AnnotationDbi&quot;</span>, <span class="st">&quot;Biostrings&quot;</span>, <span class="st">&quot;ComplexHeatmap&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>, <span class="st">&quot;EnsDb.Hsapiens.v86&quot;</span>,</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;FlowSorted.DLPFC.450k&quot;</span>, <span class="st">&quot;GeneNet&quot;</span>, <span class="st">&quot;GenomicFeatures&quot;</span>, <span class="st">&quot;IlluminaHumanMethylation450kmanifest&quot;</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;OUTRIDER&quot;</span>,<span class="st">&quot;RColorBrewer&quot;</span>, <span class="st">&quot;RMariaDB&quot;</span>, <span class="st">&quot;Rgraphviz&quot;</span>, <span class="st">&quot;S4Vectors&quot;</span>, <span class="st">&quot;SummarizedExperiment&quot;</span>,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;TxDb.Hsapiens.UCSC.hg38.knownGene&quot;</span>, <span class="st">&quot;bladderbatch&quot;</span>, <span class="st">&quot;clusterProfiler&quot;</span>,</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;corpcor&quot;</span>, <span class="st">&quot;ensembldb&quot;</span>, <span class="st">&quot;fdrtool&quot;</span>, <span class="st">&quot;graph&quot;</span>, <span class="st">&quot;graphite&quot;</span>, <span class="st">&quot;heatmaply&quot;</span>,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;minfi&quot;</span>, <span class="st">&quot;org.Hs.eg.db&quot;</span>, <span class="st">&quot;plyr&quot;</span>, <span class="st">&quot;quantro&quot;</span>, <span class="st">&quot;recount3&quot;</span>, <span class="st">&quot;sva&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (p <span class="cf">in</span> pkgs) <span class="cf">if</span> (<span class="fu">length</span>(<span class="fu">grep</span>(<span class="fu">paste</span>(<span class="st">&quot;^package:&quot;</span>, p, <span class="st">&quot;$&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), <span class="fu">search</span>())) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(p)) <span class="fu">warning</span>(<span class="fu">paste0</span>(<span class="st">&quot;This vignette needs package `&quot;</span>, p, <span class="st">&quot;&#39;; please install&quot;</span>))</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">suppressMessages</span>(<span class="fu">lapply</span>(pkgs, require, <span class="at">character.only =</span> <span class="cn">TRUE</span>)))</span></code></pre></div>
<div id="liftover" class="section level1" number="1">
<h1><span class="header-section-number">1</span> liftover</h1>
<p>See <code>inst/turboman</code>in the source, <a href="https://github.com/jinghuazhao/pQTLtools/tree/master/inst/turboman" class="uri">https://github.com/jinghuazhao/pQTLtools/tree/master/inst/turboman</a>, or <code>turboman/</code> directory in the installed package.</p>
</div>
<div id="normalisation" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Normalisation</h1>
<div id="combat" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> ComBat</h2>
<p>This is the documentation example, based on Bioconductor 3.14.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(bladderdata, <span class="at">package=</span><span class="st">&quot;bladderbatch&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>edat <span class="ot">&lt;-</span> bladderEset[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> Biobase<span class="sc">::</span><span class="fu">pData</span>(edat)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>batch <span class="ot">&lt;-</span> pheno<span class="sc">$</span>batch</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(batch)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; batch</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  1  2  3  4  5 </span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11 18  4  5 19</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>quantro<span class="sc">::</span><span class="fu">matboxplot</span>(edat,batch,<span class="at">cex.axis=</span><span class="fl">0.6</span>,<span class="at">notch=</span><span class="cn">TRUE</span>,<span class="at">pch=</span><span class="dv">19</span>,<span class="at">ylab=</span><span class="st">&quot;Expression&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:ComBat-1"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/ComBat-1.png" alt="ComBat example" width="672" />
<p class="caption">
Figure 2.1: ComBat example
</p>
</div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>quantro<span class="sc">::</span><span class="fu">matdensity</span>(edat,batch,<span class="at">xlab=</span><span class="st">&quot; &quot;</span>,<span class="at">ylab=</span><span class="st">&quot;density&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>,<span class="at">legend=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="at">col=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="at">lty=</span><span class="dv">1</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:ComBat-2"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/ComBat-2.png" alt="ComBat example" width="672" />
<p class="caption">
Figure 2.2: ComBat example
</p>
</div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. parametric adjustment</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>combat_edata1 <span class="ot">&lt;-</span> sva<span class="sc">::</span><span class="fu">ComBat</span>(<span class="at">dat=</span>edat, <span class="at">batch=</span>batch, <span class="at">par.prior=</span><span class="cn">TRUE</span>, <span class="at">prior.plots=</span><span class="cn">TRUE</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Found5batches</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adjusting for0covariate(s) or covariate level(s)</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Standardizing Data across genes</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Finding parametric adjustments</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adjusting the Data</span></span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:ComBat-3"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/ComBat-3.png" alt="ComBat example" width="672" />
<p class="caption">
Figure 2.3: ComBat example
</p>
</div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. non-parametric adjustment, mean-only version</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>combat_edata2 <span class="ot">&lt;-</span> sva<span class="sc">::</span><span class="fu">ComBat</span>(<span class="at">dat=</span>edat, <span class="at">batch=</span>batch, <span class="at">par.prior=</span><span class="cn">FALSE</span>, <span class="at">mean.only=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Using the &#39;mean only&#39; version of ComBat</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Found5batches</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adjusting for0covariate(s) or covariate level(s)</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Standardizing Data across genes</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Finding nonparametric adjustments</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adjusting the Data</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. reference-batch version, with covariates</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="fu">as.factor</span>(cancer), <span class="at">data=</span>pheno)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>combat_edata3 <span class="ot">&lt;-</span> sva<span class="sc">::</span><span class="fu">ComBat</span>(<span class="at">dat=</span>edat, <span class="at">batch=</span>batch, <span class="at">mod=</span>mod, <span class="at">par.prior=</span><span class="cn">TRUE</span>, <span class="at">ref.batch=</span><span class="dv">3</span>, <span class="at">prior.plots=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Using batch =3as a reference batch (this batch won&#39;t change)</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Found5batches</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adjusting for2covariate(s) or covariate level(s)</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Standardizing Data across genes</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fitting L/S model and finding priors</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Finding parametric adjustments</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Adjusting the Data</span></span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:ComBat-4"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/ComBat-4.png" alt="ComBat example" width="672" />
<p class="caption">
Figure 2.4: ComBat example
</p>
</div>
</div>
<div id="quantro" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> quantro</h2>
<p>This is also adapted from the package vignette but with <code>FlowSorted.DLPFC.450k</code> in place of <code>FlowSorted</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(FlowSorted.DLPFC<span class="fl">.450</span>k,<span class="at">package=</span><span class="st">&quot;FlowSorted.DLPFC.450k&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">getBeta</span>(FlowSorted.DLPFC<span class="fl">.450</span>k,<span class="at">offset=</span><span class="dv">100</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">&lt;-</span> Biobase<span class="sc">::</span><span class="fu">pData</span>(FlowSorted.DLPFC<span class="fl">.450</span>k)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>quantro<span class="sc">::</span><span class="fu">matboxplot</span>(p, <span class="at">groupFactor =</span> pd<span class="sc">$</span>CellType, <span class="at">xaxt =</span> <span class="st">&quot;n&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Beta Values&quot;</span>, <span class="at">pch=</span><span class="dv">19</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:quantro-1"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/quantro-1.png" alt="quantro example" width="672" />
<p class="caption">
Figure 2.5: quantro example
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>quantro<span class="sc">::</span><span class="fu">matdensity</span>(p, <span class="at">groupFactor =</span> pd<span class="sc">$</span>CellType, <span class="at">xlab =</span> <span class="st">&quot; &quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;density&quot;</span>,</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">main =</span> <span class="st">&quot;Beta Values&quot;</span>, <span class="at">brewer.n =</span> <span class="dv">8</span>, <span class="at">brewer.name =</span> <span class="st">&quot;Dark2&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&#39;top&#39;</span>, <span class="fu">c</span>(<span class="st">&quot;NeuN_neg&quot;</span>, <span class="st">&quot;NeuN_pos&quot;</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:quantro-2"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/quantro-2.png" alt="quantro example" width="672" />
<p class="caption">
Figure 2.6: quantro example
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>qtest <span class="ot">&lt;-</span> quantro<span class="sc">::</span><span class="fu">quantro</span>(<span class="at">object =</span> p, <span class="at">groupFactor =</span> pd<span class="sc">$</span>CellType)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [quantro] Average medians of the distributions are </span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                         not equal across groups.</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [quantro] Calculating the quantro test statistic.</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [quantro] No permutation testing performed. </span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                          Use B &gt; 0 for permutation testing.</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="cn">FALSE</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  doParallel<span class="sc">::</span><span class="fu">registerDoParallel</span>(<span class="at">cores=</span><span class="dv">10</span>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  qtestPerm <span class="ot">&lt;-</span> quantro<span class="sc">::</span><span class="fu">quantro</span>(p, <span class="at">groupFactor =</span> pd<span class="sc">$</span>CellType, <span class="at">B =</span> <span class="dv">1000</span>)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  quantro<span class="sc">::</span><span class="fu">quantroPlot</span>(qtestPerm)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="outlier-detection-in-rna-seq" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Outlier detection in RNA-Seq</h1>
<p>The following is adapted from package OUTRIDER,</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>ctsFile <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&#39;extdata&#39;</span>, <span class="st">&#39;KremerNBaderSmall.tsv&#39;</span>, <span class="at">package=</span><span class="st">&#39;OUTRIDER&#39;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>ctsTable <span class="ot">&lt;-</span> <span class="fu">read.table</span>(ctsFile, <span class="at">check.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>ods <span class="ot">&lt;-</span> OUTRIDER<span class="sc">::</span><span class="fu">OutriderDataSet</span>(<span class="at">countData=</span>ctsTable)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>ods <span class="ot">&lt;-</span> OUTRIDER<span class="sc">::</span><span class="fu">filterExpression</span>(ods, <span class="at">minCounts=</span><span class="cn">TRUE</span>, <span class="at">filterGenes=</span><span class="cn">TRUE</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 229 genes did not pass the filter due to zero counts. This is 22.9% of the genes.</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>ods <span class="ot">&lt;-</span> OUTRIDER<span class="sc">::</span><span class="fu">OUTRIDER</span>(ods)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tue May 14 21:26:18 2024: SizeFactor estimation ...</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tue May 14 21:26:18 2024: Controlling for confounders ...</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Using estimated q with: 23</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tue May 14 21:26:18 2024: Using the autoencoder implementation for controlling.</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:23 2024: Initial PCA loss: 4.73997327486604&quot;</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:28 2024: Iteration: 1 loss: 4.19495955308718&quot;</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:32 2024: Iteration: 2 loss: 4.175558109168&quot;</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:35 2024: Iteration: 3 loss: 4.16627394855922&quot;</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:37 2024: Iteration: 4 loss: 4.16188545534749&quot;</span></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:40 2024: Iteration: 5 loss: 4.15785003075681&quot;</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:42 2024: Iteration: 6 loss: 4.15514657702473&quot;</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:44 2024: Iteration: 7 loss: 4.15361195345535&quot;</span></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:46 2024: Iteration: 8 loss: 4.1519827021534&quot;</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:48 2024: Iteration: 9 loss: 4.15111658632385&quot;</span></span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:50 2024: Iteration: 10 loss: 4.14989393180947&quot;</span></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:51 2024: Iteration: 11 loss: 4.14980710469775&quot;</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:52 2024: Iteration: 12 loss: 4.14920776796145&quot;</span></span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:54 2024: Iteration: 13 loss: 4.14899084050906&quot;</span></span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:55 2024: Iteration: 14 loss: 4.14820997972724&quot;</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:57 2024: Iteration: 15 loss: 4.14814964413254&quot;</span></span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Time difference of 32.14789 secs</span></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Tue May 14 21:26:57 2024: 15 Final nb-AE loss: 4.14814964413254&quot;</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tue May 14 21:26:57 2024: Used the autoencoder implementation for controlling.</span></span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tue May 14 21:26:57 2024: P-value calculation ...</span></span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Tue May 14 21:27:00 2024: Zscore calculation ...</span></span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> OUTRIDER<span class="sc">::</span><span class="fu">results</span>(ods)</span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(res,<span class="at">caption=</span><span class="st">&quot;A check list of outliers&quot;</span>)</span></code></pre></div>
<table style="width:100%;">
<caption><span id="tab:ods">Table 3.1: </span>A check list of outliers</caption>
<colgroup>
<col width="7%" />
<col width="5%" />
<col width="4%" />
<col width="6%" />
<col width="4%" />
<col width="3%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="4%" />
<col width="5%" />
<col width="10%" />
<col width="9%" />
<col width="6%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">geneID</th>
<th align="left">sampleID</th>
<th align="right">pValue</th>
<th align="right">padjust</th>
<th align="right">zScore</th>
<th align="right">l2fc</th>
<th align="right">rawcounts</th>
<th align="right">normcounts</th>
<th align="right">meanCorrected</th>
<th align="right">theta</th>
<th align="left">aberrant</th>
<th align="right">AberrantBySample</th>
<th align="right">AberrantByGene</th>
<th align="right">padj_rank</th>
<th align="left">FDR_set</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ATAD3C</td>
<td align="left">MUC1360</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000002</td>
<td align="right">5.28</td>
<td align="right">1.87</td>
<td align="right">948</td>
<td align="right">247.03</td>
<td align="right">67.31</td>
<td align="right">16.60</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="even">
<td align="left">NBPF15</td>
<td align="left">MUC1351</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000037</td>
<td align="right">5.78</td>
<td align="right">0.78</td>
<td align="right">7591</td>
<td align="right">7054.78</td>
<td align="right">4121.48</td>
<td align="right">110.75</td>
<td align="left">TRUE</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="odd">
<td align="left">MSTO1</td>
<td align="left">MUC1367</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000215</td>
<td align="right">-6.23</td>
<td align="right">-0.81</td>
<td align="right">761</td>
<td align="right">728.28</td>
<td align="right">1276.06</td>
<td align="right">151.81</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="even">
<td align="left">HDAC1</td>
<td align="left">MUC1350</td>
<td align="right">0.0e+00</td>
<td align="right">0.0000764</td>
<td align="right">-5.95</td>
<td align="right">-0.78</td>
<td align="right">2215</td>
<td align="right">2122.51</td>
<td align="right">3648.82</td>
<td align="right">137.72</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="odd">
<td align="left">DCAF6</td>
<td align="left">MUC1374</td>
<td align="right">1.0e-07</td>
<td align="right">0.0003708</td>
<td align="right">-5.69</td>
<td align="right">-0.62</td>
<td align="right">2348</td>
<td align="right">3082.74</td>
<td align="right">4724.20</td>
<td align="right">197.18</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="even">
<td align="left">NBPF16</td>
<td align="left">MUC1351</td>
<td align="right">2.0e-07</td>
<td align="right">0.0006284</td>
<td align="right">4.84</td>
<td align="right">0.68</td>
<td align="right">4014</td>
<td align="right">3836.96</td>
<td align="right">2402.50</td>
<td align="right">106.52</td>
<td align="left">TRUE</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="odd">
<td align="left">FAM102B</td>
<td align="left">MUC1363</td>
<td align="right">1.2e-06</td>
<td align="right">0.0068396</td>
<td align="right">-5.37</td>
<td align="right">-1.28</td>
<td align="right">455</td>
<td align="right">443.55</td>
<td align="right">1076.64</td>
<td align="right">42.53</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="even">
<td align="left">LOC100288142</td>
<td align="left">MUC1361</td>
<td align="right">4.0e-06</td>
<td align="right">0.0222251</td>
<td align="right">4.20</td>
<td align="right">0.84</td>
<td align="right">637</td>
<td align="right">618.68</td>
<td align="right">345.77</td>
<td align="right">57.25</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="odd">
<td align="left">SERINC2</td>
<td align="left">MUC1343</td>
<td align="right">7.5e-06</td>
<td align="right">0.0416750</td>
<td align="right">-6.10</td>
<td align="right">-4.91</td>
<td align="right">42</td>
<td align="right">45.84</td>
<td align="right">1405.65</td>
<td align="right">4.52</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
<tr class="even">
<td align="left">TARDBP</td>
<td align="left">MUC0486</td>
<td align="right">7.6e-06</td>
<td align="right">0.0423348</td>
<td align="right">-4.55</td>
<td align="right">-0.32</td>
<td align="right">5911</td>
<td align="right">4452.35</td>
<td align="right">5565.42</td>
<td align="right">466.01</td>
<td align="left">TRUE</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">transcriptome-wide</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>OUTRIDER<span class="sc">::</span><span class="fu">plotQQ</span>(ods, res[<span class="st">&quot;geneID&quot;</span>],<span class="at">global=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:ods"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/ods-1.png" alt="Q-Q plot for Kremer data" width="672" />
<p class="caption">
Figure 3.1: Q-Q plot for Kremer data
</p>
</div>
</div>
<div id="differential-expression" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Differential expression</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>ex <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">makeExampleDESeqDataSet</span>(<span class="at">m=</span><span class="dv">4</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ex)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; estimating size factors</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; mean-dispersion relationship</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; final dispersion estimates</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">results</span>(dds, <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;condition&quot;</span>,<span class="st">&quot;B&quot;</span>,<span class="st">&quot;A&quot;</span>))</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>rld <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">rlogTransformation</span>(ex, <span class="at">blind=</span><span class="cn">TRUE</span>)</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="fu">c</span>(<span class="st">&quot;condition&quot;</span>),<span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; using ntop=500 top features by variance</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">attr</span>(dat,<span class="st">&quot;percentVar&quot;</span>))</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(dat, ggplot2<span class="sc">::</span><span class="fu">aes</span>(PC1, PC2, <span class="at">color=</span>condition, <span class="at">shape=</span>condition)) <span class="sc">+</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC1:&quot;</span>,percentVar[<span class="dv">1</span>],<span class="st">&quot;% variance&quot;</span>)) <span class="sc">+</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">&quot;PC2:&quot;</span>,percentVar[<span class="dv">2</span>],<span class="st">&quot;% variance&quot;</span>))</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:DESeq2"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/DESeq2-1.png" alt="DESeq2 example" width="672" />
<p class="caption">
Figure 4.1: DESeq2 example
</p>
</div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>ex<span class="sc">$</span>condition <span class="ot">&lt;-</span> <span class="fu">relevel</span>(ex<span class="sc">$</span>condition, <span class="at">ref=</span><span class="st">&quot;B&quot;</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>dds2 <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(dds)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; using pre-existing size factors</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; estimating dispersions</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; found already estimated dispersions, replacing these</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; mean-dispersion relationship</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; final dispersion estimates</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; fitting model and testing</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">results</span>(dds2)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">as.data.frame</span>(res)))</span></code></pre></div>
<table>
<colgroup>
<col width="8%" />
<col width="13%" />
<col width="20%" />
<col width="13%" />
<col width="15%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">stat</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">gene1</td>
<td align="right">277.46471</td>
<td align="right">0.0164732</td>
<td align="right">0.4477271</td>
<td align="right">0.0367929</td>
<td align="right">0.9706501</td>
<td align="right">0.9998477</td>
</tr>
<tr class="even">
<td align="left">gene2</td>
<td align="right">10.51017</td>
<td align="right">0.9062864</td>
<td align="right">1.2748781</td>
<td align="right">0.7108808</td>
<td align="right">0.4771581</td>
<td align="right">0.9998477</td>
</tr>
<tr class="odd">
<td align="left">gene3</td>
<td align="right">37.01996</td>
<td align="right">0.4181164</td>
<td align="right">0.9271126</td>
<td align="right">0.4509877</td>
<td align="right">0.6519984</td>
<td align="right">0.9998477</td>
</tr>
<tr class="even">
<td align="left">gene4</td>
<td align="right">78.76033</td>
<td align="right">-0.9719377</td>
<td align="right">0.6074553</td>
<td align="right">-1.6000153</td>
<td align="right">0.1095952</td>
<td align="right">0.9998477</td>
</tr>
<tr class="odd">
<td align="left">gene5</td>
<td align="right">14.43047</td>
<td align="right">-0.5318057</td>
<td align="right">1.4614592</td>
<td align="right">-0.3638868</td>
<td align="right">0.7159426</td>
<td align="right">0.9998477</td>
</tr>
<tr class="even">
<td align="left">gene6</td>
<td align="right">41.86626</td>
<td align="right">1.2435933</td>
<td align="right">0.7644820</td>
<td align="right">1.6267136</td>
<td align="right">0.1037979</td>
<td align="right">0.9998477</td>
</tr>
</tbody>
</table>
<p>See the package in action from a snakemake workflow<span class="citation"><sup><a href="#ref-koster21" role="doc-biblioref">1</a></sup></span>.</p>
</div>
<div id="gene-co-expression-and-network-analysis" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Gene co-expression and network analysis</h1>
<p>A simple network is furnished with the <code>GeneNet</code> documentation example,</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## A random network with 40 nodes </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># it contains 780=40*39/2 edges of which 5 percent (=39) are non-zero</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>true.pcor <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">ggm.simulate.pcor</span>(<span class="dv">40</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># A data set with 40 observations</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>m.sim <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">ggm.simulate.data</span>(<span class="dv">40</span>, true.pcor)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co"># A simple estimate of partial correlations</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>estimated.pcor <span class="ot">&lt;-</span> corpcor<span class="sc">::</span><span class="fu">cor2pcor</span>( <span class="fu">cor</span>(m.sim) )</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co"># A comparison of estimated and true values</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>((true.pcor<span class="sc">-</span>estimated.pcor)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 593.2851</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="co"># A slightly better estimate ...</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>estimated.pcor<span class="fl">.2</span> <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">ggm.estimate.pcor</span>(m.sim)</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating optimal shrinkage intensity lambda (correlation matrix): 0.3689</span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>((true.pcor<span class="sc">-</span>estimated.pcor<span class="fl">.2</span>)<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 10.85778</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="do">## ecoli data </span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ecoli, <span class="at">package=</span><span class="st">&quot;GeneNet&quot;</span>)</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="co"># partial correlation matrix </span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>inferred.pcor <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">ggm.estimate.pcor</span>(ecoli)</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimating optimal shrinkage intensity lambda (correlation matrix): 0.1804</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a><span class="co"># p-values, q-values and posterior probabilities for each potential edge </span></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>test.results <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">network.test.edges</span>(inferred.pcor)</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Estimate (local) false discovery rates (partial correlations):</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 1... determine cutoff point</span></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 2... estimate parameters of null distribution and eta0</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 3... compute p-values and estimate empirical PDF/CDF</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 4... compute q-values and local fdr</span></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 5... prepare for plotting</span></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a><span class="co"># best 20 edges (strongest correlation)</span></span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>test.results[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,]</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a><span class="co"># network containing edges with prob &gt; 0.9 (i.e. local fdr &lt; 0.1)</span></span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">extract.network</span>(test.results, <span class="at">cutoff.ggm=</span><span class="fl">0.9</span>)</span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Significant edges:  65 </span></span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Corresponding to  1.26 %  of possible edges</span></span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a>net</span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span id="cb13-73"><a href="#cb13-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span id="cb13-74"><a href="#cb13-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span id="cb13-88"><a href="#cb13-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 21  0.09576091    89    95 1.443542e-05 3.354777e-03 0.9891317</span></span>
<span id="cb13-89"><a href="#cb13-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 22  0.09473210     7    51 1.784126e-05 3.864825e-03 0.9891317</span></span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 23 -0.09386896    53    58 2.127622e-05 4.313590e-03 0.9891317</span></span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 24 -0.09366615    29    83 2.217013e-05 4.421099e-03 0.9891317</span></span>
<span id="cb13-92"><a href="#cb13-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 25 -0.09341148    21    89 2.334321e-05 4.556947e-03 0.9810727</span></span>
<span id="cb13-93"><a href="#cb13-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 26 -0.09156391    49    93 3.380043e-05 5.955972e-03 0.9810727</span></span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 27 -0.09150710    80    90 3.418363e-05 6.002083e-03 0.9810727</span></span>
<span id="cb13-95"><a href="#cb13-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 28  0.09101505     7    53 3.767966e-05 6.408102e-03 0.9810727</span></span>
<span id="cb13-96"><a href="#cb13-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 29  0.09050688    21    84 4.164471e-05 6.838782e-03 0.9810727</span></span>
<span id="cb13-97"><a href="#cb13-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 30  0.08965490    72    73 4.919365e-05 7.581866e-03 0.9810727</span></span>
<span id="cb13-98"><a href="#cb13-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 31 -0.08934025    29    99 5.229604e-05 7.861416e-03 0.9810727</span></span>
<span id="cb13-99"><a href="#cb13-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 32 -0.08906819     9    95 5.512708e-05 8.104759e-03 0.9810727</span></span>
<span id="cb13-100"><a href="#cb13-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 33  0.08888345     2    49 5.713144e-05 8.270673e-03 0.9810727</span></span>
<span id="cb13-101"><a href="#cb13-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 34  0.08850681    86    90 6.143363e-05 8.610161e-03 0.9810727</span></span>
<span id="cb13-102"><a href="#cb13-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 35  0.08805868    17    53 6.695170e-05 9.015175e-03 0.9810727</span></span>
<span id="cb13-103"><a href="#cb13-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 36  0.08790809    28    48 6.890884e-05 9.151291e-03 0.9810727</span></span>
<span id="cb13-104"><a href="#cb13-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 37  0.08783471    33    58 6.988211e-05 9.217597e-03 0.9682377</span></span>
<span id="cb13-105"><a href="#cb13-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 38 -0.08705796     7    49 8.101244e-05 1.021362e-02 0.9682377</span></span>
<span id="cb13-106"><a href="#cb13-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 39  0.08645033    20    46 9.086547e-05 1.102466e-02 0.9682377</span></span>
<span id="cb13-107"><a href="#cb13-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 40  0.08609950    48    86 9.705862e-05 1.150392e-02 0.9682377</span></span>
<span id="cb13-108"><a href="#cb13-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 41  0.08598769    21    52 9.911458e-05 1.165816e-02 0.9682377</span></span>
<span id="cb13-109"><a href="#cb13-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 42  0.08555275    32    95 1.075099e-04 1.226435e-02 0.9682377</span></span>
<span id="cb13-110"><a href="#cb13-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 43  0.08548231    17    51 1.089311e-04 1.236337e-02 0.9424721</span></span>
<span id="cb13-111"><a href="#cb13-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 44  0.08470370    80    83 1.258659e-04 1.382356e-02 0.9424721</span></span>
<span id="cb13-112"><a href="#cb13-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 45  0.08442510    80    82 1.325062e-04 1.437068e-02 0.9174573</span></span>
<span id="cb13-113"><a href="#cb13-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 46  0.08271606    80    93 1.810275e-04 1.845632e-02 0.9174573</span></span>
<span id="cb13-114"><a href="#cb13-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 47  0.08235175    46    91 1.933329e-04 1.941579e-02 0.9174573</span></span>
<span id="cb13-115"><a href="#cb13-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 48  0.08217787    25    95 1.994788e-04 1.988432e-02 0.9174573</span></span>
<span id="cb13-116"><a href="#cb13-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 49 -0.08170331    29    87 2.171999e-04 2.119715e-02 0.9174573</span></span>
<span id="cb13-117"><a href="#cb13-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 50  0.08123632    19    29 2.360716e-04 2.253606e-02 0.9174573</span></span>
<span id="cb13-118"><a href="#cb13-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 51  0.08101702    51    84 2.454547e-04 2.318024e-02 0.9174573</span></span>
<span id="cb13-119"><a href="#cb13-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 52  0.08030748    16    93 2.782643e-04 2.532796e-02 0.9174573</span></span>
<span id="cb13-120"><a href="#cb13-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 53  0.08006503    28    52 2.903870e-04 2.608271e-02 0.9174573</span></span>
<span id="cb13-121"><a href="#cb13-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 54 -0.07941656    41    80 3.252833e-04 2.814824e-02 0.9174573</span></span>
<span id="cb13-122"><a href="#cb13-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 55  0.07941410    54    89 3.254229e-04 2.815620e-02 0.9174573</span></span>
<span id="cb13-123"><a href="#cb13-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 56 -0.07934653    28    80 3.292784e-04 2.837511e-02 0.9174573</span></span>
<span id="cb13-124"><a href="#cb13-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 57  0.07916783    29    92 3.396802e-04 2.895702e-02 0.9174573</span></span>
<span id="cb13-125"><a href="#cb13-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 58 -0.07866905    17    86 3.703635e-04 3.060293e-02 0.9174573</span></span>
<span id="cb13-126"><a href="#cb13-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 59  0.07827749    16    29 3.962446e-04 3.191462e-02 0.9174573</span></span>
<span id="cb13-127"><a href="#cb13-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 60 -0.07808262    73    89 4.097452e-04 3.257290e-02 0.9174573</span></span>
<span id="cb13-128"><a href="#cb13-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 61  0.07766261    52    67 4.403165e-04 3.400207e-02 0.9174573</span></span>
<span id="cb13-129"><a href="#cb13-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 62  0.07762917    25    87 4.428396e-04 3.411637e-02 0.9174573</span></span>
<span id="cb13-130"><a href="#cb13-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 63 -0.07739378     9    93 4.609872e-04 3.492295e-02 0.9174573</span></span>
<span id="cb13-131"><a href="#cb13-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 64  0.07738885    31    80 4.613747e-04 3.493988e-02 0.9174573</span></span>
<span id="cb13-132"><a href="#cb13-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 65 -0.07718681    80    94 4.775136e-04 3.563444e-02 0.9174573</span></span>
<span id="cb13-133"><a href="#cb13-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-134"><a href="#cb13-134" aria-hidden="true" tabindex="-1"></a><span class="co"># significant based on FDR cutoff Q=0.05?</span></span>
<span id="cb13-135"><a href="#cb13-135" aria-hidden="true" tabindex="-1"></a>num.significant<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">sum</span>(test.results<span class="sc">$</span>qval <span class="sc">&lt;=</span> <span class="fl">0.05</span>)</span>
<span id="cb13-136"><a href="#cb13-136" aria-hidden="true" tabindex="-1"></a>test.results[<span class="dv">1</span><span class="sc">:</span>num.significant<span class="fl">.1</span>,]</span>
<span id="cb13-137"><a href="#cb13-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span id="cb13-138"><a href="#cb13-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-139"><a href="#cb13-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-140"><a href="#cb13-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-141"><a href="#cb13-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span id="cb13-142"><a href="#cb13-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span id="cb13-143"><a href="#cb13-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span id="cb13-144"><a href="#cb13-144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span id="cb13-145"><a href="#cb13-145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span id="cb13-146"><a href="#cb13-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span id="cb13-147"><a href="#cb13-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span id="cb13-148"><a href="#cb13-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span id="cb13-149"><a href="#cb13-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span id="cb13-150"><a href="#cb13-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span id="cb13-151"><a href="#cb13-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span id="cb13-152"><a href="#cb13-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span id="cb13-153"><a href="#cb13-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span id="cb13-154"><a href="#cb13-154" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span id="cb13-155"><a href="#cb13-155" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span id="cb13-156"><a href="#cb13-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span id="cb13-157"><a href="#cb13-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span id="cb13-158"><a href="#cb13-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 21  0.09576091    89    95 1.443542e-05 3.354777e-03 0.9891317</span></span>
<span id="cb13-159"><a href="#cb13-159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 22  0.09473210     7    51 1.784126e-05 3.864825e-03 0.9891317</span></span>
<span id="cb13-160"><a href="#cb13-160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 23 -0.09386896    53    58 2.127622e-05 4.313590e-03 0.9891317</span></span>
<span id="cb13-161"><a href="#cb13-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 24 -0.09366615    29    83 2.217013e-05 4.421099e-03 0.9891317</span></span>
<span id="cb13-162"><a href="#cb13-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 25 -0.09341148    21    89 2.334321e-05 4.556947e-03 0.9810727</span></span>
<span id="cb13-163"><a href="#cb13-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 26 -0.09156391    49    93 3.380043e-05 5.955972e-03 0.9810727</span></span>
<span id="cb13-164"><a href="#cb13-164" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 27 -0.09150710    80    90 3.418363e-05 6.002083e-03 0.9810727</span></span>
<span id="cb13-165"><a href="#cb13-165" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 28  0.09101505     7    53 3.767966e-05 6.408102e-03 0.9810727</span></span>
<span id="cb13-166"><a href="#cb13-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 29  0.09050688    21    84 4.164471e-05 6.838782e-03 0.9810727</span></span>
<span id="cb13-167"><a href="#cb13-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 30  0.08965490    72    73 4.919365e-05 7.581866e-03 0.9810727</span></span>
<span id="cb13-168"><a href="#cb13-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 31 -0.08934025    29    99 5.229604e-05 7.861416e-03 0.9810727</span></span>
<span id="cb13-169"><a href="#cb13-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 32 -0.08906819     9    95 5.512708e-05 8.104759e-03 0.9810727</span></span>
<span id="cb13-170"><a href="#cb13-170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 33  0.08888345     2    49 5.713144e-05 8.270673e-03 0.9810727</span></span>
<span id="cb13-171"><a href="#cb13-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 34  0.08850681    86    90 6.143363e-05 8.610161e-03 0.9810727</span></span>
<span id="cb13-172"><a href="#cb13-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 35  0.08805868    17    53 6.695170e-05 9.015175e-03 0.9810727</span></span>
<span id="cb13-173"><a href="#cb13-173" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 36  0.08790809    28    48 6.890884e-05 9.151291e-03 0.9810727</span></span>
<span id="cb13-174"><a href="#cb13-174" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 37  0.08783471    33    58 6.988211e-05 9.217597e-03 0.9682377</span></span>
<span id="cb13-175"><a href="#cb13-175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 38 -0.08705796     7    49 8.101244e-05 1.021362e-02 0.9682377</span></span>
<span id="cb13-176"><a href="#cb13-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 39  0.08645033    20    46 9.086547e-05 1.102466e-02 0.9682377</span></span>
<span id="cb13-177"><a href="#cb13-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 40  0.08609950    48    86 9.705862e-05 1.150392e-02 0.9682377</span></span>
<span id="cb13-178"><a href="#cb13-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 41  0.08598769    21    52 9.911458e-05 1.165816e-02 0.9682377</span></span>
<span id="cb13-179"><a href="#cb13-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 42  0.08555275    32    95 1.075099e-04 1.226435e-02 0.9682377</span></span>
<span id="cb13-180"><a href="#cb13-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 43  0.08548231    17    51 1.089311e-04 1.236337e-02 0.9424721</span></span>
<span id="cb13-181"><a href="#cb13-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 44  0.08470370    80    83 1.258659e-04 1.382356e-02 0.9424721</span></span>
<span id="cb13-182"><a href="#cb13-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 45  0.08442510    80    82 1.325062e-04 1.437068e-02 0.9174573</span></span>
<span id="cb13-183"><a href="#cb13-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 46  0.08271606    80    93 1.810275e-04 1.845632e-02 0.9174573</span></span>
<span id="cb13-184"><a href="#cb13-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 47  0.08235175    46    91 1.933329e-04 1.941579e-02 0.9174573</span></span>
<span id="cb13-185"><a href="#cb13-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 48  0.08217787    25    95 1.994788e-04 1.988432e-02 0.9174573</span></span>
<span id="cb13-186"><a href="#cb13-186" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 49 -0.08170331    29    87 2.171999e-04 2.119715e-02 0.9174573</span></span>
<span id="cb13-187"><a href="#cb13-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 50  0.08123632    19    29 2.360716e-04 2.253606e-02 0.9174573</span></span>
<span id="cb13-188"><a href="#cb13-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 51  0.08101702    51    84 2.454547e-04 2.318024e-02 0.9174573</span></span>
<span id="cb13-189"><a href="#cb13-189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 52  0.08030748    16    93 2.782643e-04 2.532796e-02 0.9174573</span></span>
<span id="cb13-190"><a href="#cb13-190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 53  0.08006503    28    52 2.903870e-04 2.608271e-02 0.9174573</span></span>
<span id="cb13-191"><a href="#cb13-191" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 54 -0.07941656    41    80 3.252833e-04 2.814824e-02 0.9174573</span></span>
<span id="cb13-192"><a href="#cb13-192" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 55  0.07941410    54    89 3.254229e-04 2.815620e-02 0.9174573</span></span>
<span id="cb13-193"><a href="#cb13-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 56 -0.07934653    28    80 3.292784e-04 2.837511e-02 0.9174573</span></span>
<span id="cb13-194"><a href="#cb13-194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 57  0.07916783    29    92 3.396802e-04 2.895702e-02 0.9174573</span></span>
<span id="cb13-195"><a href="#cb13-195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 58 -0.07866905    17    86 3.703635e-04 3.060293e-02 0.9174573</span></span>
<span id="cb13-196"><a href="#cb13-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 59  0.07827749    16    29 3.962446e-04 3.191462e-02 0.9174573</span></span>
<span id="cb13-197"><a href="#cb13-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 60 -0.07808262    73    89 4.097452e-04 3.257290e-02 0.9174573</span></span>
<span id="cb13-198"><a href="#cb13-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 61  0.07766261    52    67 4.403165e-04 3.400207e-02 0.9174573</span></span>
<span id="cb13-199"><a href="#cb13-199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 62  0.07762917    25    87 4.428396e-04 3.411637e-02 0.9174573</span></span>
<span id="cb13-200"><a href="#cb13-200" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 63 -0.07739378     9    93 4.609872e-04 3.492295e-02 0.9174573</span></span>
<span id="cb13-201"><a href="#cb13-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 64  0.07738885    31    80 4.613747e-04 3.493988e-02 0.9174573</span></span>
<span id="cb13-202"><a href="#cb13-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 65 -0.07718681    80    94 4.775136e-04 3.563444e-02 0.9174573</span></span>
<span id="cb13-203"><a href="#cb13-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 66  0.07706275    27    58 4.876831e-04 3.606179e-02 0.8297811</span></span>
<span id="cb13-204"><a href="#cb13-204" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 67 -0.07610709    16    83 5.730532e-04 4.085920e-02 0.8297811</span></span>
<span id="cb13-205"><a href="#cb13-205" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 68  0.07550557    53    84 6.337143e-04 4.406472e-02 0.8297811</span></span>
<span id="cb13-206"><a href="#cb13-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-207"><a href="#cb13-207" aria-hidden="true" tabindex="-1"></a><span class="co"># significant based on &quot;local fdr&quot; cutoff (prob &gt; 0.9)?</span></span>
<span id="cb13-208"><a href="#cb13-208" aria-hidden="true" tabindex="-1"></a>num.significant<span class="fl">.2</span> <span class="ot">&lt;-</span> <span class="fu">sum</span>(test.results<span class="sc">$</span>prob <span class="sc">&gt;</span> <span class="fl">0.9</span>)</span>
<span id="cb13-209"><a href="#cb13-209" aria-hidden="true" tabindex="-1"></a>test.results[test.results<span class="sc">$</span>prob <span class="sc">&gt;</span> <span class="fl">0.9</span>,]</span>
<span id="cb13-210"><a href="#cb13-210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;           pcor node1 node2         pval         qval      prob</span></span>
<span id="cb13-211"><a href="#cb13-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1   0.23185664    51    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-212"><a href="#cb13-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2   0.22405545    52    53 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-213"><a href="#cb13-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3   0.21507824    51    52 2.220446e-16 3.612205e-13 1.0000000</span></span>
<span id="cb13-214"><a href="#cb13-214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4   0.17328863     7    93 3.108624e-15 3.792816e-12 0.9999945</span></span>
<span id="cb13-215"><a href="#cb13-215" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  -0.13418892    29    86 1.120812e-09 1.093997e-06 0.9999516</span></span>
<span id="cb13-216"><a href="#cb13-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6   0.12594697    21    72 1.103836e-08 8.978563e-06 0.9998400</span></span>
<span id="cb13-217"><a href="#cb13-217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7   0.11956105    28    86 5.890924e-08 3.853590e-05 0.9998400</span></span>
<span id="cb13-218"><a href="#cb13-218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8  -0.11723897    26    80 1.060526e-07 5.816172e-05 0.9998400</span></span>
<span id="cb13-219"><a href="#cb13-219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9  -0.11711625    72    89 1.093655e-07 5.930499e-05 0.9972804</span></span>
<span id="cb13-220"><a href="#cb13-220" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10  0.10658013    20    21 1.366610e-06 5.925275e-04 0.9972804</span></span>
<span id="cb13-221"><a href="#cb13-221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11  0.10589778    21    73 1.596859e-06 6.678429e-04 0.9972804</span></span>
<span id="cb13-222"><a href="#cb13-222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12  0.10478689    20    91 2.053403e-06 8.024425e-04 0.9972804</span></span>
<span id="cb13-223"><a href="#cb13-223" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13  0.10420836     7    52 2.338382e-06 8.778605e-04 0.9944557</span></span>
<span id="cb13-224"><a href="#cb13-224" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14  0.10236077    87    95 3.525186e-06 1.224964e-03 0.9944557</span></span>
<span id="cb13-225"><a href="#cb13-225" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15  0.10113550    27    95 4.610444e-06 1.500047e-03 0.9920084</span></span>
<span id="cb13-226"><a href="#cb13-226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16  0.09928954    21    51 6.868357e-06 2.046549e-03 0.9920084</span></span>
<span id="cb13-227"><a href="#cb13-227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 17  0.09791914    21    88 9.192373e-06 2.520616e-03 0.9920084</span></span>
<span id="cb13-228"><a href="#cb13-228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 18  0.09719685    18    95 1.070232e-05 2.790102e-03 0.9920084</span></span>
<span id="cb13-229"><a href="#cb13-229" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 19  0.09621791    28    90 1.313007e-05 3.171817e-03 0.9920084</span></span>
<span id="cb13-230"><a href="#cb13-230" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 20  0.09619099    12    80 1.320374e-05 3.182526e-03 0.9920084</span></span>
<span id="cb13-231"><a href="#cb13-231" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 21  0.09576091    89    95 1.443542e-05 3.354777e-03 0.9891317</span></span>
<span id="cb13-232"><a href="#cb13-232" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 22  0.09473210     7    51 1.784126e-05 3.864825e-03 0.9891317</span></span>
<span id="cb13-233"><a href="#cb13-233" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 23 -0.09386896    53    58 2.127622e-05 4.313590e-03 0.9891317</span></span>
<span id="cb13-234"><a href="#cb13-234" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 24 -0.09366615    29    83 2.217013e-05 4.421099e-03 0.9891317</span></span>
<span id="cb13-235"><a href="#cb13-235" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 25 -0.09341148    21    89 2.334321e-05 4.556947e-03 0.9810727</span></span>
<span id="cb13-236"><a href="#cb13-236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 26 -0.09156391    49    93 3.380043e-05 5.955972e-03 0.9810727</span></span>
<span id="cb13-237"><a href="#cb13-237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 27 -0.09150710    80    90 3.418363e-05 6.002083e-03 0.9810727</span></span>
<span id="cb13-238"><a href="#cb13-238" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 28  0.09101505     7    53 3.767966e-05 6.408102e-03 0.9810727</span></span>
<span id="cb13-239"><a href="#cb13-239" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 29  0.09050688    21    84 4.164471e-05 6.838782e-03 0.9810727</span></span>
<span id="cb13-240"><a href="#cb13-240" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 30  0.08965490    72    73 4.919365e-05 7.581866e-03 0.9810727</span></span>
<span id="cb13-241"><a href="#cb13-241" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 31 -0.08934025    29    99 5.229604e-05 7.861416e-03 0.9810727</span></span>
<span id="cb13-242"><a href="#cb13-242" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 32 -0.08906819     9    95 5.512708e-05 8.104759e-03 0.9810727</span></span>
<span id="cb13-243"><a href="#cb13-243" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 33  0.08888345     2    49 5.713144e-05 8.270673e-03 0.9810727</span></span>
<span id="cb13-244"><a href="#cb13-244" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 34  0.08850681    86    90 6.143363e-05 8.610161e-03 0.9810727</span></span>
<span id="cb13-245"><a href="#cb13-245" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 35  0.08805868    17    53 6.695170e-05 9.015175e-03 0.9810727</span></span>
<span id="cb13-246"><a href="#cb13-246" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 36  0.08790809    28    48 6.890884e-05 9.151291e-03 0.9810727</span></span>
<span id="cb13-247"><a href="#cb13-247" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 37  0.08783471    33    58 6.988211e-05 9.217597e-03 0.9682377</span></span>
<span id="cb13-248"><a href="#cb13-248" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 38 -0.08705796     7    49 8.101244e-05 1.021362e-02 0.9682377</span></span>
<span id="cb13-249"><a href="#cb13-249" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 39  0.08645033    20    46 9.086547e-05 1.102466e-02 0.9682377</span></span>
<span id="cb13-250"><a href="#cb13-250" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 40  0.08609950    48    86 9.705862e-05 1.150392e-02 0.9682377</span></span>
<span id="cb13-251"><a href="#cb13-251" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 41  0.08598769    21    52 9.911458e-05 1.165816e-02 0.9682377</span></span>
<span id="cb13-252"><a href="#cb13-252" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 42  0.08555275    32    95 1.075099e-04 1.226435e-02 0.9682377</span></span>
<span id="cb13-253"><a href="#cb13-253" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 43  0.08548231    17    51 1.089311e-04 1.236337e-02 0.9424721</span></span>
<span id="cb13-254"><a href="#cb13-254" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 44  0.08470370    80    83 1.258659e-04 1.382356e-02 0.9424721</span></span>
<span id="cb13-255"><a href="#cb13-255" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 45  0.08442510    80    82 1.325062e-04 1.437068e-02 0.9174573</span></span>
<span id="cb13-256"><a href="#cb13-256" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 46  0.08271606    80    93 1.810275e-04 1.845632e-02 0.9174573</span></span>
<span id="cb13-257"><a href="#cb13-257" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 47  0.08235175    46    91 1.933329e-04 1.941579e-02 0.9174573</span></span>
<span id="cb13-258"><a href="#cb13-258" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 48  0.08217787    25    95 1.994788e-04 1.988432e-02 0.9174573</span></span>
<span id="cb13-259"><a href="#cb13-259" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 49 -0.08170331    29    87 2.171999e-04 2.119715e-02 0.9174573</span></span>
<span id="cb13-260"><a href="#cb13-260" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 50  0.08123632    19    29 2.360716e-04 2.253606e-02 0.9174573</span></span>
<span id="cb13-261"><a href="#cb13-261" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 51  0.08101702    51    84 2.454547e-04 2.318024e-02 0.9174573</span></span>
<span id="cb13-262"><a href="#cb13-262" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 52  0.08030748    16    93 2.782643e-04 2.532796e-02 0.9174573</span></span>
<span id="cb13-263"><a href="#cb13-263" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 53  0.08006503    28    52 2.903870e-04 2.608271e-02 0.9174573</span></span>
<span id="cb13-264"><a href="#cb13-264" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 54 -0.07941656    41    80 3.252833e-04 2.814824e-02 0.9174573</span></span>
<span id="cb13-265"><a href="#cb13-265" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 55  0.07941410    54    89 3.254229e-04 2.815620e-02 0.9174573</span></span>
<span id="cb13-266"><a href="#cb13-266" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 56 -0.07934653    28    80 3.292784e-04 2.837511e-02 0.9174573</span></span>
<span id="cb13-267"><a href="#cb13-267" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 57  0.07916783    29    92 3.396802e-04 2.895702e-02 0.9174573</span></span>
<span id="cb13-268"><a href="#cb13-268" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 58 -0.07866905    17    86 3.703635e-04 3.060293e-02 0.9174573</span></span>
<span id="cb13-269"><a href="#cb13-269" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 59  0.07827749    16    29 3.962446e-04 3.191462e-02 0.9174573</span></span>
<span id="cb13-270"><a href="#cb13-270" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 60 -0.07808262    73    89 4.097452e-04 3.257290e-02 0.9174573</span></span>
<span id="cb13-271"><a href="#cb13-271" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 61  0.07766261    52    67 4.403165e-04 3.400207e-02 0.9174573</span></span>
<span id="cb13-272"><a href="#cb13-272" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 62  0.07762917    25    87 4.428396e-04 3.411637e-02 0.9174573</span></span>
<span id="cb13-273"><a href="#cb13-273" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 63 -0.07739378     9    93 4.609872e-04 3.492295e-02 0.9174573</span></span>
<span id="cb13-274"><a href="#cb13-274" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 64  0.07738885    31    80 4.613747e-04 3.493988e-02 0.9174573</span></span>
<span id="cb13-275"><a href="#cb13-275" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 65 -0.07718681    80    94 4.775136e-04 3.563444e-02 0.9174573</span></span>
<span id="cb13-276"><a href="#cb13-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-277"><a href="#cb13-277" aria-hidden="true" tabindex="-1"></a><span class="co"># parameters of the mixture distribution used to compute p-values etc.</span></span>
<span id="cb13-278"><a href="#cb13-278" aria-hidden="true" tabindex="-1"></a>c <span class="ot">&lt;-</span> fdrtool<span class="sc">::</span><span class="fu">fdrtool</span>(corpcor<span class="sc">::</span><span class="fu">sm2vec</span>(inferred.pcor), <span class="at">statistic=</span><span class="st">&quot;correlation&quot;</span>)</span>
<span id="cb13-279"><a href="#cb13-279" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 1... determine cutoff point</span></span>
<span id="cb13-280"><a href="#cb13-280" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 2... estimate parameters of null distribution and eta0</span></span>
<span id="cb13-281"><a href="#cb13-281" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 3... compute p-values and estimate empirical PDF/CDF</span></span>
<span id="cb13-282"><a href="#cb13-282" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 4... compute q-values and local fdr</span></span>
<span id="cb13-283"><a href="#cb13-283" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Step 5... prepare for plotting</span></span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:GeneNet-1"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/GeneNet-1.png" alt="GeneNet example" width="672" />
<p class="caption">
Figure 5.1: GeneNet example
</p>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>c<span class="sc">$</span>param</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;          cutoff N.cens      eta0     eta0.SE    kappa kappa.SE</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1,] 0.03553068   4352 0.9474623 0.005656465 2043.377 94.72267</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="do">## A random network with 20 nodes and 10 percent (=19) edges</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>true.pcor <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">ggm.simulate.pcor</span>(<span class="dv">20</span>, <span class="fl">0.1</span>)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to edge list</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>test.results <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">ggm.list.edges</span>(true.pcor)[<span class="dv">1</span><span class="sc">:</span><span class="dv">19</span>,]</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>nlab <span class="ot">&lt;-</span> LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co"># graphviz</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="co"># network.make.dot(filename=&quot;test.dot&quot;, test.results, nlab, main = &quot;A graph&quot;)</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co"># system(&quot;fdp -T svg -o test.svg test.dot&quot;)</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Rgraphviz</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>gr <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">network.make.graph</span>( test.results, nlab)</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>gr</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A graphNEL graph with directed edges</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of Nodes = 20 </span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of Edges = 38</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="fu">num.nodes</span>(gr)</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 20</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a><span class="fu">edge.info</span>(gr)</span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $weight</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      A~T      A~Q      A~J      A~L      B~N      B~J      C~R      D~N </span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -0.06270  0.19879  0.40924  0.62571  0.19421  0.52032 -0.69851  0.38015 </span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      D~H      F~Q      F~M      G~K      H~I      J~K      K~M      M~T </span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  0.61005 -0.23242  0.52489  0.48304  0.67365  0.25697  0.44416  0.18103 </span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      M~S      N~Q      R~T </span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  0.31499 -0.43478 -0.59535 </span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $dir</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    A~T    A~Q    A~J    A~L    B~N    B~J    C~R    D~N    D~H    F~Q    F~M </span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; </span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    G~K    H~I    J~K    K~M    M~T    M~S    N~Q    R~T </span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot;</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>gr2 <span class="ot">&lt;-</span> GeneNet<span class="sc">::</span><span class="fu">network.make.graph</span>( test.results, nlab, <span class="at">drop.singles=</span><span class="cn">TRUE</span>)</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>gr2</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; A graphNEL graph with directed edges</span></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of Nodes = 17 </span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of Edges = 38</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>GeneNet<span class="sc">::</span><span class="fu">num.nodes</span>(gr2)</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 17</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>GeneNet<span class="sc">::</span><span class="fu">edge.info</span>(gr2)</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $weight</span></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      A~T      A~Q      A~J      A~L      B~N      B~J      C~R      D~N </span></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -0.06270  0.19879  0.40924  0.62571  0.19421  0.52032 -0.69851  0.38015 </span></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      D~H      F~Q      F~M      G~K      H~I      J~K      K~M      M~T </span></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  0.61005 -0.23242  0.52489  0.48304  0.67365  0.25697  0.44416  0.18103 </span></span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      M~S      N~Q      R~T </span></span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  0.31499 -0.43478 -0.59535 </span></span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $dir</span></span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    A~T    A~Q    A~J    A~L    B~N    B~J    C~R    D~N    D~H    F~Q    F~M </span></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; </span></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    G~K    H~I    J~K    K~M    M~T    M~S    N~Q    R~T </span></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot; &quot;none&quot;</span></span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a><span class="co"># plot network</span></span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(gr, <span class="st">&quot;fdp&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:GeneNet-2"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/GeneNet-2.png" alt="GeneNet example" width="672" />
<p class="caption">
Figure 5.2: GeneNet example
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(gr2, <span class="st">&quot;fdp&quot;</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:GeneNet-3"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/GeneNet-3.png" alt="GeneNet example" width="672" />
<p class="caption">
Figure 5.3: GeneNet example
</p>
</div>
<p>A side-by-side heatmaps</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123454321</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">runif</span>(<span class="dv">2500</span>),<span class="dv">50</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>r <span class="ot">&lt;-</span> <span class="fu">cor</span>(m)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(r<span class="sc">&gt;=</span><span class="fl">0.7</span>)<span class="sc">+</span><span class="dv">0</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>f1 <span class="ot">&lt;-</span> ComplexHeatmap<span class="sc">::</span><span class="fu">Heatmap</span>(r)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>f2 <span class="ot">&lt;-</span> ComplexHeatmap<span class="sc">::</span><span class="fu">Heatmap</span>(g)</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>f <span class="ot">&lt;-</span> f1<span class="sc">+</span>f2</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>ComplexHeatmap<span class="sc">::</span><span class="fu">draw</span>(f)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:heatmap"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/heatmap-1.png" alt="Heatmaps" width="672" />
<p class="caption">
Figure 5.4: Heatmaps
</p>
</div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> heatmaply<span class="sc">::</span><span class="fu">normalize</span>(mtcars)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>hm <span class="ot">&lt;-</span> heatmaply<span class="sc">::</span><span class="fu">heatmaply</span>(df,<span class="at">k_col=</span><span class="dv">5</span>,<span class="at">k_row=</span><span class="dv">5</span>,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colors =</span> grDevices<span class="sc">::</span><span class="fu">colorRampPalette</span>(RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">&quot;RdBu&quot;</span>))(<span class="dv">256</span>))</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>htmlwidgets<span class="sc">::</span><span class="fu">saveWidget</span>(hm,<span class="at">file=</span><span class="st">&quot;heatmaply.html&quot;</span>)</span></code></pre></div>
<p>so we have <a href="heatmaply.html">heatmaply.html</a> and a module analysis with WGCNA,</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>pwr <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="fu">seq</span>(<span class="at">from=</span><span class="dv">12</span>, <span class="at">to=</span><span class="dv">30</span>, <span class="at">by=</span><span class="dv">2</span>))</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>sft <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">pickSoftThreshold</span>(dat, <span class="at">powerVector=</span>pwr, <span class="at">verbose=</span><span class="dv">5</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>ADJ <span class="ot">&lt;-</span> <span class="fu">abs</span>(<span class="fu">cor</span>(dat, <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>, <span class="at">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>))<span class="sc">^</span><span class="dv">6</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>dissADJ <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">-</span>ADJ</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>dissTOM <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">TOMdist</span>(ADJ)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>TOM <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">TOMsimilarityFromExpr</span>(dat)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>Tree <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">as.dist</span>(<span class="dv">1</span><span class="sc">-</span>TOM), <span class="at">method=</span><span class="st">&quot;average&quot;</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> pwr)</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  pam_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;pam&quot;</span>,j)</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(pam_name, cluster<span class="sc">::</span><span class="fu">pam</span>(<span class="fu">as.dist</span>(dissADJ),j))</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  pamTOM_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;pamTOM&quot;</span>,j)</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">assign</span>(pamTOM_name,cluster<span class="sc">::</span><span class="fu">pam</span>(<span class="fu">as.dist</span>(dissTOM),j))</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  tc <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">get</span>(pam_name)<span class="sc">$</span>clustering,<span class="fu">get</span>(pamTOM_name)<span class="sc">$</span>clustering)</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(tc)</span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">diag</span>(tc))</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>colorStaticTOM <span class="ot">&lt;-</span> <span class="fu">as.character</span>(WGCNA<span class="sc">::</span><span class="fu">cutreeStaticColor</span>(Tree,<span class="at">cutHeight=</span>.<span class="dv">99</span>,<span class="at">minSize=</span><span class="dv">5</span>))</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>colorDynamicTOM <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">labels2colors</span>(<span class="fu">cutreeDynamic</span>(Tree,<span class="at">method=</span><span class="st">&quot;tree&quot;</span>,<span class="at">minClusterSize=</span><span class="dv">5</span>))</span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>Colors <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pamTOM6<span class="sc">$</span>clustering,colorStaticTOM,colorDynamicTOM)</span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>WGCNA<span class="sc">::</span><span class="fu">plotDendroAndColors</span>(Tree, Colors, <span class="at">dendroLabels=</span><span class="cn">FALSE</span>, <span class="at">hang=</span><span class="fl">0.03</span>, <span class="at">addGuide=</span><span class="cn">TRUE</span>, <span class="at">guideHang=</span><span class="fl">0.05</span>)</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>meg <span class="ot">&lt;-</span> WGCNA<span class="sc">::</span><span class="fu">moduleEigengenes</span>(dat, <span class="at">color=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(dat), <span class="at">softPower=</span><span class="dv">6</span>)</span></code></pre></div>
</div>
<div id="meta-data" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Meta-data</h1>
<p>This section is based on package <code>recount3</code>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>hs <span class="ot">&lt;-</span> recount3<span class="sc">::</span><span class="fu">available_projects</span>()</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">subset</span>(hs,file_source<span class="sc">==</span><span class="st">&quot;gtex&quot;</span>))</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>recount3<span class="sc">::</span><span class="fu">annotation_options</span>(<span class="st">&quot;human&quot;</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>blood_rse <span class="ot">&lt;-</span> recount3<span class="sc">::</span><span class="fu">create_rse</span>(<span class="fu">subset</span>(hs,project<span class="sc">==</span><span class="st">&quot;BLOOD&quot;</span>))</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>S4Vectors<span class="sc">::</span><span class="fu">metadata</span>(blood_rse)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>SummarizedExperiment<span class="sc">::</span><span class="fu">rowRanges</span>(blood_rse)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(blood_rse))[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>recount3<span class="sc">::</span><span class="fu">expand_sra_attributes</span>(blood_rse)</span></code></pre></div>
</div>
<div id="pathway-and-enrichment-analysis" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Pathway and enrichment analysis</h1>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>reactome <span class="ot">&lt;-</span> graphite<span class="sc">::</span><span class="fu">pathways</span>(<span class="st">&quot;hsapiens&quot;</span>, <span class="st">&quot;reactome&quot;</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> graphite<span class="sc">::</span><span class="fu">pathways</span>(<span class="st">&quot;hsapiens&quot;</span>,<span class="st">&quot;kegg&quot;</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>pharmgkb <span class="ot">&lt;-</span> graphite<span class="sc">::</span><span class="fu">pathways</span>(<span class="st">&quot;hsapiens&quot;</span>,<span class="st">&quot;pharmgkb&quot;</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">nodes</span>(kegg[[<span class="dv">21</span>]])</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;ENTREZID:102724560&quot; &quot;ENTREZID:10993&quot;     &quot;ENTREZID:113675&quot;   </span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] &quot;ENTREZID:132158&quot;    &quot;ENTREZID:1610&quot;      &quot;ENTREZID:1738&quot;     </span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;ENTREZID:1757&quot;      &quot;ENTREZID:189&quot;       &quot;ENTREZID:211&quot;      </span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] &quot;ENTREZID:212&quot;       &quot;ENTREZID:23464&quot;     &quot;ENTREZID:2593&quot;     </span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] &quot;ENTREZID:26227&quot;     &quot;ENTREZID:2628&quot;      &quot;ENTREZID:27232&quot;    </span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] &quot;ENTREZID:2731&quot;      &quot;ENTREZID:275&quot;       &quot;ENTREZID:29958&quot;    </span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] &quot;ENTREZID:29968&quot;     &quot;ENTREZID:441531&quot;    &quot;ENTREZID:501&quot;      </span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [22] &quot;ENTREZID:51268&quot;     &quot;ENTREZID:5223&quot;      &quot;ENTREZID:5224&quot;     </span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] &quot;ENTREZID:55349&quot;     &quot;ENTREZID:5723&quot;      &quot;ENTREZID:635&quot;      </span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [28] &quot;ENTREZID:63826&quot;     &quot;ENTREZID:6470&quot;      &quot;ENTREZID:6472&quot;     </span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] &quot;ENTREZID:64902&quot;     &quot;ENTREZID:669&quot;       &quot;ENTREZID:875&quot;      </span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [34] &quot;ENTREZID:9380&quot;      &quot;ENTREZID:1491&quot;</span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>kegg_t2g <span class="ot">&lt;-</span> <span class="fu">ldply</span>(<span class="fu">lapply</span>(kegg, nodes), data.frame)</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(kegg_t2g) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;gs_name&quot;</span>, <span class="st">&quot;gene_symbol&quot;</span>)</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>VEGF <span class="ot">&lt;-</span> <span class="fu">subset</span>(kegg_t2g,gs_name<span class="sc">==</span><span class="st">&quot;VEGF signaling pathway&quot;</span>)[[<span class="dv">2</span>]]</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>eKEGG <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">enricher</span>(<span class="at">gene=</span>VEGF, <span class="at">TERM2GENE =</span> kegg_t2g,</span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">universe=</span>,</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">pvalueCutoff =</span> <span class="fl">0.1</span>, <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">500</span>)</span></code></pre></div>
</div>
<div id="peptide-sequence" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Peptide sequence</h1>
<p>Here is an example for PROC_HUMAN, which is handled by the <code>Biostrings</code> package,</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>fasta_file_path <span class="ot">&lt;-</span> <span class="st">&#39;https://rest.uniprot.org/uniprotkb/P04070.fasta&#39;</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>fasta_sequences <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readAAStringSet</span>(fasta_file_path, <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>AA_sequence <span class="ot">&lt;-</span> fasta_sequences[[<span class="dv">1</span>]]</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Sequence:&quot;</span>, <span class="fu">toString</span>(AA_sequence), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Sequence: MWQLTSLLLFVATWGISGTPAPLDSVFSSSERAHQVLRIRKRANSFLEELRHSSLERECIEEICDFEEAKEIFQNVDDTLAFWSKHVDGDQCLVLPLEHPCASLCCGHGTCIDGIGSFSCDCRSGWEGRFCQREVSFLNCSLDNGGCTHYCLEEVGWRRCSCAPGYKLGDDLLQCHPAVKFPCGRPWKRMEKKRSHLKRDTEDQEDQVDPRLIDGKMTRRGDSPWQVVLLDSKKKLACGAVLIHPSWVLTAAHCMDESKKLLVRLGEYDLRRWEKWELDLDIKEVFVHPNYSKSTTDNDIALLHLAQPATLSQTIVPICLPDSGLAERELNQAGQETLVTGWGYHSSREKEAKRNRTFVLNFIKIPVVPHNECSEVMSNMVSENMLCAGILGDRQDACEGDSGGPMVASFHGTWFLVGLVSWGEGCGLLHNYGVYTKVSRYLDWIHGHIRDKEAPQKSWAP</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>iso_442688365 <span class="ot">&lt;-</span> <span class="st">&#39;TDGEGALSEPSATVTIEELAAPPPPVLMHHGESSQVLHPGNK&#39;</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>match_position <span class="ot">&lt;-</span> <span class="fu">regexpr</span>(iso_442688365, AA_sequence)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>match_position</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] -1</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;match.length&quot;)</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] -1</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;index.type&quot;)</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;chars&quot;</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;useBytes&quot;)</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>mp <span class="ot">&lt;-</span> <span class="fu">matchPattern</span>(iso_442688365,AA_sequence)</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>mp</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Views on a 461-letter AAString subject</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; subject: [48;5;224m[30mM[39m[49m[48;5;194m[30mW[39m[49m[48;5;159m[30mQ[39m[49m[48;5;224m[30mL[39m[49m[48;5;159m[30mT[39m[49m[48;5;159m[30mS[39m[49m[48;5;224m[30mL[39m[49m[48;5;224m[30mL[39m[49m[48;5;224m[30mL[39m[49m[48;5;194m[30mF[39m[49m[48;5;224m[30mV[39m[49m[48;5;224m[30mA[39m[49m[48;5;159m[30mT[39m[49m[48;5;194m[30mW[39m[49m[48;5;189m[30mG[39m[49m[48;5;224m[30mI[39m[49m[48;5;159m[30mS[39m[49m[48;5;189m[30mG[39m[49m[48;5;159m[30mT[39m[49m[48;5;189m[30mP[39m[49m[48;5;224m[30mA[39m[49m[48;5;189m[30mP[39m[49m[48;5;224m[30mL[39m[49m[48;5;225m[30mD[39m[49m[48;5;159m[30mS[39m[49m[48;5;224m[30mV[39m[49m[48;5;194m[30mF[39m[49m[48;5;159m[30mS[39m[49m[48;5;159m[30mS[39m[49m[48;5;159m[30mS[39m[49m[48;5;225m[30mE[39m[49m[48;5;153m[30mR[39m[49m[48;5;224m[30mA[39m[49m[48;5;153m[30mH[39m[49m...[48;5;224m[30mL[39m[49m[48;5;224m[30mL[39m[49m[48;5;153m[30mH[39m[49m[48;5;159m[30mN[39m[49m[48;5;194m[30mY[39m[49m[48;5;189m[30mG[39m[49m[48;5;224m[30mV[39m[49m[48;5;194m[30mY[39m[49m[48;5;159m[30mT[39m[49m[48;5;153m[30mK[39m[49m[48;5;224m[30mV[39m[49m[48;5;159m[30mS[39m[49m[48;5;153m[30mR[39m[49m[48;5;194m[30mY[39m[49m[48;5;224m[30mL[39m[49m[48;5;225m[30mD[39m[49m[48;5;194m[30mW[39m[49m[48;5;224m[30mI[39m[49m[48;5;153m[30mH[39m[49m[48;5;189m[30mG[39m[49m[48;5;153m[30mH[39m[49m[48;5;224m[30mI[39m[49m[48;5;153m[30mR[39m[49m[48;5;225m[30mD[39m[49m[48;5;153m[30mK[39m[49m[48;5;225m[30mE[39m[49m[48;5;224m[30mA[39m[49m[48;5;189m[30mP[39m[49m[48;5;159m[30mQ[39m[49m[48;5;153m[30mK[39m[49m[48;5;159m[30mS[39m[49m[48;5;194m[30mW[39m[49m[48;5;224m[30mA[39m[49m[48;5;189m[30mP[39m[49m</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; views: NONE</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>protein <span class="ot">&lt;-</span> <span class="st">&quot;PROC&quot;</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>cistrans <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/pQTLtools/tests&quot;</span>,<span class="st">&quot;/&quot;</span>,protein,<span class="st">&quot;.cis.vs.trans&quot;</span>))</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/pQTLtools/tests/&quot;</span>,protein,<span class="st">&quot;.rda&quot;</span>))</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>pQTLtools<span class="sc">::</span><span class="fu">peptideAssociationPlot</span>(protein,cistrans)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1m[22mJoining with `by = join_by(Modified.Peptide.Sequence)`</span></span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:a1bg"></span>
<img src="/rds/user/jhz22/hpc-work/pQTLtools/vignettes/bioconductor/a1bg-1.png" alt="peptide association plot" width="960" />
<p class="caption">
Figure 8.1: peptide association plot
</p>
</div>
</div>
<div id="transcript-databases" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Transcript databases</h1>
<p>An overview of annotation is available<span class="citation"><sup><a href="#ref-carlson16" role="doc-biblioref">2</a></sup></span>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">width=</span><span class="dv">200</span>)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co"># columns(org.Hs.eg.db)</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># keyref &lt;- keys(org.Hs.eg.db, keytype=&quot;ENTREZID&quot;)</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co"># symbol_uniprot &lt;- select(org.Hs.eg.db,keys=keyref,columns = c(&quot;SYMBOL&quot;,&quot;UNIPROT&quot;))</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="co"># subset(symbol_uniprot,SYMBOL==&quot;MC4R&quot;)</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> EnsDb.Hsapiens.v86</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">listColumns</span>(x, <span class="st">&quot;protein&quot;</span>, <span class="at">skip.keys=</span><span class="cn">TRUE</span>)</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;tx_id&quot;            &quot;protein_id&quot;       &quot;protein_sequence&quot;</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">listGenebiotypes</span>(x)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;protein_coding&quot;                     &quot;unitary_pseudogene&quot;                 &quot;unprocessed_pseudogene&quot;             &quot;processed_pseudogene&quot;               &quot;processed_transcript&quot;              </span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [6] &quot;transcribed_unprocessed_pseudogene&quot; &quot;antisense&quot;                          &quot;transcribed_unitary_pseudogene&quot;     &quot;polymorphic_pseudogene&quot;             &quot;lincRNA&quot;                           </span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] &quot;sense_intronic&quot;                     &quot;transcribed_processed_pseudogene&quot;   &quot;sense_overlapping&quot;                  &quot;IG_V_pseudogene&quot;                    &quot;pseudogene&quot;                        </span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] &quot;TR_V_gene&quot;                          &quot;3prime_overlapping_ncRNA&quot;           &quot;IG_V_gene&quot;                          &quot;bidirectional_promoter_lncRNA&quot;      &quot;snRNA&quot;                             </span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] &quot;miRNA&quot;                              &quot;misc_RNA&quot;                           &quot;snoRNA&quot;                             &quot;rRNA&quot;                               &quot;Mt_tRNA&quot;                           </span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [26] &quot;Mt_rRNA&quot;                            &quot;IG_C_gene&quot;                          &quot;IG_J_gene&quot;                          &quot;TR_J_gene&quot;                          &quot;TR_C_gene&quot;                         </span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] &quot;TR_V_pseudogene&quot;                    &quot;TR_J_pseudogene&quot;                    &quot;IG_D_gene&quot;                          &quot;ribozyme&quot;                           &quot;IG_C_pseudogene&quot;                   </span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [36] &quot;TR_D_gene&quot;                          &quot;TEC&quot;                                &quot;IG_J_pseudogene&quot;                    &quot;scRNA&quot;                              &quot;scaRNA&quot;                            </span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] &quot;vaultRNA&quot;                           &quot;sRNA&quot;                               &quot;macro_lncRNA&quot;                       &quot;non_coding&quot;                         &quot;IG_pseudogene&quot;                     </span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [46] &quot;LRG_gene&quot;</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">listTxbiotypes</span>(x)</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;protein_coding&quot;                     &quot;processed_transcript&quot;               &quot;nonsense_mediated_decay&quot;            &quot;retained_intron&quot;                    &quot;unitary_pseudogene&quot;                </span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [6] &quot;TEC&quot;                                &quot;miRNA&quot;                              &quot;misc_RNA&quot;                           &quot;non_stop_decay&quot;                     &quot;unprocessed_pseudogene&quot;            </span></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] &quot;processed_pseudogene&quot;               &quot;transcribed_unprocessed_pseudogene&quot; &quot;lincRNA&quot;                            &quot;antisense&quot;                          &quot;transcribed_unitary_pseudogene&quot;    </span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] &quot;polymorphic_pseudogene&quot;             &quot;sense_intronic&quot;                     &quot;transcribed_processed_pseudogene&quot;   &quot;sense_overlapping&quot;                  &quot;IG_V_pseudogene&quot;                   </span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] &quot;pseudogene&quot;                         &quot;TR_V_gene&quot;                          &quot;3prime_overlapping_ncRNA&quot;           &quot;IG_V_gene&quot;                          &quot;bidirectional_promoter_lncRNA&quot;     </span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [26] &quot;snRNA&quot;                              &quot;snoRNA&quot;                             &quot;rRNA&quot;                               &quot;Mt_tRNA&quot;                            &quot;Mt_rRNA&quot;                           </span></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] &quot;IG_C_gene&quot;                          &quot;IG_J_gene&quot;                          &quot;TR_J_gene&quot;                          &quot;TR_C_gene&quot;                          &quot;TR_V_pseudogene&quot;                   </span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [36] &quot;TR_J_pseudogene&quot;                    &quot;IG_D_gene&quot;                          &quot;ribozyme&quot;                           &quot;IG_C_pseudogene&quot;                    &quot;TR_D_gene&quot;                         </span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [41] &quot;IG_J_pseudogene&quot;                    &quot;scRNA&quot;                              &quot;scaRNA&quot;                             &quot;vaultRNA&quot;                           &quot;sRNA&quot;                              </span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [46] &quot;macro_lncRNA&quot;                       &quot;non_coding&quot;                         &quot;IG_pseudogene&quot;                      &quot;LRG_gene&quot;</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">listTables</span>(x)</span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $gene</span></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;gene_id&quot;          &quot;gene_name&quot;        &quot;gene_biotype&quot;     &quot;gene_seq_start&quot;   &quot;gene_seq_end&quot;     &quot;seq_name&quot;         &quot;seq_strand&quot;       &quot;seq_coord_system&quot; &quot;symbol&quot;          </span></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $tx</span></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;tx_id&quot;            &quot;tx_biotype&quot;       &quot;tx_seq_start&quot;     &quot;tx_seq_end&quot;       &quot;tx_cds_seq_start&quot; &quot;tx_cds_seq_end&quot;   &quot;gene_id&quot;          &quot;tx_name&quot;         </span></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $tx2exon</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;tx_id&quot;    &quot;exon_id&quot;  &quot;exon_idx&quot;</span></span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $exon</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;exon_id&quot;        &quot;exon_seq_start&quot; &quot;exon_seq_end&quot;  </span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $chromosome</span></span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;seq_name&quot;    &quot;seq_length&quot;  &quot;is_circular&quot;</span></span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $protein</span></span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;tx_id&quot;            &quot;protein_id&quot;       &quot;protein_sequence&quot;</span></span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $uniprot</span></span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;protein_id&quot;           &quot;uniprot_id&quot;           &quot;uniprot_db&quot;           &quot;uniprot_mapping_type&quot;</span></span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $protein_domain</span></span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;protein_id&quot;            &quot;protein_domain_id&quot;     &quot;protein_domain_source&quot; &quot;interpro_accession&quot;    &quot;prot_dom_start&quot;        &quot;prot_dom_end&quot;         </span></span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $entrezgene</span></span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;gene_id&quot;  &quot;entrezid&quot;</span></span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $metadata</span></span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;name&quot;  &quot;value&quot;</span></span>
<span id="cb22-63"><a href="#cb22-63" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">metadata</span>(x)</span>
<span id="cb22-64"><a href="#cb22-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                  name                               value</span></span>
<span id="cb22-65"><a href="#cb22-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1             Db type                               EnsDb</span></span>
<span id="cb22-66"><a href="#cb22-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2     Type of Gene ID                     Ensembl Gene ID</span></span>
<span id="cb22-67"><a href="#cb22-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3  Supporting package                           ensembldb</span></span>
<span id="cb22-68"><a href="#cb22-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4       Db created by ensembldb package from Bioconductor</span></span>
<span id="cb22-69"><a href="#cb22-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5      script_version                               0.3.0</span></span>
<span id="cb22-70"><a href="#cb22-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6       Creation time            Thu May 18 16:32:27 2017</span></span>
<span id="cb22-71"><a href="#cb22-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7     ensembl_version                                  86</span></span>
<span id="cb22-72"><a href="#cb22-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8        ensembl_host                           localhost</span></span>
<span id="cb22-73"><a href="#cb22-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9            Organism                        homo_sapiens</span></span>
<span id="cb22-74"><a href="#cb22-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10        taxonomy_id                                9606</span></span>
<span id="cb22-75"><a href="#cb22-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11       genome_build                              GRCh38</span></span>
<span id="cb22-76"><a href="#cb22-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12    DBSCHEMAVERSION                                 2.0</span></span>
<span id="cb22-77"><a href="#cb22-77" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">organism</span>(x)</span>
<span id="cb22-78"><a href="#cb22-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;Homo sapiens&quot;</span></span>
<span id="cb22-79"><a href="#cb22-79" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">returnFilterColumns</span>(x)</span>
<span id="cb22-80"><a href="#cb22-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb22-81"><a href="#cb22-81" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">seqinfo</span>(x)</span>
<span id="cb22-82"><a href="#cb22-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Seqinfo object with 357 sequences (1 circular) from GRCh38 genome:</span></span>
<span id="cb22-83"><a href="#cb22-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   seqnames seqlengths isCircular genome</span></span>
<span id="cb22-84"><a href="#cb22-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   X         156040895      FALSE GRCh38</span></span>
<span id="cb22-85"><a href="#cb22-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   20         64444167      FALSE GRCh38</span></span>
<span id="cb22-86"><a href="#cb22-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   1         248956422      FALSE GRCh38</span></span>
<span id="cb22-87"><a href="#cb22-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   6         170805979      FALSE GRCh38</span></span>
<span id="cb22-88"><a href="#cb22-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   3         198295559      FALSE GRCh38</span></span>
<span id="cb22-89"><a href="#cb22-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ...             ...        ...    ...</span></span>
<span id="cb22-90"><a href="#cb22-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   LRG_239      114904      FALSE GRCh38</span></span>
<span id="cb22-91"><a href="#cb22-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   LRG_311      115492      FALSE GRCh38</span></span>
<span id="cb22-92"><a href="#cb22-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   LRG_721       33396      FALSE GRCh38</span></span>
<span id="cb22-93"><a href="#cb22-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   LRG_741      231167      FALSE GRCh38</span></span>
<span id="cb22-94"><a href="#cb22-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   LRG_93        22459      FALSE GRCh38</span></span>
<span id="cb22-95"><a href="#cb22-95" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">seqlevels</span>(x)</span>
<span id="cb22-96"><a href="#cb22-96" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [1] &quot;1&quot;                                      &quot;10&quot;                                     &quot;11&quot;                                     &quot;12&quot;                                    </span></span>
<span id="cb22-97"><a href="#cb22-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [5] &quot;13&quot;                                     &quot;14&quot;                                     &quot;15&quot;                                     &quot;16&quot;                                    </span></span>
<span id="cb22-98"><a href="#cb22-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   [9] &quot;17&quot;                                     &quot;18&quot;                                     &quot;19&quot;                                     &quot;2&quot;                                     </span></span>
<span id="cb22-99"><a href="#cb22-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [13] &quot;20&quot;                                     &quot;21&quot;                                     &quot;22&quot;                                     &quot;3&quot;                                     </span></span>
<span id="cb22-100"><a href="#cb22-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [17] &quot;4&quot;                                      &quot;5&quot;                                      &quot;6&quot;                                      &quot;7&quot;                                     </span></span>
<span id="cb22-101"><a href="#cb22-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [21] &quot;8&quot;                                      &quot;9&quot;                                      &quot;CHR_HG107_PATCH&quot;                        &quot;CHR_HG126_PATCH&quot;                       </span></span>
<span id="cb22-102"><a href="#cb22-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [25] &quot;CHR_HG1311_PATCH&quot;                       &quot;CHR_HG1342_HG2282_PATCH&quot;                &quot;CHR_HG1362_PATCH&quot;                       &quot;CHR_HG142_HG150_NOVEL_TEST&quot;            </span></span>
<span id="cb22-103"><a href="#cb22-103" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [29] &quot;CHR_HG151_NOVEL_TEST&quot;                   &quot;CHR_HG1651_PATCH&quot;                       &quot;CHR_HG1832_PATCH&quot;                       &quot;CHR_HG2021_PATCH&quot;                      </span></span>
<span id="cb22-104"><a href="#cb22-104" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [33] &quot;CHR_HG2022_PATCH&quot;                       &quot;CHR_HG2023_PATCH&quot;                       &quot;CHR_HG2030_PATCH&quot;                       &quot;CHR_HG2058_PATCH&quot;                      </span></span>
<span id="cb22-105"><a href="#cb22-105" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [37] &quot;CHR_HG2062_PATCH&quot;                       &quot;CHR_HG2063_PATCH&quot;                       &quot;CHR_HG2066_PATCH&quot;                       &quot;CHR_HG2072_PATCH&quot;                      </span></span>
<span id="cb22-106"><a href="#cb22-106" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [41] &quot;CHR_HG2095_PATCH&quot;                       &quot;CHR_HG2104_PATCH&quot;                       &quot;CHR_HG2116_PATCH&quot;                       &quot;CHR_HG2128_PATCH&quot;                      </span></span>
<span id="cb22-107"><a href="#cb22-107" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [45] &quot;CHR_HG2191_PATCH&quot;                       &quot;CHR_HG2213_PATCH&quot;                       &quot;CHR_HG2217_PATCH&quot;                       &quot;CHR_HG2232_PATCH&quot;                      </span></span>
<span id="cb22-108"><a href="#cb22-108" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [49] &quot;CHR_HG2233_PATCH&quot;                       &quot;CHR_HG2235_PATCH&quot;                       &quot;CHR_HG2239_PATCH&quot;                       &quot;CHR_HG2247_PATCH&quot;                      </span></span>
<span id="cb22-109"><a href="#cb22-109" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [53] &quot;CHR_HG2249_PATCH&quot;                       &quot;CHR_HG2288_HG2289_PATCH&quot;                &quot;CHR_HG2290_PATCH&quot;                       &quot;CHR_HG2291_PATCH&quot;                      </span></span>
<span id="cb22-110"><a href="#cb22-110" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [57] &quot;CHR_HG2334_PATCH&quot;                       &quot;CHR_HG26_PATCH&quot;                         &quot;CHR_HG986_PATCH&quot;                        &quot;CHR_HSCHR10_1_CTG1&quot;                    </span></span>
<span id="cb22-111"><a href="#cb22-111" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [61] &quot;CHR_HSCHR10_1_CTG2&quot;                     &quot;CHR_HSCHR10_1_CTG3&quot;                     &quot;CHR_HSCHR10_1_CTG4&quot;                     &quot;CHR_HSCHR10_1_CTG6&quot;                    </span></span>
<span id="cb22-112"><a href="#cb22-112" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [65] &quot;CHR_HSCHR11_1_CTG1_2&quot;                   &quot;CHR_HSCHR11_1_CTG5&quot;                     &quot;CHR_HSCHR11_1_CTG6&quot;                     &quot;CHR_HSCHR11_1_CTG7&quot;                    </span></span>
<span id="cb22-113"><a href="#cb22-113" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [69] &quot;CHR_HSCHR11_1_CTG8&quot;                     &quot;CHR_HSCHR11_2_CTG1&quot;                     &quot;CHR_HSCHR11_2_CTG1_1&quot;                   &quot;CHR_HSCHR11_3_CTG1&quot;                    </span></span>
<span id="cb22-114"><a href="#cb22-114" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [73] &quot;CHR_HSCHR12_1_CTG1&quot;                     &quot;CHR_HSCHR12_1_CTG2_1&quot;                   &quot;CHR_HSCHR12_2_CTG1&quot;                     &quot;CHR_HSCHR12_2_CTG2&quot;                    </span></span>
<span id="cb22-115"><a href="#cb22-115" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [77] &quot;CHR_HSCHR12_2_CTG2_1&quot;                   &quot;CHR_HSCHR12_3_CTG2&quot;                     &quot;CHR_HSCHR12_3_CTG2_1&quot;                   &quot;CHR_HSCHR12_4_CTG2&quot;                    </span></span>
<span id="cb22-116"><a href="#cb22-116" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [81] &quot;CHR_HSCHR12_4_CTG2_1&quot;                   &quot;CHR_HSCHR12_5_CTG2&quot;                     &quot;CHR_HSCHR12_5_CTG2_1&quot;                   &quot;CHR_HSCHR12_6_CTG2_1&quot;                  </span></span>
<span id="cb22-117"><a href="#cb22-117" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [85] &quot;CHR_HSCHR13_1_CTG1&quot;                     &quot;CHR_HSCHR13_1_CTG3&quot;                     &quot;CHR_HSCHR13_1_CTG5&quot;                     &quot;CHR_HSCHR13_1_CTG8&quot;                    </span></span>
<span id="cb22-118"><a href="#cb22-118" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [89] &quot;CHR_HSCHR14_1_CTG1&quot;                     &quot;CHR_HSCHR14_2_CTG1&quot;                     &quot;CHR_HSCHR14_3_CTG1&quot;                     &quot;CHR_HSCHR14_7_CTG1&quot;                    </span></span>
<span id="cb22-119"><a href="#cb22-119" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [93] &quot;CHR_HSCHR15_1_CTG1&quot;                     &quot;CHR_HSCHR15_1_CTG3&quot;                     &quot;CHR_HSCHR15_1_CTG8&quot;                     &quot;CHR_HSCHR15_2_CTG3&quot;                    </span></span>
<span id="cb22-120"><a href="#cb22-120" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [97] &quot;CHR_HSCHR15_2_CTG8&quot;                     &quot;CHR_HSCHR15_3_CTG3&quot;                     &quot;CHR_HSCHR15_3_CTG8&quot;                     &quot;CHR_HSCHR15_4_CTG8&quot;                    </span></span>
<span id="cb22-121"><a href="#cb22-121" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [101] &quot;CHR_HSCHR15_5_CTG8&quot;                     &quot;CHR_HSCHR15_6_CTG8&quot;                     &quot;CHR_HSCHR16_1_CTG1&quot;                     &quot;CHR_HSCHR16_1_CTG3_1&quot;                  </span></span>
<span id="cb22-122"><a href="#cb22-122" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [105] &quot;CHR_HSCHR16_2_CTG3_1&quot;                   &quot;CHR_HSCHR16_3_CTG1&quot;                     &quot;CHR_HSCHR16_4_CTG1&quot;                     &quot;CHR_HSCHR16_4_CTG3_1&quot;                  </span></span>
<span id="cb22-123"><a href="#cb22-123" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [109] &quot;CHR_HSCHR16_5_CTG1&quot;                     &quot;CHR_HSCHR16_CTG2&quot;                       &quot;CHR_HSCHR17_10_CTG4&quot;                    &quot;CHR_HSCHR17_1_CTG1&quot;                    </span></span>
<span id="cb22-124"><a href="#cb22-124" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [113] &quot;CHR_HSCHR17_1_CTG2&quot;                     &quot;CHR_HSCHR17_1_CTG4&quot;                     &quot;CHR_HSCHR17_1_CTG5&quot;                     &quot;CHR_HSCHR17_1_CTG9&quot;                    </span></span>
<span id="cb22-125"><a href="#cb22-125" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [117] &quot;CHR_HSCHR17_2_CTG1&quot;                     &quot;CHR_HSCHR17_2_CTG2&quot;                     &quot;CHR_HSCHR17_2_CTG4&quot;                     &quot;CHR_HSCHR17_2_CTG5&quot;                    </span></span>
<span id="cb22-126"><a href="#cb22-126" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [121] &quot;CHR_HSCHR17_3_CTG2&quot;                     &quot;CHR_HSCHR17_3_CTG4&quot;                     &quot;CHR_HSCHR17_4_CTG4&quot;                     &quot;CHR_HSCHR17_5_CTG4&quot;                    </span></span>
<span id="cb22-127"><a href="#cb22-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [125] &quot;CHR_HSCHR17_6_CTG4&quot;                     &quot;CHR_HSCHR17_7_CTG4&quot;                     &quot;CHR_HSCHR17_8_CTG4&quot;                     &quot;CHR_HSCHR17_9_CTG4&quot;                    </span></span>
<span id="cb22-128"><a href="#cb22-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [129] &quot;CHR_HSCHR18_1_CTG1_1&quot;                   &quot;CHR_HSCHR18_1_CTG2_1&quot;                   &quot;CHR_HSCHR18_2_CTG1_1&quot;                   &quot;CHR_HSCHR18_2_CTG2&quot;                    </span></span>
<span id="cb22-129"><a href="#cb22-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [133] &quot;CHR_HSCHR18_2_CTG2_1&quot;                   &quot;CHR_HSCHR18_3_CTG2_1&quot;                   &quot;CHR_HSCHR18_5_CTG1_1&quot;                   &quot;CHR_HSCHR18_ALT21_CTG2_1&quot;              </span></span>
<span id="cb22-130"><a href="#cb22-130" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [137] &quot;CHR_HSCHR18_ALT2_CTG2_1&quot;                &quot;CHR_HSCHR19KIR_ABC08_A1_HAP_CTG3_1&quot;     &quot;CHR_HSCHR19KIR_ABC08_AB_HAP_C_P_CTG3_1&quot; &quot;CHR_HSCHR19KIR_ABC08_AB_HAP_T_P_CTG3_1&quot;</span></span>
<span id="cb22-131"><a href="#cb22-131" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [141] &quot;CHR_HSCHR19KIR_FH05_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_FH05_B_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_FH06_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_FH06_BA1_HAP_CTG3_1&quot;    </span></span>
<span id="cb22-132"><a href="#cb22-132" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [145] &quot;CHR_HSCHR19KIR_FH08_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_FH08_BAX_HAP_CTG3_1&quot;     &quot;CHR_HSCHR19KIR_FH13_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_FH13_BA2_HAP_CTG3_1&quot;    </span></span>
<span id="cb22-133"><a href="#cb22-133" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [149] &quot;CHR_HSCHR19KIR_FH15_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_FH15_B_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_G085_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_G085_BA1_HAP_CTG3_1&quot;    </span></span>
<span id="cb22-134"><a href="#cb22-134" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [153] &quot;CHR_HSCHR19KIR_G248_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_G248_BA2_HAP_CTG3_1&quot;     &quot;CHR_HSCHR19KIR_GRC212_AB_HAP_CTG3_1&quot;    &quot;CHR_HSCHR19KIR_GRC212_BA1_HAP_CTG3_1&quot;  </span></span>
<span id="cb22-135"><a href="#cb22-135" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [157] &quot;CHR_HSCHR19KIR_LUCE_A_HAP_CTG3_1&quot;       &quot;CHR_HSCHR19KIR_LUCE_BDEL_HAP_CTG3_1&quot;    &quot;CHR_HSCHR19KIR_RP5_B_HAP_CTG3_1&quot;        &quot;CHR_HSCHR19KIR_RSH_A_HAP_CTG3_1&quot;       </span></span>
<span id="cb22-136"><a href="#cb22-136" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [161] &quot;CHR_HSCHR19KIR_RSH_BA2_HAP_CTG3_1&quot;      &quot;CHR_HSCHR19KIR_T7526_A_HAP_CTG3_1&quot;      &quot;CHR_HSCHR19KIR_T7526_BDEL_HAP_CTG3_1&quot;   &quot;CHR_HSCHR19LRC_COX1_CTG3_1&quot;            </span></span>
<span id="cb22-137"><a href="#cb22-137" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [165] &quot;CHR_HSCHR19LRC_COX2_CTG3_1&quot;             &quot;CHR_HSCHR19LRC_LRC_I_CTG3_1&quot;            &quot;CHR_HSCHR19LRC_LRC_J_CTG3_1&quot;            &quot;CHR_HSCHR19LRC_LRC_S_CTG3_1&quot;           </span></span>
<span id="cb22-138"><a href="#cb22-138" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [169] &quot;CHR_HSCHR19LRC_LRC_T_CTG3_1&quot;            &quot;CHR_HSCHR19LRC_PGF1_CTG3_1&quot;             &quot;CHR_HSCHR19LRC_PGF2_CTG3_1&quot;             &quot;CHR_HSCHR19_1_CTG2&quot;                    </span></span>
<span id="cb22-139"><a href="#cb22-139" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [173] &quot;CHR_HSCHR19_1_CTG3_1&quot;                   &quot;CHR_HSCHR19_2_CTG2&quot;                     &quot;CHR_HSCHR19_2_CTG3_1&quot;                   &quot;CHR_HSCHR19_3_CTG2&quot;                    </span></span>
<span id="cb22-140"><a href="#cb22-140" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [177] &quot;CHR_HSCHR19_3_CTG3_1&quot;                   &quot;CHR_HSCHR19_4_CTG2&quot;                     &quot;CHR_HSCHR19_4_CTG3_1&quot;                   &quot;CHR_HSCHR19_5_CTG2&quot;                    </span></span>
<span id="cb22-141"><a href="#cb22-141" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [181] &quot;CHR_HSCHR1_1_CTG11&quot;                     &quot;CHR_HSCHR1_1_CTG3&quot;                      &quot;CHR_HSCHR1_1_CTG31&quot;                     &quot;CHR_HSCHR1_1_CTG32_1&quot;                  </span></span>
<span id="cb22-142"><a href="#cb22-142" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [185] &quot;CHR_HSCHR1_2_CTG3&quot;                      &quot;CHR_HSCHR1_2_CTG31&quot;                     &quot;CHR_HSCHR1_2_CTG32_1&quot;                   &quot;CHR_HSCHR1_3_CTG3&quot;                     </span></span>
<span id="cb22-143"><a href="#cb22-143" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [189] &quot;CHR_HSCHR1_3_CTG31&quot;                     &quot;CHR_HSCHR1_3_CTG32_1&quot;                   &quot;CHR_HSCHR1_4_CTG3&quot;                      &quot;CHR_HSCHR1_4_CTG31&quot;                    </span></span>
<span id="cb22-144"><a href="#cb22-144" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [193] &quot;CHR_HSCHR1_5_CTG3&quot;                      &quot;CHR_HSCHR1_5_CTG32_1&quot;                   &quot;CHR_HSCHR1_ALT2_1_CTG32_1&quot;              &quot;CHR_HSCHR20_1_CTG1&quot;                    </span></span>
<span id="cb22-145"><a href="#cb22-145" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [197] &quot;CHR_HSCHR20_1_CTG2&quot;                     &quot;CHR_HSCHR20_1_CTG3&quot;                     &quot;CHR_HSCHR20_1_CTG4&quot;                     &quot;CHR_HSCHR21_2_CTG1_1&quot;                  </span></span>
<span id="cb22-146"><a href="#cb22-146" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [201] &quot;CHR_HSCHR21_3_CTG1_1&quot;                   &quot;CHR_HSCHR21_4_CTG1_1&quot;                   &quot;CHR_HSCHR21_5_CTG2&quot;                     &quot;CHR_HSCHR21_6_CTG1_1&quot;                  </span></span>
<span id="cb22-147"><a href="#cb22-147" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [205] &quot;CHR_HSCHR21_8_CTG1_1&quot;                   &quot;CHR_HSCHR22_1_CTG1&quot;                     &quot;CHR_HSCHR22_1_CTG2&quot;                     &quot;CHR_HSCHR22_1_CTG3&quot;                    </span></span>
<span id="cb22-148"><a href="#cb22-148" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [209] &quot;CHR_HSCHR22_1_CTG4&quot;                     &quot;CHR_HSCHR22_1_CTG5&quot;                     &quot;CHR_HSCHR22_1_CTG6&quot;                     &quot;CHR_HSCHR22_1_CTG7&quot;                    </span></span>
<span id="cb22-149"><a href="#cb22-149" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [213] &quot;CHR_HSCHR22_2_CTG1&quot;                     &quot;CHR_HSCHR22_3_CTG1&quot;                     &quot;CHR_HSCHR22_4_CTG1&quot;                     &quot;CHR_HSCHR22_5_CTG1&quot;                    </span></span>
<span id="cb22-150"><a href="#cb22-150" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [217] &quot;CHR_HSCHR22_6_CTG1&quot;                     &quot;CHR_HSCHR22_7_CTG1&quot;                     &quot;CHR_HSCHR22_8_CTG1&quot;                     &quot;CHR_HSCHR2_1_CTG1&quot;                     </span></span>
<span id="cb22-151"><a href="#cb22-151" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [221] &quot;CHR_HSCHR2_1_CTG15&quot;                     &quot;CHR_HSCHR2_1_CTG5&quot;                      &quot;CHR_HSCHR2_1_CTG7&quot;                      &quot;CHR_HSCHR2_1_CTG7_2&quot;                   </span></span>
<span id="cb22-152"><a href="#cb22-152" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [225] &quot;CHR_HSCHR2_2_CTG1&quot;                      &quot;CHR_HSCHR2_2_CTG15&quot;                     &quot;CHR_HSCHR2_2_CTG7&quot;                      &quot;CHR_HSCHR2_2_CTG7_2&quot;                   </span></span>
<span id="cb22-153"><a href="#cb22-153" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [229] &quot;CHR_HSCHR2_3_CTG1&quot;                      &quot;CHR_HSCHR2_3_CTG15&quot;                     &quot;CHR_HSCHR2_3_CTG7_2&quot;                    &quot;CHR_HSCHR2_4_CTG1&quot;                     </span></span>
<span id="cb22-154"><a href="#cb22-154" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [233] &quot;CHR_HSCHR2_6_CTG7_2&quot;                    &quot;CHR_HSCHR3_1_CTG1&quot;                      &quot;CHR_HSCHR3_1_CTG2_1&quot;                    &quot;CHR_HSCHR3_1_CTG3&quot;                     </span></span>
<span id="cb22-155"><a href="#cb22-155" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [237] &quot;CHR_HSCHR3_2_CTG2_1&quot;                    &quot;CHR_HSCHR3_2_CTG3&quot;                      &quot;CHR_HSCHR3_3_CTG1&quot;                      &quot;CHR_HSCHR3_3_CTG3&quot;                     </span></span>
<span id="cb22-156"><a href="#cb22-156" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [241] &quot;CHR_HSCHR3_4_CTG2_1&quot;                    &quot;CHR_HSCHR3_4_CTG3&quot;                      &quot;CHR_HSCHR3_5_CTG2_1&quot;                    &quot;CHR_HSCHR3_5_CTG3&quot;                     </span></span>
<span id="cb22-157"><a href="#cb22-157" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [245] &quot;CHR_HSCHR3_6_CTG3&quot;                      &quot;CHR_HSCHR3_7_CTG3&quot;                      &quot;CHR_HSCHR3_8_CTG3&quot;                      &quot;CHR_HSCHR3_9_CTG3&quot;                     </span></span>
<span id="cb22-158"><a href="#cb22-158" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [249] &quot;CHR_HSCHR4_11_CTG12&quot;                    &quot;CHR_HSCHR4_1_CTG12&quot;                     &quot;CHR_HSCHR4_1_CTG4&quot;                      &quot;CHR_HSCHR4_1_CTG6&quot;                     </span></span>
<span id="cb22-159"><a href="#cb22-159" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [253] &quot;CHR_HSCHR4_1_CTG9&quot;                      &quot;CHR_HSCHR4_2_CTG12&quot;                     &quot;CHR_HSCHR4_2_CTG4&quot;                      &quot;CHR_HSCHR4_3_CTG12&quot;                    </span></span>
<span id="cb22-160"><a href="#cb22-160" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [257] &quot;CHR_HSCHR4_4_CTG12&quot;                     &quot;CHR_HSCHR4_5_CTG12&quot;                     &quot;CHR_HSCHR4_6_CTG12&quot;                     &quot;CHR_HSCHR4_7_CTG12&quot;                    </span></span>
<span id="cb22-161"><a href="#cb22-161" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [261] &quot;CHR_HSCHR4_8_CTG12&quot;                     &quot;CHR_HSCHR4_9_CTG12&quot;                     &quot;CHR_HSCHR5_1_CTG1&quot;                      &quot;CHR_HSCHR5_1_CTG1_1&quot;                   </span></span>
<span id="cb22-162"><a href="#cb22-162" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [265] &quot;CHR_HSCHR5_1_CTG5&quot;                      &quot;CHR_HSCHR5_2_CTG1&quot;                      &quot;CHR_HSCHR5_2_CTG1_1&quot;                    &quot;CHR_HSCHR5_2_CTG5&quot;                     </span></span>
<span id="cb22-163"><a href="#cb22-163" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [269] &quot;CHR_HSCHR5_3_CTG1&quot;                      &quot;CHR_HSCHR5_3_CTG5&quot;                      &quot;CHR_HSCHR5_4_CTG1&quot;                      &quot;CHR_HSCHR5_4_CTG1_1&quot;                   </span></span>
<span id="cb22-164"><a href="#cb22-164" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [273] &quot;CHR_HSCHR5_5_CTG1&quot;                      &quot;CHR_HSCHR5_6_CTG1&quot;                      &quot;CHR_HSCHR5_7_CTG1&quot;                      &quot;CHR_HSCHR6_1_CTG10&quot;                    </span></span>
<span id="cb22-165"><a href="#cb22-165" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [277] &quot;CHR_HSCHR6_1_CTG2&quot;                      &quot;CHR_HSCHR6_1_CTG3&quot;                      &quot;CHR_HSCHR6_1_CTG4&quot;                      &quot;CHR_HSCHR6_1_CTG5&quot;                     </span></span>
<span id="cb22-166"><a href="#cb22-166" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [281] &quot;CHR_HSCHR6_1_CTG6&quot;                      &quot;CHR_HSCHR6_1_CTG7&quot;                      &quot;CHR_HSCHR6_1_CTG8&quot;                      &quot;CHR_HSCHR6_1_CTG9&quot;                     </span></span>
<span id="cb22-167"><a href="#cb22-167" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [285] &quot;CHR_HSCHR6_8_CTG1&quot;                      &quot;CHR_HSCHR6_MHC_APD_CTG1&quot;                &quot;CHR_HSCHR6_MHC_COX_CTG1&quot;                &quot;CHR_HSCHR6_MHC_DBB_CTG1&quot;               </span></span>
<span id="cb22-168"><a href="#cb22-168" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [289] &quot;CHR_HSCHR6_MHC_MANN_CTG1&quot;               &quot;CHR_HSCHR6_MHC_MCF_CTG1&quot;                &quot;CHR_HSCHR6_MHC_QBL_CTG1&quot;                &quot;CHR_HSCHR6_MHC_SSTO_CTG1&quot;              </span></span>
<span id="cb22-169"><a href="#cb22-169" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [293] &quot;CHR_HSCHR7_1_CTG1&quot;                      &quot;CHR_HSCHR7_1_CTG4_4&quot;                    &quot;CHR_HSCHR7_1_CTG6&quot;                      &quot;CHR_HSCHR7_1_CTG7&quot;                     </span></span>
<span id="cb22-170"><a href="#cb22-170" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [297] &quot;CHR_HSCHR7_2_CTG1&quot;                      &quot;CHR_HSCHR7_2_CTG4_4&quot;                    &quot;CHR_HSCHR7_2_CTG6&quot;                      &quot;CHR_HSCHR7_2_CTG7&quot;                     </span></span>
<span id="cb22-171"><a href="#cb22-171" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [301] &quot;CHR_HSCHR7_3_CTG6&quot;                      &quot;CHR_HSCHR8_1_CTG1&quot;                      &quot;CHR_HSCHR8_1_CTG6&quot;                      &quot;CHR_HSCHR8_1_CTG7&quot;                     </span></span>
<span id="cb22-172"><a href="#cb22-172" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [305] &quot;CHR_HSCHR8_2_CTG1&quot;                      &quot;CHR_HSCHR8_2_CTG7&quot;                      &quot;CHR_HSCHR8_3_CTG1&quot;                      &quot;CHR_HSCHR8_3_CTG7&quot;                     </span></span>
<span id="cb22-173"><a href="#cb22-173" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [309] &quot;CHR_HSCHR8_4_CTG1&quot;                      &quot;CHR_HSCHR8_4_CTG7&quot;                      &quot;CHR_HSCHR8_5_CTG1&quot;                      &quot;CHR_HSCHR8_5_CTG7&quot;                     </span></span>
<span id="cb22-174"><a href="#cb22-174" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [313] &quot;CHR_HSCHR8_6_CTG1&quot;                      &quot;CHR_HSCHR8_7_CTG1&quot;                      &quot;CHR_HSCHR8_8_CTG1&quot;                      &quot;CHR_HSCHR8_9_CTG1&quot;                     </span></span>
<span id="cb22-175"><a href="#cb22-175" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [317] &quot;CHR_HSCHR9_1_CTG1&quot;                      &quot;CHR_HSCHR9_1_CTG2&quot;                      &quot;CHR_HSCHR9_1_CTG3&quot;                      &quot;CHR_HSCHR9_1_CTG4&quot;                     </span></span>
<span id="cb22-176"><a href="#cb22-176" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [321] &quot;CHR_HSCHR9_1_CTG5&quot;                      &quot;CHR_HSCHR9_1_CTG6&quot;                      &quot;CHR_HSCHRX_1_CTG3&quot;                      &quot;CHR_HSCHRX_2_CTG12&quot;                    </span></span>
<span id="cb22-177"><a href="#cb22-177" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [325] &quot;CHR_HSCHRX_2_CTG3&quot;                      &quot;GL000009.2&quot;                             &quot;GL000194.1&quot;                             &quot;GL000195.1&quot;                            </span></span>
<span id="cb22-178"><a href="#cb22-178" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [329] &quot;GL000205.2&quot;                             &quot;GL000213.1&quot;                             &quot;GL000216.2&quot;                             &quot;GL000218.1&quot;                            </span></span>
<span id="cb22-179"><a href="#cb22-179" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [333] &quot;GL000219.1&quot;                             &quot;GL000220.1&quot;                             &quot;GL000225.1&quot;                             &quot;KI270442.1&quot;                            </span></span>
<span id="cb22-180"><a href="#cb22-180" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [337] &quot;KI270711.1&quot;                             &quot;KI270713.1&quot;                             &quot;KI270721.1&quot;                             &quot;KI270726.1&quot;                            </span></span>
<span id="cb22-181"><a href="#cb22-181" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [341] &quot;KI270727.1&quot;                             &quot;KI270728.1&quot;                             &quot;KI270731.1&quot;                             &quot;KI270733.1&quot;                            </span></span>
<span id="cb22-182"><a href="#cb22-182" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [345] &quot;KI270734.1&quot;                             &quot;KI270744.1&quot;                             &quot;KI270750.1&quot;                             &quot;LRG_183&quot;                               </span></span>
<span id="cb22-183"><a href="#cb22-183" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [349] &quot;LRG_187&quot;                                &quot;LRG_239&quot;                                &quot;LRG_311&quot;                                &quot;LRG_721&quot;                               </span></span>
<span id="cb22-184"><a href="#cb22-184" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [353] &quot;LRG_741&quot;                                &quot;LRG_93&quot;                                 &quot;MT&quot;                                     &quot;X&quot;                                     </span></span>
<span id="cb22-185"><a href="#cb22-185" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [357] &quot;Y&quot;</span></span>
<span id="cb22-186"><a href="#cb22-186" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">updateEnsDb</span>(x)</span>
<span id="cb22-187"><a href="#cb22-187" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; EnsDb for Ensembl:</span></span>
<span id="cb22-188"><a href="#cb22-188" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Backend: SQLite</span></span>
<span id="cb22-189"><a href="#cb22-189" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Db type: EnsDb</span></span>
<span id="cb22-190"><a href="#cb22-190" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Type of Gene ID: Ensembl Gene ID</span></span>
<span id="cb22-191"><a href="#cb22-191" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Supporting package: ensembldb</span></span>
<span id="cb22-192"><a href="#cb22-192" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Db created by: ensembldb package from Bioconductor</span></span>
<span id="cb22-193"><a href="#cb22-193" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |script_version: 0.3.0</span></span>
<span id="cb22-194"><a href="#cb22-194" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Creation time: Thu May 18 16:32:27 2017</span></span>
<span id="cb22-195"><a href="#cb22-195" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |ensembl_version: 86</span></span>
<span id="cb22-196"><a href="#cb22-196" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |ensembl_host: localhost</span></span>
<span id="cb22-197"><a href="#cb22-197" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Organism: homo_sapiens</span></span>
<span id="cb22-198"><a href="#cb22-198" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |taxonomy_id: 9606</span></span>
<span id="cb22-199"><a href="#cb22-199" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |genome_build: GRCh38</span></span>
<span id="cb22-200"><a href="#cb22-200" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |DBSCHEMAVERSION: 2.0</span></span>
<span id="cb22-201"><a href="#cb22-201" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; | No. of genes: 63970.</span></span>
<span id="cb22-202"><a href="#cb22-202" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; | No. of transcripts: 216741.</span></span>
<span id="cb22-203"><a href="#cb22-203" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; |Protein data available.</span></span>
<span id="cb22-204"><a href="#cb22-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-205"><a href="#cb22-205" aria-hidden="true" tabindex="-1"></a>ensembldb<span class="sc">::</span><span class="fu">genes</span>(x, <span class="at">columns=</span><span class="fu">c</span>(<span class="st">&quot;gene_name&quot;</span>),</span>
<span id="cb22-206"><a href="#cb22-206" aria-hidden="true" tabindex="-1"></a>             <span class="at">filter=</span><span class="fu">list</span>(<span class="fu">SeqNameFilter</span>(<span class="st">&quot;X&quot;</span>), AnnotationFilter<span class="sc">::</span><span class="fu">GeneBiotypeFilter</span>(<span class="st">&quot;protein_coding&quot;</span>)))</span>
<span id="cb22-207"><a href="#cb22-207" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; GRanges object with 841 ranges and 3 metadata columns:</span></span>
<span id="cb22-208"><a href="#cb22-208" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                   seqnames              ranges strand |   gene_name         gene_id   gene_biotype</span></span>
<span id="cb22-209"><a href="#cb22-209" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;     &lt;character&gt;    &lt;character&gt;</span></span>
<span id="cb22-210"><a href="#cb22-210" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000182378        X       276322-303356      + |      PLCXD1 ENSG00000182378 protein_coding</span></span>
<span id="cb22-211"><a href="#cb22-211" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000178605        X       304529-318819      - |      GTPBP6 ENSG00000178605 protein_coding</span></span>
<span id="cb22-212"><a href="#cb22-212" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000167393        X       333963-386955      - |     PPP2R3B ENSG00000167393 protein_coding</span></span>
<span id="cb22-213"><a href="#cb22-213" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000185960        X       624344-659411      + |        SHOX ENSG00000185960 protein_coding</span></span>
<span id="cb22-214"><a href="#cb22-214" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000205755        X     1187549-1212750      - |       CRLF2 ENSG00000205755 protein_coding</span></span>
<span id="cb22-215"><a href="#cb22-215" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;               ...      ...                 ...    ... .         ...             ...            ...</span></span>
<span id="cb22-216"><a href="#cb22-216" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000277745        X 155459415-155460005      - |      H2AFB3 ENSG00000277745 protein_coding</span></span>
<span id="cb22-217"><a href="#cb22-217" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000185973        X 155490115-155669944      - |       TMLHE ENSG00000185973 protein_coding</span></span>
<span id="cb22-218"><a href="#cb22-218" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000168939        X 155767812-155782459      + |       SPRY3 ENSG00000168939 protein_coding</span></span>
<span id="cb22-219"><a href="#cb22-219" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000124333        X 155881293-155943769      + |       VAMP7 ENSG00000124333 protein_coding</span></span>
<span id="cb22-220"><a href="#cb22-220" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENSG00000124334        X 155997581-156010817      + |        IL9R ENSG00000124334 protein_coding</span></span>
<span id="cb22-221"><a href="#cb22-221" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   -------</span></span>
<span id="cb22-222"><a href="#cb22-222" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   seqinfo: 1 sequence from GRCh38 genome</span></span>
<span id="cb22-223"><a href="#cb22-223" aria-hidden="true" tabindex="-1"></a>ensembldb <span class="sc">::</span><span class="fu">transcripts</span>(x, <span class="at">columns=</span>ensembldb<span class="sc">::</span><span class="fu">listColumns</span>(x, <span class="st">&quot;tx&quot;</span>),</span>
<span id="cb22-224"><a href="#cb22-224" aria-hidden="true" tabindex="-1"></a>                        <span class="at">filter =</span> AnnotationFilter<span class="sc">::</span><span class="fu">AnnotationFilterList</span>(), <span class="at">order.type =</span> <span class="st">&quot;asc&quot;</span>, <span class="at">return.type =</span> <span class="st">&quot;GRanges&quot;</span>)</span>
<span id="cb22-225"><a href="#cb22-225" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; GRanges object with 216741 ranges and 6 metadata columns:</span></span>
<span id="cb22-226"><a href="#cb22-226" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                   seqnames            ranges strand |           tx_id             tx_biotype tx_cds_seq_start tx_cds_seq_end         gene_id         tx_name</span></span>
<span id="cb22-227"><a href="#cb22-227" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                      &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;            &lt;character&gt;        &lt;integer&gt;      &lt;integer&gt;     &lt;character&gt;     &lt;character&gt;</span></span>
<span id="cb22-228"><a href="#cb22-228" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000456328        1       11869-14409      + | ENST00000456328   processed_transcript             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000456328</span></span>
<span id="cb22-229"><a href="#cb22-229" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000450305        1       12010-13670      + | ENST00000450305 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000450305</span></span>
<span id="cb22-230"><a href="#cb22-230" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000488147        1       14404-29570      - | ENST00000488147 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000227232 ENST00000488147</span></span>
<span id="cb22-231"><a href="#cb22-231" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000619216        1       17369-17436      - | ENST00000619216                  miRNA             &lt;NA&gt;           &lt;NA&gt; ENSG00000278267 ENST00000619216</span></span>
<span id="cb22-232"><a href="#cb22-232" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000473358        1       29554-31097      + | ENST00000473358                lincRNA             &lt;NA&gt;           &lt;NA&gt; ENSG00000243485 ENST00000473358</span></span>
<span id="cb22-233"><a href="#cb22-233" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;               ...      ...               ...    ... .             ...                    ...              ...            ...             ...             ...</span></span>
<span id="cb22-234"><a href="#cb22-234" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000420810        Y 26549425-26549743      + | ENST00000420810   processed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000224240 ENST00000420810</span></span>
<span id="cb22-235"><a href="#cb22-235" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000456738        Y 26586642-26591601      - | ENST00000456738 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000227629 ENST00000456738</span></span>
<span id="cb22-236"><a href="#cb22-236" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000435945        Y 26594851-26634652      - | ENST00000435945 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000237917 ENST00000435945</span></span>
<span id="cb22-237"><a href="#cb22-237" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000435741        Y 26626520-26627159      - | ENST00000435741   processed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000231514 ENST00000435741</span></span>
<span id="cb22-238"><a href="#cb22-238" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ENST00000431853        Y 56855244-56855488      + | ENST00000431853   processed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000235857 ENST00000431853</span></span>
<span id="cb22-239"><a href="#cb22-239" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   -------</span></span>
<span id="cb22-240"><a href="#cb22-240" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   seqinfo: 357 sequences (1 circular) from GRCh38 genome</span></span>
<span id="cb22-241"><a href="#cb22-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-242"><a href="#cb22-242" aria-hidden="true" tabindex="-1"></a>txdbEnsemblGRCh38 <span class="ot">&lt;-</span> GenomicFeatures<span class="sc">::</span><span class="fu">makeTxDbFromEnsembl</span>(<span class="at">organism=</span><span class="st">&quot;Homo sapiens&quot;</span>, <span class="at">release=</span><span class="dv">98</span>)</span>
<span id="cb22-243"><a href="#cb22-243" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in download.file(url, destfile, method, quiet = TRUE): downloaded length 57402 != reported length 0</span></span>
<span id="cb22-244"><a href="#cb22-244" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fetch transcripts and genes from Ensembl ... OK</span></span>
<span id="cb22-245"><a href="#cb22-245" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   (fetched 250194 transcripts from 67946 genes)</span></span>
<span id="cb22-246"><a href="#cb22-246" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fetch exons and CDS from Ensembl ... OK</span></span>
<span id="cb22-247"><a href="#cb22-247" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Fetch chromosome names and lengths from Ensembl ...OK</span></span>
<span id="cb22-248"><a href="#cb22-248" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Gather the metadata ... OK</span></span>
<span id="cb22-249"><a href="#cb22-249" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Make the TxDb object ... OK</span></span>
<span id="cb22-250"><a href="#cb22-250" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> <span class="fu">as.list</span>(txdbEnsemblGRCh38)</span>
<span id="cb22-251"><a href="#cb22-251" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(txdb,head)</span>
<span id="cb22-252"><a href="#cb22-252" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $transcripts</span></span>
<span id="cb22-253"><a href="#cb22-253" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   tx_id         tx_name        tx_type        tx_chrom tx_strand tx_start  tx_end</span></span>
<span id="cb22-254"><a href="#cb22-254" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1     1 ENST00000636745         lncRNA CHR_HG107_PATCH         +  1049876 1055745</span></span>
<span id="cb22-255"><a href="#cb22-255" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2     2 ENST00000636387         lncRNA CHR_HG107_PATCH         +  1052607 1055745</span></span>
<span id="cb22-256"><a href="#cb22-256" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3     3 ENST00000643422 protein_coding CHR_HG107_PATCH         +  1075018 1112365</span></span>
<span id="cb22-257"><a href="#cb22-257" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4     4 ENST00000645631 protein_coding CHR_HG107_PATCH         +  1075018 1112365</span></span>
<span id="cb22-258"><a href="#cb22-258" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5     5 ENST00000636567 protein_coding CHR_HG107_PATCH         +  1159911 1203106</span></span>
<span id="cb22-259"><a href="#cb22-259" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6     6 ENST00000636545 protein_coding CHR_HG107_PATCH         -  1012823 1036718</span></span>
<span id="cb22-260"><a href="#cb22-260" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-261"><a href="#cb22-261" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $splicings</span></span>
<span id="cb22-262"><a href="#cb22-262" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   tx_id exon_rank exon_id       exon_name      exon_chrom exon_strand exon_start exon_end cds_id        cds_name cds_start cds_end</span></span>
<span id="cb22-263"><a href="#cb22-263" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1     1         1       1 ENSE00003797146 CHR_HG107_PATCH           +    1049876  1049958     NA            &lt;NA&gt;        NA      NA</span></span>
<span id="cb22-264"><a href="#cb22-264" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2     1         2       2 ENSE00003795151 CHR_HG107_PATCH           +    1051619  1051839     NA            &lt;NA&gt;        NA      NA</span></span>
<span id="cb22-265"><a href="#cb22-265" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3     1         3       4 ENSE00003793692 CHR_HG107_PATCH           +    1054235  1054388     NA            &lt;NA&gt;        NA      NA</span></span>
<span id="cb22-266"><a href="#cb22-266" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4     1         4       5 ENSE00003797325 CHR_HG107_PATCH           +    1055110  1055745     NA            &lt;NA&gt;        NA      NA</span></span>
<span id="cb22-267"><a href="#cb22-267" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5     2         1       3 ENSE00003798310 CHR_HG107_PATCH           +    1052607  1055745     NA            &lt;NA&gt;        NA      NA</span></span>
<span id="cb22-268"><a href="#cb22-268" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6     3         1       6 ENSE00003815958 CHR_HG107_PATCH           +    1075018  1075093      1 ENSP00000494473   1075018 1075093</span></span>
<span id="cb22-269"><a href="#cb22-269" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-270"><a href="#cb22-270" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $genes</span></span>
<span id="cb22-271"><a href="#cb22-271" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   tx_id         gene_id</span></span>
<span id="cb22-272"><a href="#cb22-272" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1     1 ENSG00000283640</span></span>
<span id="cb22-273"><a href="#cb22-273" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2     2 ENSG00000283640</span></span>
<span id="cb22-274"><a href="#cb22-274" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3     3 ENSG00000284971</span></span>
<span id="cb22-275"><a href="#cb22-275" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4     4 ENSG00000284971</span></span>
<span id="cb22-276"><a href="#cb22-276" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5     5 ENSG00000283158</span></span>
<span id="cb22-277"><a href="#cb22-277" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6     6 ENSG00000283350</span></span>
<span id="cb22-278"><a href="#cb22-278" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb22-279"><a href="#cb22-279" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; $chrominfo</span></span>
<span id="cb22-280"><a href="#cb22-280" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;              chrom    length is_circular</span></span>
<span id="cb22-281"><a href="#cb22-281" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1  CHR_HG107_PATCH 135088590          NA</span></span>
<span id="cb22-282"><a href="#cb22-282" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2  CHR_HG109_PATCH  58617934          NA</span></span>
<span id="cb22-283"><a href="#cb22-283" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3  CHR_HG126_PATCH 198295908          NA</span></span>
<span id="cb22-284"><a href="#cb22-284" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4 CHR_HG1277_PATCH 133754853          NA</span></span>
<span id="cb22-285"><a href="#cb22-285" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5 CHR_HG1296_PATCH 190208697          NA</span></span>
<span id="cb22-286"><a href="#cb22-286" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6 CHR_HG1298_PATCH 190196285          NA</span></span>
<span id="cb22-287"><a href="#cb22-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-288"><a href="#cb22-288" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> TxDb.Hsapiens.UCSC.hg38.knownGene</span>
<span id="cb22-289"><a href="#cb22-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-290"><a href="#cb22-290" aria-hidden="true" tabindex="-1"></a><span class="co"># liverExprs &lt;- quantifyExpressionsFromBWs(txdb = txdb,BWfiles=,experimentalDesign=)</span></span></code></pre></div>
</div>
<div id="bionconductor-forum" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Bionconductor forum</h1>
<p>Web: <a href="https://support.bioconductor.org/">https://support.bioconductor.org/</a></p>
</div>
<div id="bioconductorcran-packages" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Bioconductor/CRAN packages</h1>
<table>
<colgroup>
<col width="40%" />
<col width="60%" />
</colgroup>
<thead>
<tr class="header">
<th>Package</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Bioconductor</strong></td>
<td></td>
</tr>
<tr class="even">
<td>AnnotationDbi</td>
<td>AnnotationDb objects and their progeny, methods etc.</td>
</tr>
<tr class="odd">
<td>Biobase</td>
<td>Base functions for Bioconductor</td>
</tr>
<tr class="even">
<td>Biostrings</td>
<td>Efficient manipulation of biological strings</td>
</tr>
<tr class="odd">
<td>clusterProfiler</td>
<td>Functional profiles for genes and gene clusters</td>
</tr>
<tr class="even">
<td>ComplexHeatmap</td>
<td>Make complex heatmaps</td>
</tr>
<tr class="odd">
<td>DESSeq2</td>
<td>Differential gene expression analysis based on the negative binomial distribution</td>
</tr>
<tr class="even">
<td>edgeR</td>
<td>Empirical analysis of digital gene expression</td>
</tr>
<tr class="odd">
<td>EnsDb.Hsapiens.v86</td>
<td>Exposes an annotation databases generated from Ensembl</td>
</tr>
<tr class="even">
<td>ensembldb</td>
<td>Retrieve annotation data from an Ensembl based package</td>
</tr>
<tr class="odd">
<td>FlowSorted.DLPFC.450k</td>
<td>Illumina HumanMethylation data on sorted frontal cortex cell populations</td>
</tr>
<tr class="even">
<td>graphite</td>
<td>GRAPH Interaction from pathway topological environment</td>
</tr>
<tr class="odd">
<td>IlluminaHumanMethylation450kmanifest</td>
<td>Annotation for Illuminaâ€™s 450k methylation arrays</td>
</tr>
<tr class="even">
<td>INSPEcT</td>
<td>Quantification of the intronic and exonic gene features and the post-transcriptional regulation analysis</td>
</tr>
<tr class="odd">
<td>org.Hs.eg.db</td>
<td>Conversion of Entrez ID â€“ gene symbols</td>
</tr>
<tr class="even">
<td>OUTRIDER</td>
<td>OUTlier in RNA-Seq fInDER</td>
</tr>
<tr class="odd">
<td>Pi</td>
<td>Priority index, leveraging genetic evidence to prioritise drug targets at the gene and pathway level</td>
</tr>
<tr class="even">
<td>quantro</td>
<td>A test for when to use quantile normalisation</td>
</tr>
<tr class="odd">
<td>recount3</td>
<td>Interface to uniformly processed RNA-seq data</td>
</tr>
<tr class="even">
<td>Rgraphiz</td>
<td>Interfaces R with the AT&amp;T graphviz library for plotting R graph objects from the graph package</td>
</tr>
<tr class="odd">
<td>sva</td>
<td>Surrogate Variable Analysis</td>
</tr>
<tr class="even">
<td>TxDb.Hsapiens.UCSC.hg38.knownGene</td>
<td>Annotation of the human genome</td>
</tr>
<tr class="odd">
<td><strong>CRAN</strong></td>
<td></td>
</tr>
<tr class="even">
<td>doParallel</td>
<td>Foreach Parallel Adaptor for the â€˜parallelâ€™ Package</td>
</tr>
<tr class="odd">
<td>GeneNet</td>
<td>Modeling and Inferring Gene Networks</td>
</tr>
<tr class="even">
<td>ggplot2</td>
<td>Data Visualisations Using the grammar of graphics</td>
</tr>
<tr class="odd">
<td>heatmaply</td>
<td>Interactive Cluster Heat Maps Using <code>plotly</code> and <code>ggplot2</code></td>
</tr>
<tr class="even">
<td>pheatmap</td>
<td>results visualisation</td>
</tr>
<tr class="odd">
<td>plyr</td>
<td>Splitting, applying and combining data</td>
</tr>
<tr class="even">
<td>RColorBrewer</td>
<td>ColorBrewer Palettes</td>
</tr>
<tr class="odd">
<td>WGCNA</td>
<td>Weighted correlation network analysis</td>
</tr>
</tbody>
</table>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-koster21" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">KÃ¶ster, J., Forster, J., Schmeier, S. &amp; Salazar, V. <span class="nocase">snakemake-workflows/rna-seq-star-deseq2: Version 1.2.0</span>. (2021) doi:<a href="https://doi.org/10.5281/zenodo.5245549">10.5281/zenodo.5245549</a>.</div>
</div>
<div id="ref-carlson16" class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Carlson, M. R. J., PagÃ¨s, H., Arora, S., Obenchain, V. &amp; Morgan, M. Genomic annotation resources in <span>R/Bioconductor</span>. in <em>Statistical genomics: Methods and protocols</em> (eds. MathÃ©, E. &amp; Davis, S.) 67â€“69 (Springer New York, New York, NY, 2016). doi:<a href="https://doi.org/10.1007/978-1-4939-3578-9_4">10.1007/978-1-4939-3578-9_4</a>.</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>



    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer></div>

  

  

  </body></html>
