<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An Overview of pQTLtools • pQTLtools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An Overview of pQTLtools">
<meta property="og:description" content="pQTLtools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pQTLtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jinghuazhao/pQTLtools">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An Overview of pQTLtools</h1>
                        <h4 class="author">Jing Hua Zhao</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/master/vignettes/pQTLtools.Rmd"><code>vignettes/pQTLtools.Rmd</code></a></small>
      <div class="hidden name"><code>pQTLtools.Rmd</code></div>

    </div>

    
    
<div id="cistrans-classification" class="section level2">
<h2 class="hasAnchor">
<a href="#cistrans-classification" class="anchor"></a>cis/trans classification</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/R">gap</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: gap</span>
<span class="co">#&gt; gap version 1.2.3-2</span>
<span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span>
<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"work"</span>,<span class="st">"INF1.merge"</span><span class="op">)</span>
<span class="va">clumped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">f</span>,as.is<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">clumped</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CHR"</span>,<span class="st">"POS"</span>,<span class="st">"MarkerName"</span>,<span class="st">"prot"</span>,<span class="st">"log10p"</span><span class="op">)</span><span class="op">]</span>,
              <span class="va">inf1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"uniprot"</span><span class="op">)</span><span class="op">]</span>,by<span class="op">=</span><span class="st">"prot"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"Chr"</span>,<span class="st">"bp"</span>,<span class="st">"SNP"</span>,<span class="st">"log10p"</span>,<span class="st">"uniprot"</span><span class="op">)</span>
<span class="va">cistrans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gap/man/gap-internal.html">cis.vs.trans.classification</a></span><span class="op">(</span><span class="va">hits</span>,<span class="va">inf1</span>,<span class="st">"uniprot"</span><span class="op">)</span>
<span class="va">cis.vs.trans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">cistrans</span>,<span class="va">data</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">cistrans</span>,<span class="va">table</span><span class="op">)</span>
<span class="co">#&gt;           cis trans total</span>
<span class="co">#&gt; ADA         1     0     1</span>
<span class="co">#&gt; CASP8       1     0     1</span>
<span class="co">#&gt; CCL11       1     4     5</span>
<span class="co">#&gt; CCL13       1     3     4</span>
<span class="co">#&gt; CCL19       1     3     4</span>
<span class="co">#&gt; CCL2        0     3     3</span>
<span class="co">#&gt; CCL20       1     1     2</span>
<span class="co">#&gt; CCL23       1     0     1</span>
<span class="co">#&gt; CCL25       1     3     4</span>
<span class="co">#&gt; CCL3        1     1     2</span>
<span class="co">#&gt; CCL4        1     1     2</span>
<span class="co">#&gt; CCL7        1     2     3</span>
<span class="co">#&gt; CCL8        1     1     2</span>
<span class="co">#&gt; CD244       1     2     3</span>
<span class="co">#&gt; CD274       1     0     1</span>
<span class="co">#&gt; CD40        1     0     1</span>
<span class="co">#&gt; CD5         1     2     3</span>
<span class="co">#&gt; CD6         1     1     2</span>
<span class="co">#&gt; CDCP1       1     2     3</span>
<span class="co">#&gt; CSF1        1     0     1</span>
<span class="co">#&gt; CST5        1     3     4</span>
<span class="co">#&gt; CX3CL1      1     2     3</span>
<span class="co">#&gt; CXCL1       1     0     1</span>
<span class="co">#&gt; CXCL10      1     1     2</span>
<span class="co">#&gt; CXCL11      1     3     4</span>
<span class="co">#&gt; CXCL5       1     3     4</span>
<span class="co">#&gt; CXCL6       1     1     2</span>
<span class="co">#&gt; CXCL9       1     2     3</span>
<span class="co">#&gt; DNER        1     0     1</span>
<span class="co">#&gt; EIF4EBP1    0     1     1</span>
<span class="co">#&gt; FGF19       0     3     3</span>
<span class="co">#&gt; FGF21       1     2     3</span>
<span class="co">#&gt; FGF23       0     2     2</span>
<span class="co">#&gt; FGF5        1     0     1</span>
<span class="co">#&gt; FLT3LG      0     6     6</span>
<span class="co">#&gt; GDNF        1     0     1</span>
<span class="co">#&gt; HGF         1     1     2</span>
<span class="co">#&gt; IL10        1     2     3</span>
<span class="co">#&gt; IL10RB      1     1     2</span>
<span class="co">#&gt; IL12B       1     7     8</span>
<span class="co">#&gt; IL15RA      1     0     1</span>
<span class="co">#&gt; IL17C       1     0     1</span>
<span class="co">#&gt; IL18        1     1     2</span>
<span class="co">#&gt; IL18R1      1     1     2</span>
<span class="co">#&gt; IL1A        0     1     1</span>
<span class="co">#&gt; IL6         0     1     1</span>
<span class="co">#&gt; IL7         1     0     1</span>
<span class="co">#&gt; IL8         1     0     1</span>
<span class="co">#&gt; KITLG       0     7     7</span>
<span class="co">#&gt; LIFR        0     1     1</span>
<span class="co">#&gt; LTA         1     2     3</span>
<span class="co">#&gt; MMP1        1     2     3</span>
<span class="co">#&gt; MMP10       1     1     2</span>
<span class="co">#&gt; NGF         1     1     2</span>
<span class="co">#&gt; NTF3        0     1     1</span>
<span class="co">#&gt; OSM         0     2     2</span>
<span class="co">#&gt; PLAU        1     5     6</span>
<span class="co">#&gt; S100A12     1     0     1</span>
<span class="co">#&gt; SIRT2       1     0     1</span>
<span class="co">#&gt; SLAMF1      1     4     5</span>
<span class="co">#&gt; SULT1A1     1     1     2</span>
<span class="co">#&gt; TGFA        1     0     1</span>
<span class="co">#&gt; TGFB1       1     0     1</span>
<span class="co">#&gt; TNFRSF11B   1     1     2</span>
<span class="co">#&gt; TNFRSF9     1     1     2</span>
<span class="co">#&gt; TNFSF10     1     7     8</span>
<span class="co">#&gt; TNFSF11     1     5     6</span>
<span class="co">#&gt; TNFSF12     1     4     5</span>
<span class="co">#&gt; TNFSF14     1     0     1</span>
<span class="co">#&gt; VEGFA       1     3     4</span>
<span class="co">#&gt; total      59   121   180</span>
<span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">cistrans</span>,<span class="va">total</span><span class="op">)</span>
<span class="co">#&gt; [1] 180</span>
<span class="fu"><a href="https://rdrr.io/pkg/gap/man/gap-internal.html">circos.cis.vs.trans.plot</a></span><span class="op">(</span>hits<span class="op">=</span><span class="va">f</span>,<span class="va">inf1</span>,<span class="st">"uniprot"</span><span class="op">)</span></code></pre></div>
<p><img src="pQTLtools/unnamed-chunk-1-1.png" width="700" style="display:block; margin: auto"> This is used to generate 2d-Manhattan plot</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"work"</span>,<span class="st">"INF1.merge.cis.vs.trans"</span><span class="op">)</span>,as.is<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">t2d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gap/man/gap-internal.html">mhtplot2d</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></code></pre></div>
<p><img src="pQTLtools/unnamed-chunk-2-1.png" width="700" style="display:block; margin: auto"> and ideogram</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span>
<span class="va">INF1_merge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"work"</span>,<span class="st">"INF1.merge"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Chrom"</span>,<span class="st">"Start"</span>,<span class="st">"End"</span>,<span class="st">"prot"</span>,<span class="st">"MarkerName"</span><span class="op">)</span><span class="op">]</span>
<span class="va">singletons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">INF1_merge</span>,<span class="va">End</span><span class="op">-</span><span class="va">Start</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span>
<span class="va">INF1_merge</span><span class="op">[</span><span class="va">singletons</span>,<span class="st">"Start"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">INF1_merge</span><span class="op">[</span><span class="va">singletons</span>,<span class="st">"Start"</span><span class="op">]</span> <span class="op">-</span> <span class="fl">1e6</span>
<span class="va">INF1_merge</span><span class="op">[</span><span class="va">singletons</span>,<span class="st">"End"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">INF1_merge</span><span class="op">[</span><span class="va">singletons</span>,<span class="st">"End"</span><span class="op">]</span> <span class="op">+</span> <span class="fl">1e6</span>
<span class="va">small</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">INF1_merge</span>, <span class="va">Start</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span>
<span class="va">INF1_merge</span><span class="op">[</span><span class="va">small</span>,<span class="st">"Start"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>
<span class="va">cvt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"work"</span>,<span class="st">"/INF1.merge.cis.vs.trans"</span><span class="op">)</span>,as.is<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">INF1_merge_cvt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">INF1_merge</span>,<span class="va">cvt</span>,by.x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"MarkerName"</span><span class="op">)</span>,by.y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"SNP"</span><span class="op">)</span><span class="op">)</span>
<span class="va">ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">INF1_merge_cvt</span>,<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="va">bp</span><span class="op">)</span><span class="op">)</span>
<span class="va">INF1_merge_cvt</span> <span class="op">&lt;-</span> <span class="va">INF1_merge_cvt</span><span class="op">[</span><span class="va">ord</span>,<span class="op">]</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://httr.r-lib.org/">httr</a></span><span class="op">)</span>
<span class="fu"><a href="https://httr.r-lib.org/reference/set_config.html">set_config</a></span><span class="op">(</span><span class="fu"><a href="https://httr.r-lib.org/reference/config.html">config</a></span><span class="op">(</span>ssl_verifypeer <span class="op">=</span> <span class="fl">0L</span><span class="op">)</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomaRt</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">regioneR</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ, clusterExport, clusterMap, parApply, parCapply, parLapply, parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,</span>
<span class="co">#&gt;     Map, mapply, match, mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply, union, unique, unsplit,</span>
<span class="co">#&gt;     which, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html">useMart</a></span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"ensembl"</span>, dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span>
<span class="va">bm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">INF1_merge_cvt</span>, <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html">getBM</a></span><span class="op">(</span>attributes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"hgnc_symbol"</span>, <span class="st">"chromosome_name"</span>, <span class="st">"start_position"</span>, <span class="st">"end_position"</span>, <span class="st">"band"</span><span class="op">)</span>,
           filters<span class="op">=</span><span class="st">"chromosomal_region"</span>, values<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="st">":"</span>,<span class="va">bp</span>,<span class="st">":"</span>,<span class="va">bp</span><span class="op">)</span>, mart<span class="op">=</span><span class="va">mart</span><span class="op">)</span><span class="op">)</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">bm</span>,<span class="fu"><a href="https://rdrr.io/pkg/regioneR/man/toGRanges.html">toGRanges</a></span><span class="op">(</span><span class="va">chromosome_name</span>,<span class="va">start_position</span>,<span class="va">end_position</span>,labels<span class="op">=</span><span class="va">hgnc_symbol</span><span class="op">)</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/bernatgel/karyoploteR">karyoploteR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/attach.html">attach</a></span><span class="op">(</span><span class="va">INF1_merge_cvt</span><span class="op">)</span>
<span class="va">sentinels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regioneR/man/toGRanges.html">toGRanges</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="va">bp</span><span class="op">-</span><span class="fl">1</span>,<span class="va">bp</span>,labels<span class="op">=</span><span class="va">p.gene</span><span class="op">)</span>
<span class="va">cis.regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regioneR/man/toGRanges.html">toGRanges</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="va">cis.start</span>,<span class="va">cis.end</span><span class="op">)</span>
<span class="va">loci</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regioneR/man/toGRanges.html">toGRanges</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="va">Start</span>,<span class="va">End</span><span class="op">)</span>
<span class="va">panel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/regioneR/man/toGRanges.html">toGRanges</a></span><span class="op">(</span><span class="va">p.chr</span>,<span class="va">p.start</span>,<span class="va">p.end</span>,labels<span class="op">=</span><span class="va">p.gene</span><span class="op">)</span>
<span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"blue"</span><span class="op">)</span>
<span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">sentinels</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span>
<span class="va">kp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/karyoploteR/man/plotKaryotype.html">plotKaryotype</a></span><span class="op">(</span>genome<span class="op">=</span><span class="st">"hg19"</span>,chromosomes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">sentinels</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/karyoploteR/man/kpAddBaseNumbers.html">kpAddBaseNumbers</a></span><span class="op">(</span><span class="va">kp</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/karyoploteR/man/kpPlotRegions.html">kpPlotRegions</a></span><span class="op">(</span><span class="va">kp</span>,data<span class="op">=</span><span class="va">loci</span>,r0<span class="op">=</span><span class="fl">0.05</span>,r1<span class="op">=</span><span class="fl">0.15</span>,border<span class="op">=</span><span class="st">"black"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/karyoploteR/man/kpPlotMarkers.html">kpPlotMarkers</a></span><span class="op">(</span><span class="va">kp</span>, data<span class="op">=</span><span class="va">sentinels</span>, labels<span class="op">=</span><span class="va">p.gene</span>, text.orientation<span class="op">=</span><span class="st">"vertical"</span>,
              cex<span class="op">=</span><span class="fl">0.45</span>, y<span class="op">=</span><span class="fl">0.3</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_along</a></span><span class="op">(</span><span class="va">p.gene</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">p.gene</span><span class="op">)</span>, srt<span class="op">=</span><span class="fl">30</span>, ignore.chromosome.ends<span class="op">=</span><span class="cn">TRUE</span>,
              adjust.label.position<span class="op">=</span><span class="cn">TRUE</span>, label.color<span class="op">=</span><span class="va">colors</span><span class="op">[</span><span class="fl">2</span><span class="op">-</span><span class="va">cis</span><span class="op">]</span>, label.dist<span class="op">=</span><span class="fl">0.002</span>,
              cex.axis<span class="op">=</span><span class="fl">3</span>, cex.lab<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span><span class="op">(</span><span class="st">"bottomright"</span>, bty<span class="op">=</span><span class="st">"n"</span>, pch<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">19</span>,<span class="fl">19</span><span class="op">)</span>, col<span class="op">=</span><span class="va">colors</span>, pt.cex<span class="op">=</span><span class="fl">0.4</span>, legend<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cis"</span>, <span class="st">"trans"</span><span class="op">)</span>, text.col<span class="op">=</span><span class="va">colors</span>, cex<span class="op">=</span><span class="fl">0.8</span>, horiz<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="pQTLtools/unnamed-chunk-3-1.png" width="700" style="display:block; margin: auto"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># kpPlotLinks(kp, data=loci, data2=panel, col=colors[2-cis])</span>
<span class="kw"><a href="https://rdrr.io/r/base/detach.html">detach</a></span><span class="op">(</span><span class="va">INF1_merge_cvt</span><span class="op">)</span></code></pre></div>
</div>
<div id="eqtl-catalogue-for-colocalization-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#eqtl-catalogue-for-colocalization-analysis" class="anchor"></a>eQTL Catalogue for colocalization analysis</h2>
<p>See example associated with <code>import_eQTLCatalogue</code>. A related function is <code>import_OpenGWAS</code> used to fetch data from OpenGWAS.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/pQTLtools">pQTLtools</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'pQTLtools'</span>
<span class="co">#&gt; The following object is masked from 'package:gap':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     inf1</span>
<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html">path.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"eQTL-Catalogue"</span>,<span class="st">"hg19ToHg38.over.chain"</span><span class="op">)</span>
<span class="va">chain</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/Chain-class.html">import.chain</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dplyr"</span>, <span class="st">"ggplot2"</span>, <span class="st">"readr"</span>, <span class="st">"coloc"</span>, <span class="st">"GenomicRanges"</span>,<span class="st">"seqminer"</span><span class="op">)</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:GenomicRanges':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, union</span>
<span class="co">#&gt; The following object is masked from 'package:GenomeInfoDb':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect</span>
<span class="co">#&gt; The following objects are masked from 'package:IRanges':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse, desc, intersect, setdiff, slice, union</span>
<span class="co">#&gt; The following objects are masked from 'package:S4Vectors':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     first, intersect, rename, setdiff, setequal, union</span>
<span class="co">#&gt; The following objects are masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine, intersect, setdiff, union</span>
<span class="co">#&gt; The following object is masked from 'package:biomaRt':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     select</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Loading required package: readr</span>
<span class="co">#&gt; Loading required package: coloc</span>
<span class="co">#&gt; Loading required package: seqminer</span>
<span class="va">HPC_WORK</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html">Sys.getenv</a></span><span class="op">(</span><span class="st">"HPC_WORK"</span><span class="op">)</span>
<span class="fu">gwasvcf</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gwasvcf/man/set_bcftools.html">set_bcftools</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">HPC_WORK</span>,<span class="st">"bin"</span>,<span class="st">"bcftools"</span><span class="op">)</span><span class="op">)</span>
<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html">path.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"eQTL-Catalogue"</span>,<span class="st">"tabix_ftp_paths.tsv"</span><span class="op">)</span>
<span class="va">tabix_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">f</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">HOME</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html">Sys.getenv</a></span><span class="op">(</span><span class="st">"HOME"</span><span class="op">)</span>
<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html">path.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"eQTL-Catalogue"</span>,<span class="st">"tabix_ftp_paths_imported.tsv"</span><span class="op">)</span>
<span class="va">imported_tabix_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">within</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">f</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span>,
      <span class="op">{</span><span class="va">ftp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"ftp://ftp.ebi.ac.uk/pub/databases/spot/eQTL/csv/GTEx_V8/ge"</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">HOME</span>,<span class="st">"/rds/public_databases/GTEx/csv"</span><span class="op">)</span>,<span class="va">ftp_path</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>
<span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span>
<span class="va">M</span> <span class="op">&lt;-</span> <span class="fl">1e6</span>
<span class="va">sentinels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"work"</span>,<span class="st">"INF1.merge.cis.vs.trans"</span><span class="op">)</span><span class="op">)</span>,<span class="va">cis</span><span class="op">)</span>
<span class="va">liftRegion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">chain</span><span class="op">)</span>
<span class="op">{</span>
  <span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va">GenomicRanges</span><span class="op">)</span>
  <span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">x</span>,<span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="va">chr</span>,<span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html">IRanges</a></span><span class="op">(</span><span class="va">start</span>,<span class="va">end</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span>
  <span class="va">gr38</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/liftOver.html">liftOver</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">chain</span><span class="op">)</span>
  <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">gr38</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">""</span>,<span class="va">chr</span><span class="op">)</span>
  <span class="va">start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">gr38</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">end</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">gr38</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>chr<span class="op">=</span><span class="va">chr</span>,start<span class="op">=</span><span class="va">start</span>,end<span class="op">=</span><span class="va">end</span>,region<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">chr</span>,<span class="st">":"</span>,<span class="va">start</span>,<span class="st">"-"</span>,<span class="va">end</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">coloc_sumstats</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">prot</span>,<span class="va">chr</span>,<span class="va">region37</span><span class="op">)</span>
<span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"GWAS sumstats\n"</span><span class="op">)</span>
  <span class="va">gwas_stats</span> <span class="op">&lt;-</span> <span class="fu">gwasvcf</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gwasvcf/man/query_gwas.html">query_gwas</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"METAL/gwas2vcf"</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">".vcf.gz"</span><span class="op">)</span><span class="op">)</span>, chrompos <span class="op">=</span> <span class="va">region37</span><span class="op">)</span>
  <span class="va">gwas_stats</span> <span class="op">&lt;-</span> <span class="fu">gwasvcf</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gwasvcf/man/vcf_to_granges.html">vcf_to_granges</a></span><span class="op">(</span><span class="va">gwas_stats</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">keepSeqlevels</span><span class="op">(</span><span class="va">chr</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">renameSeqlevels</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="va">chr</span><span class="op">)</span><span class="op">)</span>
  <span class="va">gwas_stats_hg38</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/liftOver.html">liftOver</a></span><span class="op">(</span><span class="va">gwas_stats</span>, <span class="va">chain</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">renameSeqlevels</span><span class="op">(</span><span class="va">chr</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">transmute</a></span><span class="op">(</span>chromosome <span class="op">=</span> <span class="va">seqnames</span>, position <span class="op">=</span> <span class="va">start</span>, <span class="va">AF</span>, <span class="va">ES</span>, <span class="va">SE</span>, <span class="va">LP</span>, <span class="va">SS</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">chromosome</span>, <span class="va">position</span>, sep <span class="op">=</span> <span class="st">":"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>MAF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">pmin</a></span><span class="op">(</span><span class="va">AF</span>, <span class="fl">1</span><span class="op">-</span><span class="va">AF</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>row_count <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">row_count</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span>
  <span class="fu">ggplot</span><span class="op">(</span><span class="va">gwas_stats_hg38</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">position</span>, y <span class="op">=</span> <span class="va">LP</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span>
  <span class="va">gwas_stats_hg38</span>
<span class="op">}</span>

<span class="va">coloc_c</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gwas_stats_hg38</span>,<span class="va">ensGene</span>,<span class="va">region38</span><span class="op">)</span>
<span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"c. GTEx_v8 imported eQTL datasets\n"</span><span class="op">)</span>
  <span class="va">rnaseq_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">imported_tabix_paths</span>, <span class="va">quant_method</span> <span class="op">==</span> <span class="st">"ge"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>qtl_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">study</span>, <span class="va">qtl_group</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span>
  <span class="va">ftp_path_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">rnaseq_df</span><span class="op">$</span><span class="va">ftp_path</span><span class="op">)</span>, <span class="va">rnaseq_df</span><span class="op">$</span><span class="va">qtl_id</span><span class="op">)</span>
  <span class="va">hdr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html">path.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"eQTL-Catalogue"</span>,<span class="st">"column_names.GTEx"</span><span class="op">)</span>
  <span class="va">column_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">hdr</span><span class="op">)</span><span class="op">)</span>
  <span class="va">safe_import</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/safely.html">safely</a></span><span class="op">(</span><span class="va">import_eQTLCatalogue</span><span class="op">)</span>
  <span class="va">summary_list</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">ftp_path_list</span>, <span class="op">~</span><span class="fu">safe_import</span><span class="op">(</span><span class="va">.</span>, <span class="va">region38</span>, selected_gene_id <span class="op">=</span> <span class="va">ensGene</span>, <span class="va">column_names</span><span class="op">)</span><span class="op">)</span>
  <span class="va">result_list</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">summary_list</span>, <span class="op">~</span><span class="va">.</span><span class="op">$</span><span class="va">result</span><span class="op">)</span>
  <span class="va">result_list</span> <span class="op">&lt;-</span> <span class="va">result_list</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">result_list</span>, <span class="va">is.null</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
  <span class="va">result_filtered</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map</a></span><span class="op">(</span><span class="va">result_list</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">result_list</span>,<span class="va">nrow</span><span class="op">)</span><span class="op">!=</span><span class="fl">0</span><span class="op">]</span>, <span class="op">~</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">.</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="va">coloc_df_imported</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html">map_df</a></span><span class="op">(</span><span class="va">result_filtered</span>, <span class="op">~</span><span class="fu"><a href="../reference/pQTLtools-internal.html">run_coloc</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">gwas_stats_hg38</span><span class="op">)</span>, .id <span class="op">=</span> <span class="st">"qtl_id"</span><span class="op">)</span>
<span class="op">}</span>

<span class="va">mixed_coloc</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">prot</span>,<span class="va">chr</span>,<span class="va">ensGene</span>,<span class="va">chain</span>,<span class="va">region37</span>,<span class="va">region38</span>,<span class="va">out</span><span class="op">)</span>
<span class="op">{</span>
  <span class="va">gwas_stats_hg38</span> <span class="op">&lt;-</span> <span class="fu">coloc_sumstats</span><span class="op">(</span><span class="va">prot</span>,<span class="va">chr</span>,<span class="va">region37</span><span class="op">)</span>
  <span class="va">coloc_df_imported</span> <span class="op">&lt;-</span> <span class="fu">coloc_c</span><span class="op">(</span><span class="va">gwas_stats_hg38</span>,<span class="va">ensGene</span>,<span class="va">region38</span><span class="op">)</span>
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html">exists</a></span><span class="op">(</span><span class="st">"coloc_df_imported"</span><span class="op">)</span><span class="op">)</span>
  <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">coloc_df_imported</span>,file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">out</span>,<span class="st">".RDS"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">coloc_df_imported</span>, <span class="op">-</span><span class="va">PP.H4.abf</span><span class="op">)</span>
    <span class="fu">ggplot</span><span class="op">(</span><span class="va">coloc_df_imported</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">PP.H4.abf</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">}</span>

<span class="va">single_run</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">r</span><span class="op">)</span>
<span class="op">{</span>
  <span class="va">sentinel</span> <span class="op">&lt;-</span> <span class="va">sentinels</span><span class="op">[</span><span class="va">r</span>,<span class="op">]</span>
  <span class="va">isnpid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">within</a></span><span class="op">(</span><span class="fu">gap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gap/man/gap-internal.html">inv_chr_pos_a1_a2</a></span><span class="op">(</span><span class="va">sentinel</span><span class="op">[[</span><span class="st">"SNP"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,
  <span class="op">{</span>
    <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">""</span>,<span class="va">chr</span><span class="op">)</span>
    <span class="va">pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="va">pos</span><span class="op">)</span>
    <span class="va">start</span> <span class="op">&lt;-</span> <span class="va">pos</span><span class="op">-</span><span class="va">M</span>
    <span class="kw">if</span> <span class="op">(</span><span class="va">start</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span> <span class="va">start</span> <span class="op">&lt;-</span> <span class="fl">0</span>
    <span class="va">end</span> <span class="op">&lt;-</span> <span class="va">pos</span><span class="op">+</span><span class="va">M</span>
  <span class="op">}</span><span class="op">)</span>
  <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">isnpid</span>,<span class="va">chr</span><span class="op">)</span>
  <span class="va">region37</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">isnpid</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">chr</span>,<span class="st">":"</span>,<span class="va">start</span>,<span class="st">"-"</span>,<span class="va">end</span><span class="op">)</span><span class="op">)</span>
  <span class="va">ensRegion37</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">inf1</span>,<span class="va">prot</span><span class="op">==</span><span class="va">sentinel</span><span class="op">[[</span><span class="st">"prot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">chr</span>,<span class="st">":"</span>,<span class="va">start</span>,<span class="st">"-"</span>,<span class="va">end</span><span class="op">)</span><span class="op">)</span>
  <span class="va">region38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu">liftRegion</span><span class="op">(</span><span class="va">isnpid</span>,<span class="va">chain</span><span class="op">)</span>,<span class="va">region</span><span class="op">)</span>
  <span class="va">ensGene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">inf1</span>,<span class="va">prot</span><span class="op">==</span><span class="va">sentinel</span><span class="op">[[</span><span class="st">"prot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="st">"ensembl_gene_id"</span><span class="op">]</span><span class="op">]</span>
  <span class="va">ensRegion38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu">liftRegion</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">inf1</span>,<span class="va">prot</span><span class="op">==</span><span class="va">sentinel</span><span class="op">[[</span><span class="st">"prot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,<span class="va">chain</span><span class="op">)</span>,<span class="va">region</span><span class="op">)</span>
  <span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"coloc"</span>,<span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">sentinel</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">SNP</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="va">chr</span>,<span class="va">region37</span>,<span class="va">region38</span>,<span class="va">ensGene</span>,<span class="va">ensRegion37</span>,<span class="va">ensRegion38</span>,<span class="st">"\n"</span><span class="op">)</span>
  <span class="fu">mixed_coloc</span><span class="op">(</span><span class="va">sentinel</span><span class="op">[[</span><span class="st">"prot"</span><span class="op">]</span><span class="op">]</span>,<span class="va">chr</span>,<span class="va">ensGene</span>,<span class="va">chain</span>,<span class="va">region37</span>,<span class="va">region38</span>,<span class="va">f</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">single_run</span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="co">#&gt; 8 8:119081031-121081031 8:118068792-120068792 ENSG00000164761 8:119935796-119964439 8:118923557-118952200 </span>
<span class="co">#&gt; GWAS sumstats</span>
<span class="co">#&gt; c. GTEx_v8 imported eQTL datasets</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Adipose_Subcutaneous.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Adipose_Visceral_Omentum.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Adrenal_Gland.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Artery_Aorta.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Artery_Coronary.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Artery_Tibial.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Amygdala.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Anterior_cingulate_cortex_BA24.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Caudate_basal_ganglia.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Cerebellar_Hemisphere.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Cerebellum.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Cortex.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Frontal_Cortex_BA9.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Hippocampus.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Hypothalamus.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Nucleus_accumbens_basal_ganglia.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Putamen_basal_ganglia.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Spinal_cord_cervical_c-1.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Brain_Substantia_nigra.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Breast_Mammary_Tissue.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Cells_Cultured_fibroblasts.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Cells_EBV-transformed_lymphocytes.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Colon_Sigmoid.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Colon_Transverse.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Esophagus_Gastroesophageal_Junction.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Esophagus_Mucosa.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Esophagus_Muscularis.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Heart_Atrial_Appendage.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Heart_Left_Ventricle.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Kidney_Cortex.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Liver.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Lung.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Minor_Salivary_Gland.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Muscle_Skeletal.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Nerve_Tibial.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Ovary.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Pancreas.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Pituitary.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Prostate.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Skin_Not_Sun_Exposed_Suprapubic.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Skin_Sun_Exposed_Lower_leg.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Small_Intestine_Terminal_Ileum.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Spleen.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Stomach.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Testis.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Thyroid.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Uterus.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Vagina.tsv.gz"</span>
<span class="co">#&gt; [1] "/home/jhz22/rds/public_databases/GTEx/csv/Whole_Blood.tsv.gz"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  2.43e-49  6.57e-44  3.69e-06  1.00e+00  1.66e-07 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 1.66e-05%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  5.76e-45  5.97e-44  8.77e-02  9.09e-01  3.64e-03 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.364%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.02e-44  2.38e-44  4.59e-01  3.62e-01  1.79e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 17.9%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.91e-44  2.49e-44  5.95e-01  3.79e-01  2.56e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 2.56%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.18e-44  3.14e-44  4.84e-01  4.78e-01  3.73e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.73%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.91e-44  2.20e-44  5.94e-01  3.34e-01  7.17e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 7.17%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.69e-44  2.37e-44  5.61e-01  3.60e-01  7.92e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 7.92%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.48e-44  2.85e-44  5.29e-01  4.33e-01  3.75e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.75%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.69e-44  2.66e-44  5.62e-01  4.05e-01  3.26e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.26%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.58e-44  2.72e-44  5.44e-01  4.13e-01  4.22e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.22%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.58e-44  2.48e-44  5.45e-01  3.77e-01  7.79e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 7.79%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.14e-44  3.07e-44  4.78e-01  4.67e-01  5.41e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 5.41%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.13e-44  3.12e-44  4.77e-01  4.75e-01  4.85e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.85%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.62e-44  2.27e-44  5.51e-01  3.45e-01  1.05e-01 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 10.5%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.30e-44  2.96e-44  5.03e-01  4.50e-01  4.74e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.74%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.70e-44  2.47e-44  5.63e-01  3.76e-01  6.07e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 6.07%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.79e-44  2.50e-44  5.77e-01  3.81e-01  4.22e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.22%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.73e-44  2.54e-44  5.67e-01  3.86e-01  4.68e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.68%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.81e-44  2.51e-44  5.80e-01  3.82e-01  3.88e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.88%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.62e-44  2.77e-44  5.51e-01  4.21e-01  2.74e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 2.74%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  5.04e-55  6.57e-44  7.67e-12  1.00e+00  3.72e-13 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.72e-11%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.18e-44  3.19e-44  4.84e-01  4.85e-01  3.11e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.11%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.45e-44  2.87e-44  5.25e-01  4.37e-01  3.84e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.84%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  4.94e-46  6.52e-44  7.51e-03  9.92e-01  3.49e-04 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.0349%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.80e-44  2.60e-44  5.78e-01  3.95e-01  2.72e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 2.72%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  2.41e-44  4.05e-44  3.67e-01  6.16e-01  1.71e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 1.71%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  7.06e-46  6.43e-44  1.07e-02  9.79e-01  1.05e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 1.05%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.32e-47  6.57e-44  5.05e-04  9.99e-01  2.32e-05 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.00232%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  1.62e-45  6.40e-44  2.47e-02  9.74e-01  1.19e-03 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.119%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.51e-44  2.76e-44  5.35e-01  4.21e-01  4.49e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.49%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.08e-44  3.12e-44  4.68e-01  4.74e-01  5.78e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 5.78%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  6.83e-46  6.50e-44  1.04e-02  9.89e-01  1.04e-03 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.104%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.78e-44  2.42e-44  5.75e-01  3.68e-01  5.76e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 5.76%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  4.02e-44  2.40e-44  6.12e-01  3.66e-01  2.24e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 2.24%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  1.22e-44  5.30e-44  1.86e-01  8.06e-01  7.92e-03 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.792%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.61e-44  2.72e-44  5.50e-01  4.13e-01  3.69e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.69%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  1.24e-45  6.44e-44  1.89e-02  9.80e-01  9.57e-04 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.0957%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  4.02e-44  2.32e-44  6.12e-01  3.52e-01  3.60e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.6%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.37e-44  2.98e-44  5.12e-01  4.53e-01  3.45e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.45%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.62e-44  2.81e-44  5.51e-01  4.27e-01  2.18e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 2.18%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  2.81e-44  3.65e-44  4.28e-01  5.55e-01  1.78e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 1.78%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.74e-44  2.32e-44  5.69e-01  3.53e-01  7.80e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 7.8%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.58e-44  2.68e-44  5.45e-01  4.08e-01  4.70e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.7%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.82e-44  2.48e-44  5.80e-01  3.77e-01  4.23e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.23%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  4.04e-44  2.32e-44  6.15e-01  3.53e-01  3.20e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.2%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  9.86e-45  5.54e-44  1.50e-01  8.44e-01  6.55e-03 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 0.655%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.76e-44  2.56e-44  5.72e-01  3.90e-01  3.87e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 3.87%"</span>
<span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span>
<span class="co">#&gt;  3.91e-44  2.39e-44  5.95e-01  3.64e-01  4.14e-02 </span>
<span class="co">#&gt; [1] "PP abf for shared variant: 4.14%"</span>
<span class="co">#&gt; `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</span></code></pre></div>
<p><img src="pQTLtools/unnamed-chunk-4-1.png" width="700" style="display:block; margin: auto"></p>
</div>
<div id="pqtl-based-mendelian-randomization-mr" class="section level2">
<h2 class="hasAnchor">
<a href="#pqtl-based-mendelian-randomization-mr" class="anchor"></a>pQTL-based Mendelian Randomization (MR)</h2>
<p>The function <code>pqtlMR</code> is derived from <span class="citation">Zheng et al. (2020)</span>. It has an attractive feature that multiple pQTLs can be used together for conducting MR with a list of outcomes from MR-Base. To simplify matters while maintaining greater flexibility, a composite input is needed to run the function which is illustrated below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TwoSampleMR</span><span class="op">)</span>
<span class="co">#&gt; TwoSampleMR version 0.5.6 </span>
<span class="co">#&gt; [&gt;] New: Option to use non-European LD reference panels for clumping etc</span>
<span class="co">#&gt; [&gt;] Some studies temporarily quarantined to verify effect allele</span>
<span class="co">#&gt; [&gt;] See news(package='TwoSampleMR') and https://gwas.mrcieu.ac.uk for further details</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'TwoSampleMR'</span>
<span class="co">#&gt; The following object is masked from 'package:GenomicRanges':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     trim</span>
<span class="co">#&gt; The following object is masked from 'package:IRanges':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     trim</span>
<span class="va">ins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html">path.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"Ins.csv"</span><span class="op">)</span><span class="op">)</span>
<span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ieu-a-7"</span>,<span class="st">"ebi-a-GCST007432"</span><span class="op">)</span>
<span class="va">Ins</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/format_data.html">format_data</a></span><span class="op">(</span><span class="va">ins</span><span class="op">)</span>
<span class="va">Ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html">extract_outcome_data</a></span><span class="op">(</span>snps<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">Ins</span>,<span class="va">SNP</span><span class="op">)</span>,outcomes<span class="op">=</span><span class="va">ids</span><span class="op">)</span>
<span class="co">#&gt; API: public: http://gwas-api.mrcieu.ac.uk/</span>
<span class="co">#&gt; Extracting data for 2 SNP(s) from 2 GWAS(s)</span>
<span class="va">harmonise</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html">harmonise_data</a></span><span class="op">(</span><span class="va">Ins</span>,<span class="va">Ids</span><span class="op">)</span>
<span class="co">#&gt; Harmonising ABO (dU3C0x) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span>
<span class="co">#&gt; Harmonising LIFR (YmGdhW) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span>
<span class="co">#&gt; Harmonising ABO (dU3C0x) and Coronary heart disease || id:ieu-a-7 (ieu-a-7)</span>
<span class="co">#&gt; Harmonising LIFR (YmGdhW) and Coronary heart disease || id:ieu-a-7 (ieu-a-7)</span>
<span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">Ins</span>, <span class="va">Ids</span>, <span class="va">harmonise</span><span class="op">)</span>
<span class="fu"><a href="../reference/pqtlMR.html">pqtlMR</a></span><span class="op">(</span><span class="va">input</span>,plot<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; Analysing 'dU3C0x' on 'ebi-a-GCST007432'</span>
<span class="co">#&gt; Analysing 'dU3C0x' on 'ieu-a-7'</span>
<span class="co">#&gt; Analysing 'YmGdhW' on 'ebi-a-GCST007432'</span>
<span class="co">#&gt; Analysing 'YmGdhW' on 'ieu-a-7'</span>
<span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="st">"pQTL-combined-result.txt"</span><span class="op">)</span><span class="op">)</span>
<span class="op">{</span>
  <span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="st">"pQTL-combined-result.txt"</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">result</span>,caption<span class="op">=</span><span class="st">"ABO/LIFR variants and CHD/FEV1"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<table class="table">
<caption>ABO/LIFR variants and CHD/FEV1</caption>
<colgroup>
<col width="9%">
<col width="13%">
<col width="37%">
<col width="7%">
<col width="8%">
<col width="4%">
<col width="6%">
<col width="6%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="left">outcome</th>
<th align="left">exposure</th>
<th align="left">method</th>
<th align="right">nsnp</th>
<th align="right">b</th>
<th align="right">se</th>
<th align="right">pval</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">dU3C0x</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ABO</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">-0.0109</td>
<td align="right">0.00193</td>
<td align="right">0.0e+00</td>
</tr>
<tr class="even">
<td align="left">dU3C0x</td>
<td align="left">ieu-a-7</td>
<td align="left">Coronary heart disease || id:ieu-a-7</td>
<td align="left">ABO</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">0.0334</td>
<td align="right">0.00730</td>
<td align="right">4.6e-06</td>
</tr>
<tr class="odd">
<td align="left">YmGdhW</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">LIFR</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">0.0620</td>
<td align="right">0.01000</td>
<td align="right">0.0e+00</td>
</tr>
<tr class="even">
<td align="left">YmGdhW</td>
<td align="left">ieu-a-7</td>
<td align="left">Coronary heart disease || id:ieu-a-7</td>
<td align="left">LIFR</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">-0.2572</td>
<td align="right">0.03904</td>
<td align="right">0.0e+00</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"result"</span>,<span class="st">"single"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"pQTL-combined-"</span>,<span class="va">x</span>,<span class="st">".txt"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="run_twosamplemr" class="section level2">
<h2 class="hasAnchor">
<a href="#run_twosamplemr" class="anchor"></a>run_TwoSampleMR</h2>
<p>The function has similiarity to <span class="citation">Dimou and Tsilidis (2018)</span>. The documentation example is quoted here,</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TwoSampleMR</span><span class="op">)</span>
<span class="va">outcomes</span> <span class="op">&lt;-</span> <span class="st">"ebi-a-GCST007432"</span>
<span class="va">prot</span> <span class="op">&lt;-</span> <span class="st">"MMP.10"</span>
<span class="va">type</span> <span class="op">&lt;-</span> <span class="st">"cis"</span>
<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">type</span>,<span class="st">".mrx"</span><span class="op">)</span>
<span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html">path.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="va">f</span><span class="op">)</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">exposure</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/format_data.html">format_data</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">within</a></span><span class="op">(</span><span class="va">d</span>,<span class="op">{</span><span class="va">P</span><span class="op">=</span><span class="fl">10</span><span class="op">^</span><span class="va">logP</span><span class="op">}</span><span class="op">)</span>, phenotype_col<span class="op">=</span><span class="st">"prot"</span>, snp_col<span class="op">=</span><span class="st">"rsid"</span>,
                        chr_col<span class="op">=</span><span class="st">"Chromosome"</span>, pos_col<span class="op">=</span><span class="st">"Posistion"</span>,
                        effect_allele_col<span class="op">=</span><span class="st">"Allele1"</span>, other_allele_col<span class="op">=</span><span class="st">"Allele2"</span>,
                        eaf_col<span class="op">=</span><span class="st">"Freq1"</span>, beta_col<span class="op">=</span><span class="st">"Effect"</span>, se_col<span class="op">=</span><span class="st">"StdErr"</span>,
                        pval_col<span class="op">=</span><span class="st">"P"</span>, log_pval<span class="op">=</span><span class="cn">FALSE</span>,
                        samplesize_col<span class="op">=</span><span class="st">"N"</span><span class="op">)</span>
<span class="va">clump</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/clump_data.html">clump_data</a></span><span class="op">(</span><span class="va">exposure</span><span class="op">)</span>
<span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="co">#&gt; Clumping ZoHewj, 1106 variants, using EUR population reference</span>
<span class="co">#&gt; Removing 1102 of 1106 variants due to LD with other variants or absence from LD reference panel</span>
<span class="va">outcome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html">extract_outcome_data</a></span><span class="op">(</span>snps<span class="op">=</span><span class="va">exposure</span><span class="op">$</span><span class="va">SNP</span>,outcomes<span class="op">=</span><span class="va">outcomes</span><span class="op">)</span>
<span class="co">#&gt; Extracting data for 1106 SNP(s) from 1 GWAS(s)</span>
<span class="co">#&gt; Finding proxies for 155 SNPs in outcome ebi-a-GCST007432</span>
<span class="co">#&gt; Extracting data for 155 SNP(s) from 1 GWAS(s)</span>
<span class="va">harmonise</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html">harmonise_data</a></span><span class="op">(</span><span class="va">clump</span>,<span class="va">outcome</span><span class="op">)</span>
<span class="co">#&gt; Harmonising MMP.10 (ZoHewj) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span>
<span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">exposure</span>,<span class="va">outcome</span>,<span class="va">clump</span>,<span class="va">harmonise</span><span class="op">)</span>
<span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">outcomes</span>,<span class="va">prot</span>,<span class="va">type</span>,sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span>
<span class="fu"><a href="../reference/run_TwoSampleMR.html">run_TwoSampleMR</a></span><span class="op">(</span><span class="va">input</span>, prefix<span class="op">=</span><span class="va">prefix</span><span class="op">)</span>
<span class="co">#&gt; r.exposure and/or r.outcome not present.</span>
<span class="co">#&gt; Calculating approximate SNP-exposure and/or SNP-outcome correlations, assuming all are quantitative traits. Please pre-calculate r.exposure and/or r.outcome using get_r_from_lor() for any binary traits</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Estimating correlation for quantitative trait.</span>
<span class="co">#&gt; This method is an approximation, and may be numerically unstable.</span>
<span class="co">#&gt; Ideally you should estimate r directly from independent replication samples.</span>
<span class="co">#&gt; Use get_r_from_lor for binary traits.</span>
<span class="co">#&gt; Analysing 'ZoHewj' on 'ebi-a-GCST007432'</span></code></pre></div>
<p><img src="pQTLtools/unnamed-chunk-6-1.png" width="700" style="display:block; margin: auto"><img src="pQTLtools/unnamed-chunk-6-2.png" width="700" style="display:block; margin: auto"><img src="pQTLtools/unnamed-chunk-6-3.png" width="700" style="display:block; margin: auto"><img src="pQTLtools/unnamed-chunk-6-4.png" width="700" style="display:block; margin: auto"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-result.txt"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">{</span>
  <span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-result.txt"</span><span class="op">)</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span><span class="op">(</span><span class="va">result</span>,caption<span class="op">=</span><span class="st">"MMP.10 variants and FEV1"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/invisible.html">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"result"</span>,<span class="st">"heterogeneity"</span>,<span class="st">"pleiotropy"</span>,<span class="st">"single"</span>,<span class="st">"loo"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-"</span>,<span class="va">x</span>,<span class="st">".txt"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The output is contained in individual files <code>ukb-b-19732-MMP.10-cis-result.txt</code>, etc. The scatter, forest, funnel and leave-one-out plots is contained in <code>ukb-b-19732-MMP.10-cis.pdf</code>.</p>
</div>
<div id="literature-on-pqtls" class="section level2">
<h2 class="hasAnchor">
<a href="#literature-on-pqtls" class="anchor"></a>Literature on pQTLs</h2>
<p>Those as identified by <span class="citation">Sun et al. (2018)</span> and <span class="citation">Suhre, McCarthy, and Schwenk (2020)</span> are included as EndNote libraries.</p>
</div>
<div id="uniprot-ids" class="section level2">
<h2 class="hasAnchor">
<a href="#uniprot-ids" class="anchor"></a>UniProt IDs</h2>
<p>The function <code>uniprot2ids</code> converts UniProt IDs to others.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references">
<div id="ref-dt18">
<p>Dimou, Niki L., and Konstantinos K. Tsilidis. 2018. “A Primer in Mendelian Randomization Methodology with a Focus on Utilizing Published Summary Association Data.” In <em>Genetic Epidemiology: Methods and Protocols</em>, edited by Evangelos Evangelou, 211–30. New York: Springer Science+Business Media, LLC, part of Springer Nature.</p>
</div>
<div id="ref-suhre20">
<p>Suhre, K., M. I. McCarthy, and J. M. Schwenk. 2020. “Genetics Meets Proteomics: Perspectives for Large Population-Based Studies.” Journal Article. <em>Nat Rev Genet</em>. <a href="https://doi.org/10.1038/s41576-020-0268-2" class="uri">https://doi.org/10.1038/s41576-020-0268-2</a>.</p>
</div>
<div id="ref-sun18">
<p>Sun, B. B., J. C. Maranville, J. E. Peters, D. Stacey, J. R. Staley, J. Blackshaw, S. Burgess, et al. 2018. “Genomic Atlas of the Human Plasma Proteome.” Journal Article. <em>Nature</em> 558 (7708):73–79. <a href="https://doi.org/10.1038/s41586-018-0175-2" class="uri">https://doi.org/10.1038/s41586-018-0175-2</a>.</p>
</div>
<div id="ref-zheng20">
<p>Zheng, J., V. Haberland, D. Baird, V. Walker, P. C. Haycock, M. R. Hurle, A. Gutteridge, et al. 2020. “Phenome-Wide Mendelian Randomization Mapping the Influence of the Plasma Proteome on Complex Diseases.” Journal Article. <em>Nature Genetics</em> 52 (10):1122–31. <a href="https://doi.org/10.1038/s41588-020-0682-6" class="uri">https://doi.org/10.1038/s41588-020-0682-6</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jing hua Zhao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
