<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="pQTLtools">
<title>An Overview of pQTLtools â€¢ pQTLtools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="An Overview of pQTLtools">
<meta property="og:description" content="pQTLtools">
<meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Vignettes:</h6>
    <a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Articles:</h6>
    <a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a>
    <a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a>
    <a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a>
    <a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/pQTLtools.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://jinghuazhao.github.io/R/">
    <span class="fab fa fab fa-r-project"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/">
    <span class="fas fa fas fa-fast-backward"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-urls">URLs</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-urls">
    <a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a>
    <a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a>
    <a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a>
    <a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a>
    <a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a>
    <a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a>
    <a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a>
    <a class="external-link dropdown-item" href="https://metascape.org/">metascape</a>
    <a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a>
    <a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a>
    <a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a>
    <a class="external-link dropdown-item" href="https://string-db.org/">STRING</a>
    <a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a>
    <a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.uniprot.org/">
    <span class="fas fa fas fa-fast-forward"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>An Overview of pQTLtools</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/vignettes/pQTLtools.Rmd" class="external-link"><code>vignettes/pQTLtools.Rmd</code></a></small>
      <div class="d-none name"><code>pQTLtools.Rmd</code></div>
    </div>

    
    
<p>The examples here are based on the SCALLOP work<span class="citation"><sup>1</sup></span>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GenomicRanges"</span>, <span class="st">"TwoSampleMR"</span>, <span class="st">"biomaRt"</span>, <span class="st">"coloc"</span>, <span class="st">"dplyr"</span>, <span class="st">"gap"</span>, <span class="st">"ggplot2"</span>, <span class="st">"httr"</span>,</span>
<span>          <span class="st">"karyoploteR"</span>, <span class="st">"circlize"</span>, <span class="st">"knitr"</span>, <span class="st">"plotly"</span>, <span class="st">"pQTLtools"</span>, <span class="st">"rGREAT"</span>, <span class="st">"readr"</span>, <span class="st">"regioneR"</span>,</span>
<span>          <span class="st">"seqminer"</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">p</span> <span class="kw">in</span> <span class="va">pkgs</span><span class="op">)</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"^package:"</span>, <span class="va">p</span>, <span class="st">"$"</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/search.html" class="external-link">search</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/warning.html" class="external-link">warning</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"This vignette needs package `"</span>, <span class="va">p</span>, <span class="st">"'; please install"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">pkgs</span>, <span class="va">require</span>, character.only<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2" number="1">
<h2 id="forest-plots">
<span class="header-section-number">1</span> Forest plots<a class="anchor" aria-label="anchor" href="#forest-plots"></a>
</h2>
<p>We start with results on osteoprotegerin (OPG)<span class="citation"><sup>2</sup></span>,</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">OPG</span>,package<span class="op">=</span><span class="st">"gap.datasets"</span><span class="op">)</span></span>
<span><span class="fu">meta</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/meta/man/settings.meta.html" class="external-link">settings.meta</a></span><span class="op">(</span>method.tau<span class="op">=</span><span class="st">"DL"</span><span class="op">)</span></span>
<span><span class="fu">METAL_forestplot</span><span class="op">(</span><span class="va">OPGtbl</span>,<span class="va">OPGall</span>,<span class="va">OPGrsid</span>,width<span class="op">=</span><span class="fl">6.75</span>,height<span class="op">=</span><span class="fl">5</span>,digits.TE<span class="op">=</span><span class="fl">2</span>,digits.se<span class="op">=</span><span class="fl">2</span>,</span>
<span>                 col.diamond<span class="op">=</span><span class="st">"black"</span>,col.inside<span class="op">=</span><span class="st">"black"</span>,col.square<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(MarkerName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(MarkerName)`</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:forest-1"></span>
<img src="pQTLtools/forest-1.png" alt="Forest plots" width="864" style="display:block; margin: auto"><p class="caption">
Figure 1.1: Forest plots
</p>
</div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:forest-2"></span>
<img src="pQTLtools/forest-2.png" alt="Forest plots" width="864" style="display:block; margin: auto"><p class="caption">
Figure 1.2: Forest plots
</p>
</div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">METAL_forestplot</span><span class="op">(</span><span class="va">OPGtbl</span>,<span class="va">OPGall</span>,<span class="va">OPGrsid</span>,package<span class="op">=</span><span class="st">"metafor"</span>,method<span class="op">=</span><span class="st">"FE"</span>,xlab<span class="op">=</span><span class="st">"Effect"</span>,</span>
<span>                 showweights<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(MarkerName)`</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(MarkerName)`</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:forest-3"></span>
<img src="pQTLtools/forest-3.png" alt="Forest plots" width="864" style="display:block; margin: auto"><p class="caption">
Figure 1.3: Forest plots
</p>
</div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:forest-4"></span>
<img src="pQTLtools/forest-4.png" alt="Forest plots" width="864" style="display:block; margin: auto"><p class="caption">
Figure 1.4: Forest plots
</p>
</div>
<p>involving both a cis and a trans pQTLs. As <code>meta</code> inherently includes random effects, we use a fixed effects (FE) model from <code>metafor</code>.</p>
</div>
<div class="section level2" number="2">
<h2 id="cistrans-classification">
<span class="header-section-number">2</span> cis/trans classification<a class="anchor" aria-label="anchor" href="#cistrans-classification"></a>
</h2>
<div class="section level3" number="2.1">
<h3 id="pqtl-signals-and-classification-table">
<span class="header-section-number">2.1</span> pQTL signals and classification table<a class="anchor" aria-label="anchor" href="#pqtl-signals-and-classification-table"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"INF1.merge"</span><span class="op">)</span></span>
<span><span class="va">merged</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">f</span>,as.is<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">merged</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHR"</span>,<span class="st">"POS"</span>,<span class="st">"MarkerName"</span>,<span class="st">"prot"</span>,<span class="st">"log10p"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>              <span class="fu">pQTLdata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/pQTLdata/man/inf1.html" class="external-link">inf1</a></span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"uniprot"</span><span class="op">)</span><span class="op">]</span>,by<span class="op">=</span><span class="st">"prot"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>log10p<span class="op">=</span><span class="op">-</span><span class="va">log10p</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hits</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"Chr"</span>,<span class="st">"bp"</span>,<span class="st">"SNP"</span>,<span class="st">"log10p"</span>,<span class="st">"uniprot"</span><span class="op">)</span></span>
<span><span class="va">cistrans</span> <span class="op">&lt;-</span> <span class="fu">cis.vs.trans.classification</span><span class="op">(</span><span class="va">hits</span>,<span class="fu">pQTLdata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/pQTLdata/man/inf1.html" class="external-link">inf1</a></span>,<span class="st">"uniprot"</span><span class="op">)</span></span>
<span><span class="va">cis.vs.trans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">cistrans</span>,<span class="va">data</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">cistrans</span>,<span class="va">table</span><span class="op">)</span>,caption<span class="op">=</span><span class="st">"cis/trans classification"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:ctdata">Table 2.1: </span>cis/trans classification</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">cis</th>
<th align="right">trans</th>
<th align="right">total</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ADA</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">CASP8</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">CCL11</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">CCL13</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">CCL19</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">CCL2</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">CCL20</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">CCL23</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">CCL25</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">CCL3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">CCL4</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">CCL7</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">CCL8</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">CD244</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">CD274</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">CD40</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">CD5</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">CD6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">CDCP1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">CSF1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">CST5</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">CX3CL1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">CXCL1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">CXCL10</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">CXCL11</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">CXCL5</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">CXCL6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">CXCL9</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">DNER</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">EIF4EBP1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">FGF19</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">FGF21</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">FGF23</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">FGF5</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">FLT3LG</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">GDNF</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">HGF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">IL10</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">IL10RB</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">IL12B</td>
<td align="right">1</td>
<td align="right">7</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">IL15RA</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">IL17C</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">IL18</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">IL18R1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">IL1A</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">IL6</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">IL7</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">IL8</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">KITLG</td>
<td align="right">0</td>
<td align="right">7</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">LIFR</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">LTA</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">MMP1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">MMP10</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">NGF</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">NTF3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">OSM</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">PLAU</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">S100A12</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">SIRT2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">SLAMF1</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">SULT1A1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">TGFA</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">TGFB1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">TNFRSF11B</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">TNFRSF9</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">TNFSF10</td>
<td align="right">1</td>
<td align="right">7</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">TNFSF11</td>
<td align="right">1</td>
<td align="right">5</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">TNFSF12</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">TNFSF14</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">VEGFA</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">total</td>
<td align="right">59</td>
<td align="right">121</td>
<td align="right">180</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">cistrans</span>,<span class="va">total</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 180</span></span>
<span><span class="cn">T</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">cistrans</span>,<span class="va">table</span><span class="op">)</span></span>
<span><span class="va">H</span> <span class="op">&lt;-</span> <span class="cn">T</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="cn">T</span><span class="op">)</span><span class="op">!=</span><span class="st">"total"</span>,<span class="st">"total"</span><span class="op">]</span></span>
<span><span class="va">merge</span> <span class="op">&lt;-</span> <span class="va">merged</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chrom"</span>,<span class="st">"Start"</span>,<span class="st">"End"</span>,<span class="st">"prot"</span>,<span class="st">"MarkerName"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">merge_cvt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">merge</span>,<span class="va">cis.vs.trans</span>,by.x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"MarkerName"</span><span class="op">)</span>,by.y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"SNP"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ord</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">merge_cvt</span>,<span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="va">bp</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">merge_cvt</span> <span class="op">&lt;-</span> <span class="va">merge_cvt</span><span class="op">[</span><span class="va">ord</span>,<span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3" number="2.2">
<h3 id="genomic-associations">
<span class="header-section-number">2.2</span> Genomic associations<a class="anchor" aria-label="anchor" href="#genomic-associations"></a>
</h3>
<p>This is visualised via a circos plot, highlighting the likely causal genes for pQTLs,</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pQTLs</span> <span class="op">&lt;-</span> <span class="fu">transmute</span><span class="op">(</span><span class="va">merge_cvt</span>,chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="va">Chr</span><span class="op">)</span>,start<span class="op">=</span><span class="va">bp</span>,end<span class="op">=</span><span class="va">bp</span>,<span class="va">log10p</span><span class="op">)</span></span>
<span><span class="va">cis.pQTLs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">merge_cvt</span>,<span class="va">cis</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">transmute</span><span class="op">(</span>chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="va">p.chr</span><span class="op">)</span>,start<span class="op">=</span><span class="va">p.start</span>,end<span class="op">=</span><span class="va">p.end</span>,gene<span class="op">=</span><span class="va">p.gene</span>,cols<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></span>
<span><span class="va">pQTL_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"pQTL_genes.txt"</span><span class="op">)</span>,</span>
<span>                         col.names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">"start"</span>,<span class="st">"end"</span>,<span class="st">"gene"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">mutate</span><span class="op">(</span>chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"hs"</span>,<span class="st">"chr"</span>,<span class="va">chr</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">left_join</span><span class="op">(</span><span class="va">cis.pQTLs</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>              <span class="fu">mutate</span><span class="op">(</span>cols<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">cols</span><span class="op">)</span>,<span class="st">"blue"</span>,<span class="va">cols</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(chr, start, end, gene)`</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>cex<span class="op">=</span><span class="fl">0.7</span><span class="op">)</span></span>
<span><span class="fu">gap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gap/man/circos.mhtplot2.html" class="external-link">circos.mhtplot2</a></span><span class="op">(</span><span class="va">pQTLs</span>,<span class="va">pQTL_genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Some of the regions have end position values larger than the end of the chromosomes.</span></span>
<span><span class="co">#&gt; Note: 7 points are out of plotting region in sector 'chr1', track '5'.</span></span>
<span><span class="co">#&gt; Note: 3 points are out of plotting region in sector 'chr2', track '5'.</span></span>
<span><span class="co">#&gt; Note: 10 points are out of plotting region in sector 'chr3', track '5'.</span></span>
<span><span class="co">#&gt; Note: 7 points are out of plotting region in sector 'chr4', track '5'.</span></span>
<span><span class="co">#&gt; Note: 2 points are out of plotting region in sector 'chr5', track '5'.</span></span>
<span><span class="co">#&gt; Note: 4 points are out of plotting region in sector 'chr6', track '5'.</span></span>
<span><span class="co">#&gt; Note: 2 points are out of plotting region in sector 'chr8', track '5'.</span></span>
<span><span class="co">#&gt; Note: 2 points are out of plotting region in sector 'chr9', track '5'.</span></span>
<span><span class="co">#&gt; Note: 2 points are out of plotting region in sector 'chr10', track '5'.</span></span>
<span><span class="co">#&gt; Note: 4 points are out of plotting region in sector 'chr11', track '5'.</span></span>
<span><span class="co">#&gt; Note: 1 point is out of plotting region in sector 'chr12', track '5'.</span></span>
<span><span class="co">#&gt; Note: 1 point is out of plotting region in sector 'chr13', track '5'.</span></span>
<span><span class="co">#&gt; Note: 1 point is out of plotting region in sector 'chr14', track '5'.</span></span>
<span><span class="co">#&gt; Note: 2 points are out of plotting region in sector 'chr16', track '5'.</span></span>
<span><span class="co">#&gt; Note: 8 points are out of plotting region in sector 'chr17', track '5'.</span></span>
<span><span class="co">#&gt; Note: 8 points are out of plotting region in sector 'chr19', track '5'.</span></span>
<span><span class="co">#&gt; Note: 4 points are out of plotting region in sector 'chr20', track '5'.</span></span>
<span><span class="co">#&gt; Note: 1 point is out of plotting region in sector 'chr21', track '5'.</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:pQTLs"></span>
<img src="pQTLtools/pQTLs-1.png" alt="Genomic associations" width="679.68" style="display:block; margin: auto"><p class="caption">
Figure 2.1: Genomic associations
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>cex<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>where the red and blue colours indicate cis/trans classifications.</p>
</div>
<div class="section level3" number="2.3">
<h3 id="bar-chart-and-circos-plot">
<span class="header-section-number">2.3</span> Bar chart and circos plot<a class="anchor" aria-label="anchor" href="#bar-chart-and-circos-plot"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">H</span><span class="op">)</span>,xlab<span class="op">=</span><span class="st">"No. of pQTL regions"</span>,ylab<span class="op">=</span><span class="st">"No. of proteins"</span>,</span>
<span>        ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">25</span><span class="op">)</span>,col<span class="op">=</span><span class="st">"darkgrey"</span>,border<span class="op">=</span><span class="st">"black"</span>,cex<span class="op">=</span><span class="fl">0.8</span>,cex.axis<span class="op">=</span><span class="fl">2</span>,cex.names<span class="op">=</span><span class="fl">2</span>,las<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:ctbar"></span>
<img src="pQTLtools/ctbar-1.png" alt="Bar chart" width="672" style="display:block; margin: auto"><p class="caption">
Figure 2.2: Bar chart
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">circos.cis.vs.trans.plot</span><span class="op">(</span>hits<span class="op">=</span><span class="va">f</span>,<span class="fu">pQTLdata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/pQTLdata/man/inf1.html" class="external-link">inf1</a></span>,<span class="st">"uniprot"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:ctcircos"></span>
<img src="pQTLtools/ctcircos-1.png" alt="circos plot" width="672" style="display:block; margin: auto"><p class="caption">
Figure 2.3: circos plot
</p>
</div>
<p>The circos plot is based on target genes (those encoding proteins) and somewhat too busy.</p>
</div>
<div class="section level3" number="2.4">
<h3 id="sh2b3">
<span class="header-section-number">2.4</span> <em>SH2B3</em><a class="anchor" aria-label="anchor" href="#sh2b3"></a>
</h3>
<p>Here we focus on <em>SH2B3</em>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="va">HOTSPOT</span> <span class="op">&lt;-</span> <span class="st">"chr12:111884608_C_T"</span></span>
<span>  <span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>chr<span class="op">=</span><span class="st">"chr12"</span>,start<span class="op">=</span><span class="fl">111884607</span>,end<span class="op">=</span><span class="fl">111884608</span>,gene<span class="op">=</span><span class="st">"SH2B3"</span><span class="op">)</span></span>
<span>  <span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">cis.vs.trans</span>,<span class="va">SNP</span><span class="op">==</span><span class="va">HOTSPOT</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>       <span class="fu">mutate</span><span class="op">(</span>p.chr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="va">p.chr</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>       <span class="fu">rename</span><span class="op">(</span>chr<span class="op">=</span><span class="va">p.chr</span>,start<span class="op">=</span><span class="va">p.start</span>,end<span class="op">=</span><span class="va">p.end</span>,gene<span class="op">=</span><span class="va">p.gene</span>,cistrans<span class="op">=</span><span class="va">cis.trans</span><span class="op">)</span></span>
<span>  <span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">12</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">b</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">cols</span><span class="op">[</span><span class="va">b</span><span class="op">[[</span><span class="st">"cis"</span><span class="op">]</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span>  <span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="fu">select</span><span class="op">(</span><span class="va">b</span>,<span class="va">chr</span>,<span class="va">start</span>,<span class="va">end</span>,<span class="va">gene</span><span class="op">)</span>,<span class="va">a</span><span class="op">)</span></span>
<span>  <span class="fu">circos.clear</span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="fu">circos.par</span><span class="op">(</span>start.degree<span class="op">=</span><span class="fl">90</span>, track.height<span class="op">=</span><span class="fl">0.1</span>, cell.padding<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu">circos.initializeWithIdeogram</span><span class="op">(</span>species<span class="op">=</span><span class="st">"hg19"</span>, track.height<span class="op">=</span><span class="fl">0.05</span>, ideogram.height<span class="op">=</span><span class="fl">0.06</span><span class="op">)</span></span>
<span>  <span class="fu">circos.genomicLabels</span><span class="op">(</span><span class="va">labels</span>, labels.column<span class="op">=</span><span class="fl">4</span>, cex<span class="op">=</span><span class="fl">1.1</span>, font<span class="op">=</span><span class="fl">3</span>, side<span class="op">=</span><span class="st">"inside"</span><span class="op">)</span></span>
<span>  <span class="fu">circos.genomicLink</span><span class="op">(</span><span class="fu">bind_rows</span><span class="op">(</span><span class="va">a</span>,<span class="va">a</span>,<span class="va">a</span>,<span class="va">a</span>,<span class="va">a</span>,<span class="va">a</span><span class="op">)</span>, <span class="va">b</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">"start"</span>,<span class="st">"end"</span><span class="op">)</span><span class="op">]</span>, col<span class="op">=</span><span class="va">cols</span>,</span>
<span>                     directional<span class="op">=</span><span class="fl">1</span>, border<span class="op">=</span><span class="fl">10</span>, lwd<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:sh2b3"></span>
<img src="pQTLtools/sh2b3-1.png" alt="*SH2B3* hotspot" width="672" style="display:block; margin: auto"><p class="caption">
Figure 2.4: <em>SH2B3</em> hotspot
</p>
</div>
<p>A more recent implementation is the <code>qtlClassifier</code> function. For this example we have,</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span>
<span><span class="va">geneSNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">merged</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"MarkerName"</span><span class="op">)</span><span class="op">]</span>,<span class="fu">pQTLdata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/pQTLdata/man/inf1.html" class="external-link">inf1</a></span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"gene"</span><span class="op">)</span><span class="op">]</span>,by<span class="op">=</span><span class="st">"prot"</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>,<span class="st">"MarkerName"</span>,<span class="st">"prot"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">SNPPos</span> <span class="op">&lt;-</span> <span class="va">merged</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MarkerName"</span>,<span class="st">"CHR"</span>,<span class="st">"POS"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">genePos</span> <span class="op">&lt;-</span> <span class="fu">pQTLdata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/pQTLdata/man/inf1.html" class="external-link">inf1</a></span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>,<span class="st">"chr"</span>,<span class="st">"start"</span>,<span class="st">"end"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">cvt</span> <span class="op">&lt;-</span> <span class="fu">qtlClassifier</span><span class="op">(</span><span class="va">geneSNP</span>,<span class="va">SNPPos</span>,<span class="va">genePos</span>,<span class="fl">1e6</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cvt</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="3%">
<col width="9%">
<col width="20%">
<col width="9%">
<col width="10%">
<col width="10%">
<col width="10%">
<col width="9%">
<col width="10%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">gene</th>
<th align="left">MarkerName</th>
<th align="left">prot</th>
<th align="right">geneChrom</th>
<th align="right">geneStart</th>
<th align="right">geneEnd</th>
<th align="right">SNPChrom</th>
<th align="right">SNPPos</th>
<th align="left">Type</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">2</td>
<td align="left">EIF4EBP1</td>
<td align="left">chr4:187158034_A_G</td>
<td align="left">4E.BP1</td>
<td align="right">8</td>
<td align="right">37887859</td>
<td align="right">37917883</td>
<td align="right">4</td>
<td align="right">187158034</td>
<td align="left">trans</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">ADA</td>
<td align="left">chr20:43255220_C_T</td>
<td align="left">ADA</td>
<td align="right">20</td>
<td align="right">43248163</td>
<td align="right">43280874</td>
<td align="right">20</td>
<td align="right">43255220</td>
<td align="left">cis</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">NGF</td>
<td align="left">chr1:115829943_A_C</td>
<td align="left">Beta.NGF</td>
<td align="right">1</td>
<td align="right">115828539</td>
<td align="right">115880857</td>
<td align="right">1</td>
<td align="right">115829943</td>
<td align="left">cis</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">NGF</td>
<td align="left">chr9:90362040_C_T</td>
<td align="left">Beta.NGF</td>
<td align="right">1</td>
<td align="right">115828539</td>
<td align="right">115880857</td>
<td align="right">9</td>
<td align="right">90362040</td>
<td align="left">trans</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">CASP8</td>
<td align="left">chr2:202164805_C_G</td>
<td align="left">CASP.8</td>
<td align="right">2</td>
<td align="right">202098166</td>
<td align="right">202152434</td>
<td align="right">2</td>
<td align="right">202164805</td>
<td align="left">cis</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">CCL11</td>
<td align="left">chr1:159175354_A_G</td>
<td align="left">CCL11</td>
<td align="right">17</td>
<td align="right">32612687</td>
<td align="right">32615353</td>
<td align="right">1</td>
<td align="right">159175354</td>
<td align="left">trans</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cistrans.check</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">cvt</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>,<span class="st">"MarkerName"</span>,<span class="st">"Type"</span><span class="op">)</span><span class="op">]</span>,<span class="va">cis.vs.trans</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"p.gene"</span>,<span class="st">"SNP"</span>,<span class="st">"cis.trans"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                        by.x<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>,<span class="st">"MarkerName"</span><span class="op">)</span>,by.y<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"p.gene"</span>,<span class="st">"SNP"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">cistrans.check</span>,<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">Type</span>,<span class="va">cis.trans</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;        cis.trans</span></span>
<span><span class="co">#&gt; Type    cis trans</span></span>
<span><span class="co">#&gt;   cis    59     0</span></span>
<span><span class="co">#&gt;   trans   0   121</span></span></code></pre></div>
</div>
<div class="section level3" number="2.5">
<h3 id="pqtl-gene-plot">
<span class="header-section-number">2.5</span> pQTL-gene plot<a class="anchor" aria-label="anchor" href="#pqtl-gene-plot"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">t2d</span> <span class="op">&lt;-</span> <span class="fu">qtl2dplot</span><span class="op">(</span><span class="va">cis.vs.trans</span>,xlab<span class="op">=</span><span class="st">"pQTL position"</span>,ylab<span class="op">=</span><span class="st">"Gene position"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:qtl2dplot"></span>
<img src="pQTLtools/qtl2dplot-1.png" alt="pQTL-gene plot" width="768" style="display:block; margin: auto"><p class="caption">
Figure 2.5: pQTL-gene plot
</p>
</div>
<p>whose result can be also viewed in a 2-d plotly style, <a href="fig2d.html">fig2d.html</a>,</p>
</div>
<div class="section level3" number="2.6">
<h3 id="plotly">
<span class="header-section-number">2.6</span> plotly<a class="anchor" aria-label="anchor" href="#plotly"></a>
</h3>
<p>These plots are responsive.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fig2d</span> <span class="op">&lt;-</span> <span class="fu">qtl2dplotly</span><span class="op">(</span><span class="va">cis.vs.trans</span>,xlab<span class="op">=</span><span class="st">"pQTL position"</span>,ylab<span class="op">=</span><span class="st">"Gene position"</span><span class="op">)</span></span>
<span><span class="fu">htmlwidgets</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/htmlwidgets/man/saveWidget.html" class="external-link">saveWidget</a></span><span class="op">(</span><span class="va">fig2d</span>,file<span class="op">=</span><span class="st">"fig2d.html"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">fig2d</span><span class="op">)</span></span></code></pre></div>
<p>and 3-d counterpart, <a href="fig3d.html">fig3d.html</a>,</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fig3d</span> <span class="op">&lt;-</span> <span class="fu">qtl3dplotly</span><span class="op">(</span><span class="va">cis.vs.trans</span>,zmax<span class="op">=</span><span class="fl">300</span>,qtl.prefix<span class="op">=</span><span class="st">"pQTL:"</span>,xlab<span class="op">=</span><span class="st">"pQTL position"</span>,ylab<span class="op">=</span><span class="st">"Gene position"</span><span class="op">)</span></span>
<span><span class="fu">htmlwidgets</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/htmlwidgets/man/saveWidget.html" class="external-link">saveWidget</a></span><span class="op">(</span><span class="va">fig3d</span>,file<span class="op">=</span><span class="st">"fig3d.html"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">fig3d</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="2.7">
<h3 id="karyoplot">
<span class="header-section-number">2.7</span> Karyoplot<a class="anchor" aria-label="anchor" href="#karyoplot"></a>
</h3>
<p>As <code>biomaRt</code> is not always on, we keep a copy of hgnc.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">set_config</span><span class="op">(</span><span class="fu">config</span><span class="op">(</span>ssl_verifypeer <span class="op">=</span> <span class="fl">0L</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu">useMart</span><span class="op">(</span>biomart <span class="op">=</span> <span class="st">"ensembl"</span>, dataset <span class="op">=</span> <span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span>
<span><span class="va">attrs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hgnc_symbol"</span>, <span class="st">"chromosome_name"</span>, <span class="st">"start_position"</span>, <span class="st">"end_position"</span>, <span class="st">"band"</span><span class="op">)</span></span>
<span><span class="va">hgnc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">vector</a></span><span class="op">(</span><span class="st">"character"</span>,<span class="fl">180</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">180</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">merge_cvt</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">Chr</span>,<span class="st">":"</span>,<span class="va">bp</span>,<span class="st">":"</span>,<span class="va">bp</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu">getBM</span><span class="op">(</span>attributes <span class="op">=</span> <span class="va">attrs</span>, filters<span class="op">=</span><span class="st">"chromosomal_region"</span>, values<span class="op">=</span><span class="va">v</span>, mart<span class="op">=</span><span class="va">mart</span><span class="op">)</span>,<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">hgnc_symbol</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">hgnc</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">g</span><span class="op">[[</span><span class="st">"hgnc_symbol"</span><span class="op">]</span><span class="op">]</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">i</span>,<span class="va">g</span><span class="op">[[</span><span class="st">"hgnc_symbol"</span><span class="op">]</span><span class="op">]</span>,<span class="va">hgnc</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,<span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">hgnc</span>,file<span class="op">=</span><span class="st">"hgnc.rda"</span>,compress<span class="op">=</span><span class="st">"xz"</span><span class="op">)</span></span></code></pre></div>
<p>We now proceed with</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"hgnc.rda"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">merge_cvt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">merge_cvt</span>,<span class="op">{</span></span>
<span>  <span class="va">hgnc</span> <span class="op">&lt;-</span> <span class="va">hgnc</span></span>
<span>  <span class="va">hgnc</span><span class="op">[</span><span class="va">cis</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">p.gene</span><span class="op">[</span><span class="va">cis</span><span class="op">]</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">merge_cvt</span>, <span class="op">{</span></span>
<span>  <span class="va">sentinels</span> <span class="op">&lt;-</span> <span class="fu">toGRanges</span><span class="op">(</span><span class="va">Chr</span>,<span class="va">bp</span><span class="op">-</span><span class="fl">1</span>,<span class="va">bp</span>,labels<span class="op">=</span><span class="va">hgnc</span><span class="op">)</span></span>
<span>  <span class="va">cis.regions</span> <span class="op">&lt;-</span> <span class="fu">toGRanges</span><span class="op">(</span><span class="va">Chr</span>,<span class="va">cis.start</span>,<span class="va">cis.end</span><span class="op">)</span></span>
<span>  <span class="va">loci</span> <span class="op">&lt;-</span> <span class="fu">toGRanges</span><span class="op">(</span><span class="va">Chr</span>,<span class="va">Start</span>,<span class="va">End</span><span class="op">)</span></span>
<span>  <span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"blue"</span><span class="op">)</span></span>
<span>  <span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">sentinels</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span>
<span>  <span class="va">kp</span> <span class="op">&lt;-</span> <span class="fu">plotKaryotype</span><span class="op">(</span>genome<span class="op">=</span><span class="st">"hg19"</span>,chromosomes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">sentinels</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu">kpAddBaseNumbers</span><span class="op">(</span><span class="va">kp</span><span class="op">)</span></span>
<span>  <span class="fu">kpPlotRegions</span><span class="op">(</span><span class="va">kp</span>, data<span class="op">=</span><span class="va">loci</span>,r0<span class="op">=</span><span class="fl">0.05</span>,r1<span class="op">=</span><span class="fl">0.15</span>,border<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span>
<span>  <span class="fu">kpPlotMarkers</span><span class="op">(</span><span class="va">kp</span>, data<span class="op">=</span><span class="va">sentinels</span>, labels<span class="op">=</span><span class="va">hgnc</span>, text.orientation<span class="op">=</span><span class="st">"vertical"</span>,</span>
<span>                cex<span class="op">=</span><span class="fl">0.5</span>, y<span class="op">=</span><span class="fl">0.3</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">hgnc</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hgnc</span><span class="op">)</span>, srt<span class="op">=</span><span class="fl">30</span>,</span>
<span>                ignore.chromosome.ends<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                adjust.label.position<span class="op">=</span><span class="cn">TRUE</span>, label.color<span class="op">=</span><span class="va">colors</span><span class="op">[</span><span class="fl">2</span><span class="op">-</span><span class="va">cis</span><span class="op">]</span>, label.dist<span class="op">=</span><span class="fl">0.002</span>,</span>
<span>                cex.axis<span class="op">=</span><span class="fl">3</span>, cex.lab<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"bottomright"</span>, bty<span class="op">=</span><span class="st">"n"</span>, pch<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">19</span>,<span class="fl">19</span><span class="op">)</span>, col<span class="op">=</span><span class="va">colors</span>, pt.cex<span class="op">=</span><span class="fl">0.4</span>,</span>
<span>         legend<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cis"</span>, <span class="st">"trans"</span><span class="op">)</span>, text.col<span class="op">=</span><span class="va">colors</span>, cex<span class="op">=</span><span class="fl">0.8</span>, horiz<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># panel &lt;- toGRanges(p.chr,p.start,p.end,labels=p.gene)</span></span>
<span><span class="co"># kpPlotLinks(kp, data=loci, data2=panel, col=colors[2-cis])</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:karyoplot"></span>
<img src="pQTLtools/karyoplot-1.png" alt="Karyoplot of cis/trans pQTLs" width="672" style="display:block; margin: auto"><p class="caption">
Figure 2.6: Karyoplot of cis/trans pQTLs
</p>
</div>
</div>
</div>
<div class="section level2" number="3">
<h2 id="genomic-regions-enrichment-analysis">
<span class="header-section-number">3</span> Genomic regions enrichment analysis<a class="anchor" aria-label="anchor" href="#genomic-regions-enrichment-analysis"></a>
</h2>
<p>It is now considerably easier with Genomic Regions Enrichment of Annotations Tool (GREAT).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">post</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">regions</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">job</span> <span class="op">&lt;-</span> <span class="fu">submitGreatJob</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span>, species<span class="op">=</span><span class="st">"hg19"</span>, version<span class="op">=</span><span class="st">"3.0.0"</span><span class="op">)</span></span>
<span>  <span class="va">et</span> <span class="op">&lt;-</span> <span class="fu">getEnrichmentTables</span><span class="op">(</span><span class="va">job</span>,download_by <span class="op">=</span> <span class="st">'tsv'</span><span class="op">)</span></span>
<span>  <span class="va">tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">'rbind'</span>,<span class="va">et</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">tb</span>,file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">regions</span>,<span class="st">".tsv"</span><span class="op">)</span>,quote<span class="op">=</span><span class="cn">FALSE</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>job<span class="op">=</span><span class="va">job</span>,tb<span class="op">=</span><span class="va">tb</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">M</span> <span class="op">&lt;-</span> <span class="fl">1e+6</span></span>
<span><span class="va">merge</span> <span class="op">&lt;-</span> <span class="va">merged</span> <span class="op">%&gt;%</span></span>
<span>         <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>chr<span class="op">=</span><span class="va">Chrom</span>, start<span class="op">=</span><span class="va">POS</span><span class="op">-</span><span class="va">M</span>, end<span class="op">=</span><span class="va">POS</span><span class="op">+</span><span class="va">M</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>         <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>start<span class="op">=</span><span class="fu">if_else</span><span class="op">(</span><span class="va">start</span><span class="op">&lt;</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="va">start</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>         <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">prot</span>,<span class="va">MarkerName</span>,<span class="va">chr</span>,<span class="va">start</span>,<span class="va">end</span><span class="op">)</span></span>
<span><span class="va">cistrans</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">merge</span>, <span class="va">chr</span>,<span class="va">start</span>,<span class="va">end</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">chr</span>,<span class="va">start</span>,<span class="va">end</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co"># All regions</span></span>
<span><span class="va">cistrans.post</span> <span class="op">&lt;-</span> <span class="fu">post</span><span class="op">(</span><span class="st">"cistrans"</span><span class="op">)</span></span>
<span><span class="va">job</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">cistrans.post</span>,<span class="va">job</span><span class="op">)</span></span>
<span><span class="fu">plotRegionGeneAssociationGraphs</span><span class="op">(</span><span class="va">job</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:GREAT-1"></span>
<img src="pQTLtools/GREAT-1.png" alt="GREAT plots" width="768" style="display:block; margin: auto"><p class="caption">
Figure 3.1: GREAT plots
</p>
</div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">availableOntologies</span><span class="op">(</span><span class="va">job</span><span class="op">)</span></span>
<span><span class="co"># plot of the top term</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfcol<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">plotRegionGeneAssociationGraphs</span><span class="op">(</span><span class="va">job</span>, ontology<span class="op">=</span><span class="st">"GO Molecular Function"</span><span class="op">)</span></span>
<span><span class="fu">plotRegionGeneAssociationGraphs</span><span class="op">(</span><span class="va">job</span>, ontology<span class="op">=</span><span class="st">"GO Biological Process"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:GREAT-2"></span>
<img src="pQTLtools/GREAT-2.png" alt="GREAT plots" width="768" style="display:block; margin: auto"><p class="caption">
Figure 3.2: GREAT plots
</p>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotRegionGeneAssociationGraphs</span><span class="op">(</span><span class="va">job</span>, ontology<span class="op">=</span><span class="st">"GO Cellular Component"</span><span class="op">)</span></span>
<span><span class="co"># Specific regions</span></span>
<span><span class="va">IL12B</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">merge</span>,<span class="va">prot</span><span class="op">==</span><span class="st">"IL.12B"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">chr</span>,<span class="va">start</span>,<span class="va">end</span><span class="op">)</span></span>
<span><span class="va">KITLG</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">merge</span>,<span class="va">prot</span><span class="op">==</span><span class="st">"SCF"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">chr</span>,<span class="va">start</span>,<span class="va">end</span><span class="op">)</span></span>
<span><span class="va">TNFSF10</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">merge</span>,<span class="va">prot</span><span class="op">==</span><span class="st">"TRAIL"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">chr</span>,<span class="va">start</span>,<span class="va">end</span><span class="op">)</span></span>
<span><span class="va">tb_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">r</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IL12B"</span>,<span class="st">"KITLG"</span>,<span class="st">"TNFSF10"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">r.post</span> <span class="op">&lt;-</span> <span class="fu">post</span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span>  <span class="va">tb_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">tb_all</span>,<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene<span class="op">=</span><span class="va">r</span>,<span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r.post</span>,<span class="va">tb</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; Don't make too frequent requests. The time break is 60s.</span></span>
<span><span class="co">#&gt; Please wait for 54s for the next request.</span></span>
<span><span class="co">#&gt; The time break can be set by `request_interval` argument.</span></span>
<span><span class="co">#&gt; Don't make too frequent requests. The time break is 60s.</span></span>
<span><span class="co">#&gt; Please wait for 57s for the next request.</span></span>
<span><span class="co">#&gt; The time break can be set by `request_interval` argument.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Don't make too frequent requests. The time break is 60s.</span></span>
<span><span class="co">#&gt; Please wait for 57s for the next request.</span></span>
<span><span class="co">#&gt; The time break can be set by `request_interval` argument.</span></span></code></pre></div>
<p>The top terms at Binomial p=1e-5 could be extracted as follows,</p>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-2">Table 3.1: </span>GREAT IL12B-KITLG-TNFSF10 results</caption>
<colgroup>
<col width="5%">
<col width="1%">
<col width="4%">
<col width="2%">
<col width="8%">
<col width="1%">
<col width="1%">
<col width="2%">
<col width="1%">
<col width="3%">
<col width="2%">
<col width="2%">
<col width="2%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="2%">
<col width="1%">
<col width="2%">
<col width="1%">
<col width="1%">
<col width="2%">
<col width="2%">
<col width="1%">
<col width="29%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">gene</th>
<th align="left">Ontology</th>
<th align="left">ID</th>
<th align="left">Desc</th>
<th align="right">BinomRank</th>
<th align="right">BinomP</th>
<th align="right">BinomBonfP</th>
<th align="right">BinomFdrQ</th>
<th align="right">RegionFoldEnrich</th>
<th align="right">ExpRegions</th>
<th align="right">ObsRegions</th>
<th align="right">GenomeFrac</th>
<th align="right">SetCov</th>
<th align="right">HyperRank</th>
<th align="right">HyperP</th>
<th align="right">HyperBonfP</th>
<th align="right">HyperFdrQ</th>
<th align="right">GeneFoldEnrich</th>
<th align="right">ExpGenes</th>
<th align="right">ObsGenes</th>
<th align="right">TotalGenes</th>
<th align="right">GeneSetCov</th>
<th align="right">TermCov</th>
<th align="left">Regions</th>
<th align="left">Genes</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">GO Biological Process.1100</td>
<td align="left">KITLG</td>
<td align="left">GO Biological Process</td>
<td align="left">GO: 0010874</td>
<td align="left">regulation of cholesterol efflux</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.001</td>
<td align="right">0.001</td>
<td align="right">272.460</td>
<td align="right">0.011</td>
<td align="right">3</td>
<td align="right">0.002</td>
<td align="right">0.429</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.002</td>
<td align="right">0.002</td>
<td align="right">265.309</td>
<td align="right">0.011</td>
<td align="right">3</td>
<td align="right">17</td>
<td align="right">0.250</td>
<td align="right">0.176</td>
<td align="left">/chr16:55993160-57993161,/chr7:93953894-95953895,/chr9:106661741-108661742</td>
<td align="left">ABCA1,CETP,PON1</td>
</tr>
<tr class="even">
<td align="left">GO Biological Process.2100</td>
<td align="left">KITLG</td>
<td align="left">GO Biological Process</td>
<td align="left">GO: 0032374</td>
<td align="left">regulation of cholesterol transport</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0.004</td>
<td align="right">0.002</td>
<td align="right">185.189</td>
<td align="right">0.016</td>
<td align="right">3</td>
<td align="right">0.002</td>
<td align="right">0.429</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0.012</td>
<td align="right">0.006</td>
<td align="right">140.945</td>
<td align="right">0.021</td>
<td align="right">3</td>
<td align="right">32</td>
<td align="right">0.250</td>
<td align="right">0.094</td>
<td align="left">/chr16:55993160-57993161,/chr7:93953894-95953895,/chr9:106661741-108661742</td>
<td align="left">ABCA1,CETP,PON1</td>
</tr>
<tr class="odd">
<td align="left">GO Biological Process.3100</td>
<td align="left">KITLG</td>
<td align="left">GO Biological Process</td>
<td align="left">GO: 0032368</td>
<td align="left">regulation of lipid transport</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0.094</td>
<td align="right">0.031</td>
<td align="right">67.046</td>
<td align="right">0.045</td>
<td align="right">3</td>
<td align="right">0.006</td>
<td align="right">0.429</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0.115</td>
<td align="right">0.038</td>
<td align="right">66.327</td>
<td align="right">0.045</td>
<td align="right">3</td>
<td align="right">68</td>
<td align="right">0.250</td>
<td align="right">0.044</td>
<td align="left">/chr16:55993160-57993161,/chr7:93953894-95953895,/chr9:106661741-108661742</td>
<td align="left">ABCA1,CETP,PON1</td>
</tr>
<tr class="even">
<td align="left">GO Biological Process.1102</td>
<td align="left">TNFSF10</td>
<td align="left">GO Biological Process</td>
<td align="left">GO: 0030195</td>
<td align="left">negative regulation of blood coagulation</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.006</td>
<td align="right">0.006</td>
<td align="right">170.502</td>
<td align="right">0.018</td>
<td align="right">3</td>
<td align="right">0.002</td>
<td align="right">0.375</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.034</td>
<td align="right">0.034</td>
<td align="right">100.228</td>
<td align="right">0.030</td>
<td align="right">3</td>
<td align="right">36</td>
<td align="right">0.200</td>
<td align="right">0.083</td>
<td align="left">/chr17:63224774-65224775,/chr19:43153099-45153100,/chr3:185449121-187449122</td>
<td align="left">APOH,KNG1,PLAUR</td>
</tr>
<tr class="odd">
<td align="left">GO Biological Process.2102</td>
<td align="left">TNFSF10</td>
<td align="left">GO Biological Process</td>
<td align="left">GO: 0050819</td>
<td align="left">negative regulation of coagulation</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0.014</td>
<td align="right">0.007</td>
<td align="right">129.538</td>
<td align="right">0.023</td>
<td align="right">3</td>
<td align="right">0.003</td>
<td align="right">0.375</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0.047</td>
<td align="right">0.024</td>
<td align="right">90.205</td>
<td align="right">0.033</td>
<td align="right">3</td>
<td align="right">40</td>
<td align="right">0.200</td>
<td align="right">0.075</td>
<td align="left">/chr17:63224774-65224775,/chr19:43153099-45153100,/chr3:185449121-187449122</td>
<td align="left">APOH,KNG1,PLAUR</td>
</tr>
<tr class="even">
<td align="left">GO Biological Process.3102</td>
<td align="left">TNFSF10</td>
<td align="left">GO Biological Process</td>
<td align="left">GO: 0072376</td>
<td align="left">protein activation cascade</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0.046</td>
<td align="right">0.015</td>
<td align="right">87.207</td>
<td align="right">0.034</td>
<td align="right">3</td>
<td align="right">0.004</td>
<td align="right">0.375</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0.196</td>
<td align="right">0.065</td>
<td align="right">56.378</td>
<td align="right">0.053</td>
<td align="right">3</td>
<td align="right">64</td>
<td align="right">0.200</td>
<td align="right">0.047</td>
<td align="left">/chr17:63224774-65224775,/chr1:195710915-197710916,/chr3:185449121-187449122</td>
<td align="left">APOH,CFH,KNG1</td>
</tr>
<tr class="odd">
<td align="left">GO Cellular Component.119</td>
<td align="left">TNFSF10</td>
<td align="left">GO Cellular Component</td>
<td align="left">GO: 0005615</td>
<td align="left">extracellular space</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.008</td>
<td align="right">0.008</td>
<td align="right">9.342</td>
<td align="right">0.642</td>
<td align="right">6</td>
<td align="right">0.080</td>
<td align="right">0.750</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">10.934</td>
<td align="right">0.732</td>
<td align="right">8</td>
<td align="right">880</td>
<td align="right">0.533</td>
<td align="right">0.009</td>
<td align="left">/chr14:93844946-95844947,/chr17:63224774-65224775,/chr18:28804862-30804863,/chr1:195710915-197710916,/chr3:171274231-173274232,/chr3:185449121-187449122</td>
<td align="left">APOH,CFH,CFHR3,KNG1,MEP1B,SERPINA1,SERPINA6,TNFSF10</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2" number="4">
<h2 id="eqtl-catalog-for-colocalization-analysis">
<span class="header-section-number">4</span> eQTL Catalog for colocalization analysis<a class="anchor" aria-label="anchor" href="#eqtl-catalog-for-colocalization-analysis"></a>
</h2>
<p>See example associated with <code>import_eQTLCatalogue</code>. A related function is <code>import_OpenGWAS</code> used to fetch data from OpenGWAS. The cis-pQTLs and 1e+6
flanking regions were considered and data are actually fetched from files stored locally. Only the first sentinel was used (r=1).</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">liftRegion</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,<span class="va">chain</span>,<span class="va">flanking</span><span class="op">=</span><span class="fl">1e6</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span>  <span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">x</span>,<span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="va">chr</span>,<span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="va">start</span>,<span class="va">end</span><span class="op">)</span><span class="op">)</span><span class="op">+</span><span class="va">flanking</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></span>
<span>  <span class="va">gr38</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/liftOver.html" class="external-link">liftOver</a></span><span class="op">(</span><span class="va">gr</span>, <span class="va">chain</span><span class="op">)</span></span>
<span>  <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqnames</a></span><span class="op">(</span><span class="va">gr38</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">start</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">gr38</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">end</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">end</a></span><span class="op">(</span><span class="va">gr38</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>chr<span class="op">=</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,start<span class="op">=</span><span class="va">start</span>,end<span class="op">=</span><span class="va">end</span>,region<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">chr</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,<span class="st">":"</span>,<span class="va">start</span>,<span class="st">"-"</span>,<span class="va">end</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">sumstats</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">prot</span>,<span class="va">chr</span>,<span class="va">region37</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"GWAS sumstats\n"</span><span class="op">)</span></span>
<span>  <span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"METAL/gwas2vcf"</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">".vcf.gz"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">gwas_stats</span> <span class="op">&lt;-</span> <span class="fu">gwasvcf</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gwasvcf/man/query_gwas.html" class="external-link">query_gwas</a></span><span class="op">(</span><span class="va">vcf</span>, chrompos <span class="op">=</span> <span class="va">region37</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu">gwasvcf</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gwasvcf/man/vcf_to_granges.html" class="external-link">vcf_to_granges</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html" class="external-link">keepSeqlevels</a></span><span class="op">(</span><span class="va">chr</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevels-wrappers.html" class="external-link">renameSeqlevels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="va">chr</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">gwas_stats_hg38</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/liftOver.html" class="external-link">liftOver</a></span><span class="op">(</span><span class="va">gwas_stats</span>, <span class="va">chain</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/transmute.html" class="external-link">transmute</a></span><span class="op">(</span>chromosome <span class="op">=</span> <span class="va">seqnames</span>,</span>
<span>                     position <span class="op">=</span> <span class="va">start</span>, <span class="va">REF</span>, <span class="va">ALT</span>, <span class="va">AF</span>, <span class="va">ES</span>, <span class="va">SE</span>, <span class="va">LP</span>, <span class="va">SS</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">chromosome</span>, <span class="va">position</span>, sep <span class="op">=</span> <span class="st">":"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>MAF <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/Extremes.html" class="external-link">pmin</a></span><span class="op">(</span><span class="va">AF</span>, <span class="fl">1</span><span class="op">-</span><span class="va">AF</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>row_count <span class="op">=</span> <span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">row_count</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>chromosome<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">""</span>,<span class="va">chromosome</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">s</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gwas_stats_hg38</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">position</span>, y <span class="op">=</span> <span class="va">LP</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">sentinel</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">SNP</span>,<span class="st">" association plot"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">s</span></span>
<span>  <span class="va">gwas_stats_hg38</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">gtex</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">gwas_stats_hg38</span>,<span class="va">ensGene</span>,<span class="va">region38</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"c. GTEx_v8 imported eQTL datasets\n"</span><span class="op">)</span></span>
<span>  <span class="va">fp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"eQTL-Catalogue"</span>,<span class="st">"tabix_ftp_paths_gtex.tsv"</span><span class="op">)</span></span>
<span>  <span class="va">web</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">fp</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">local</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">web</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        <span class="op">{</span></span>
<span>          <span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">ftp_path</span>,<span class="st">"/imported/|/ge/"</span><span class="op">)</span>,<span class="st">"["</span>,<span class="fl">3</span><span class="op">)</span>;</span>
<span>          <span class="va">ftp_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"~/rds/public_databases/GTEx/csv/"</span>,<span class="va">f</span><span class="op">)</span></span>
<span>        <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="va">gtex_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">local</span>, <span class="va">quant_method</span> <span class="op">==</span> <span class="st">"ge"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>             <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>qtl_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">study</span>, <span class="va">qtl_group</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">ftp_path_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">gtex_df</span><span class="op">$</span><span class="va">ftp_path</span><span class="op">)</span>, <span class="va">gtex_df</span><span class="op">$</span><span class="va">qtl_id</span><span class="op">)</span></span>
<span>  <span class="va">hdr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"eQTL-Catalogue"</span>,<span class="st">"column_names.GTEx"</span><span class="op">)</span></span>
<span>  <span class="va">column_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">hdr</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">safe_import</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/safely.html" class="external-link">safely</a></span><span class="op">(</span><span class="va">import_eQTLCatalogue</span><span class="op">)</span></span>
<span>  <span class="va">summary_list</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">ftp_path_list</span>,</span>
<span>                             <span class="op">~</span><span class="fu">safe_import</span><span class="op">(</span><span class="va">.</span>, <span class="va">region38</span>, selected_gene_id <span class="op">=</span> <span class="va">ensGene</span>, <span class="va">column_names</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">result_list</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">summary_list</span>, <span class="op">~</span><span class="va">.</span><span class="op">$</span><span class="va">result</span><span class="op">)</span></span>
<span>  <span class="va">result_list</span> <span class="op">&lt;-</span> <span class="va">result_list</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">result_list</span>, <span class="va">is.null</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">result_filtered</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">result_list</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">result_list</span>,<span class="va">nrow</span><span class="op">)</span><span class="op">!=</span><span class="fl">0</span><span class="op">]</span>,</span>
<span>                                <span class="op">~</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">.</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map_dfr.html" class="external-link">map_df</a></span><span class="op">(</span><span class="va">result_filtered</span>, <span class="op">~</span><span class="fu"><a href="../reference/run_coloc.html">run_coloc</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">gwas_stats_hg38</span><span class="op">)</span>, .id <span class="op">=</span> <span class="st">"qtl_id"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">gtex_coloc</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">prot</span>,<span class="va">chr</span>,<span class="va">ensGene</span>,<span class="va">chain</span>,<span class="va">region37</span>,<span class="va">region38</span>,<span class="va">out</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">gwas_stats_hg38</span> <span class="op">&lt;-</span> <span class="fu">sumstats</span><span class="op">(</span><span class="va">prot</span>,<span class="va">chr</span>,<span class="va">region37</span><span class="op">)</span></span>
<span>  <span class="va">df_gtex</span> <span class="op">&lt;-</span> <span class="fu">gtex</span><span class="op">(</span><span class="va">gwas_stats_hg38</span>,<span class="va">ensGene</span>,<span class="va">region38</span><span class="op">)</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/exists.html" class="external-link">exists</a></span><span class="op">(</span><span class="st">"df_gtex"</span><span class="op">)</span><span class="op">)</span> <span class="va">return</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">df_gtex</span>,file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">out</span>,<span class="st">".RDS"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">df_gtex</span>, <span class="op">-</span><span class="va">PP.H4.abf</span><span class="op">)</span></span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df_gtex</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">PP.H4.abf</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">ggtitle</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">sentinel</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">SNP</span>,<span class="st">" PP4 histogram"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>       <span class="fu">xlab</span><span class="op">(</span><span class="st">"PP4"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Frequency"</span><span class="op">)</span></span>
<span>  <span class="va">p</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">single_run</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">sentinel</span>,<span class="va">Chr</span><span class="op">)</span></span>
<span>  <span class="va">ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">inf1</span>,<span class="va">prot</span><span class="op">==</span><span class="va">sentinel</span><span class="op">[[</span><span class="st">"prot"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="va">ensRegion37</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">ss</span>,</span>
<span>                      <span class="op">{</span></span>
<span>                        <span class="va">start</span> <span class="op">&lt;-</span> <span class="va">start</span><span class="op">-</span><span class="va">M</span></span>
<span>                        <span class="kw">if</span> <span class="op">(</span><span class="va">start</span><span class="op">&lt;</span><span class="fl">0</span><span class="op">)</span> <span class="va">start</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span>                        <span class="va">end</span> <span class="op">&lt;-</span> <span class="va">end</span><span class="op">+</span><span class="va">M</span></span>
<span>                        <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">chr</span>,<span class="st">":"</span>,<span class="va">start</span>,<span class="st">"-"</span>,<span class="va">end</span><span class="op">)</span></span>
<span>                      <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="va">ensGene</span> <span class="op">&lt;-</span> <span class="va">ss</span><span class="op">[[</span><span class="st">"ensembl_gene_id"</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">ensRegion38</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="fu">liftRegion</span><span class="op">(</span><span class="va">ss</span>,<span class="va">chain</span><span class="op">)</span>,<span class="va">region</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">chr</span>,<span class="va">ensGene</span>,<span class="va">ensRegion37</span>,<span class="va">ensRegion38</span>,<span class="st">"\n"</span><span class="op">)</span></span>
<span>  <span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">sentinel</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">SNP</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu">gtex_coloc</span><span class="op">(</span><span class="va">sentinel</span><span class="op">[[</span><span class="st">"prot"</span><span class="op">]</span><span class="op">]</span>,<span class="va">chr</span>,<span class="va">ensGene</span>,<span class="va">chain</span>,<span class="va">ensRegion37</span>,<span class="va">ensRegion38</span>,<span class="va">f</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span>
<span><span class="va">HOME</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"HOME"</span><span class="op">)</span></span>
<span><span class="va">HPC_WORK</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"HPC_WORK"</span><span class="op">)</span></span>
<span><span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span></span>
<span><span class="va">M</span> <span class="op">&lt;-</span> <span class="fl">1e6</span></span>
<span><span class="va">sentinels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">cis.vs.trans</span>,<span class="va">cis</span><span class="op">)</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"eQTL-Catalogue"</span>,<span class="st">"hg19ToHg38.over.chain"</span><span class="op">)</span></span>
<span><span class="va">chain</span> <span class="op">&lt;-</span> <span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/Chain-class.html" class="external-link">import.chain</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span>
<span><span class="fu">gwasvcf</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gwasvcf/man/set_bcftools.html" class="external-link">set_bcftools</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">HPC_WORK</span>,<span class="st">"bin"</span>,<span class="st">"bcftools"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">sentinel</span> <span class="op">&lt;-</span> <span class="va">sentinels</span><span class="op">[</span><span class="va">r</span>,<span class="op">]</span></span>
<span><span class="fu">single_run</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; 8 ENSG00000164761 8:118935796-120964439 8:117923557-119952199 </span></span>
<span><span class="co">#&gt; GWAS sumstats</span></span>
<span><span class="co">#&gt; c. GTEx_v8 imported eQTL datasets</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Adipose_Subcutaneous.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Adipose_Visceral_Omentum.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Adrenal_Gland.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Artery_Aorta.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Artery_Coronary.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Artery_Tibial.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Amygdala.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Anterior_cingulate_cortex_BA24.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Caudate_basal_ganglia.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Cerebellar_Hemisphere.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Cerebellum.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Cortex.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Frontal_Cortex_BA9.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Hippocampus.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Hypothalamus.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Nucleus_accumbens_basal_ganglia.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Putamen_basal_ganglia.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Spinal_cord_cervical_c-1.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Brain_Substantia_nigra.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Breast_Mammary_Tissue.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Cells_Cultured_fibroblasts.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Cells_EBV-transformed_lymphocytes.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Colon_Sigmoid.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Colon_Transverse.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Esophagus_Gastroesophageal_Junction.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Esophagus_Mucosa.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Esophagus_Muscularis.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Heart_Atrial_Appendage.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Heart_Left_Ventricle.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Kidney_Cortex.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Liver.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Lung.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Minor_Salivary_Gland.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Muscle_Skeletal.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Nerve_Tibial.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Ovary.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Pancreas.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Pituitary.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Prostate.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Skin_Not_Sun_Exposed_Suprapubic.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Skin_Sun_Exposed_Lower_leg.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Small_Intestine_Terminal_Ileum.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Spleen.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Stomach.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Testis.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Thyroid.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Uterus.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Vagina.tsv.gz"</span></span>
<span><span class="co">#&gt; [1] "~/rds/public_databases/GTEx/csv/Whole_Blood.tsv.gz"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  2.43e-49  6.58e-44  3.69e-06  1.00e+00  1.66e-07 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 1.66e-05%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  5.76e-45  5.98e-44  8.76e-02  9.09e-01  3.63e-03 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.363%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  2.99e-44  2.42e-44  4.55e-01  3.68e-01  1.77e-01 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 17.7%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.86e-44  2.55e-44  5.87e-01  3.88e-01  2.53e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 2.53%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.13e-44  3.20e-44  4.77e-01  4.87e-01  3.67e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.67%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.87e-44  2.24e-44  5.88e-01  3.41e-01  7.10e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 7.1%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.65e-44  2.42e-44  5.54e-01  3.67e-01  7.83e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 7.83%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.44e-44  2.89e-44  5.23e-01  4.40e-01  3.71e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.71%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.65e-44  2.71e-44  5.55e-01  4.12e-01  3.22e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.22%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.51e-44  2.79e-44  5.34e-01  4.25e-01  4.14e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.14%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.50e-44  2.58e-44  5.32e-01  3.92e-01  7.61e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 7.61%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.11e-44  3.12e-44  4.73e-01  4.74e-01  5.35e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 5.35%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.10e-44  3.16e-44  4.72e-01  4.80e-01  4.80e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.8%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.56e-44  2.34e-44  5.42e-01  3.55e-01  1.03e-01 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 10.3%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.26e-44  3.01e-44  4.96e-01  4.57e-01  4.68e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.68%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.64e-44  2.55e-44  5.53e-01  3.87e-01  5.96e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 5.96%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.75e-44  2.56e-44  5.69e-01  3.89e-01  4.16e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.16%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.69e-44  2.59e-44  5.60e-01  3.93e-01  4.63e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.63%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.76e-44  2.57e-44  5.72e-01  3.90e-01  3.82e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.82%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.57e-44  2.83e-44  5.43e-01  4.30e-01  2.70e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 2.7%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  5.07e-55  6.58e-44  7.70e-12  1.00e+00  3.74e-13 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.74e-11%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.08e-44  3.31e-44  4.68e-01  5.02e-01  3.00e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.40e-44  2.92e-44  5.18e-01  4.45e-01  3.79e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.79%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  4.95e-46  6.53e-44  7.53e-03  9.92e-01  3.50e-04 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.035%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.74e-44  2.66e-44  5.69e-01  4.04e-01  2.68e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 2.68%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  2.39e-44  4.07e-44  3.64e-01  6.19e-01  1.69e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 1.69%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  7.06e-46  6.44e-44  1.07e-02  9.79e-01  1.05e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 1.05%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.32e-47  6.57e-44  5.04e-04  9.99e-01  2.32e-05 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.00232%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  1.62e-45  6.41e-44  2.46e-02  9.74e-01  1.19e-03 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.119%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.47e-44  2.81e-44  5.28e-01  4.28e-01  4.44e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.44%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.05e-44  3.15e-44  4.64e-01  4.79e-01  5.72e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 5.72%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  6.84e-46  6.50e-44  1.04e-02  9.89e-01  1.04e-03 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.104%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.73e-44  2.47e-44  5.68e-01  3.76e-01  5.68e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 5.68%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.98e-44  2.46e-44  6.04e-01  3.74e-01  2.21e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 2.21%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  1.22e-44  5.30e-44  1.86e-01  8.06e-01  7.91e-03 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.791%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.58e-44  2.76e-44  5.44e-01  4.19e-01  3.65e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.65%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  1.25e-45  6.45e-44  1.90e-02  9.80e-01  9.63e-04 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.0963%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.97e-44  2.37e-44  6.04e-01  3.60e-01  3.55e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.55%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.34e-44  3.02e-44  5.07e-01  4.58e-01  3.42e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.42%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.59e-44  2.84e-44  5.46e-01  4.32e-01  2.16e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 2.16%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  2.79e-44  3.67e-44  4.25e-01  5.58e-01  1.77e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 1.77%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.69e-44  2.38e-44  5.61e-01  3.62e-01  7.70e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 7.7%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.52e-44  2.76e-44  5.35e-01  4.19e-01  4.61e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.61%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.74e-44  2.56e-44  5.69e-01  3.90e-01  4.14e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.14%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.99e-44  2.38e-44  6.06e-01  3.62e-01  3.16e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.16%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  9.86e-45  5.55e-44  1.50e-01  8.44e-01  6.55e-03 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 0.655%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.70e-44  2.63e-44  5.63e-01  3.99e-01  3.81e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 3.81%"</span></span>
<span><span class="co">#&gt; PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf </span></span>
<span><span class="co">#&gt;  3.86e-44  2.45e-44  5.86e-01  3.73e-01  4.08e-02 </span></span>
<span><span class="co">#&gt; [1] "PP abf for shared variant: 4.08%"</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:coloc"></span>
<img src="pQTLtools/coloc-1.png" alt="Association plot &amp; histogram" width="768" style="display:block; margin: auto"><p class="caption">
Figure 4.1: Association plot &amp; histogram
</p>
</div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The results are also loadable as follows.</span></span>
<span><span class="va">coloc_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"OPG-chr8:120081031_C_T.RDS"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>Tissue<span class="op">=</span><span class="va">qtl_id</span>, H0<span class="op">=</span><span class="va">PP.H0.abf</span>,H1<span class="op">=</span><span class="va">PP.H1.abf</span>,</span>
<span>                          H2<span class="op">=</span><span class="va">PP.H2.abf</span>,H3<span class="op">=</span><span class="va">PP.H3.abf</span>,H4<span class="op">=</span><span class="va">PP.H4.abf</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>            <span class="fu">mutate</span><span class="op">(</span>Tissue<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"GTEx_V8_"</span>,<span class="st">""</span>,<span class="va">Tissue</span><span class="op">)</span>,</span>
<span>                   H0<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">H0</span>,<span class="fl">2</span><span class="op">)</span>,H1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">H1</span>,<span class="fl">2</span><span class="op">)</span>,H2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">H2</span>,<span class="fl">2</span><span class="op">)</span>,H3<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">H3</span>,<span class="fl">2</span><span class="op">)</span>,H4<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">H4</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                   <span class="fu">arrange</span><span class="op">(</span><span class="op">-</span><span class="va">H4</span><span class="op">)</span></span>
<span><span class="va">ktitle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">sentinel</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Colocalization results for "</span>,<span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">SNP</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">coloc_df</span>,caption<span class="op">=</span><span class="va">ktitle</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:coloc">Table 4.1: </span>Colocalization results for OPG-chr8:120081031_C_T</caption>
<colgroup>
<col width="58%">
<col width="9%">
<col width="4%">
<col width="4%">
<col width="7%">
<col width="7%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">Tissue</th>
<th align="right">nsnps</th>
<th align="right">H0</th>
<th align="right">H1</th>
<th align="right">H2</th>
<th align="right">H3</th>
<th align="right">H4</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Adrenal_Gland</td>
<td align="right">6741</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.46</td>
<td align="right">0.37</td>
<td align="right">0.18</td>
</tr>
<tr class="even">
<td align="left">Brain_Hippocampus</td>
<td align="right">6733</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.54</td>
<td align="right">0.36</td>
<td align="right">0.10</td>
</tr>
<tr class="odd">
<td align="left">Brain_Amygdala</td>
<td align="right">6701</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.55</td>
<td align="right">0.37</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">Brain_Cerebellum</td>
<td align="right">6738</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.53</td>
<td align="right">0.39</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">Small_Intestine_Terminal_Ileum</td>
<td align="right">6738</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.56</td>
<td align="right">0.36</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">Artery_Tibial</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.59</td>
<td align="right">0.34</td>
<td align="right">0.07</td>
</tr>
<tr class="odd">
<td align="left">Brain_Nucleus_accumbens_basal_ganglia</td>
<td align="right">6741</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.55</td>
<td align="right">0.39</td>
<td align="right">0.06</td>
</tr>
<tr class="even">
<td align="left">Liver</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.46</td>
<td align="right">0.48</td>
<td align="right">0.06</td>
</tr>
<tr class="odd">
<td align="left">Minor_Salivary_Gland</td>
<td align="right">6721</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.57</td>
<td align="right">0.38</td>
<td align="right">0.06</td>
</tr>
<tr class="even">
<td align="left">Brain_Cortex</td>
<td align="right">6741</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.47</td>
<td align="right">0.47</td>
<td align="right">0.05</td>
</tr>
<tr class="odd">
<td align="left">Brain_Frontal_Cortex_BA9</td>
<td align="right">6736</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.47</td>
<td align="right">0.48</td>
<td align="right">0.05</td>
</tr>
<tr class="even">
<td align="left">Brain_Hypothalamus</td>
<td align="right">6737</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.50</td>
<td align="right">0.46</td>
<td align="right">0.05</td>
</tr>
<tr class="odd">
<td align="left">Brain_Spinal_cord_cervical_c-1</td>
<td align="right">6725</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.56</td>
<td align="right">0.39</td>
<td align="right">0.05</td>
</tr>
<tr class="even">
<td align="left">Spleen</td>
<td align="right">6740</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.53</td>
<td align="right">0.42</td>
<td align="right">0.05</td>
</tr>
<tr class="odd">
<td align="left">Artery_Coronary</td>
<td align="right">6740</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.48</td>
<td align="right">0.49</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">Brain_Anterior_cingulate_cortex_BA24</td>
<td align="right">6728</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.52</td>
<td align="right">0.44</td>
<td align="right">0.04</td>
</tr>
<tr class="odd">
<td align="left">Brain_Cerebellar_Hemisphere</td>
<td align="right">6738</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.53</td>
<td align="right">0.42</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">Brain_Putamen_basal_ganglia</td>
<td align="right">6732</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.57</td>
<td align="right">0.39</td>
<td align="right">0.04</td>
</tr>
<tr class="odd">
<td align="left">Brain_Substantia_nigra</td>
<td align="right">6706</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.57</td>
<td align="right">0.39</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">Colon_Sigmoid</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.52</td>
<td align="right">0.44</td>
<td align="right">0.04</td>
</tr>
<tr class="odd">
<td align="left">Kidney_Cortex</td>
<td align="right">6625</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.53</td>
<td align="right">0.43</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">Ovary</td>
<td align="right">6739</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.54</td>
<td align="right">0.42</td>
<td align="right">0.04</td>
</tr>
<tr class="odd">
<td align="left">Pituitary</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.60</td>
<td align="right">0.36</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">Stomach</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.57</td>
<td align="right">0.39</td>
<td align="right">0.04</td>
</tr>
<tr class="odd">
<td align="left">Uterus</td>
<td align="right">6735</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.56</td>
<td align="right">0.40</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">Vagina</td>
<td align="right">6719</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.59</td>
<td align="right">0.37</td>
<td align="right">0.04</td>
</tr>
<tr class="odd">
<td align="left">Artery_Aorta</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.59</td>
<td align="right">0.39</td>
<td align="right">0.03</td>
</tr>
<tr class="even">
<td align="left">Brain_Caudate_basal_ganglia</td>
<td align="right">6741</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.56</td>
<td align="right">0.41</td>
<td align="right">0.03</td>
</tr>
<tr class="odd">
<td align="left">Breast_Mammary_Tissue</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.54</td>
<td align="right">0.43</td>
<td align="right">0.03</td>
</tr>
<tr class="even">
<td align="left">Cells_EBV-transformed_lymphocytes</td>
<td align="right">6730</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.47</td>
<td align="right">0.50</td>
<td align="right">0.03</td>
</tr>
<tr class="odd">
<td align="left">Esophagus_Gastroesophageal_Junction</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.57</td>
<td align="right">0.40</td>
<td align="right">0.03</td>
</tr>
<tr class="even">
<td align="left">Prostate</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.51</td>
<td align="right">0.46</td>
<td align="right">0.03</td>
</tr>
<tr class="odd">
<td align="left">Testis</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.61</td>
<td align="right">0.36</td>
<td align="right">0.03</td>
</tr>
<tr class="even">
<td align="left">Esophagus_Mucosa</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.36</td>
<td align="right">0.62</td>
<td align="right">0.02</td>
</tr>
<tr class="odd">
<td align="left">Muscle_Skeletal</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.60</td>
<td align="right">0.37</td>
<td align="right">0.02</td>
</tr>
<tr class="even">
<td align="left">Skin_Not_Sun_Exposed_Suprapubic</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.55</td>
<td align="right">0.43</td>
<td align="right">0.02</td>
</tr>
<tr class="odd">
<td align="left">Skin_Sun_Exposed_Lower_leg</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.42</td>
<td align="right">0.56</td>
<td align="right">0.02</td>
</tr>
<tr class="even">
<td align="left">Esophagus_Muscularis</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.01</td>
<td align="right">0.98</td>
<td align="right">0.01</td>
</tr>
<tr class="odd">
<td align="left">Nerve_Tibial</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.19</td>
<td align="right">0.81</td>
<td align="right">0.01</td>
</tr>
<tr class="even">
<td align="left">Thyroid</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.15</td>
<td align="right">0.84</td>
<td align="right">0.01</td>
</tr>
<tr class="odd">
<td align="left">Adipose_Subcutaneous</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.00</td>
<td align="right">1.00</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td align="left">Adipose_Visceral_Omentum</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.09</td>
<td align="right">0.91</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">Cells_Cultured_fibroblasts</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.00</td>
<td align="right">1.00</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td align="left">Colon_Transverse</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.01</td>
<td align="right">0.99</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">Heart_Atrial_Appendage</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.00</td>
<td align="right">1.00</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td align="left">Heart_Left_Ventricle</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.02</td>
<td align="right">0.97</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">Lung</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.01</td>
<td align="right">0.99</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td align="left">Pancreas</td>
<td align="right">6742</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0.02</td>
<td align="right">0.98</td>
<td align="right">0.00</td>
</tr>
</tbody>
</table>
<p>The function <code>sumstats()</code> obtained meta-analysis summary statistics (in build 37 and therefore lifted over to build 38) to be used in colocalization
analysis. The output are saved in the <code>.RDS</code> files. Note that <code>ftp_path</code> changes from eQTL Catalog to local files.</p>
</div>
<div class="section level2" number="5">
<h2 id="mendelian-randomisation-mr">
<span class="header-section-number">5</span> Mendelian Randomisation (MR)<a class="anchor" aria-label="anchor" href="#mendelian-randomisation-mr"></a>
</h2>
<div class="section level3" number="5.1">
<h3 id="pqtl-based-mr">
<span class="header-section-number">5.1</span> pQTL-based MR<a class="anchor" aria-label="anchor" href="#pqtl-based-mr"></a>
</h3>
<p>The function <code>pqtlMR</code> has an attractive feature that multiple pQTLs
can be used together for conducting MR with a list of outcomes from MR-Base. For generic applications
the <code><a href="../reference/run_TwoSampleMR.html">run_TwoSampleMR()</a></code> function can be used.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"Ins.csv"</span><span class="op">)</span></span>
<span><span class="va">ivs</span> <span class="op">&lt;-</span> <span class="fu">format_data</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">caption4</span> <span class="op">&lt;-</span> <span class="st">"ABO/LIFR variants and CHD/FEV1"</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">ivs</span>, caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption4</span>,<span class="st">"(instruments)"</span><span class="op">)</span>,digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-3">Table 5.1: </span>ABO/LIFR variants and CHD/FEV1 (instruments)</caption>
<colgroup>
<col width="5%">
<col width="13%">
<col width="13%">
<col width="7%">
<col width="8%">
<col width="7%">
<col width="8%">
<col width="5%">
<col width="10%">
<col width="12%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">SNP</th>
<th align="left">effect_allele.exposure</th>
<th align="left">other_allele.exposure</th>
<th align="right">eaf.exposure</th>
<th align="right">beta.exposure</th>
<th align="right">se.exposure</th>
<th align="right">pval.exposure</th>
<th align="left">exposure</th>
<th align="left">mr_keep.exposure</th>
<th align="left">pval_origin.exposure</th>
<th align="left">id.exposure</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rs505922</td>
<td align="left">C</td>
<td align="left">T</td>
<td align="right">0.313</td>
<td align="right">1.298</td>
<td align="right">0.014</td>
<td align="right">0</td>
<td align="left">ABO</td>
<td align="left">TRUE</td>
<td align="left">reported</td>
<td align="left">dU3C0x</td>
</tr>
<tr class="even">
<td align="left">rs635634</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.180</td>
<td align="right">-0.300</td>
<td align="right">0.032</td>
<td align="right">0</td>
<td align="left">LIFR</td>
<td align="left">TRUE</td>
<td align="left">reported</td>
<td align="left">YmGdhW</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ieu-a-7"</span>,<span class="st">"ebi-a-GCST007432"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/pqtlMR.html">pqtlMR</a></span><span class="op">(</span><span class="va">ivs</span>, <span class="va">ids</span><span class="op">)</span></span>
<span><span class="co">#&gt; API: public: http://gwas-api.mrcieu.ac.uk/</span></span>
<span><span class="co">#&gt; Extracting data for 2 SNP(s) from 2 GWAS(s)</span></span>
<span><span class="co">#&gt; Harmonising ABO (dU3C0x) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span></span>
<span><span class="co">#&gt; Harmonising LIFR (YmGdhW) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span></span>
<span><span class="co">#&gt; Harmonising ABO (dU3C0x) and Coronary heart disease || id:ieu-a-7 (ieu-a-7)</span></span>
<span><span class="co">#&gt; Harmonising LIFR (YmGdhW) and Coronary heart disease || id:ieu-a-7 (ieu-a-7)</span></span>
<span><span class="co">#&gt; Analysing 'dU3C0x' on 'ebi-a-GCST007432'</span></span>
<span><span class="co">#&gt; Analysing 'dU3C0x' on 'ieu-a-7'</span></span>
<span><span class="co">#&gt; Analysing 'YmGdhW' on 'ebi-a-GCST007432'</span></span>
<span><span class="co">#&gt; Analysing 'YmGdhW' on 'ieu-a-7'</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"pQTL-combined-result.txt"</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="va">result</span>,caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption4</span>, <span class="st">"(result)"</span><span class="op">)</span>,digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>
<span id="tab:unnamed-chunk-3">Table 5.1: </span>ABO/LIFR variants and CHD/FEV1 (result)</caption>
<colgroup>
<col width="10%">
<col width="14%">
<col width="39%">
<col width="7%">
<col width="9%">
<col width="4%">
<col width="5%">
<col width="5%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="left">outcome</th>
<th align="left">exposure</th>
<th align="left">method</th>
<th align="right">nsnp</th>
<th align="right">b</th>
<th align="right">se</th>
<th align="right">pval</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">dU3C0x</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ABO</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">-0.011</td>
<td align="right">0.002</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">dU3C0x</td>
<td align="left">ieu-a-7</td>
<td align="left">Coronary heart disease || id:ieu-a-7</td>
<td align="left">ABO</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">0.033</td>
<td align="right">0.007</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">YmGdhW</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">LIFR</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">0.062</td>
<td align="right">0.010</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">YmGdhW</td>
<td align="left">ieu-a-7</td>
<td align="left">Coronary heart disease || id:ieu-a-7</td>
<td align="left">LIFR</td>
<td align="left">Wald ratio</td>
<td align="right">1</td>
<td align="right">-0.257</td>
<td align="right">0.039</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">single</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"pQTL-combined-single.txt"</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">single</span>,<span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"All"</span>,<span class="va">SNP</span><span class="op">)</span><span class="op">)</span>, caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption4</span>, <span class="st">"(single)"</span><span class="op">)</span>,digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-3">Table 5.1: </span>ABO/LIFR variants and CHD/FEV1 (single)</caption>
<colgroup>
<col width="2%">
<col width="6%">
<col width="35%">
<col width="9%">
<col width="12%">
<col width="8%">
<col width="6%">
<col width="6%">
<col width="6%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">exposure</th>
<th align="left">outcome</th>
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="right">samplesize</th>
<th align="left">SNP</th>
<th align="left">b</th>
<th align="left">se</th>
<th align="left">p</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">ABO</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">dU3C0x</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs505922</td>
<td align="left">-0.0109</td>
<td align="left">0.00193</td>
<td align="left">1.35e-08</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">ABO</td>
<td align="left">Coronary heart disease || id:ieu-a-7</td>
<td align="left">dU3C0x</td>
<td align="left">ieu-a-7</td>
<td align="right">184305</td>
<td align="left">rs505922</td>
<td align="left">0.0334</td>
<td align="left">0.00730</td>
<td align="left">4.63e-06</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">LIFR</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">YmGdhW</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs635634</td>
<td align="left">0.0620</td>
<td align="left">0.01000</td>
<td align="left">5.65e-10</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">LIFR</td>
<td align="left">Coronary heart disease || id:ieu-a-7</td>
<td align="left">YmGdhW</td>
<td align="left">ieu-a-7</td>
<td align="right">184305</td>
<td align="left">rs635634</td>
<td align="left">-0.2572</td>
<td align="left">0.03904</td>
<td align="left">4.47e-11</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"harmonise"</span>,<span class="st">"result"</span>,<span class="st">"single"</span><span class="op">)</span>,</span>
<span>                 <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pQTL-combined-"</span>,<span class="va">x</span>,<span class="st">".txt"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="5.2">
<h3 id="two-sample-mr">
<span class="header-section-number">5.2</span> Two-sample MR<a class="anchor" aria-label="anchor" href="#two-sample-mr"></a>
</h3>
<p>The documentation example is quoted here,</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outcomes</span> <span class="op">&lt;-</span> <span class="st">"ebi-a-GCST007432"</span></span>
<span><span class="va">prot</span> <span class="op">&lt;-</span> <span class="st">"MMP.10"</span></span>
<span><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">"cis"</span></span>
<span><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">type</span>,<span class="st">".mrx"</span><span class="op">)</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="va">f</span><span class="op">)</span>,</span>
<span>                header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">exposure</span> <span class="op">&lt;-</span> <span class="fu">format_data</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">d</span>,<span class="op">{</span><span class="va">P</span><span class="op">=</span><span class="fl">10</span><span class="op">^</span><span class="va">logP</span><span class="op">}</span><span class="op">)</span>, phenotype_col<span class="op">=</span><span class="st">"prot"</span>, snp_col<span class="op">=</span><span class="st">"rsid"</span>,</span>
<span>                        chr_col<span class="op">=</span><span class="st">"Chromosome"</span>, pos_col<span class="op">=</span><span class="st">"Posistion"</span>,</span>
<span>                        effect_allele_col<span class="op">=</span><span class="st">"Allele1"</span>, other_allele_col<span class="op">=</span><span class="st">"Allele2"</span>,</span>
<span>                        eaf_col<span class="op">=</span><span class="st">"Freq1"</span>, beta_col<span class="op">=</span><span class="st">"Effect"</span>, se_col<span class="op">=</span><span class="st">"StdErr"</span>,</span>
<span>                        pval_col<span class="op">=</span><span class="st">"P"</span>, log_pval<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>                        samplesize_col<span class="op">=</span><span class="st">"N"</span><span class="op">)</span></span>
<span><span class="va">clump</span> <span class="op">&lt;-</span> <span class="fu">clump_data</span><span class="op">(</span><span class="va">exposure</span><span class="op">)</span></span>
<span><span class="co">#&gt; Please look at vignettes for options on running this locally if you need to run many instances of this command.</span></span>
<span><span class="co">#&gt; Clumping ZoHewj, 1106 variants, using EUR population reference</span></span>
<span><span class="co">#&gt; Removing 1102 of 1106 variants due to LD with other variants or absence from LD reference panel</span></span>
<span><span class="va">outcome</span> <span class="op">&lt;-</span> <span class="fu">extract_outcome_data</span><span class="op">(</span>snps<span class="op">=</span><span class="va">exposure</span><span class="op">$</span><span class="va">SNP</span>,outcomes<span class="op">=</span><span class="va">outcomes</span><span class="op">)</span></span>
<span><span class="co">#&gt; Extracting data for 1106 SNP(s) from 1 GWAS(s)</span></span>
<span><span class="co">#&gt; Finding proxies for 155 SNPs in outcome ebi-a-GCST007432</span></span>
<span><span class="co">#&gt; Extracting data for 155 SNP(s) from 1 GWAS(s)</span></span>
<span><span class="va">harmonise</span> <span class="op">&lt;-</span> <span class="fu">harmonise_data</span><span class="op">(</span><span class="va">clump</span>,<span class="va">outcome</span><span class="op">)</span></span>
<span><span class="co">#&gt; Harmonising MMP.10 (ZoHewj) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span></span>
<span><span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">outcomes</span>,<span class="va">prot</span>,<span class="va">type</span>,sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/run_TwoSampleMR.html">run_TwoSampleMR</a></span><span class="op">(</span><span class="va">harmonise</span>, mr_plot<span class="op">=</span><span class="st">"pQTLtools"</span>, prefix<span class="op">=</span><span class="va">prefix</span><span class="op">)</span></span>
<span><span class="co">#&gt; Analysing 'ZoHewj' on 'ebi-a-GCST007432'</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runTwoSampleMR-1"></span>
<img src="pQTLtools/runTwoSampleMR-1.png" alt="Two-sample MR" width="672" style="display:block; margin: auto"><p class="caption">
Figure 5.1: Two-sample MR
</p>
</div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runTwoSampleMR-2"></span>
<img src="pQTLtools/runTwoSampleMR-2.png" alt="Two-sample MR" width="672" style="display:block; margin: auto"><p class="caption">
Figure 5.2: Two-sample MR
</p>
</div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runTwoSampleMR-3"></span>
<img src="pQTLtools/runTwoSampleMR-3.png" alt="Two-sample MR" width="672" style="display:block; margin: auto"><p class="caption">
Figure 5.3: Two-sample MR
</p>
</div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runTwoSampleMR-4"></span>
<img src="pQTLtools/runTwoSampleMR-4.png" alt="Two-sample MR" width="672" style="display:block; margin: auto"><p class="caption">
Figure 5.4: Two-sample MR
</p>
</div>
<p>The output is contained in individual <code>.txt</code> files, together with the scatter, forest, funnel and leave-one-out plots.</p>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-4">Table 5.2: </span>MMP.10 variants and FEV1 (result)</caption>
<colgroup>
<col width="9%">
<col width="13%">
<col width="30%">
<col width="7%">
<col width="20%">
<col width="3%">
<col width="5%">
<col width="4%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="left">outcome</th>
<th align="left">exposure</th>
<th align="left">method</th>
<th align="right">nsnp</th>
<th align="right">b</th>
<th align="right">se</th>
<th align="right">pval</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="left">MR Egger</td>
<td align="right">4</td>
<td align="right">-0.001</td>
<td align="right">0.010</td>
<td align="right">0.959</td>
</tr>
<tr class="even">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="left">Weighted median</td>
<td align="right">4</td>
<td align="right">0.002</td>
<td align="right">0.006</td>
<td align="right">0.750</td>
</tr>
<tr class="odd">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="left">Inverse variance weighted</td>
<td align="right">4</td>
<td align="right">0.002</td>
<td align="right">0.006</td>
<td align="right">0.728</td>
</tr>
<tr class="even">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="left">Simple mode</td>
<td align="right">4</td>
<td align="right">0.004</td>
<td align="right">0.010</td>
<td align="right">0.704</td>
</tr>
<tr class="odd">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="left">Weighted mode</td>
<td align="right">4</td>
<td align="right">0.001</td>
<td align="right">0.007</td>
<td align="right">0.883</td>
</tr>
</tbody>
</table>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-4">Table 5.2: </span>MMP.10 variants and FEV1 (heterogeneity)</caption>
<colgroup>
<col width="10%">
<col width="14%">
<col width="31%">
<col width="7%">
<col width="21%">
<col width="5%">
<col width="4%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="left">outcome</th>
<th align="left">exposure</th>
<th align="left">method</th>
<th align="right">Q</th>
<th align="right">Q_df</th>
<th align="right">Q_pval</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="left">MR Egger</td>
<td align="right">0.114</td>
<td align="right">2</td>
<td align="right">0.944</td>
</tr>
<tr class="even">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="left">Inverse variance weighted</td>
<td align="right">0.222</td>
<td align="right">3</td>
<td align="right">0.974</td>
</tr>
</tbody>
</table>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-4">Table 5.2: </span>MMP.10 variants and FEV1 (pleiotropy)</caption>
<colgroup>
<col width="11%">
<col width="16%">
<col width="36%">
<col width="8%">
<col width="15%">
<col width="5%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="left">outcome</th>
<th align="left">exposure</th>
<th align="right">egger_intercept</th>
<th align="right">se</th>
<th align="right">pval</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">MMP.10</td>
<td align="right">0.001</td>
<td align="right">0.004</td>
<td align="right">0.774</td>
</tr></tbody>
</table>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-4">Table 5.2: </span>MMP.10 variants and FEV1 (single)</caption>
<colgroup>
<col width="6%">
<col width="27%">
<col width="8%">
<col width="12%">
<col width="7%">
<col width="23%">
<col width="5%">
<col width="4%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">exposure</th>
<th align="left">outcome</th>
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="right">samplesize</th>
<th align="left">SNP</th>
<th align="right">b</th>
<th align="right">se</th>
<th align="right">p</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs11225354</td>
<td align="right">0.006</td>
<td align="right">0.012</td>
<td align="right">0.598</td>
</tr>
<tr class="even">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs142915220</td>
<td align="right">-0.004</td>
<td align="right">0.029</td>
<td align="right">0.889</td>
</tr>
<tr class="odd">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs17099622</td>
<td align="right">0.004</td>
<td align="right">0.022</td>
<td align="right">0.845</td>
</tr>
<tr class="even">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs17860955</td>
<td align="right">0.001</td>
<td align="right">0.007</td>
<td align="right">0.934</td>
</tr>
<tr class="odd">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">All - Inverse variance weighted</td>
<td align="right">0.002</td>
<td align="right">0.006</td>
<td align="right">0.728</td>
</tr>
<tr class="even">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">All - MR Egger</td>
<td align="right">-0.001</td>
<td align="right">0.010</td>
<td align="right">0.959</td>
</tr>
</tbody>
</table>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-4">Table 5.2: </span>MMP.10 variants and FEV1 (loo)</caption>
<colgroup>
<col width="7%">
<col width="32%">
<col width="10%">
<col width="14%">
<col width="9%">
<col width="10%">
<col width="5%">
<col width="5%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">exposure</th>
<th align="left">outcome</th>
<th align="left">id.exposure</th>
<th align="left">id.outcome</th>
<th align="right">samplesize</th>
<th align="left">SNP</th>
<th align="right">b</th>
<th align="right">se</th>
<th align="right">p</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs11225354</td>
<td align="right">0.001</td>
<td align="right">0.007</td>
<td align="right">0.918</td>
</tr>
<tr class="even">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs142915220</td>
<td align="right">0.002</td>
<td align="right">0.006</td>
<td align="right">0.700</td>
</tr>
<tr class="odd">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs17099622</td>
<td align="right">0.002</td>
<td align="right">0.006</td>
<td align="right">0.759</td>
</tr>
<tr class="even">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">rs17860955</td>
<td align="right">0.005</td>
<td align="right">0.010</td>
<td align="right">0.636</td>
</tr>
<tr class="odd">
<td align="left">MMP.10</td>
<td align="left">FEV1 || id:ebi-a-GCST007432</td>
<td align="left">ZoHewj</td>
<td align="left">ebi-a-GCST007432</td>
<td align="right">321047</td>
<td align="left">All</td>
<td align="right">0.002</td>
<td align="right">0.006</td>
<td align="right">0.728</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3" number="5.3">
<h3 id="mr-using-cis-trans-and-cistrans-pan-instruments">
<span class="header-section-number">5.3</span> MR using cis, trans and cis+trans (pan) instruments<a class="anchor" aria-label="anchor" href="#mr-using-cis-trans-and-cistrans-pan-instruments"></a>
</h3>
<p>This is illustrated with IL-12B.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">efo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"efo.txt"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">d3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"IL.12B.txt"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>MRBASEID<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">outcome</span>,<span class="st">"id:"</span><span class="op">)</span>,<span class="st">"["</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>,y<span class="op">=</span><span class="va">b</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">outcome</span>,<span class="op">-</span><span class="va">method</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">left_join</span><span class="op">(</span><span class="va">efo</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">cistrans</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(MRBASEID)`</span></span>
<span><span class="fu">kable</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">d3</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MRBASEID"</span>,<span class="st">"trait"</span>,<span class="st">"cistrans"</span>,<span class="st">"nsnp"</span>,<span class="st">"b"</span>,<span class="st">"se"</span>,<span class="st">"pval"</span><span class="op">)</span><span class="op">]</span>,<span class="fl">29</span><span class="op">)</span>,caption<span class="op">=</span><span class="st">"MR with IL-12B variants"</span>,digits<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:mr">Table 5.3: </span>MR with IL-12B variants</caption>
<colgroup>
<col width="25%">
<col width="35%">
<col width="10%">
<col width="5%">
<col width="8%">
<col width="7%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">MRBASEID</th>
<th align="left">trait</th>
<th align="left">cistrans</th>
<th align="right">nsnp</th>
<th align="right">b</th>
<th align="right">se</th>
<th align="right">pval</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">bbj-a-123</td>
<td align="left">Graves disease</td>
<td align="left">cis</td>
<td align="right">21</td>
<td align="right">0.008</td>
<td align="right">0.101</td>
<td align="right">0.938</td>
</tr>
<tr class="even">
<td align="left">ebi-a-GCST003156</td>
<td align="left">systemic lupus erythematosus</td>
<td align="left">cis</td>
<td align="right">39</td>
<td align="right">-0.096</td>
<td align="right">0.068</td>
<td align="right">0.157</td>
</tr>
<tr class="odd">
<td align="left">ebi-a-GCST005528</td>
<td align="left">juvenile idiopathic arthritis</td>
<td align="left">cis</td>
<td align="right">1</td>
<td align="right">-0.167</td>
<td align="right">0.114</td>
<td align="right">0.140</td>
</tr>
<tr class="even">
<td align="left">ebi-a-GCST005581</td>
<td align="left">primary biliary cirrhosis</td>
<td align="left">cis</td>
<td align="right">2</td>
<td align="right">0.006</td>
<td align="right">0.163</td>
<td align="right">0.972</td>
</tr>
<tr class="odd">
<td align="left">finn-a-AB1_HIV</td>
<td align="left">HIV infection</td>
<td align="left">cis</td>
<td align="right">29</td>
<td align="right">-0.129</td>
<td align="right">0.280</td>
<td align="right">0.645</td>
</tr>
<tr class="even">
<td align="left">finn-a-D3_SARCOIDOSIS</td>
<td align="left">Sarcoidosis</td>
<td align="left">cis</td>
<td align="right">29</td>
<td align="right">-0.060</td>
<td align="right">0.115</td>
<td align="right">0.601</td>
</tr>
<tr class="odd">
<td align="left">finn-a-M13_SJOGREN</td>
<td align="left">Sjogren syndrome</td>
<td align="left">cis</td>
<td align="right">29</td>
<td align="right">-0.248</td>
<td align="right">0.141</td>
<td align="right">0.080</td>
</tr>
<tr class="even">
<td align="left">finn-a-MENINGITIS</td>
<td align="left">meningitis infection</td>
<td align="left">cis</td>
<td align="right">29</td>
<td align="right">-0.217</td>
<td align="right">0.191</td>
<td align="right">0.257</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-1081</td>
<td align="left">IGA glomerulonephritis</td>
<td align="left">cis</td>
<td align="right">3</td>
<td align="right">0.210</td>
<td align="right">0.177</td>
<td align="right">0.237</td>
</tr>
<tr class="even">
<td align="left">ieu-a-1112</td>
<td align="left">sclerosing cholangitis</td>
<td align="left">cis</td>
<td align="right">32</td>
<td align="right">0.052</td>
<td align="right">0.068</td>
<td align="right">0.446</td>
</tr>
<tr class="odd">
<td align="left">ieu-a-276</td>
<td align="left">celiac disease</td>
<td align="left">cis</td>
<td align="right">3</td>
<td align="right">-0.017</td>
<td align="right">0.121</td>
<td align="right">0.887</td>
</tr>
<tr class="even">
<td align="left">ieu-a-31</td>
<td align="left">inflammatory bowel disease</td>
<td align="left">cis</td>
<td align="right">56</td>
<td align="right">0.390</td>
<td align="right">0.035</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ieu-b-18</td>
<td align="left">multiple sclerosis</td>
<td align="left">cis</td>
<td align="right">26</td>
<td align="right">-0.026</td>
<td align="right">0.043</td>
<td align="right">0.548</td>
</tr>
<tr class="even">
<td align="left">ieu-b-69</td>
<td align="left">sepsis</td>
<td align="left">cis</td>
<td align="right">56</td>
<td align="right">-0.033</td>
<td align="right">0.027</td>
<td align="right">0.209</td>
</tr>
<tr class="odd">
<td align="left">ukb-a-115</td>
<td align="left">Vitiligo</td>
<td align="left">cis</td>
<td align="right">54</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.410</td>
</tr>
<tr class="even">
<td align="left">ukb-a-552</td>
<td align="left">Crohnâ€™s disease</td>
<td align="left">cis</td>
<td align="right">54</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ukb-b-10537</td>
<td align="left">psoriasis</td>
<td align="left">cis</td>
<td align="right">17</td>
<td align="right">-0.004</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ukb-b-13251</td>
<td align="left">gout</td>
<td align="left">cis</td>
<td align="right">20</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.764</td>
</tr>
<tr class="odd">
<td align="left">ukb-b-15606</td>
<td align="left">pneumonia</td>
<td align="left">cis</td>
<td align="right">7</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.376</td>
</tr>
<tr class="even">
<td align="left">ukb-b-15622</td>
<td align="left">Tuberculosis</td>
<td align="left">cis</td>
<td align="right">7</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.668</td>
</tr>
<tr class="odd">
<td align="left">ukb-b-16499</td>
<td align="left">allergic rhinitis</td>
<td align="left">cis</td>
<td align="right">40</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.788</td>
</tr>
<tr class="even">
<td align="left">ukb-b-16702</td>
<td align="left">allergy</td>
<td align="left">cis</td>
<td align="right">7</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.265</td>
</tr>
<tr class="odd">
<td align="left">ukb-b-18194</td>
<td align="left">ankylosing spondylitis</td>
<td align="left">cis</td>
<td align="right">5</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.710</td>
</tr>
<tr class="even">
<td align="left">ukb-b-19386</td>
<td align="left">ulcerative colitis</td>
<td align="left">cis</td>
<td align="right">7</td>
<td align="right">0.001</td>
<td align="right">0.000</td>
<td align="right">0.039</td>
</tr>
<tr class="odd">
<td align="left">ukb-b-19732</td>
<td align="left">hypothyroidism</td>
<td align="left">cis</td>
<td align="right">37</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.952</td>
</tr>
<tr class="even">
<td align="left">ukb-b-20141</td>
<td align="left">atopic eczema</td>
<td align="left">cis</td>
<td align="right">26</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.988</td>
</tr>
<tr class="odd">
<td align="left">ukb-b-20208</td>
<td align="left">asthma</td>
<td align="left">cis</td>
<td align="right">7</td>
<td align="right">0.000</td>
<td align="right">0.000</td>
<td align="right">0.756</td>
</tr>
<tr class="even">
<td align="left">ukb-b-8814</td>
<td align="left">urinary tract infection</td>
<td align="left">cis</td>
<td align="right">18</td>
<td align="right">-0.001</td>
<td align="right">0.000</td>
<td align="right">0.279</td>
</tr>
<tr class="odd">
<td align="left">ukb-b-9125</td>
<td align="left">rheumatoid arthritis</td>
<td align="left">cis</td>
<td align="right">17</td>
<td align="right">0.000</td>
<td align="right">0.001</td>
<td align="right">0.845</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">d3</span>,<span class="fu">aes</span><span class="op">(</span>y <span class="op">=</span> <span class="va">trait</span>, x <span class="op">=</span> <span class="va">y</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>     <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>     <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>     <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">cistrans</span>,ncol<span class="op">=</span><span class="fl">3</span>,scales<span class="op">=</span><span class="st">"free_x"</span><span class="op">)</span><span class="op">+</span></span>
<span>     <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_segment.html" class="external-link">geom_segment</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">b</span><span class="op">-</span><span class="fl">1.96</span><span class="op">*</span><span class="va">se</span>, xend <span class="op">=</span> <span class="va">b</span><span class="op">+</span><span class="fl">1.96</span><span class="op">*</span><span class="va">se</span>, yend <span class="op">=</span> <span class="va">trait</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>     <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>lty<span class="op">=</span><span class="fl">2</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xintercept<span class="op">=</span><span class="fl">0</span><span class="op">)</span>, colour <span class="op">=</span> <span class="st">'red'</span><span class="op">)</span><span class="op">+</span></span>
<span>     <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Effect size"</span><span class="op">)</span><span class="op">+</span></span>
<span>     <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:mr"></span>
<img src="pQTLtools/mr-1.png" alt="MR with cis, trans and cis+trans variants of IL-12B" width="672" style="display:block; margin: auto"><p class="caption">
Figure 5.5: MR with cis, trans and cis+trans variants of IL-12B
</p>
</div>
</div>
<div class="section level3" number="5.4">
<h3 id="gsmr-forest-plots">
<span class="header-section-number">5.4</span> GSMR forest plots<a class="anchor" aria-label="anchor" href="#gsmr-forest-plots"></a>
</h3>
<p>This is based on GSMR results of TNFB and a range of immune-mediated traits,</p>
<table class="table">
<caption>
<span id="tab:tnfb">Table 5.4: </span>GSMR results for TNFB and immune-mediated traits</caption>
<thead><tr class="header">
<th align="left">Outcome</th>
<th align="right">Effect</th>
<th align="right">StdErr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Multiple sclerosis</td>
<td align="right">0.691</td>
<td align="right">0.059</td>
</tr>
<tr class="even">
<td align="left">Systemic lupus erythematosus</td>
<td align="right">0.767</td>
<td align="right">0.079</td>
</tr>
<tr class="odd">
<td align="left">Sclerosing cholangitis</td>
<td align="right">0.627</td>
<td align="right">0.076</td>
</tr>
<tr class="even">
<td align="left">Juvenile idiopathic arthritis</td>
<td align="right">-1.176</td>
<td align="right">0.160</td>
</tr>
<tr class="odd">
<td align="left">Psoriasis</td>
<td align="right">0.006</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">Rheumatoid arthritis</td>
<td align="right">-0.004</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">Inflammatory bowel disease</td>
<td align="right">-0.143</td>
<td align="right">0.025</td>
</tr>
<tr class="even">
<td align="left">Ankylosing spondylitis</td>
<td align="right">-0.003</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">Hypothyroidism</td>
<td align="right">-0.004</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">Allergic rhinitis</td>
<td align="right">0.004</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">IgA glomerulonephritis</td>
<td align="right">-0.327</td>
<td align="right">0.105</td>
</tr>
<tr class="even">
<td align="left">Atopic eczema</td>
<td align="right">-0.002</td>
<td align="right">0.001</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu">mr_forestplot</span><span class="op">(</span><span class="va">tnfb</span>,colgap.forest.left<span class="op">=</span><span class="st">"0.05cm"</span>, fontsize<span class="op">=</span><span class="fl">14</span>,</span>
<span>              leftcols<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"studlab"</span><span class="op">)</span>, leftlabs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Outcome"</span><span class="op">)</span>,</span>
<span>              plotwidth<span class="op">=</span><span class="st">"5inch"</span>, sm<span class="op">=</span><span class="st">"OR"</span>, sortvar<span class="op">=</span><span class="va">tnfb</span><span class="op">[[</span><span class="st">"Effect"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>              rightcols<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"effect"</span>,<span class="st">"ci"</span>,<span class="st">"pval"</span><span class="op">)</span>, rightlabs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"OR"</span>,<span class="st">"95%CI"</span>,<span class="st">"GSMR P"</span><span class="op">)</span>,</span>
<span>              digits<span class="op">=</span><span class="fl">3</span>, digits.pval<span class="op">=</span><span class="fl">2</span>, scientific.pval<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>              common<span class="op">=</span><span class="cn">FALSE</span>, random<span class="op">=</span><span class="cn">FALSE</span>, print.I2<span class="op">=</span><span class="cn">FALSE</span>, print.pval.Q<span class="op">=</span><span class="cn">FALSE</span>, print.tau2<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span>              addrow<span class="op">=</span><span class="cn">TRUE</span>, backtransf<span class="op">=</span><span class="cn">TRUE</span>, spacing<span class="op">=</span><span class="fl">1.6</span>, col.inside<span class="op">=</span><span class="st">"black"</span>, col.square<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<span style="display:block;" id="fig:mrforestplot"></span>
<img src="pQTLtools/mrforestplot-1.png" alt="Forest plots based on MR results on TNFB" width="1152" style="display:block; margin: auto"><p class="caption">
Figure 5.6: Forest plots based on MR results on TNFB
</p>
</div>
</div>
</div>
<div class="section level2" number="6">
<h2 id="literature-on-pqtls">
<span class="header-section-number">6</span> Literature on pQTLs<a class="anchor" aria-label="anchor" href="#literature-on-pqtls"></a>
</h2>
<p>References<span class="citation"><sup>3</sup></span> <span class="citation"><sup>4</sup></span> are included as EndNote libraries which is now part of pQTLdata package.</p>
</div>
<div class="section level2" number="7">
<h2 id="uniprot-ids">
<span class="header-section-number">7</span> UniProt IDs<a class="anchor" aria-label="anchor" href="#uniprot-ids"></a>
</h2>
<p>The function <code>uniprot2ids</code> converts UniProt IDs to others.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-scallop_inf" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">The SCALLOP consortium. Mapping pQTLs of circulating inflammatory proteins identifies drivers of immune-related disease risk and novel therapeutic targets. <em>medRxiv</em> 2023.03.24.23287680 (2023) doi:<a href="https://doi.org/10.1101/2023.03.24.23287680" class="external-link">10.1101/2023.03.24.23287680</a>.</div>
</div>
<div id="ref-kwan14" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Kwan, J. S. H. <em>et al.</em> <a href="https://doi.org/10.1093/hmg/ddu386" class="external-link">Meta-analysis of genome-wide association studies identifies two loci associated with circulating osteoprotegerin levels</a>. <em>Human Molecular Genetics</em> <strong>23</strong>, 6684â€“6693 (2014).</div>
</div>
<div id="ref-sun18" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Sun, B. B. <em>et al.</em> <a href="https://doi.org/10.1038/s41586-018-0175-2" class="external-link">Genomic atlas of the human plasma proteome</a>. <em>Nature</em> <strong>558</strong>, 73â€“79 (2018).</div>
</div>
<div id="ref-suhre20" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Suhre, K., McCarthy, M. I. &amp; Schwenk, J. M. Genetics meets proteomics: Perspectives for large population-based studies. <em>Nat Rev Genet</em> (2020) doi:<a href="https://doi.org/10.1038/s41576-020-0268-2" class="external-link">10.1038/s41576-020-0268-2</a>.</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Crafted by <a href="https://jinghuazhao.github.io/" class="external-link"><img src="https://jinghuazhao.github.io/bees.svg" width="24"> Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
