<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="pQTLtools">
<title>SCALLOP-INF scripts â€¢ pQTLtools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SCALLOP-INF scripts">
<meta property="og:description" content="pQTLtools">
<meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Vignettes:</h6>
    <a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Articles:</h6>
    <a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a>
    <a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a>
    <a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a>
    <a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://jinghuazhao.github.io/R/">
    <span class="fab fa fab fa-r-project"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/">
    <span class="fas fa fas fa-fast-backward"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-urls">URLs</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-urls">
    <a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a>
    <a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a>
    <a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a>
    <a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a>
    <a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a>
    <a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a>
    <a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a>
    <a class="external-link dropdown-item" href="https://metascape.org/">metascape</a>
    <a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a>
    <a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a>
    <a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a>
    <a class="external-link dropdown-item" href="https://string-db.org/">STRING</a>
    <a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a>
    <a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.uniprot.org/">
    <span class="fas fa fas fa-fast-forward"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>SCALLOP-INF scripts</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/vignettes/SCALLOP-INF.Rmd" class="external-link"><code>vignettes/SCALLOP-INF.Rmd</code></a></small>
      <div class="d-none name"><code>SCALLOP-INF.Rmd</code></div>
    </div>

    
    
<p>This article provides details of pQTL/disease overlap and colocalization analysis.</p>
<div class="section level2" number="1">
<h2 id="pqtldisease-overlap">
<span class="header-section-number">1</span> pQTL/disease overlap<a class="anchor" aria-label="anchor" href="#pqtldisease-overlap"></a>
</h2>
<p>The ontology of traits/disease is available through Experimental Factor Ontology (EFO)<span class="citation"><sup>1</sup></span>, which can be used to build lists of diseases and immune-mediated traits and filter search results from PhenoScanner<span class="citation"><sup>2</sup></span>.</p>
<div class="section level3" number="1.1">
<h3 id="diseases">
<span class="header-section-number">1.1</span> Diseases<a class="anchor" aria-label="anchor" href="#diseases"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ontologyIndex</span><span class="op">)</span></span>
<span><span class="co"># http://www.ebi.ac.uk/efo/efo.obo</span></span>
<span><span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"ebi"</span>,<span class="st">"efo-3.26.0"</span>,<span class="st">"efo.obo"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_relation_names.html" class="external-link">get_relation_names</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span>
<span><span class="va">efo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_ontology.html" class="external-link">get_ontology</a></span><span class="op">(</span><span class="va">file</span>, extract_tags<span class="op">=</span><span class="st">"everything"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">id</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ontology</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ontology</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ontology</span><span class="op">$</span><span class="va">id</span><span class="op">)</span></span>
<span>  <span class="va">inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span>,pattern<span class="op">=</span><span class="st">"immune|inflammatory"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>id<span class="op">=</span><span class="va">ontology</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="va">inf</span><span class="op">]</span>,name<span class="op">=</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">inf</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">goidname</span> <span class="op">&lt;-</span> <span class="fu">id</span><span class="op">(</span><span class="va">go</span><span class="op">)</span></span>
<span><span class="va">efoidname</span> <span class="op">&lt;-</span> <span class="fu">id</span><span class="op">(</span><span class="va">efo</span><span class="op">)</span></span>
<span><span class="co"># all diseases</span></span>
<span><span class="va">efo_diseases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_descendants.html" class="external-link">get_descendants</a></span><span class="op">(</span><span class="va">efo</span>,<span class="st">"EFO:0000408"</span><span class="op">)</span></span>
<span><span class="va">diseases_name</span> <span class="op">&lt;-</span> <span class="va">efo</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">efo_diseases</span><span class="op">]</span></span>
<span><span class="va">diseases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">efo_diseases</span>,<span class="va">diseases_name</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">diseases</span>,file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"ebi"</span>,<span class="st">"efo-3.26.0"</span>,<span class="st">"efo_diseases.csv"</span><span class="op">)</span>,col.names<span class="op">=</span><span class="cn">FALSE</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">","</span><span class="op">)</span></span>
<span><span class="co"># immune system diseases (isd)</span></span>
<span><span class="va">efo_0000540</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_descendants.html" class="external-link">get_descendants</a></span><span class="op">(</span><span class="va">efo</span>,<span class="st">"EFO:0000540"</span><span class="op">)</span></span>
<span><span class="va">efo_0000540name</span> <span class="op">&lt;-</span> <span class="va">efo</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">efo_0000540</span><span class="op">]</span></span>
<span><span class="va">isd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">efo_0000540</span>,<span class="va">efo_0000540name</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ontologyPlot</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ontologyPlot/man/onto_plot.html" class="external-link">onto_plot</a></span><span class="op">(</span><span class="va">efo</span>,<span class="va">efo_0000540</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="1.2">
<h3 id="lookup">
<span class="header-section-number">1.2</span> Lookup<a class="anchor" aria-label="anchor" href="#lookup"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/R" class="external-link">gap</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/pQTLtools" class="external-link">pQTLtools</a></span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">inf1_prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">vector</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">92</span><span class="op">)</span> <span class="va">inf1_prot</span><span class="op">[</span><span class="va">inf1</span><span class="op">[</span><span class="va">i</span>,<span class="st">"prot"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="va">inf1</span><span class="op">[</span><span class="va">i</span>,<span class="op">]</span>,target.short<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">alt_name</span><span class="op">)</span>,<span class="va">alt_name</span>,<span class="va">target.short</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="st">"target.short"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">INF1_metal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"~"</span>,<span class="st">"pQTLtools"</span>,<span class="st">"tests"</span>,<span class="st">"INF1.METAL"</span><span class="op">)</span>,as.is<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,<span class="op">{</span></span>
<span>                    <span class="va">hg19_coordinates</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="va">Chromosome</span>,<span class="st">":"</span>,<span class="va">Position</span><span class="op">)</span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>INF1_rsid<span class="op">=</span><span class="va">rsid</span>, Total<span class="op">=</span><span class="va">N</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="fu">pQTLdata</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/pQTLdata/man/inf1.html" class="external-link">inf1</a></span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prot"</span>,<span class="st">"gene"</span>,<span class="st">"target.short"</span>,<span class="st">"alt_name"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>target.short<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">alt_name</span><span class="op">)</span>,<span class="va">alt_name</span>,<span class="va">target.short</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">alt_name</span><span class="op">)</span></span>
<span><span class="va">INF1_aggr</span> <span class="op">&lt;-</span> <span class="va">INF1_metal</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>             <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Chromosome</span>,<span class="va">Position</span>,<span class="va">target.short</span>,<span class="va">gene</span>,<span class="va">hg19_coordinates</span>,</span>
<span>                    <span class="va">MarkerName</span>,<span class="va">Allele1</span>,<span class="va">Allele2</span>,<span class="va">Freq1</span>,<span class="va">Effect</span>,<span class="va">StdErr</span>,<span class="va">log.P.</span>,<span class="va">cis.trans</span>,<span class="va">INF1_rsid</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>             <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Chromosome</span>,<span class="va">Position</span>,<span class="va">MarkerName</span>,<span class="va">INF1_rsid</span>,<span class="va">hg19_coordinates</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>             <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>nprots<span class="op">=</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                       prots<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">target.short</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                       Allele1<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="va">Allele1</span><span class="op">)</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                       Allele2<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="va">Allele2</span><span class="op">)</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                       EAF<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">Freq1</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                       Effects<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">Effect</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                       SEs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">StdErr</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                       log10P<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">log.P.</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                       cistrans<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">cis.trans</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>             <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">rsid</span> <span class="op">&lt;-</span> <span class="va">INF1_aggr</span><span class="op">[[</span><span class="st">"INF1_rsid"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">catalogue</span> <span class="op">&lt;-</span> <span class="st">"GWAS"</span></span>
<span><span class="va">proxies</span> <span class="op">&lt;-</span> <span class="st">"EUR"</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fl">5e-8</span></span>
<span><span class="va">r2</span> <span class="op">&lt;-</span> <span class="fl">0.8</span></span>
<span><span class="va">build</span> <span class="op">&lt;-</span> <span class="fl">37</span></span>
<span></span>
<span><span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span></span>
<span><span class="va">efo_diseases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"ebi"</span>,<span class="st">"efo-3.26.0"</span>,<span class="st">"efo_diseases.csv"</span><span class="op">)</span>,col.names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"efo"</span>,<span class="st">"disease"</span><span class="op">)</span>,as.is<span class="op">=</span><span class="cn">TRUE</span>,sep<span class="op">=</span><span class="st">","</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>                <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>efo<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">":"</span>, <span class="st">"_"</span>, <span class="va">efo</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/snpqueries.html">snpqueries</a></span><span class="op">(</span><span class="va">rsid</span>, catalogue<span class="op">=</span><span class="va">catalogue</span>, proxies<span class="op">=</span><span class="va">proxies</span>, p<span class="op">=</span><span class="va">p</span>, r2<span class="op">=</span><span class="va">r2</span>, build<span class="op">=</span><span class="va">build</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">dim</span><span class="op">)</span></span>
<span><span class="va">snps_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">right_join</a></span><span class="op">(</span><span class="va">snps</span>,<span class="va">results</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">snps_results</span>,select<span class="op">=</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">hg38_coordinates</span>,<span class="va">ref_hg38_coordinates</span>,<span class="va">pos_hg38</span>,<span class="va">ref_pos_hg38</span>,<span class="va">dprime</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">aggr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">INF1_aggr</span>,<span class="op">{</span><span class="va">HLA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">Chromosome</span><span class="op">==</span><span class="fl">6</span> <span class="op">&amp;</span> <span class="va">Position</span> <span class="op">&gt;=</span> <span class="fl">25392021</span> <span class="op">&amp;</span> <span class="va">Position</span> <span class="op">&lt;=</span> <span class="fl">33392022</span><span class="op">)</span><span class="op">}</span><span class="op">)</span>,</span>
<span>               select<span class="op">=</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">Chromosome</span>,<span class="va">Position</span>,<span class="va">INF1_rsid</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">short</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">aggr</span>,<span class="va">ps</span>,by<span class="op">=</span><span class="st">"hg19_coordinates"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gwas</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">short</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">aggr</span>,<span class="va">ps</span>,by<span class="op">=</span><span class="st">"hg19_coordinates"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>           <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">efo</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">efo_diseases</span>,<span class="va">efo</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>           <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">efo_diseases</span><span class="op">)</span></span>
<span>  <span class="va">v</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"prots"</span>,<span class="st">"hgnc"</span>,<span class="st">"MarkerName"</span>,<span class="st">"cistrans"</span>,<span class="st">"Effects"</span>,<span class="st">"Allele1"</span>,<span class="st">"Allele2"</span>,<span class="st">"rsid"</span>,<span class="st">"a1"</span>,<span class="st">"a2"</span>,<span class="st">"efo"</span>,</span>
<span>         <span class="st">"ref_rsid"</span>,<span class="st">"ref_a1"</span>,<span class="st">"ref_a2"</span>,<span class="st">"proxy"</span>,<span class="st">"r2"</span>,</span>
<span>         <span class="st">"HLA"</span>,<span class="st">"beta"</span>,<span class="st">"se"</span>,<span class="st">"p"</span>,<span class="st">"disease"</span>,<span class="st">"n_cases"</span>,<span class="st">"n_controls"</span>,<span class="st">"unit"</span>,<span class="st">"ancestry"</span>,<span class="st">"pmid"</span>,<span class="st">"study"</span><span class="op">)</span></span>
<span>  <span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">short</span><span class="op">[</span><span class="va">v</span><span class="op">]</span>,</span>
<span>  <span class="op">{</span></span>
<span>    <span class="va">flag</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">HLA</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span></span>
<span>    <span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prots</span>,<span class="st">"-"</span>,<span class="va">rsid</span><span class="op">)</span></span>
<span>    <span class="va">prefix</span><span class="op">[</span><span class="va">flag</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span><span class="op">[</span><span class="va">flag</span><span class="op">]</span>,<span class="st">"*"</span><span class="op">)</span></span>
<span>    <span class="va">rsidProts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">" ("</span>,<span class="va">hgnc</span>,<span class="st">")"</span><span class="op">)</span></span>
<span>    <span class="va">efoTraits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\b(^[a-z])"</span>,<span class="st">"\\U\\1"</span>,<span class="va">disease</span>,perl<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>    <span class="va">qtl_direction</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sign.html" class="external-link">sign</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">beta</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="va">combined</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">mat</span>,<span class="va">efoTraits</span>,<span class="va">rsidProts</span>,<span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">n_cases</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>              <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>direction<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">qtl_direction</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                        betas<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">beta</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                        units<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">unit</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                        studies<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">study</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                        diseases<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">disease</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span>,</span>
<span>                        cases<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">n_cases</span>,collapse<span class="op">=</span><span class="st">";"</span><span class="op">)</span></span>
<span>                       <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>              <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="va">rxc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">combined</span>,<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">efoTraits</span>,<span class="va">rsidProts</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">cn</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rxc</span><span class="op">)</span><span class="op">)</span> <span class="kw">for</span><span class="op">(</span><span class="va">rn</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">rxc</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>     <span class="va">cnrn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">combined</span>,<span class="va">efoTraits</span><span class="op">==</span><span class="va">rn</span> <span class="op">&amp;</span> <span class="va">rsidProts</span><span class="op">==</span><span class="va">cn</span><span class="op">)</span></span>
<span>     <span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">cnrn</span><span class="op">)</span><span class="op">==</span><span class="fl">0</span><span class="op">)</span> <span class="kw">next</span></span>
<span>     <span class="va">rxc</span><span class="op">[</span><span class="va">rn</span>,<span class="va">cn</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">cnrn</span><span class="op">[[</span><span class="st">"direction"</span><span class="op">]</span><span class="op">]</span>,<span class="st">";"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">mat</span>,file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"work"</span>,<span class="st">"pQTL-disease-GWAS.csv"</span><span class="op">)</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,quote<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">","</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">combined</span>,file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"work"</span>,<span class="st">"pQTL-disease-GWAS-combined.csv"</span><span class="op">)</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,quote<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">","</span><span class="op">)</span></span>
<span>  <span class="va">rxc</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">rxc</span> <span class="op">&lt;-</span> <span class="fu">gwas</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="1.3">
<h3 id="visualization">
<span class="header-section-number">1.3</span> Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SF</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">rxc</span>, <span class="va">f</span><span class="op">=</span><span class="st">"SF-pQTL-GWAS.png"</span>, <span class="va">ch</span><span class="op">=</span><span class="fl">21</span>, <span class="va">cw</span><span class="op">=</span><span class="fl">21</span>, <span class="va">h</span><span class="op">=</span><span class="fl">13</span>, <span class="va">w</span><span class="op">=</span><span class="fl">18</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span>  <span class="va">col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#4287f5"</span>,<span class="st">"#ffffff"</span>,<span class="st">"#e32222"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">3</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="va">f</span><span class="op">)</span>,res<span class="op">=</span><span class="fl">300</span>,width<span class="op">=</span><span class="va">w</span>,height<span class="op">=</span><span class="va">h</span>,units<span class="op">=</span><span class="st">"in"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/userhooks.html" class="external-link">setHook</a></span><span class="op">(</span><span class="st">"grid.newpage"</span>, <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/grid/viewports.html" class="external-link">pushViewport</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grid/viewport.html" class="external-link">viewport</a></span><span class="op">(</span>x<span class="op">=</span><span class="fl">1</span>,y<span class="op">=</span><span class="fl">1</span>,width<span class="op">=</span><span class="fl">0.9</span>, height<span class="op">=</span><span class="fl">0.9</span>, name<span class="op">=</span><span class="st">"vp"</span>, just<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"right"</span>,<span class="st">"top"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, action<span class="op">=</span><span class="st">"prepend"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rxc</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"^[0-9]*-"</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">rxc</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">rxc</span>, legend<span class="op">=</span><span class="cn">FALSE</span>, angle_col<span class="op">=</span><span class="st">"270"</span>, border_color<span class="op">=</span><span class="st">"black"</span>, color<span class="op">=</span><span class="va">col</span>, cellheight<span class="op">=</span><span class="va">ch</span>, cellwidth<span class="op">=</span><span class="va">cw</span>, cluster_rows<span class="op">=</span><span class="cn">TRUE</span>, cluster_cols<span class="op">=</span><span class="cn">FALSE</span>, fontsize<span class="op">=</span><span class="fl">8</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/userhooks.html" class="external-link">setHook</a></span><span class="op">(</span><span class="st">"grid.newpage"</span>, <span class="cn">NULL</span>, <span class="st">"replace"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span><span class="st">"Protein(s)-pQTL (gene)"</span>, y<span class="op">=</span><span class="fl">0.03125</span>, gp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">12</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span><span class="st">"GWAS diseases"</span>, x<span class="op">=</span><span class="op">-</span><span class="fl">0.0625</span>, rot<span class="op">=</span><span class="fl">90</span>, gp<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize<span class="op">=</span><span class="fl">12</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">SF</span><span class="op">(</span><span class="va">rxc</span>,f<span class="op">=</span><span class="st">"SF-pQTL-GWAS.png"</span>,ch<span class="op">=</span><span class="fl">8</span>,cw<span class="op">=</span><span class="fl">8</span>,h<span class="op">=</span><span class="fl">11</span>,w<span class="op">=</span><span class="fl">8.6</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: left">
<span style="display:block;" id="fig:gwas"></span>
<img src="SF-pQTL-GWAS.png" alt="pQTL-disease overlap" width="1290" style="display:block; margin: auto"><p class="caption">
Figure 1.1: pQTL-disease overlap
</p>
</div>
</div>
</div>
<div class="section level2" number="2">
<h2 id="colocalization">
<span class="header-section-number">2</span> Colocalization<a class="anchor" aria-label="anchor" href="#colocalization"></a>
</h2>
<p>This is the actual script for cis-pQTL colocalization analysis on GTEx v8 for SCALLOP-INF.</p>
<div class="section level3" number="2.1">
<h3 id="data">
<span class="header-section-number">2.1</span> Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h3>
<p>The data were GWAS summary statistics in GRCh37 and VCF format, converted by <code>gwasvcf</code>. The GTEx association statistics were in GRCh38 and downloaded from the eQTL Catalogue and stored locally. Data on microarray and RNA-Seq remain on the eQTL Catalogue website.</p>
</div>
<div class="section level3" number="2.2">
<h3 id="the-driver-program">
<span class="header-section-number">2.2</span> The driver program<a class="anchor" aria-label="anchor" href="#the-driver-program"></a>
</h3>
<p>It is in Bash.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/usr/bin/bash</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> r <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">59</span><span class="dt">}</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>   <span class="bu">export</span> <span class="va">r</span><span class="op">=</span><span class="va">${r}</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>   <span class="bu">export</span> <span class="va">cvt</span><span class="op">=</span><span class="va">${INF}</span>/work/INF1.merge.cis.vs.trans</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>   <span class="bu">read</span> <span class="va">prot</span> <span class="va">MarkerName</span> <span class="op">&lt;</span> <span class="dt">\</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>                        <span class="op">&lt;(</span><span class="fu">awk</span> <span class="at">-vFS</span><span class="op">=</span><span class="st">","</span> <span class="st">'$14=="cis"'</span> <span class="va">${cvt}</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">awk</span> <span class="at">-vFS</span><span class="op">=</span><span class="st">","</span> <span class="at">-vr</span><span class="op">=</span><span class="va">${r}</span> <span class="st">'NR==r{print $2,$5}'</span><span class="op">)</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>   <span class="bu">echo</span> <span class="va">${r}</span> <span class="at">-</span> <span class="va">${prot}</span> <span class="at">-</span> <span class="va">${MarkerName}</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>   <span class="bu">export</span> <span class="va">prot</span><span class="op">=</span><span class="va">${prot}</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>   <span class="bu">export</span> <span class="va">MarkerName</span><span class="op">=</span><span class="va">${MarkerName}</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>   <span class="cf">if</span> <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-f</span> <span class="va">${INF}</span>/coloc/<span class="va">${prot}</span>-<span class="va">${MarkerName}</span>.pdf <span class="bu">]</span> <span class="kw">||</span> <span class="dt">\</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>      <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-f</span> <span class="va">${INF}</span>/coloc/<span class="va">${prot}</span>-<span class="va">${MarkerName}</span>.RDS <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>     <span class="bu">cd</span> <span class="va">${INF}</span>/coloc</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>     <span class="ex">R</span> <span class="at">--no-save</span> <span class="op">&lt;</span> <span class="va">${INF}</span>/rsid/coloc.R <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>     <span class="fu">tee</span> <span class="va">${prot}</span>-<span class="va">${MarkerName}</span>.log</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co">#    ls *tbi | xargs -I {} bash -c "rm {}"</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>     <span class="bu">cd</span> <span class="at">-</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>   <span class="cf">fi</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
</div>
<div class="section level3" number="2.3">
<h3 id="coloc-r">
<span class="header-section-number">2.3</span> coloc.R<a class="anchor" aria-label="anchor" href="#coloc-r"></a>
</h3>
<p>It contains minor modification to the documentation example,</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>liftRegion <span class="ot">&lt;-</span> <span class="cf">function</span>(x,chain,<span class="at">flanking=</span><span class="fl">1e6</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(GenomicRanges)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  gr <span class="ot">&lt;-</span> <span class="fu">with</span>(x,GenomicRanges<span class="sc">::</span><span class="fu">GRanges</span>(<span class="at">seqnames=</span>chr,IRanges<span class="sc">::</span><span class="fu">IRanges</span>(start,end))<span class="sc">+</span>flanking)</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(gr) <span class="ot">&lt;-</span> <span class="st">"UCSC"</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  gr38 <span class="ot">&lt;-</span> rtracklayer<span class="sc">::</span><span class="fu">liftOver</span>(gr, chain)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>  chr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"chr"</span>,<span class="st">""</span>,<span class="fu">colnames</span>(<span class="fu">table</span>(<span class="fu">seqnames</span>(gr38))))</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  start <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">unlist</span>(<span class="fu">start</span>(gr38)))</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  end <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">end</span>(gr38)))</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">invisible</span>(<span class="fu">list</span>(<span class="at">chr=</span>chr,<span class="at">start=</span>start,<span class="at">end=</span>end,<span class="at">region=</span><span class="fu">paste0</span>(chr,<span class="st">":"</span>,start,<span class="st">"-"</span>,end)))</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>sumstats <span class="ot">&lt;-</span> <span class="cf">function</span>(prot,chr,region37)</span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"GWAS sumstats</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  vcf <span class="ot">&lt;-</span> <span class="fu">file.path</span>(INF,<span class="st">"METAL/gwas2vcf"</span>,<span class="fu">paste0</span>(prot,<span class="st">".vcf.gz"</span>))</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  gwas_stats <span class="ot">&lt;-</span> gwasvcf<span class="sc">::</span><span class="fu">query_gwas</span>(vcf, <span class="at">chrompos =</span> region37)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  gwas_stats <span class="ot">&lt;-</span> gwasvcf<span class="sc">::</span><span class="fu">vcf_to_granges</span>(gwas_stats) <span class="sc">%&gt;%</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>                <span class="fu">keepSeqlevels</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>                <span class="fu">renameSeqlevels</span>(<span class="fu">paste0</span>(<span class="st">"chr"</span>,chr))</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  gwas_stats_hg38 <span class="ot">&lt;-</span> rtracklayer<span class="sc">::</span><span class="fu">liftOver</span>(gwas_stats, chain) <span class="sc">%&gt;%</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unlist</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">renameSeqlevels</span>(chr) <span class="sc">%&gt;%</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">as_tibble</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">transmute</span>(<span class="at">chromosome =</span> seqnames,</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>                     <span class="at">position =</span> start, REF, ALT, AF, ES, SE, LP, SS) <span class="sc">%&gt;%</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">paste</span>(chromosome, position, <span class="at">sep =</span> <span class="st">":"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">MAF =</span> <span class="fu">pmin</span>(AF, <span class="dv">1</span><span class="sc">-</span>AF)) <span class="sc">%&gt;%</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(id) <span class="sc">%&gt;%</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">row_count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(row_count <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>microarray <span class="ot">&lt;-</span> <span class="cf">function</span>(gwas_stats_hg38,ensGene,region38)</span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"a. eQTL datasets</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a>  microarray_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tabix_paths, quant_method <span class="sc">==</span> <span class="st">"microarray"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true" tabindex="-1"></a>                   dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">qtl_id =</span> <span class="fu">paste</span>(study, qtl_group, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true" tabindex="-1"></a>  ftp_path_list <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">as.list</span>(microarray_df<span class="sc">$</span>ftp_path), microarray_df<span class="sc">$</span>qtl_id[<span class="dv">1</span>])</span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true" tabindex="-1"></a>  hdr <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">path.package</span>(<span class="st">"pQTLtools"</span>),<span class="st">"eQTL-Catalogue"</span>,<span class="st">"column_names.CEDAR"</span>)</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true" tabindex="-1"></a>  column_names <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">read.delim</span>(hdr))</span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true" tabindex="-1"></a>  summary_list <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(ftp_path_list, <span class="sc">~</span><span class="fu">import_eQTLCatalogue</span>(., region38,</span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true" tabindex="-1"></a>                             <span class="at">selected_gene_id =</span> ensGene, column_names))</span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_df</span>(summary_list[<span class="fu">lapply</span>(summary_list,nrow)<span class="sc">!=</span><span class="dv">0</span>],</span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true" tabindex="-1"></a>                <span class="sc">~</span><span class="fu">run_coloc</span>(., gwas_stats_hg38), <span class="at">.id =</span> <span class="st">"qtl_id"</span>)</span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-48"><a href="#cb5-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-49"><a href="#cb5-49" aria-hidden="true" tabindex="-1"></a>rnaseq <span class="ot">&lt;-</span> <span class="cf">function</span>(gwas_stats_hg38,ensGene,region38)</span>
<span id="cb5-50"><a href="#cb5-50" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb5-51"><a href="#cb5-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"b. Uniformly processed RNA-seq datasets</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-52"><a href="#cb5-52" aria-hidden="true" tabindex="-1"></a>  rnaseq_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(tabix_paths, quant_method <span class="sc">==</span> <span class="st">"ge"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-53"><a href="#cb5-53" aria-hidden="true" tabindex="-1"></a>               dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">qtl_id =</span> <span class="fu">paste</span>(study, qtl_group, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb5-54"><a href="#cb5-54" aria-hidden="true" tabindex="-1"></a>  ftp_path_list <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">as.list</span>(rnaseq_df<span class="sc">$</span>ftp_path), rnaseq_df<span class="sc">$</span>qtl_id)</span>
<span id="cb5-55"><a href="#cb5-55" aria-hidden="true" tabindex="-1"></a>  hdr <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">path.package</span>(<span class="st">"pQTLtools"</span>),<span class="st">"eQTL-Catalogue"</span>,<span class="st">"column_names.Alasoo"</span>)</span>
<span id="cb5-56"><a href="#cb5-56" aria-hidden="true" tabindex="-1"></a>  column_names <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">read.delim</span>(hdr))</span>
<span id="cb5-57"><a href="#cb5-57" aria-hidden="true" tabindex="-1"></a>  safe_import <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">safely</span>(import_eQTLCatalogue)</span>
<span id="cb5-58"><a href="#cb5-58" aria-hidden="true" tabindex="-1"></a>  summary_list <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(ftp_path_list, <span class="sc">~</span><span class="fu">safe_import</span>(., region38,</span>
<span id="cb5-59"><a href="#cb5-59" aria-hidden="true" tabindex="-1"></a>                             <span class="at">selected_gene_id =</span> ensGene, column_names))</span>
<span id="cb5-60"><a href="#cb5-60" aria-hidden="true" tabindex="-1"></a>  result_list <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(summary_list[<span class="fu">lapply</span>(result_list,nrow)<span class="sc">!=</span><span class="dv">0</span>], <span class="sc">~</span>.<span class="sc">$</span>result)</span>
<span id="cb5-61"><a href="#cb5-61" aria-hidden="true" tabindex="-1"></a>  result_list <span class="ot">&lt;-</span> result_list[<span class="sc">!</span><span class="fu">unlist</span>(purrr<span class="sc">::</span><span class="fu">map</span>(result_list, is.null))]</span>
<span id="cb5-62"><a href="#cb5-62" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_df</span>(result_list, <span class="sc">~</span><span class="fu">run_coloc</span>(., gwas_stats_hg38), <span class="at">.id =</span> <span class="st">"qtl_id"</span>)</span>
<span id="cb5-63"><a href="#cb5-63" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-64"><a href="#cb5-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-65"><a href="#cb5-65" aria-hidden="true" tabindex="-1"></a>gtex <span class="ot">&lt;-</span> <span class="cf">function</span>(gwas_stats_hg38,ensGene,region38)</span>
<span id="cb5-66"><a href="#cb5-66" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb5-67"><a href="#cb5-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"c. GTEx_v8 imported eQTL datasets</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-68"><a href="#cb5-68" aria-hidden="true" tabindex="-1"></a>  rnaseq_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(imported_tabix_paths, quant_method <span class="sc">==</span> <span class="st">"ge"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-69"><a href="#cb5-69" aria-hidden="true" tabindex="-1"></a>               dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">qtl_id =</span> <span class="fu">paste</span>(study, qtl_group, <span class="at">sep =</span> <span class="st">"_"</span>))</span>
<span id="cb5-70"><a href="#cb5-70" aria-hidden="true" tabindex="-1"></a>  ftp_path_list <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">as.list</span>(rnaseq_df<span class="sc">$</span>ftp_path), rnaseq_df<span class="sc">$</span>qtl_id)</span>
<span id="cb5-71"><a href="#cb5-71" aria-hidden="true" tabindex="-1"></a>  hdr <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">path.package</span>(<span class="st">"pQTLtools"</span>),<span class="st">"eQTL-Catalogue"</span>,<span class="st">"column_names.GTEx"</span>)</span>
<span id="cb5-72"><a href="#cb5-72" aria-hidden="true" tabindex="-1"></a>  column_names <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">read.delim</span>(hdr))</span>
<span id="cb5-73"><a href="#cb5-73" aria-hidden="true" tabindex="-1"></a>  safe_import <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">safely</span>(import_eQTLCatalogue)</span>
<span id="cb5-74"><a href="#cb5-74" aria-hidden="true" tabindex="-1"></a>  summary_list <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(ftp_path_list,</span>
<span id="cb5-75"><a href="#cb5-75" aria-hidden="true" tabindex="-1"></a>                             <span class="sc">~</span><span class="fu">safe_import</span>(., region38, <span class="at">selected_gene_id =</span> ensGene, column_names))</span>
<span id="cb5-76"><a href="#cb5-76" aria-hidden="true" tabindex="-1"></a>  result_list <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(summary_list, <span class="sc">~</span>.<span class="sc">$</span>result)</span>
<span id="cb5-77"><a href="#cb5-77" aria-hidden="true" tabindex="-1"></a>  result_list <span class="ot">&lt;-</span> result_list[<span class="sc">!</span><span class="fu">unlist</span>(purrr<span class="sc">::</span><span class="fu">map</span>(result_list, is.null))]</span>
<span id="cb5-78"><a href="#cb5-78" aria-hidden="true" tabindex="-1"></a>  result_filtered <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(result_list[<span class="fu">lapply</span>(result_list,nrow)<span class="sc">!=</span><span class="dv">0</span>],</span>
<span id="cb5-79"><a href="#cb5-79" aria-hidden="true" tabindex="-1"></a>                                <span class="sc">~</span>dplyr<span class="sc">::</span><span class="fu">filter</span>(., <span class="sc">!</span><span class="fu">is.na</span>(se)))</span>
<span id="cb5-80"><a href="#cb5-80" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map_df</span>(result_filtered, <span class="sc">~</span><span class="fu">run_coloc</span>(., gwas_stats_hg38), <span class="at">.id =</span> <span class="st">"qtl_id"</span>)</span>
<span id="cb5-81"><a href="#cb5-81" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-82"><a href="#cb5-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-83"><a href="#cb5-83" aria-hidden="true" tabindex="-1"></a>coloc <span class="ot">&lt;-</span> <span class="cf">function</span>(prot,chr,ensGene,chain,region37,region38,out,<span class="at">run_all=</span><span class="cn">FALSE</span>)</span>
<span id="cb5-84"><a href="#cb5-84" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb5-85"><a href="#cb5-85" aria-hidden="true" tabindex="-1"></a>  gwas_stats_hg38 <span class="ot">&lt;-</span> <span class="fu">sumstats</span>(prot,chr,region37)</span>
<span id="cb5-86"><a href="#cb5-86" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (run_all)</span>
<span id="cb5-87"><a href="#cb5-87" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb5-88"><a href="#cb5-88" aria-hidden="true" tabindex="-1"></a>    df_microarray <span class="ot">&lt;-</span> <span class="fu">microarray</span>(gwas_stats_hg38,ensGene,region38)</span>
<span id="cb5-89"><a href="#cb5-89" aria-hidden="true" tabindex="-1"></a>    df_rnaseq <span class="ot">&lt;-</span> <span class="fu">rnaseq</span>(gwas_stats_hg38,ensGene,region38)</span>
<span id="cb5-90"><a href="#cb5-90" aria-hidden="true" tabindex="-1"></a>    df_gtex <span class="ot">&lt;-</span> <span class="fu">gtex</span>(gwas_stats_hg38,ensGene,region38)</span>
<span id="cb5-91"><a href="#cb5-91" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"df_microarray"</span>) <span class="sc">&amp;</span> <span class="fu">exits</span>(<span class="st">"df_rnaseq"</span>) <span class="sc">&amp;</span> <span class="fu">exists</span>(<span class="st">"df_gtex"</span>))</span>
<span id="cb5-92"><a href="#cb5-92" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb5-93"><a href="#cb5-93" aria-hidden="true" tabindex="-1"></a>      coloc_df <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(df_microarray, df_rnaseq, df_gtex)</span>
<span id="cb5-94"><a href="#cb5-94" aria-hidden="true" tabindex="-1"></a>      <span class="fu">saveRDS</span>(coloc_df, <span class="at">file=</span><span class="fu">paste0</span>(out,<span class="st">".RDS"</span>))</span>
<span id="cb5-95"><a href="#cb5-95" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(coloc_df, <span class="sc">-</span>PP.H4.abf)</span>
<span id="cb5-96"><a href="#cb5-96" aria-hidden="true" tabindex="-1"></a>      p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(coloc_df, <span class="fu">aes</span>(<span class="at">x =</span> PP.H4.abf)) <span class="sc">+</span> <span class="fu">geom_histogram</span>()</span>
<span id="cb5-97"><a href="#cb5-97" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-98"><a href="#cb5-98" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb5-99"><a href="#cb5-99" aria-hidden="true" tabindex="-1"></a>    df_gtex <span class="ot">&lt;-</span> <span class="fu">gtex</span>(gwas_stats_hg38,ensGene,region38)</span>
<span id="cb5-100"><a href="#cb5-100" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"df_gtex"</span>))</span>
<span id="cb5-101"><a href="#cb5-101" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb5-102"><a href="#cb5-102" aria-hidden="true" tabindex="-1"></a>      <span class="fu">saveRDS</span>(df_gtex,<span class="at">file=</span><span class="fu">paste0</span>(out,<span class="st">".RDS"</span>))</span>
<span id="cb5-103"><a href="#cb5-103" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(df_gtex, <span class="sc">-</span>PP.H4.abf)</span>
<span id="cb5-104"><a href="#cb5-104" aria-hidden="true" tabindex="-1"></a>      p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df_gtex, <span class="fu">aes</span>(<span class="at">x =</span> PP.H4.abf)) <span class="sc">+</span> <span class="fu">geom_histogram</span>()</span>
<span id="cb5-105"><a href="#cb5-105" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb5-106"><a href="#cb5-106" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb5-107"><a href="#cb5-107" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gwas_stats_hg38, <span class="fu">aes</span>(<span class="at">x =</span> position, <span class="at">y =</span> LP)) <span class="sc">+</span> <span class="fu">geom_point</span>()</span>
<span id="cb5-108"><a href="#cb5-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="at">plot =</span> s, <span class="at">filename =</span> <span class="fu">paste0</span>(out, <span class="st">"-assoc.pdf"</span>), <span class="at">path =</span> <span class="st">""</span>, <span class="at">device =</span> <span class="st">"pdf"</span>,</span>
<span id="cb5-109"><a href="#cb5-109" aria-hidden="true" tabindex="-1"></a>         <span class="at">height =</span> <span class="dv">15</span>, <span class="at">width =</span> <span class="dv">15</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb5-110"><a href="#cb5-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="at">plot =</span> p, <span class="at">filename =</span> <span class="fu">paste0</span>(out, <span class="st">"-hist.pdf"</span>), <span class="at">path =</span> <span class="st">""</span>, <span class="at">device =</span> <span class="st">"pdf"</span>,</span>
<span id="cb5-111"><a href="#cb5-111" aria-hidden="true" tabindex="-1"></a>         <span class="at">height =</span> <span class="dv">15</span>, <span class="at">width =</span> <span class="dv">15</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb5-112"><a href="#cb5-112" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-113"><a href="#cb5-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-114"><a href="#cb5-114" aria-hidden="true" tabindex="-1"></a>single_run <span class="ot">&lt;-</span> <span class="cf">function</span>(r)</span>
<span id="cb5-115"><a href="#cb5-115" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb5-116"><a href="#cb5-116" aria-hidden="true" tabindex="-1"></a>  sentinel <span class="ot">&lt;-</span> sentinels[r,]</span>
<span id="cb5-117"><a href="#cb5-117" aria-hidden="true" tabindex="-1"></a>  isnpid <span class="ot">&lt;-</span> <span class="fu">within</span>(gap<span class="sc">::</span><span class="fu">inv_chr_pos_a1_a2</span>(sentinel[[<span class="st">"SNP"</span>]]),</span>
<span id="cb5-118"><a href="#cb5-118" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb5-119"><a href="#cb5-119" aria-hidden="true" tabindex="-1"></a>    chr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"chr"</span>,<span class="st">""</span>,chr)</span>
<span id="cb5-120"><a href="#cb5-120" aria-hidden="true" tabindex="-1"></a>    pos <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(pos)</span>
<span id="cb5-121"><a href="#cb5-121" aria-hidden="true" tabindex="-1"></a>    start <span class="ot">&lt;-</span> pos<span class="sc">-</span>M</span>
<span id="cb5-122"><a href="#cb5-122" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (start<span class="sc">&lt;</span><span class="dv">0</span>) start <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb5-123"><a href="#cb5-123" aria-hidden="true" tabindex="-1"></a>    end <span class="ot">&lt;-</span> pos<span class="sc">+</span>M</span>
<span id="cb5-124"><a href="#cb5-124" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb5-125"><a href="#cb5-125" aria-hidden="true" tabindex="-1"></a>  chr <span class="ot">&lt;-</span> <span class="fu">with</span>(isnpid,chr)</span>
<span id="cb5-126"><a href="#cb5-126" aria-hidden="true" tabindex="-1"></a>  region37 <span class="ot">&lt;-</span> <span class="fu">with</span>(isnpid, <span class="fu">paste0</span>(chr,<span class="st">":"</span>,start,<span class="st">"-"</span>,end))</span>
<span id="cb5-127"><a href="#cb5-127" aria-hidden="true" tabindex="-1"></a>  ensRegion37 <span class="ot">&lt;-</span> <span class="fu">with</span>(<span class="fu">subset</span>(inf1,prot<span class="sc">==</span>sentinel[[<span class="st">"prot"</span>]]),</span>
<span id="cb5-128"><a href="#cb5-128" aria-hidden="true" tabindex="-1"></a>                      {</span>
<span id="cb5-129"><a href="#cb5-129" aria-hidden="true" tabindex="-1"></a>                        start <span class="ot">&lt;-</span> start<span class="sc">-</span>M</span>
<span id="cb5-130"><a href="#cb5-130" aria-hidden="true" tabindex="-1"></a>                        <span class="cf">if</span> (start<span class="sc">&lt;</span><span class="dv">0</span>) start <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb5-131"><a href="#cb5-131" aria-hidden="true" tabindex="-1"></a>                        end <span class="ot">&lt;-</span> end<span class="sc">+</span>M</span>
<span id="cb5-132"><a href="#cb5-132" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">paste0</span>(chr,<span class="st">":"</span>,start,<span class="st">"-"</span>,end)</span>
<span id="cb5-133"><a href="#cb5-133" aria-hidden="true" tabindex="-1"></a>                      })</span>
<span id="cb5-134"><a href="#cb5-134" aria-hidden="true" tabindex="-1"></a>  region38 <span class="ot">&lt;-</span> <span class="fu">with</span>(<span class="fu">liftRegion</span>(isnpid,chain),region)</span>
<span id="cb5-135"><a href="#cb5-135" aria-hidden="true" tabindex="-1"></a>  ensGene <span class="ot">&lt;-</span> <span class="fu">subset</span>(inf1,prot<span class="sc">==</span>sentinel[[<span class="st">"prot"</span>]])[[<span class="st">"ensembl_gene_id"</span>]]</span>
<span id="cb5-136"><a href="#cb5-136" aria-hidden="true" tabindex="-1"></a>  ensRegion38 <span class="ot">&lt;-</span> <span class="fu">with</span>(<span class="fu">liftRegion</span>(<span class="fu">subset</span>(inf1,prot<span class="sc">==</span>sentinel[[<span class="st">"prot"</span>]]),chain),region)</span>
<span id="cb5-137"><a href="#cb5-137" aria-hidden="true" tabindex="-1"></a>  f <span class="ot">&lt;-</span> <span class="fu">file.path</span>(INF,<span class="st">"coloc"</span>,<span class="fu">with</span>(sentinel,<span class="fu">paste0</span>(prot,<span class="st">"-"</span>,SNP)))</span>
<span id="cb5-138"><a href="#cb5-138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(chr,region37,region38,ensGene,ensRegion37,ensRegion38,<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb5-139"><a href="#cb5-139" aria-hidden="true" tabindex="-1"></a><span class="co"># coloc(sentinel[["prot"]],chr,ensGene,chain,region37,region38,f)</span></span>
<span id="cb5-140"><a href="#cb5-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coloc</span>(sentinel[[<span class="st">"prot"</span>]],chr,ensGene,chain,ensRegion37,ensRegion38,f)</span>
<span id="cb5-141"><a href="#cb5-141" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-142"><a href="#cb5-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-143"><a href="#cb5-143" aria-hidden="true" tabindex="-1"></a><span class="co"># slow with the following loop:</span></span>
<span id="cb5-144"><a href="#cb5-144" aria-hidden="true" tabindex="-1"></a>loop <span class="ot">&lt;-</span> <span class="cf">function</span>() <span class="cf">for</span> (r <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sentinels)) <span class="fu">single_run</span>(r)</span>
<span id="cb5-145"><a href="#cb5-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-146"><a href="#cb5-146" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pQTLtools)</span>
<span id="cb5-147"><a href="#cb5-147" aria-hidden="true" tabindex="-1"></a>f <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">path.package</span>(<span class="st">"pQTLtools"</span>),<span class="st">"eQTL-Catalogue"</span>,<span class="st">"hg19ToHg38.over.chain"</span>)</span>
<span id="cb5-148"><a href="#cb5-148" aria-hidden="true" tabindex="-1"></a>chain <span class="ot">&lt;-</span> rtracklayer<span class="sc">::</span><span class="fu">import.chain</span>(f)</span>
<span id="cb5-149"><a href="#cb5-149" aria-hidden="true" tabindex="-1"></a>pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"dplyr"</span>, <span class="st">"ggplot2"</span>, <span class="st">"readr"</span>, <span class="st">"coloc"</span>, <span class="st">"GenomicRanges"</span>,<span class="st">"seqminer"</span>)</span>
<span id="cb5-150"><a href="#cb5-150" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(pkgs, require, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span>
<span id="cb5-151"><a href="#cb5-151" aria-hidden="true" tabindex="-1"></a>HPC_WORK <span class="ot">&lt;-</span> <span class="fu">Sys.getenv</span>(<span class="st">"HPC_WORK"</span>)</span>
<span id="cb5-152"><a href="#cb5-152" aria-hidden="true" tabindex="-1"></a>gwasvcf<span class="sc">::</span><span class="fu">set_bcftools</span>(<span class="fu">file.path</span>(HPC_WORK,<span class="st">"bin"</span>,<span class="st">"bcftools"</span>))</span>
<span id="cb5-153"><a href="#cb5-153" aria-hidden="true" tabindex="-1"></a>f <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">path.package</span>(<span class="st">"pQTLtools"</span>),<span class="st">"eQTL-Catalogue"</span>,<span class="st">"tabix_ftp_paths.tsv"</span>)</span>
<span id="cb5-154"><a href="#cb5-154" aria-hidden="true" tabindex="-1"></a>tabix_paths <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(f, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">as_tibble</span>()</span>
<span id="cb5-155"><a href="#cb5-155" aria-hidden="true" tabindex="-1"></a>HOME <span class="ot">&lt;-</span> <span class="fu">Sys.getenv</span>(<span class="st">"HOME"</span>)</span>
<span id="cb5-156"><a href="#cb5-156" aria-hidden="true" tabindex="-1"></a>fp <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="fu">path.package</span>(<span class="st">"pQTLtools"</span>),<span class="st">"eQTL-Catalogue"</span>,<span class="st">"tabix_ftp_paths_gtex.tsv"</span>)</span>
<span id="cb5-157"><a href="#cb5-157" aria-hidden="true" tabindex="-1"></a>imported_tabix_paths <span class="ot">&lt;-</span> <span class="fu">within</span>(<span class="fu">read.delim</span>(fp, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">as_tibble</span>(),</span>
<span id="cb5-158"><a href="#cb5-158" aria-hidden="true" tabindex="-1"></a>      {</span>
<span id="cb5-159"><a href="#cb5-159" aria-hidden="true" tabindex="-1"></a>         f <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">strsplit</span>(ftp_path,<span class="st">"/csv/|/ge/"</span>),<span class="st">"["</span>,<span class="dv">2</span>)</span>
<span id="cb5-160"><a href="#cb5-160" aria-hidden="true" tabindex="-1"></a>         ftp_path <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"~/rds/public_databases/GTEx/csv"</span>),f<span class="er">)</span></span>
<span id="cb5-161"><a href="#cb5-161" aria-hidden="true" tabindex="-1"></a>      })</span>
<span id="cb5-162"><a href="#cb5-162" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">width=</span><span class="dv">200</span>)</span>
<span id="cb5-163"><a href="#cb5-163" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb5-164"><a href="#cb5-164" aria-hidden="true" tabindex="-1"></a>INF <span class="ot">&lt;-</span> <span class="fu">Sys.getenv</span>(<span class="st">"INF"</span>)</span>
<span id="cb5-165"><a href="#cb5-165" aria-hidden="true" tabindex="-1"></a>M <span class="ot">&lt;-</span> <span class="fl">1e6</span></span>
<span id="cb5-166"><a href="#cb5-166" aria-hidden="true" tabindex="-1"></a>sentinels <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">read.csv</span>(<span class="fu">file.path</span>(INF,<span class="st">"work"</span>,<span class="st">"INF1.merge.cis.vs.trans"</span>)),cis)</span>
<span id="cb5-167"><a href="#cb5-167" aria-hidden="true" tabindex="-1"></a>cvt_rsid <span class="ot">&lt;-</span> <span class="fu">file.path</span>(INF,<span class="st">"work"</span>,<span class="st">"INF1.merge.cis.vs.trans-rsid"</span>)</span>
<span id="cb5-168"><a href="#cb5-168" aria-hidden="true" tabindex="-1"></a>prot_rsid <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">read.delim</span>(cvt_rsid,<span class="at">sep=</span><span class="st">" "</span>),cis,<span class="at">select=</span><span class="fu">c</span>(prot,SNP))</span>
<span id="cb5-169"><a href="#cb5-169" aria-hidden="true" tabindex="-1"></a><span class="co"># Faster with parallel Bash runs.</span></span>
<span id="cb5-170"><a href="#cb5-170" aria-hidden="true" tabindex="-1"></a>r <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">Sys.getenv</span>(<span class="st">"r"</span>))</span>
<span id="cb5-171"><a href="#cb5-171" aria-hidden="true" tabindex="-1"></a><span class="fu">single_run</span>(r)</span></code></pre></div>
<p>where options for protein GWAS, microarray, RNA-Seq are available with respect to variant-flanking or gene regions. When no results are generated,
there would have problem with <code>dplyr::arrange(df_gtex, -PP.H4.abf);p &lt;- ggplot(df_gtex, aes(x = PP.H4.abf)) + geom_histogram()</code>.</p>
</div>
<div class="section level3" number="2.4">
<h3 id="collection-of-results">
<span class="header-section-number">2.4</span> Collection of results<a class="anchor" aria-label="anchor" href="#collection-of-results"></a>
</h3>
<p>When these are furnished we keep results as follows,</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">collect</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">df_coloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span>  <span class="kw">for</span><span class="op">(</span><span class="va">r</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sentinels</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">{</span></span>
<span>    <span class="va">prot</span> <span class="op">&lt;-</span> <span class="va">sentinels</span><span class="op">[[</span><span class="st">"prot"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">r</span><span class="op">]</span></span>
<span>    <span class="va">snpid</span> <span class="op">&lt;-</span> <span class="va">sentinels</span><span class="op">[[</span><span class="st">"SNP"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">r</span><span class="op">]</span></span>
<span>    <span class="va">rsid</span> <span class="op">&lt;-</span> <span class="va">prot_rsid</span><span class="op">[[</span><span class="st">"SNP"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">r</span><span class="op">]</span></span>
<span>    <span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"coloc"</span>,<span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">snpid</span>,<span class="st">".RDS"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span><span class="op">)</span> <span class="kw">next</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">rsid</span>,<span class="st">"\n"</span><span class="op">)</span></span>
<span>    <span class="va">rds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">f</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rds</span><span class="op">)</span><span class="op">==</span><span class="fl">0</span><span class="op">)</span> <span class="kw">next</span></span>
<span>    <span class="va">df_coloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">df_coloc</span>,<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>prot<span class="op">=</span><span class="va">prot</span>,rsid<span class="op">=</span><span class="va">rsid</span>,snpid<span class="op">=</span><span class="va">snpid</span>,<span class="va">rds</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span>  <span class="va">df_coloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">df_coloc</span>,<span class="op">{</span><span class="va">qtl_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"GTEx_V8_"</span>,<span class="st">""</span>,<span class="va">qtl_id</span><span class="op">)</span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>              <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>H0<span class="op">=</span><span class="va">PP.H0.abf</span>,H1<span class="op">=</span><span class="va">PP.H1.abf</span>,H2<span class="op">=</span><span class="va">PP.H2.abf</span>,H3<span class="op">=</span><span class="va">PP.H3.abf</span>,H4<span class="op">=</span><span class="va">PP.H4.abf</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">df_coloc</span>,<span class="va">H3</span><span class="op">+</span><span class="va">H4</span><span class="op">&gt;=</span><span class="fl">0.9</span> <span class="op">&amp;</span> <span class="va">H4</span><span class="op">/</span><span class="va">H3</span><span class="op">&gt;=</span><span class="fl">3</span><span class="op">)</span>,</span>
<span>              file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"coloc"</span>,<span class="st">"GTEx.tsv"</span><span class="op">)</span>,</span>
<span>              quote<span class="op">=</span><span class="cn">FALSE</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/compute.html" class="external-link">collect</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>i.e., PP3+PP4&gt;=0.9 and PP4/PP3&gt;=3.</p>
</div>
<div class="section level3" number="2.5">
<h3 id="parallel-computing">
<span class="header-section-number">2.5</span> Parallel computing<a class="anchor" aria-label="anchor" href="#parallel-computing"></a>
</h3>
<p>To speed up the analysis, we resort to <a href="https://slurm.schedmd.com/documentation.html" class="external-link">SLURM</a>,</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/usr/bin/bash</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --job-name=_coloc</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --account CARDIO-SL0-CPU</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --partition cardio</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --qos=cardio</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --array=1-59</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --mem=28800</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --time=5-00:00:00</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --error=/rds/user/jhz22/hpc-work/work/_coloc_%A_%a.err</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --output=/rds/user/jhz22/hpc-work/work/_coloc_%A_%a.out</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#SBATCH --export ALL</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">job</span><span class="op">=</span><span class="va">${SLURM_ARRAY_TASK_ID}</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">trait</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="at">-v</span> job=<span class="va">${job}</span> <span class="st">'NR==job {print $1}'</span> <span class="va">${INF}</span>/work/inf1.tmp<span class="va">)</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="kw">function</span><span class="fu"> gtex()</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  <span class="bu">export</span> <span class="va">r</span><span class="op">=</span><span class="va">${SLURM_ARRAY_TASK_ID}</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="bu">export</span> <span class="va">cvt</span><span class="op">=</span><span class="va">${INF}</span>/work/INF1.merge.cis.vs.trans</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="bu">read</span> <span class="va">prot</span> <span class="va">MarkerName</span> <span class="op">&lt;</span> <span class="dt">\</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>                       <span class="op">&lt;(</span><span class="fu">awk</span> <span class="at">-vFS</span><span class="op">=</span><span class="st">","</span> <span class="st">'$14=="cis"'</span> <span class="va">${cvt}</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">awk</span> <span class="at">-vFS</span><span class="op">=</span><span class="st">","</span> <span class="at">-vr</span><span class="op">=</span><span class="va">${r}</span> <span class="st">'NR==r{print $2,$5}'</span><span class="op">)</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="va">${r}</span> <span class="at">-</span> <span class="va">${prot}</span> <span class="at">-</span> <span class="va">${MarkerName}</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="bu">export</span> <span class="va">prot</span><span class="op">=</span><span class="va">${prot}</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  <span class="bu">export</span> <span class="va">MarkerName</span><span class="op">=</span><span class="va">${MarkerName}</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-f</span> <span class="va">${INF}</span>/coloc/<span class="va">${prot}</span>-<span class="va">${MarkerName}</span>.pdf <span class="bu">]</span> <span class="kw">||</span> <span class="dt">\</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>     <span class="bu">[</span> <span class="ot">!</span> <span class="ot">-f</span> <span class="va">${INF}</span>/coloc/<span class="va">${prot}</span>-<span class="va">${MarkerName}</span>.RDS <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>    <span class="bu">cd</span> <span class="va">${INF}</span>/coloc</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>    <span class="ex">R</span> <span class="at">--no-save</span> <span class="op">&lt;</span> <span class="va">${INF}</span>/rsid/coloc.R <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">tee</span> <span class="va">${prot}</span>-<span class="va">${MarkerName}</span>.log</span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>    <span class="bu">cd</span> <span class="at">-</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>  <span class="cf">fi</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a><span class="ex">gtex</span></span></code></pre></div>
<div class="section level4">
<h4 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h4>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-malone10" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Malone, J. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btq099" class="external-link">Modeling sample variables with an experimental factor ontology</a>. <em>Bioinformatics</em> <strong>26</strong>, 1112â€“1118 (2010).</div>
</div>
<div id="ref-kamat19" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Kamat, M. A. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btz469" class="external-link">PhenoScanner V2: An expanded tool for searching human genotype-phenotype associations</a>. <em>Bioinformatics</em> <strong>35</strong>, 4851â€“4853 (2019).</div>
</div>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Crafted by <a href="https://jinghuazhao.github.io/" class="external-link"><img src="https://jinghuazhao.github.io/bees.svg" width="24"> Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
