<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="pQTLtools">
<title>A snakemake showcase • pQTLtools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A snakemake showcase">
<meta property="og:description" content="pQTLtools">
<meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Vignettes:</h6>
    <a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Articles:</h6>
    <a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a>
    <a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a>
    <a class="dropdown-item" href="../articles/snakemake.html">A snakemake showcase</a>
    <a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/">
    <span class="fas fa fas fa-fast-backward"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-sites">Sites</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-sites">
    <a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a>
    <a class="external-link dropdown-item" href="https://www.bioconductor.org/">Bioconductor</a>
    <a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a>
    <a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a>
    <a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a>
    <a class="external-link dropdown-item" href="https://metascape.org/">metascape</a>
    <a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a>
    <a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a>
    <a class="external-link dropdown-item" href="https://string-db.org/">STRING</a>
    <a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a>
    <a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.uniprot.org/">
    <span class="fas fa fas fa-fast-forward"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://jinghuazhao.github.io/R/">
    <span class="fab fa fab fa-r-project"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>A snakemake showcase</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/vignettes/snakemake.Rmd" class="external-link"><code>vignettes/snakemake.Rmd</code></a></small>
      <div class="d-none name"><code>snakemake.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="mendelian-randomization-analysis">Mendelian randomization analysis<a class="anchor" aria-label="anchor" href="#mendelian-randomization-analysis"></a>
</h2>
<p>The required files are available from the <code>snakemake</code>
subdirectory inside the installed package or <code>inst/snakemake</code>
directory in the source package.</p>
<p>Steps to set up the environment are outlined below, while
<code>MendelianRandomization</code> v0.6.0 is used together with a bug
fix in <code>workflow/r/MR_functions.R</code>. The workflow has been
heavily edited for simplicity, efficiency and generality. Currently
<code>input/</code> contains data on CD40, OPG and heart failures – to
imitate additional trait, HF statistics are duplicated as HF2.</p>
<p>The code chunks below gives
<code>output</code>/<code>MR_HF.csv</code> (MR results) and
<code>Obs_HF.csv</code> (meta-analysis results based on observational
studies) and similarly for HF2.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load miniconda3/4.5.1</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">csd3path</span><span class="op">=</span>/rds/project/jmmh2/rds-jmmh2-projects/olink_proteomics/scallop</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> <span class="va">${csd3path}</span>/miniconda37/bin/activate</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. a dry run (-n).</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--dry-run</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. run (-c on [all] available cores without --use-conda option as local packages are more up-to-date)</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--cores</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. contrast with original output for OPG</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># grep OPG output/MR_HF.csv | diff - &lt;(grep OPG ${csd3path}/cvd1-hf/results/res_MR_aggregate.csv)</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. Some ancillary work in place.</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--rulegraph</span> <span class="kw">|</span> <span class="ex">dot</span> <span class="at">-Tsvg</span> <span class="op">&gt;</span> output/rulegraph.svg</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--dag</span> <span class="kw">|</span> <span class="ex">dot</span> <span class="at">-Tsvg</span> <span class="op">&gt;</span> output/dag.svg</span></code></pre></div>
<div class="figure">
<img src="../inst/snakemake/output/rulegraph.svg" alt=""><p class="caption">Figure 1. Dependency graph of rules</p>
</div>
<div class="figure">
<img src="../inst/snakemake/output/dag.svg" alt=""><p class="caption">Figure 2. Directed Acyclic Graph (DAG)</p>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p><a href="https://github.com/snakemake-workflows/" class="external-link">https://github.com/snakemake-workflows/</a></p>
<p><a href="https://www.ahajournals.org/doi/10.1161/CIRCULATIONAHA.121.056663" class="external-link"><img src="https://www.ahajournals.org/pb-assets/images/logos/circ-logo-1526571039097.svg" alt="Circulation"></a> <a href="https://zenodo.org/badge/latestdoi/429122036" class="external-link"><img src="https://zenodo.org/badge/429122036.svg" alt="DOI"></a></p>
</div>
<div class="section level2">
<h2 id="appendix-installation">Appendix: Installation<a class="anchor" aria-label="anchor" href="#appendix-installation"></a>
</h2>
<div class="section level3">
<h3 id="github">GitHub<a class="anchor" aria-label="anchor" href="#github"></a>
</h3>
<p><a href="https://github.com/snakemake/snakemake" class="external-link">GitHub</a> (<a href="https://snakemake.github.io/" class="external-link">documentation</a>, <a href="https://snakemake.readthedocs.io/en/stable/" class="external-link">stable</a>)</p>
<p>Note that in the following <code>source</code> instead of
<code>conda</code> is used to activate.</p>
</div>
<div class="section level3">
<h3 id="anaconda3">Anaconda3<a class="anchor" aria-label="anchor" href="#anaconda3"></a>
</h3>
<p><a href="https://www.anaconda.com/" class="external-link">https://www.anaconda.com/</a></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://repo.anaconda.com/archive/Anaconda3-2020.07-Linux-x86_64.sh</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> Anaconda3-2020.07-Linux-x86_64.sh</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># snakemake</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> anaconda</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> remove <span class="at">-n</span> anaconda snakemake</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install <span class="at">-c</span> bioconda snakemake</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate snakemake</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> update <span class="at">-n</span> anaconda snakemake</span></code></pre></div>
<p>The remove option is useful when resolving compatibility issues.</p>
</div>
<div class="section level3">
<h3 id="miniconda3">Miniconda3<a class="anchor" aria-label="anchor" href="#miniconda3"></a>
</h3>
<p><a href="https://docs.conda.io/en/latest/miniconda.html" class="external-link">https://docs.conda.io/en/latest/miniconda.html</a></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> Miniconda3-latest-Linux-x86_64.sh</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels bioconda</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels conda-forge</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-y</span> <span class="at">--name</span> miniconda python=3.7</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate miniconda</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install <span class="at">-c</span> bioconda fastqc</span></code></pre></div>
<p>or to do this permenantly.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># all users</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sudo ln -s /usr/local/Cluster-Apps/miniconda3/4.5.1/etc/profile.d/conda.sh /etc/profile.d/conda.sh</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># current user</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">". /usr/local/Cluster-Apps/miniconda3/4.5.1/etc/profile.d/conda.sh"</span> <span class="op">&gt;&gt;</span> ~/.bashrc</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># conda's base (root) environment on PATH</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># the base environment on PATH permanently</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"conda activate"</span> <span class="op">&gt;&gt;</span> ~/.bashrc</span></code></pre></div>
<p>such that</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--j4</span> <span class="at">--use-conda</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">--profile</span> slurm</span></code></pre></div>
<p>The first line calls for conda and the second links to slurm.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="va">$HOME</span>/.config/Snakemake/slurm</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> slurm.yaml <span class="va">$HOME</span>/.config/Snakemake/slurm</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">touch</span> <span class="va">$HOME</span>/.config/Snakemake/slurm/slurm.yaml</span></code></pre></div>
<p>Additional information is available from here, <a href="https://ucdavis-bioinformatics-training.github.io/2020-Genome_Assembly_Workshop/snakemake/snakemake_intro" class="external-link">https://ucdavis-bioinformatics-training.github.io/2020-Genome_Assembly_Workshop/snakemake/snakemake_intro</a>.
In particular, the parameters are specified through
<code>cluster.json</code></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode json"><code class="sourceCode json"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">{</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>    <span class="dt">"__default__"</span> <span class="fu">:</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">{</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"time"</span> <span class="fu">:</span> <span class="st">"0:20:00"</span><span class="fu">,</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"nodes"</span><span class="fu">:</span> <span class="dv">1</span><span class="fu">,</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"ntasks"</span><span class="fu">:</span> <span class="dv">1</span><span class="fu">,</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"cpus"</span> <span class="fu">:</span> <span class="dv">1</span><span class="fu">,</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"mem"</span><span class="fu">:</span> <span class="dv">2000</span><span class="fu">,</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"output"</span><span class="fu">:</span> <span class="st">"snakemake%A.out"</span><span class="fu">,</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"error"</span><span class="fu">:</span> <span class="st">"snakemake%A.err"</span><span class="fu">,</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"reservation"</span><span class="fu">:</span> <span class="st">"genome_workshop"</span><span class="fu">,</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"partition"</span><span class="fu">:</span> <span class="st">"production"</span><span class="fu">,</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>        <span class="dt">"account"</span><span class="fu">:</span> <span class="st">"genome_workshop"</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">}</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="fu">}</span></span></code></pre></div>
<p>and our call becomes</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">snakemake</span> <span class="at">-j</span> 99 <span class="at">--cluster-config</span> cluster.json <span class="at">--cluster</span> <span class="st">"sbatch -t {cluster.time} --output {cluster.output} --error {cluster.error} --nodes {cluster.nodes} --ntasks {cluster.ntasks} --cpus-per-task {cluster.cpus} --mem {cluster.mem} --partition {cluster.partition} --account {cluster.account} --reservation {cluster.reservation}"</span> <span class="at">--use-conda</span> <span class="at">--latency-wait</span> 50</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="further-information">Further information<a class="anchor" aria-label="anchor" href="#further-information"></a>
</h3>
<p>Edwards D (2022). Plant Bioinformatics-Methods and Protocols, 3e. <a href="https://link.springer.com/book/10.1007/978-1-0716-2067-0" class="external-link uri">https://link.springer.com/book/10.1007/978-1-0716-2067-0</a>.
<a href="https://link.springer.com/protocol/10.1007/978-1-0716-2067-0_11" class="external-link">Chapter
11</a>; <a href="https://link.springer.com/protocol/10.1007/978-1-0716-2067-0_9" class="external-link">Chapter
9</a>.</p>
<p>csd3, <a href="https://cambridge-ceu.github.io/csd3/applications/snakemake.html" class="external-link">https://cambridge-ceu.github.io/csd3/applications/snakemake.html</a>.</p>
<p>snakemake-with-R, <a href="https://github.com/fritzbayer/snakemake-with-R" class="external-link">https://github.com/fritzbayer/snakemake-with-R</a>.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Crafted by <a href="https://jinghuazhao.github.io/" class="external-link"><img src="https://jinghuazhao.github.io/bees.svg" width="24"> Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
