<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Spectrum data analysis • pQTLtools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Spectrum data analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span></a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a></li>
    <li><a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a></li>
    <li><a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a></li>
    <li><a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a></li>
    <li><a class="dropdown-item" href="../articles/spectrum.html">spectrum data analysis</a></li>
    <li><a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/pQTLtools.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/"><span class="fa fab fa-r-project"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/"><span class="fa fas fa-fast-backward"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-urls" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">URLs</button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-urls">
<li><a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a></li>
    <li><a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a></li>
    <li><a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a></li>
    <li><a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a></li>
    <li><a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://metascape.org/">metascape</a></li>
    <li><a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a></li>
    <li><a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a></li>
    <li><a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a></li>
    <li><a class="external-link dropdown-item" href="https://string-db.org/">STRING</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a></li>
    <li><a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a></li>
  </ul>
</li>
<li class="nav-item"><a class="external-link nav-link" href="https://www.uniprot.org/"><span class="fa fas fa-fast-forward"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Spectrum data analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/vignettes/spectrum.Rmd" class="external-link"><code>vignettes/spectrum.Rmd</code></a></small>
      <div class="d-none name"><code>spectrum.Rmd</code></div>
    </div>

    
    
<p>This article collects notes on peptide/protein analysis, especially with respect to spectrum data.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pkgs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Biostrings"</span>, <span class="st">"protViz"</span>, <span class="st">"rawrr"</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">p</span> <span class="kw">in</span> <span class="va">pkgs</span><span class="op">)</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"^package:"</span>, <span class="va">p</span>, <span class="st">"$"</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/search.html" class="external-link">search</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/warning.html" class="external-link">warning</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"This vignette needs package `"</span>, <span class="va">p</span>, <span class="st">"'; please install"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">pkgs</span>, <span class="va">require</span>, character.only <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2" number="1">
<h2 id="peptide-sequence">
<span class="header-section-number">1</span> Peptide sequence<a class="anchor" aria-label="anchor" href="#peptide-sequence"></a>
</h2>
<p>Here is an example for PROC_HUMAN, which is handled by the <code>Biostrings</code> package,</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fasta_file_path</span> <span class="op">&lt;-</span> <span class="st">'https://rest.uniprot.org/uniprotkb/P04070.fasta'</span></span>
<span><span class="va">fasta_sequences</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">fasta_file_path</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span></span>
<span><span class="va">AA_sequence</span> <span class="op">&lt;-</span> <span class="va">fasta_sequences</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Sequence:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/toString.html" class="external-link">toString</a></span><span class="op">(</span><span class="va">AA_sequence</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="va">iso_442688365</span> <span class="op">&lt;-</span> <span class="st">'TDGEGALSEPSATVTIEELAAPPPPVLMHHGESSQVLHPGNK'</span></span>
<span><span class="va">match_position</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">regexpr</a></span><span class="op">(</span><span class="va">iso_442688365</span>, <span class="va">AA_sequence</span><span class="op">)</span></span>
<span><span class="va">match_position</span></span>
<span><span class="va">mp</span> <span class="op">&lt;-</span> <span class="fu">matchPattern</span><span class="op">(</span><span class="va">iso_442688365</span>,<span class="va">AA_sequence</span><span class="op">)</span></span>
<span><span class="va">mp</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"~/pQTLtools/tests/PROC.rda"</span><span class="op">)</span></span>
<span><span class="fu">pQTLtools</span><span class="fu">::</span><span class="fu"><a href="../reference/peptideAssociationPlot.html">peptideAssociationPlot</a></span><span class="op">(</span><span class="va">protein</span>,<span class="va">cistrans</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(Modified.Peptide.Sequence)`</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:a1bg"></span>
<img src="spectrum/a1bg-1.png" alt="peptide association plot" width="960"><p class="caption">
Figure 1.1: peptide association plot
</p>
</div>
</div>
<div class="section level2" number="2">
<h2 id="spectrum-data-analysis">
<span class="header-section-number">2</span> Spectrum data analysis<a class="anchor" aria-label="anchor" href="#spectrum-data-analysis"></a>
</h2>
<p>This will be expanded in details in due course.</p>
<p>``r
options(width=200)
# ZWK .raw data
library(rawrr)
if (isFALSE(rawrr::.checkDllInMonoPath())){
rawrr::installRawFileReaderDLLs()
}
if (isFALSE(file.exists(rawrr:::.rawrrAssembly()))){
rawrr::installRawrrExe()
}
spectra_ZWK &lt;- “~/Caprion/pre_qc_data/spectral_library_ZWK”
raw_files &lt;- list.files(spectra_ZWK, pattern = “\.raw$”, full.names = TRUE)
## collectively
suppressMessages(library(MsBackendRawFileReader))
ZWK &lt;- Spectra::backendInitialize(MsBackendRawFileReader::MsBackendRawFileReader(),
files = raw_files)
class(ZWK)
methods(class=class(ZWK))
Spectra(ZWK)
spectraData(ZWK)
ZWK
ZWKvars &lt;- ZWK |&gt; Spectra::spectraVariables()
ZWKdata &lt;- ZWK |&gt; Spectra::spectraData()
dim(ZWKdata)
# rows with &gt;=1 non-NA value in the columns with prefix “precursor”
precursor &lt;- apply(ZWKdata[grep(“precursor”,ZWKvars)], 1, function(x) any(!is.na(x)))
ZWKdata_filtered &lt;- ZWKdata[precursor, ]
save(ZWK,file=“~/Caprion/analysis/work/ZWK.rda”)</p>
</div>
<div class="section level2" number="3">
<h2 id="zyqudp">
<span class="header-section-number">3</span> ZYQ/UDP<a class="anchor" aria-label="anchor" href="#zyqudp"></a>
</h2>
<p>library(utils)
spectra &lt;- “~/Caprion/pre_qc_data/spectra”
zip_files &lt;- dir(spectra, recursive = TRUE, full.names=TRUE)
work_dir &lt;- “~/Caprion/analysis/work”
for (zip_file in zip_files) unzip(zip_file, exdir=work_dir)
ZYQ_UDP &lt;- Spectra::backendInitialize(MsBackendRawFileReader::MsBackendRawFileReader(),
files = dir(work_dir,patt=“raw”,full.names=TRUE))
class(ZYQ_UDP)
ZYQ_UDP
ZYQ_UDP |&gt; Spectra::spectraVariables()
save(ZYQ_UDP,file=“~/Caprion/analysis/work/ZYQ_UDP.rda”)</p>
</div>
<div class="section level2" number="4">
<h2 id="mzml">
<span class="header-section-number">4</span> mzML<a class="anchor" aria-label="anchor" href="#mzml"></a>
</h2>
<p>library(mzR)
d &lt;- “/rds/project/rds-MkfvQMuSUxk/interval/caprion_proteomics/spectra”
mz &lt;- openMSfile(file.path(d,“budp082710c21101xms1.mzML”))
methods(class=“mzRpwiz”)
spec &lt;- spectra(mz)
class(spec)
length(spec)
lapply(spec,head,3)</p>
</div>
<div class="section level2" number="5">
<h2 id="msnbase">
<span class="header-section-number">5</span> MSnbase<a class="anchor" aria-label="anchor" href="#msnbase"></a>
</h2>
<p>library(MSnbase)
mzXML &lt;- readMSData(“szwk901104i19801xms1.mzXML”)
mgf &lt;- readMgfData(“szwk901104i19801xms1.mgf”)
save(mzXML,mgf,file=“szwk901104i19801xms1.rda”)</p>
<p>extractSpectraData(mzXML)
hasSpectra(“szwk901104i19801xms1.mzML,gz”)
hasChromatograms(“szwk901104i19801xms1.mzML,gz”)
plot2d(mzXML,z=“peaks.count”)
plotDensity(mzXML,z=“precursor.mz”)</p>
<p>extractSpectraData(mgf)
methods(class=“MSpectra”)
mz(mgf)
intensity(mgf)
rtime(mgf)
precursorMz(mgf)
precursorCharge(mgf)
precScanNum(mgf)
precursorIntensity(mgf)
acquisitionNum(mgf)
scanIndex(mgf)
peaksCount(mgf)
msLevel(mgf)
tic(mgf)
ionCount(mgf)
collisionEnergy(mgf)
fromFile(mgf)
polarity(mgf)
smoothed(mgf)
centroided(mgf)
isCentroided(mgf)
writeMgfData(mgf, con = “spectra.mgf”, COM = NULL, TITLE = NULL)
removePeaks(mgf, t, msLevel., …)
filterMsLevel(mgf, msLevel=2)
as.ExpressionSet(mgf)</p>
</div>
<div class="section level2" number="6">
<h2 id="this-turned-to-be-really-slow">
<span class="header-section-number">6</span> This turned to be really slow!<a class="anchor" aria-label="anchor" href="#this-turned-to-be-really-slow"></a>
</h2>
<p>sp_list &lt;- lapply(seq_along(mgf), function(i) {
intensity_i &lt;- intensity(mgf)[[i]]
mz_i &lt;- mz(mgf)[[i]]
centroided_i &lt;- centroided(mgf)[[i]]
return(new(“Spectrum1”, intensity = intensity_i, mz = mz_i, centroided = centroided_i))
})
sp1 &lt;- do.call(rbind, sp_list)
# only the first one is more manageable
sp1 &lt;- new(“Spectrum1”,intensity=intensity(mgf)[[1]],mz=mz(mgf)[[1]],centroided=centroided(mgf)[[1]])
sp2 &lt;- pickPeaks(sp1)
intensity(sp2)
plot(mz(sp1),intensity(sp1),type=“h”)
## Without m/z refinement
points(mz(sp2), intensity(sp2), col = “darkgrey”)
## Using k = 1, closest signals
sp3 &lt;- pickPeaks(sp1, refineMz = “kNeighbors”, k = 1)
points(mz(sp3), intensity(sp3), col = “green”, type = “h”)
## Using descendPeak requiring at least 50% or the centroid’s intensity
sp4 &lt;- pickPeaks(sp1, refineMz = “descendPeak”, signalPercentage = 50)
points(mz(sp4), intensity(sp4), col = “red”, type = “h”)</p>
</div>
<div class="section level2" number="7">
<h2 id="camera">
<span class="header-section-number">7</span> CAMERA<a class="anchor" aria-label="anchor" href="#camera"></a>
</h2>
<p>library(CAMERA)
xs &lt;- xcmsSet(c(“szwk901104i19801xms1.mzML”), method=“centWave”, ppm=30, peakwidth=c(5,10))
an &lt;- xsAnnotate(xs)
an &lt;- groupFWHM(an)
#For one group
peaklist &lt;- getpspectra(an, 1)
#For two groups
peaklist &lt;- getpspectra(an, c(1,2))</p>
</div>
<div class="section level2" number="8">
<h2 id="spectra">
<span class="header-section-number">8</span> Spectra<a class="anchor" aria-label="anchor" href="#spectra"></a>
</h2>
<p>suppressMessages(library(Spectra))
f &lt;- “szwk901104i19801xms1.mzML,gz”
sp &lt;- Spectra(f)
head(sp)
table(sp$msLevel)
d &lt;- computeMzDeltas(sp[1:1000])
plotMzDelta(d)</p>
</div>
<div class="section level2" number="9">
<h2 id="protviz">
<span class="header-section-number">9</span> protViz<a class="anchor" aria-label="anchor" href="#protviz"></a>
</h2>
<p>library(protViz)
protViz::fragmentIon(“TFVLNFIK”)
esd &lt;- extractSpectraData(mgf)
op &lt;- par(mfrow=c(2,1))
ms &lt;- function(i) with(esd[i,],list(title=TITLE,rtinseconds=RTINSECONDS,pepmass=PEPMASS,charge=CHARGE,
mZ=mz(mgf[[i]]),intensity=intensity(mgf[[i]])))
peakplot(“TAFDEAIAELDTLNEESYK”, ms(1))
peakplot(“TAFDEAIAELDTLSEESYK”, ms(2))
par(op)
load(“~/Caprion/pilot/ZWK.rda”)
peptides &lt;- subset(mapping_ZWK,Protein==“PROC_HUMAN”)[[“Modified.Peptide.Sequence”]] |&gt; unique()
pim &lt;- parentIonMass(peptides)
fi &lt;- fragmentIon(peptides)
df &lt;- as.data.frame(fi)
pdf(“~/Caprion/analysis/peptide/PROC/PROC-fragmention.pdf”,width=10,height=12)
op &lt;- par(mfrow=c(3,1))
for (i in 1:length(peptides)){
plot(0, 0,
xlab=‘m/Z’,
ylab=’‘,
xlim=range(c(fi[[i]]<span class="math inline">\(b,fi[[i]]\)</span>y)),
ylim=c(0,1),
type=’n’,
axes=FALSE,
sub=paste(peptides[i], “/”, pim[i], “Da”));
box()
axis(1, fi[[i]]<span class="math inline">\(b, round(fi[[i]]\)</span>b,1), las=2)
axis(1, fi[[i]]<span class="math inline">\(y, round(fi[[i]]\)</span>y,1), las=2)</p>
<pre><code><span><span class="va">pepSeq</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">peptides</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">""</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span><span class="fl">3</span>,<span class="va">fi</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">b</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"b"</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">fi</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">''</span><span class="op">)</span>,las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span><span class="fl">3</span>,<span class="va">fi</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">y</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"y"</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">fi</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">''</span><span class="op">)</span>,las<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="va">fi</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">b</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">peptides</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>,</span>
<span><span class="va">pepSeq</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,pos<span class="op">=</span><span class="fl">3</span>,cex<span class="op">=</span><span class="fl">4</span>, lwd<span class="op">=</span><span class="fl">4</span>, col<span class="op">=</span><span class="st">"#aaaaaaaa"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="va">fi</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">b</span>, col<span class="op">=</span><span class="st">'red'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="va">fi</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">y</span>, col<span class="op">=</span><span class="st">'blue'</span>,lwd<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre>
<p>}
par(op)
dev.off()</p>
</div>
<div class="section level2" number="10">
<h2 id="msstats">
<span class="header-section-number">10</span> MSstats<a class="anchor" aria-label="anchor" href="#msstats"></a>
</h2>
<p>library(MSstats)
f &lt;- “szwk901104i19801xms1.mzML,gz”
x &lt;- mzR::openMSfile(f, backend = “pwiz”)
x
nChrom(x)
head(tic(x))
head(chromatogram(x, 1L)) ## same as tic(x)
str(chromatogram(x))
p &lt;- mzR::peaks(x)
head(peaks(x, scan=4))</p>
<p>head(SRMRawData)
QuantData &lt;- dataProcess(SRMRawData, use_log_file = FALSE)
head(QuantData$FeatureLevelData)</p>
<p>quant &lt;- dataProcess(SRMRawData,
normalization = “equalizeMedians”,
summaryMethod = “TMP”,
censoredInt = “NA”,
MBimpute = TRUE,
maxQuantileforCensored = 0.999)
head(quant)
```</p>
</div>
<div class="section level2" number="11">
<h2 id="bioconductorcran-packages">
<span class="header-section-number">11</span> Bioconductor/CRAN packages<a class="anchor" aria-label="anchor" href="#bioconductorcran-packages"></a>
</h2>
<table class="table">
<thead><tr class="header">
<th>Package</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Bioconductor</strong></td>
<td></td>
</tr>
<tr class="even">
<td>Biostrings</td>
<td>Efficient manipulation of biological strings</td>
</tr>
<tr class="odd">
<td>rawrr</td>
<td>AnnotationDb objects and their progeny, methods etc.</td>
</tr>
<tr class="even">
<td><strong>CRAN</strong></td>
<td></td>
</tr>
<tr class="odd">
<td>protViz</td>
<td>Foreach Parallel Adaptor for the ‘parallel’ Package</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2" number="12">
<h2 id="references">
<span class="header-section-number">12</span> References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>





  </body>
</html>
