<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>A reference manual — pQTLtools • pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A reference manual — pQTLtools"><meta name="description" content="It seeds the following collection of data and utilities for pQTL analysis: 1. Articles linking functions for cis/trans classification, pQTL-gene plot, 2d/3d-plotly plots, forest plots among others available from gap, https://cran.r-project.org/package=gap; 2. Query on genes, regions, and SNPs via PhenoScanner, http://www.phenoscanner.medschl.cam.ac.uk/, adding functionality for pQTL novelty check; 3. Downstream analysis such as colocalization, pQTL-Mendelian Randomization via TwoSampleMR. https://mrcieu.github.io/TwoSampleMR/, mapping of UniProt IDs to other resources; 4. Bioconductor notes, showcases of LocusZoom.js, snakemake workflow, and spectrum data analysis."><meta property="og:description" content="It seeds the following collection of data and utilities for pQTL analysis: 1. Articles linking functions for cis/trans classification, pQTL-gene plot, 2d/3d-plotly plots, forest plots among others available from gap, https://cran.r-project.org/package=gap; 2. Query on genes, regions, and SNPs via PhenoScanner, http://www.phenoscanner.medschl.cam.ac.uk/, adding functionality for pQTL novelty check; 3. Downstream analysis such as colocalization, pQTL-Mendelian Randomization via TwoSampleMR. https://mrcieu.github.io/TwoSampleMR/, mapping of UniProt IDs to other resources; 4. Bioconductor notes, showcases of LocusZoom.js, snakemake workflow, and spectrum data analysis."><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a></li>
    <li><a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a></li>
    <li><a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a></li>
    <li><a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a></li>
    <li><a class="dropdown-item" href="../articles/spectrum.html">Spectrum data analysis</a></li>
    <li><a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/pQTLtools.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/"><span class="fa fab fa-r-project"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/"><span class="fa fas fa-fast-backward"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-urls" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">URLs</button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-urls"><li><a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a></li>
    <li><a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a></li>
    <li><a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a></li>
    <li><a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a></li>
    <li><a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://metascape.org/">metascape</a></li>
    <li><a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a></li>
    <li><a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a></li>
    <li><a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a></li>
    <li><a class="external-link dropdown-item" href="https://string-db.org/">STRING</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a></li>
    <li><a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a></li>
  </ul></li>
<li class="nav-item"><a class="external-link nav-link" href="https://www.uniprot.org/"><span class="fa fas fa-fast-forward"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>A reference manual</h1>
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/R/pQTLtools.R" class="external-link"><code>R/pQTLtools.R</code></a></small>
      <div class="d-none name"><code>pQTLtools.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>It seeds the following collection of data and utilities for pQTL analysis: 1. Articles linking functions for cis/trans classification, pQTL-gene plot, 2d/3d-plotly plots, forest plots among others available from gap, <a href="https://cran.r-project.org/package=gap" class="external-link">https://cran.r-project.org/package=gap</a>; 2. Query on genes, regions, and SNPs via PhenoScanner, <a href="http://www.phenoscanner.medschl.cam.ac.uk/" class="external-link">http://www.phenoscanner.medschl.cam.ac.uk/</a>, adding functionality for pQTL novelty check; 3. Downstream analysis such as colocalization, pQTL-Mendelian Randomization via TwoSampleMR. <a href="https://mrcieu.github.io/TwoSampleMR/" class="external-link">https://mrcieu.github.io/TwoSampleMR/</a>, mapping of UniProt IDs to other resources; 4. Bioconductor notes, showcases of LocusZoom.js, snakemake workflow, and spectrum data analysis.</p>
    </div>


    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Available data and functions are listed in the following table.</p><table class="table table"><tr><td>Objects</td><td>Description</td></tr><tr><td><strong>GWAS</strong></td><td> </td></tr><tr><td><code><a href="novelty_check.html">novelty_check</a></code></td><td>Locus novelty check</td></tr><tr><td><code><a href="peptideAssociationPlot.html">peptideAssociationPlot</a></code></td><td>peptide association plot</td></tr><tr><td><code><a href="peptideMapping.html">peptideMapping</a></code></td><td>peptide-to-protein mapping</td></tr><tr><td><code><a href="turboman.html">turboman</a></code></td><td>Manhattan plots</td></tr><tr><td><code><a href="turboqq.html">turboqq</a></code></td><td>QQ plots</td></tr><tr><td> </td><td> </td></tr><tr><td><strong>pGWAS</strong></td><td> </td></tr><tr><td><code><a href="csq.html">csq</a></code></td><td>Variant consequence</td></tr><tr><td><code><a href="protein_altering_variants.html">protein_altering_variants</a></code></td><td>Protein Altering Variants (PAVs)</td></tr><tr><td><strong>Expression analysis</strong></td><td> </td></tr><tr><td><code><a href="get.prop.below.LLOD.html">get.prop.below.LLOD</a></code></td><td>Limit of detection analysis</td></tr><tr><td><code><a href="import_eQTLCatalogue.html">import_eQTLCatalogue</a></code></td><td>Import eQTL Catalogue</td></tr><tr><td><code><a href="make_ExpressionSet.html">make_ExpressionSet</a></code></td><td>A call to ExpressionSet class</td></tr><tr><td><code><a href="run_coloc.html">run_coloc</a></code></td><td>Colocalisation analysis</td></tr><tr><td> </td><td> </td></tr><tr><td><strong>MR analysis</strong></td><td> </td></tr><tr><td><code><a href="import_OpenGWAS.html">import_OpenGWAS</a></code></td><td>Import OpenGWAS</td></tr><tr><td><code><a href="pqtlMR.html">pqtlMR</a></code></td><td>Bidirectional pQTL-MR analysis</td></tr><tr><td><code><a href="qtl_lookup.html">qtl_lookup</a></code></td><td>QTL lookup</td></tr><tr><td><code><a href="run_TwoSampleMR.html">run_TwoSampleMR</a></code></td><td>A generic wrapper for TwoSampleMR analysis</td></tr><tr><td> </td><td> </td></tr><tr><td><strong>PhenoScanner Utilities</strong></td><td> </td></tr><tr><td><code><a href="genequeries.html">genequeries</a></code></td><td>phenoscanner genequeries in batches</td></tr><tr><td><code><a href="regionqueries.html">regionqueries</a></code></td><td>phenoscanner regionqueries in batches</td></tr><tr><td><code><a href="snpqueries.html">snpqueries</a></code></td><td>phenoscanner snpqueries in batches</td></tr><tr><td> </td><td> </td></tr><tr><td><strong>UniProt API</strong></td><td> </td></tr><tr><td><code><a href="uniprot2ids.html">uniprot2ids</a></code></td><td>UniProt ID to others</td></tr><tr><td> </td><td> </td></tr><tr><td><strong>Functions in gap</strong></td><td> </td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/METAL_forestplot.html" class="external-link">gap::METAL_forestplot</a></code></td><td>Forest plots from metal analysis</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/ci2ms.html" class="external-link">gap::ci2ms</a></code></td><td>Effect size and standard error from confidence interval</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/cis.vs.trans.classification.html" class="external-link">gap::cis.vs.trans.classification</a></code></td><td>a cis/trans classifier</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/circos.cis.vs.trans.plot.html" class="external-link">gap::circos.cis.vs.trans.plot</a></code></td><td>circos plot of cis/trans classification</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/circos.mhtplot.html" class="external-link">gap::circos.mhtplot</a></code></td><td>circos Manhattan plot with gene annotation</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/circos.mhtplot2.html" class="external-link">gap::circos.mhtplot2</a></code></td><td>Another circos Manhattan plot</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/cs.html" class="external-link">gap::cs</a></code></td><td>Credible set</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/get_b_se.html" class="external-link">gap::get_b_se</a></code></td><td>Get b and se from AF, n, and z</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/get_pve_se.html" class="external-link">gap::get_pve_se</a></code></td><td>Get pve and its standard error from n, z</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/get_sdy.html" class="external-link">gap::get_sdy</a></code></td><td>Get sd(y) from AF, n, b, se</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/mr.html" class="external-link">gap::mr</a></code></td><td>Mendelian randomization analysis</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/invnormal.html" class="external-link">gap::invnormal</a></code></td><td>Inverse normal transformation</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/log10p.html" class="external-link">gap::log10p</a></code></td><td>log10(p) for a standard normal deviate</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/log10pvalue.html" class="external-link">gap::log10pvalue</a></code></td><td>log10(p) for a P value including its scientific format</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/logp.html" class="external-link">gap::logp</a></code></td><td>log(p) for a normal deviate</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/mhtplot.trunc.html" class="external-link">gap::mhtplot.trunc</a></code></td><td>Truncated Manhattan plot</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/miamiplot2.html" class="external-link">gap::miamiplot2</a></code></td><td>Miami plot</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/mr_forestplot.html" class="external-link">gap::mr_forestplot</a></code></td><td>Mendelian Randomization forest plot</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/pvalue.html" class="external-link">gap::pvalue</a></code></td><td>P value for a normal deviate</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/qtlClassifier.html" class="external-link">gap::qtlClassifier</a></code></td><td>A QTL cis/trans classifier</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/qtlFinder.html" class="external-link">gap::qtlFinder</a></code></td><td>Distance-based signal identification</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/qtl2dplot.html" class="external-link">gap::qtl2dplot</a></code></td><td>2D QTL plot</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/qtl2dplotly.html" class="external-link">gap::qtl2dplotly</a></code></td><td>2D QTL plotly</td></tr><tr><td><code><a href="https://rdrr.io/pkg/gap/man/qtl3dplotly.html" class="external-link">gap::qtl3dplotly</a></code></td><td>3D QTL plotly</td></tr></table></div>
    <div class="section level2">
    <h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a></h2>


<p>Vignettes on package usage:</p><ul><li><p>An Overview of pQTLtools. <code><a href="../articles/pQTLtools.html">vignette("pQTLtools")</a></code>.</p></li>
<li><p>Bioconductor Notes. <code><a href="../articles/bioconductor.html">vignette("bioconductor")</a></code>.</p></li>
<li><p>ExpressionSet usage. <code><a href="../articles/es.html">vignette("es")</a></code>.</p></li>
<li><p>LocusZoom.js. <code><a href="../articles/LocusZoom.js.html">vignette("LocusZoom.js")</a></code>.</p></li>
<li><p>snakemake showcases. <code><a href="../articles/snakemake.html">vignette("snakemake")</a></code>.</p></li>
<li><p>Spectrum data analysis. <code><a href="../articles/spectrum.html">vignette("spectrum")</a></code>.</p></li>
<li><p>SCALLOP-INF scripts. <code><a href="../articles/SCALLOP-INF.html">vignette("SCALLOP-INF")</a></code>.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Useful links:</p><ul><li><p><a href="https://github.com/jinghuazhao/pQTLtools" class="external-link">https://github.com/jinghuazhao/pQTLtools</a></p></li>
<li><p>Report bugs at <a href="https://github.com/jinghuazhao/pQTLtools/issues" class="external-link">https://github.com/jinghuazhao/pQTLtools/issues</a></p></li>
</ul></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Jing Hua Zhao in collaboration with other colleagues</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer></div>





  </body></html>

