<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='This function takes exposure and outcome data as produced by format_data() and used to perform
MR analysis against a list of outcomes; in the latter case it can be data from MR-Base, e.g.,
outcome &amp;lt;- extract_outcome_data(snps=with(exposure,SNP),outcomes=c("ieu-a-7","ebi-a-GCST007432")).'><title>Basic pQTL-MR analysis — pqtlMR • pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Basic pQTL-MR analysis — pqtlMR"><meta property="og:description" content='This function takes exposure and outcome data as produced by format_data() and used to perform
MR analysis against a list of outcomes; in the latter case it can be data from MR-Base, e.g.,
outcome &amp;lt;- extract_outcome_data(snps=with(exposure,SNP),outcomes=c("ieu-a-7","ebi-a-GCST007432")).'><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Vignettes:</h6>
    <a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Articles:</h6>
    <a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a>
    <a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a>
    <a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a>
    <a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a>
    <a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/pQTLtools.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://jinghuazhao.github.io/R/">
    <span class="fab fa fab fa-r-project"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/">
    <span class="fas fa fas fa-fast-backward"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-urls">URLs</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-urls">
    <a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a>
    <a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a>
    <a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a>
    <a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a>
    <a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a>
    <a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a>
    <a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a>
    <a class="external-link dropdown-item" href="https://metascape.org/">metascape</a>
    <a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a>
    <a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a>
    <a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a>
    <a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a>
    <a class="external-link dropdown-item" href="https://string-db.org/">STRING</a>
    <a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a>
    <a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.uniprot.org/">
    <span class="fas fa fas fa-fast-forward"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Basic pQTL-MR analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/R/pqtlMR.R" class="external-link"><code>R/pqtlMR.R</code></a></small>
      <div class="d-none name"><code>pqtlMR.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function takes exposure and outcome data as produced by <code>format_data()</code> and used to perform
MR analysis against a list of outcomes; in the latter case it can be data from MR-Base, e.g.,
<code>outcome &lt;- extract_outcome_data(snps=with(exposure,SNP),outcomes=c("ieu-a-7","ebi-a-GCST007432"))</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">pqtlMR</span><span class="op">(</span></span>
<span>  <span class="va">exposure</span>,</span>
<span>  <span class="va">outcome</span>,</span>
<span>  mr_plot <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  prefix <span class="op">=</span> <span class="st">"pQTL-combined-"</span>,</span>
<span>  reverse <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>exposure</dt>
<dd><p>exposure data.</p></dd>


<dt>outcome</dt>
<dd><p>the counterpart for outcome.</p></dd>


<dt>mr_plot</dt>
<dd><p>to produce plots.</p></dd>


<dt>prefix</dt>
<dd><p>a prefix for output files.</p></dd>


<dt>reverse</dt>
<dd><p>if TRUE, perform reverse MR.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>No value is returned but several files.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Adapted from Zheng et al. (2020)
, this function is analogous to <code><a href="run_TwoSampleMR.html">run_TwoSampleMR()</a></code>.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Adapted from script by Jie Zheng.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Zheng J, Haberland V, Baird D, Walker V, Haycock PC, Hurle MR, Gutteridge A, Erola P, Liu Y, Luo S, Robinson J, Richardson TG, Staley JR, Elsworth B, Burgess S, Sun BB, Danesh J, Runz H, Maranville JC, Martin HM, Yarmolinsky J, Laurin C, Holmes MV, Liu JZ, Estrada K, Santos R, McCarthy L, Waterworth D, Nelson MR, Smith GD, Butterworth AS, Hemani G, Scott RA, Gaunt TR (2020).
“Phenome-wide Mendelian randomization mapping the influence of the plasma proteome on complex diseases.”
<em>Nature Genetics</em>, <b>52</b>(10), 1122-1131.
<a href="https://doi.org/10.1038/s41588-020-0682-6" class="external-link">doi:10.1038/s41588-020-0682-6</a>
.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MRCIEU/TwoSampleMR" class="external-link">TwoSampleMR</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> TwoSampleMR version 0.6.0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [&gt;] New: Improved API performance and stability</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [&gt;] New: Authentication required for all OpenGWAS queries from 1st May 2024</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [&gt;] For guidance see https://mrcieu.github.io/ieugwasr/articles/guide.html#authentication</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Warning:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> You are running an old version of the TwoSampleMR package.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This version:   0.6.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Latest version: 0.6.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Please consider updating using remotes::install_github('MRCIEU/TwoSampleMR')</span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/pQTLtools" class="external-link">pQTLtools</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span>,lib.loc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/libPaths.html" class="external-link">.libPaths</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"Ins.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">exposure</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mrcieu.github.io/TwoSampleMR/reference/format_data.html" class="external-link">format_data</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">fi</span><span class="op">)</span>,type<span class="op">=</span><span class="st">"exposure"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span>,lib.loc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/libPaths.html" class="external-link">.libPaths</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="st">"Out.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">outcome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mrcieu.github.io/TwoSampleMR/reference/format_data.html" class="external-link">format_data</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">fo</span><span class="op">)</span>,type<span class="op">=</span><span class="st">"outcome"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following columns are not present but are helpful for harmonisation</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> eaf</span>
<span class="r-in"><span><span class="fu">pqtlMR</span><span class="op">(</span><span class="va">exposure</span>, <span class="va">outcome</span>, prefix<span class="op">=</span><span class="st">"IL6R-"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising IL.6 (ImbABK) and Coronary artery disease (4nec1Z)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising IL.6 (ImbABK) and Atopic dermatitis (Ku3B84)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising IL.6 (ImbABK) and Rheumatoid arthritis (SGZG4k)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysing 'ImbABK' on '4nec1Z'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysing 'ImbABK' on 'Ku3B84'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysing 'ImbABK' on 'SGZG4k'</span>
<span class="r-in"><span><span class="fu">pqtlMR</span><span class="op">(</span><span class="va">exposure</span>, <span class="va">outcome</span>, prefix<span class="op">=</span><span class="st">"IL6R_rev-"</span>,reverse<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising IL.6 (ImbABK) and Coronary artery disease (4nec1Z)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising IL.6 (ImbABK) and Atopic dermatitis (Ku3B84)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising IL.6 (ImbABK) and Rheumatoid arthritis (SGZG4k)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysing '4nec1Z' on 'ImbABK'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysing 'Ku3B84' on 'ImbABK'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysing 'SGZG4k' on 'ImbABK'</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IL6R*"</span>,<span class="st">"pQTL-combined*"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Phenotype,SNP,effect_allele,other_allele,eaf,beta,se,pval</span></span></span>
<span class="r-in"><span><span class="co"># ABO,rs505922,C,T,0.313,1.298,0.014,1.2e-1828</span></span></span>
<span class="r-in"><span><span class="co"># LIFR,rs635634,T,C,0.180,-0.300,0.032,6.00E-21</span></span></span>
<span class="r-in"><span><span class="co"># f &lt;- file.path(find.package("pQTLtools",lib.loc=.libPaths()),"tests","ms.ins")</span></span></span>
<span class="r-in"><span><span class="co"># exposure &lt;- format_data(read.table(f, header=TRUE), samplesize_col="N")</span></span></span>
<span class="r-in"><span><span class="co"># SNP Phenotype effect_allele other_allele eaf beta se pval N</span></span></span>
<span class="r-in"><span><span class="co"># rs1800693 TNFB T C 0.6033 0.0282  0.0136 0.0389045   11787</span></span></span>
<span class="r-in"><span><span class="co"># rs2364485 TNFB A C 0.1645 6514963 0.1759 1.62181e-20 11344</span></span></span>
<span class="r-in"><span><span class="co"># https://raw.githubusercontent.com/MRCIEU/epigraphdb-pqtl/master/scripts/MR-pQTL-script.R</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer></div>

  

  

  </body></html>

