<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Basic pQTL-MR analysis — pqtlMR • pQTLtools</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Basic pQTL-MR analysis — pqtlMR" />
<meta property="og:description" content="This function takes input based on pQTLs for MR analysis." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pQTLtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jinghuazhao/pQTLtools">Source</a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Basic pQTL-MR analysis</h1>
    
    <div class="hidden name"><code>pqtlMR.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function takes input based on pQTLs for MR analysis.</p>
    </div>

    <pre class="usage"><span class='fu'>pqtlMR</span><span class='op'>(</span><span class='va'>TwoSampleMRinput</span>, plot<span class='op'>=</span><span class='cn'>FALSE</span>, prefix<span class='op'>=</span><span class='st'>"pQTL-combined-"</span>,reverse<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>TwoSampleMRinput</th>
      <td><p>Harmonised data</p></td>
    </tr>
    <tr>
      <th>plot</th>
      <td><p>to produce plots</p></td>
    </tr>
    <tr>
      <th>prefix</th>
      <td><p>a prefix for output files</p></td>
    </tr>
    <tr>
      <th>reverse</th>
      <td><p>if TRUE, perform reverse MR</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function is based on TwoSampleMR.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>No value is returned but several files.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Zheng J, et al. (2020). Phenome-wide Mendelian randomization mapping the influence of the plasma proteome on complex diseases. *Nature Genetics* 52(10): 1122-1131.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Adapted from script by Jie Zheng</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>TwoSampleMR</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>TwoSampleMR version 0.5.6 </span>
#&gt; <span class='message'>[&gt;] New: Option to use non-European LD reference panels for clumping etc</span>
#&gt; <span class='message'>[&gt;] Some studies temporarily quarantined to verify effect allele</span>
#&gt; <span class='message'>[&gt;] See news(package='TwoSampleMR') and https://gwas.mrcieu.ac.uk for further details</span></div><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/jinghuazhao/pQTLtools'>pQTLtools</a></span><span class='op'>)</span>
<span class='co'># Original examples</span>
<span class='va'>ins</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.csv</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/find.package.html'>find.package</a></span><span class='op'>(</span><span class='st'>"pQTLtools"</span>,lib.loc<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/libPaths.html'>.libPaths</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>,<span class='st'>"tests"</span>,<span class='st'>"Ins.csv"</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>ids</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ieu-a-7"</span>,<span class='st'>"ebi-a-GCST007432"</span><span class='op'>)</span>
<span class='va'>Ins</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/format_data.html'>format_data</a></span><span class='op'>(</span><span class='va'>ins</span><span class='op'>)</span>
<span class='va'>Ids</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html'>extract_outcome_data</a></span><span class='op'>(</span>snps<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/with.html'>with</a></span><span class='op'>(</span><span class='va'>Ins</span>,<span class='va'>SNP</span><span class='op'>)</span>,outcomes<span class='op'>=</span><span class='va'>ids</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>API: public: http://gwas-api.mrcieu.ac.uk/</span></div><div class='output co'>#&gt; <span class='message'>Extracting data for 2 SNP(s) from 2 GWAS(s)</span></div><div class='input'><span class='va'>harmonise</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html'>harmonise_data</a></span><span class='op'>(</span><span class='va'>Ins</span>,<span class='va'>Ids</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Harmonising ABO (ImbABK) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span></div><div class='output co'>#&gt; <span class='message'>Harmonising LIFR (Ku3B84) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span></div><div class='output co'>#&gt; <span class='message'>Harmonising ABO (ImbABK) and Coronary heart disease || id:ieu-a-7 (ieu-a-7)</span></div><div class='output co'>#&gt; <span class='message'>Harmonising LIFR (Ku3B84) and Coronary heart disease || id:ieu-a-7 (ieu-a-7)</span></div><div class='input'><span class='va'>input</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='va'>Ins</span>, <span class='va'>Ids</span>, <span class='va'>harmonise</span><span class='op'>)</span>
<span class='fu'>pqtlMR</span><span class='op'>(</span><span class='va'>input</span>,plot<span class='op'>=</span><span class='cn'>FALSE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; Error in if (empty(.data)) return(.data) : 
#&gt;   missing value where TRUE/FALSE needed
#&gt; Error in TwoSampleMR::mr_singlesnp(harmonise) : 
#&gt;   "outcome" %in% names(dat) is not TRUE</div><div class='input'><span class='co'># A bidirectional analysis</span>
<span class='va'>ms</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/find.package.html'>find.package</a></span><span class='op'>(</span><span class='st'>"pQTLtools"</span>,lib.loc<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/libPaths.html'>.libPaths</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>,<span class='st'>"tests"</span>,<span class='st'>"ms.ins"</span><span class='op'>)</span>,
                 header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>Ins</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/format_data.html'>format_data</a></span><span class='op'>(</span><span class='va'>ms</span>,samplesize_col<span class='op'>=</span><span class='st'>"N"</span><span class='op'>)</span>
<span class='va'>Ids</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html'>extract_outcome_data</a></span><span class='op'>(</span>snps<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/with.html'>with</a></span><span class='op'>(</span><span class='va'>Ins</span>,<span class='va'>SNP</span><span class='op'>)</span>,outcomes<span class='op'>=</span><span class='st'>"ieu-b-18"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Extracting data for 2 SNP(s) from 1 GWAS(s)</span></div><div class='output co'>#&gt; <span class='message'>Finding proxies for 1 SNPs in outcome ieu-b-18</span></div><div class='output co'>#&gt; <span class='message'>Extracting data for 1 SNP(s) from 1 GWAS(s)</span></div><div class='input'><span class='va'>harmonise</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html'>harmonise_data</a></span><span class='op'>(</span><span class='va'>Ins</span>,<span class='va'>Ids</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Harmonising TNFB (G6vQ58) and multiple sclerosis || id:ieu-b-18 (ieu-b-18)</span></div><div class='input'><span class='co'># MR</span>
<span class='fu'>pqtlMR</span><span class='op'>(</span><span class='va'>harmonise</span>,prefix<span class='op'>=</span><span class='st'>"MS-"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Analysing 'G6vQ58' on 'ieu-b-18'</span></div><div class='input'><span class='co'># reverse MR</span>
<span class='fu'>pqtlMR</span><span class='op'>(</span><span class='va'>harmonise</span>,prefix<span class='op'>=</span><span class='st'>"MS_rev-"</span>,reverse<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Analysing 'ieu-b-18' on 'G6vQ58'</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jing hua Zhao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


