<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>QTL lookup — qtl_lookup • pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><script src="../deps/MathJax-3.2.2/tex-chtml.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="QTL lookup — qtl_lookup"><meta name="description" content="This function takes MR results (involving pQTL/QTL) to look up QTLs in trait GWASs given a P value and
linkage disequilibrium (LD) cutoffs.
The rationale is that a pQTL may not necessarily be in strong LD with QTL but some other independent signal
in the same region. The interest then lands on association signals below a given P value (p_threshold) and above
a given LD (r2_threshold) thresholds."><meta property="og:description" content="This function takes MR results (involving pQTL/QTL) to look up QTLs in trait GWASs given a P value and
linkage disequilibrium (LD) cutoffs.
The rationale is that a pQTL may not necessarily be in strong LD with QTL but some other independent signal
in the same region. The interest then lands on association signals below a given P value (p_threshold) and above
a given LD (r2_threshold) thresholds."><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html" aria-label="Home"><span class="fa fa-home"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a></li>
    <li><a class="dropdown-item" href="../articles/esse.html">ExpressionSet/SummarizedExperiment usage</a></li>
    <li><a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a></li>
    <li><a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a></li>
    <li><a class="dropdown-item" href="../articles/spectrum.html">Spectrum data analysis</a></li>
    <li><a class="dropdown-item" href="../articles/SCALLOP-Caprion.html">SCALLOP-Caprion scripts</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../reference/pQTLtools.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools" aria-label="GitHub"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/" aria-label="R packages"><span class="fa fab fa-r-project"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/" aria-label="AlphaFold"><span class="fa fas fa-fast-backward"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-urls" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">URLs</button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-urls"><li><a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a></li>
    <li><a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a></li>
    <li><a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a></li>
    <li><a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a></li>
    <li><a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.histo.fyi/">histo.fyi</a></li>
    <li><a class="external-link dropdown-item" href="https://metascape.org/">metascape</a></li>
    <li><a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a></li>
    <li><a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a></li>
    <li><a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a></li>
    <li><a class="external-link dropdown-item" href="https://string-db.org/">STRING</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a></li>
    <li><a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a></li>
  </ul></li>
<li class="nav-item"><a class="external-link nav-link" href="https://www.uniprot.org/" aria-label="UniProt"><span class="fa fas fa-fast-forward"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>QTL lookup</h1>
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/R/qtl_lookup.R" class="external-link"><code>R/qtl_lookup.R</code></a></small>
      <div class="d-none name"><code>qtl_lookup.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function takes MR results (involving pQTL/QTL) to look up QTLs in trait GWASs given a P value and
linkage disequilibrium (LD) cutoffs.
The rationale is that a pQTL may not necessarily be in strong LD with QTL but some other independent signal
in the same region. The interest then lands on association signals below a given P value (p_threshold) and above
a given LD (r2_threshold) thresholds.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">qtl_lookup</span><span class="op">(</span></span>
<span>  <span class="va">d</span>,</span>
<span>  <span class="va">dat</span>,</span>
<span>  panel <span class="op">=</span> <span class="st">"1000Genomes"</span>,</span>
<span>  p_threshold <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>  r2_threshold <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  pop <span class="op">=</span> <span class="st">"EUR"</span>,</span>
<span>  plink_bin <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  r <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  r2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  xlsx <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-d">d<a class="anchor" aria-label="anchor" href="#arg-d"></a></dt>
<dd><p>directory where <code>dat</code> (below) is held.</p></dd>


<dt id="arg-dat">dat<a class="anchor" aria-label="anchor" href="#arg-dat"></a></dt>
<dd><p>MR results (<code>protein</code>, <code>id</code>, <code>pqtl</code>,<code>p</code>, <code>qtl</code>, <code>p_qtl</code>) whose <code>proxy</code>, <code>p_proxy</code> and <code>rsq</code> variables will be updated.</p></dd>


<dt id="arg-panel">panel<a class="anchor" aria-label="anchor" href="#arg-panel"></a></dt>
<dd><p>reference panel.</p></dd>


<dt id="arg-p-threshold">p_threshold<a class="anchor" aria-label="anchor" href="#arg-p-threshold"></a></dt>
<dd><p>cutoff of QTL association from trait GWASs.</p></dd>


<dt id="arg-r-threshold">r2_threshold<a class="anchor" aria-label="anchor" href="#arg-r-threshold"></a></dt>
<dd><p>cutoff of r^2.</p></dd>


<dt id="arg-pop">pop<a class="anchor" aria-label="anchor" href="#arg-pop"></a></dt>
<dd><p>reference population from 1000Genomes for LD calculation.</p></dd>


<dt id="arg-plink-bin">plink_bin<a class="anchor" aria-label="anchor" href="#arg-plink-bin"></a></dt>
<dd><p>PLINK executable file whose binary files are indicated in <code>bfile</code> variable of <code>dat</code>.</p></dd>


<dt id="arg-r">r<a class="anchor" aria-label="anchor" href="#arg-r"></a></dt>
<dd><p>when specified, the LD(r) is output.</p></dd>


<dt id="arg-r-">r2<a class="anchor" aria-label="anchor" href="#arg-r-"></a></dt>
<dd><p>when specified, the LD(r^2) is output.</p></dd>


<dt id="arg-xlsx">xlsx<a class="anchor" aria-label="anchor" href="#arg-xlsx"></a></dt>
<dd><p>a non-null specification indicates name of an output Excel workbook.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data.frame containing the looked up loci.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">INF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"INF"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"mr"</span>,<span class="st">"gsmr"</span>,<span class="st">"trait"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">inf1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">gap.datasets</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/gap.datasets/man/inf1.html" class="external-link">inf1</a></span>,<span class="va">prot</span>,<span class="va">target.short</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">gsmr_efo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"mr"</span>,<span class="st">"gsmr"</span>,<span class="st">"gsmr-efo.txt"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>            <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">inf1</span>,by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'protein'</span><span class="op">=</span><span class="st">'target.short'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>            <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>file_gwas<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">prot</span>,<span class="va">id</span>,<span class="st">"rsid.txt"</span>,sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                   bfile<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"INTERVAL"</span>,<span class="st">"per_chr"</span>,</span></span>
<span class="r-in"><span>                                   <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"interval.imputed.olink.chr_"</span>,<span class="va">chr</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                   proxy<span class="op">=</span><span class="cn">NA</span>,p_proxy<span class="op">=</span><span class="cn">NA</span>,rsq<span class="op">=</span><span class="cn">NA</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">proxies</span> <span class="op">&lt;-</span> <span class="fu">qtl_lookup</span><span class="op">(</span><span class="va">d</span>,<span class="va">gsmr_efo</span>,plink_bin<span class="op">=</span><span class="st">"/rds/user/jhz22/hpc-work/bin/plink"</span>,</span></span>
<span class="r-in"><span>                      xlsx<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"mr"</span>,<span class="st">"gsmr"</span>,<span class="st">"r2_INTERVAL.xlsx"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>           <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">protein</span>,<span class="va">id</span>,<span class="va">Disease</span>,<span class="va">fdr</span>,<span class="va">pqtl</span>,<span class="va">p</span>,<span class="va">qtl</span>,<span class="va">p_qtl</span>,<span class="va">proxy</span>,<span class="va">p_proxy</span>,<span class="va">rsq</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">proxies</span>,file<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">INF</span>,<span class="st">"mr"</span>,<span class="st">"gsmr"</span>,<span class="st">"r2_INTERVAL.tsv"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>            row.names<span class="op">=</span><span class="cn">FALSE</span>,quote<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer></div>



  <script type="module">import mermaid from "https://unpkg.com/mermaid@10.4.0/dist/mermaid.esm.min.mjs";mermaid.initialize({});</script></body></html>

