<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Given harmonised data, this function conducts a two-sample MR analysis."><title>Basic TwoSampleMR analysis — run_TwoSampleMR • pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Basic TwoSampleMR analysis — run_TwoSampleMR"><meta property="og:description" content="Given harmonised data, this function conducts a two-sample MR analysis."><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Vignettes:</h6>
    <a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Articles:</h6>
    <a class="dropdown-item" href="../articles/es.html">ExpressionSet</a>
    <a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools">
    <span class="fa fa-github"></span>
     
    GitHub
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Basic TwoSampleMR analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/R/utils.R" class="external-link"><code>R/utils.R</code></a></small>
      <div class="d-none name"><code>run_TwoSampleMR.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Given harmonised data, this function conducts a two-sample MR analysis.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">run_TwoSampleMR</span><span class="op">(</span><span class="va">TwoSampleMRinput</span>, mr_plot <span class="op">=</span> <span class="st">"None"</span>, prefix <span class="op">=</span> <span class="st">""</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>TwoSampleMRinput</dt>
<dd><p>Harmonised data.</p></dd>
<dt>mr_plot</dt>
<dd><p>one of "None", "TwoSampleMR", "pQTLtools" for no, the original and the revised plots, respectively.</p></dd>
<dt>prefix</dt>
<dd><p>a prefix for output files.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>No value is returned but several files.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>As TwoSampleMR faces seemingly perplexing options, this function intends to simplify various steps in a two-sample MR. It is
particularly useful when a large numbher of MRs are necessary, e.g., multiple proteins and their cis/trans regions need to be examined,
in which case prefix could direct the output to various directories.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Adapted from script by Dimou NL, Tsilidis KK.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Dimou NL, Tsilidis KK. A Primer in Mendelian Randomization Methodology with a Focus on Utilizing Published Summary Association Data. In
Evangelos Evangelou (ed.), Genetic Epidemiology: Methods and Protocols, Methods in Molecular Biology, vol. 1793,
https://doi.org/10.1007/978-1-4939-7868-7_13, Springer Science+Business Media, LLC, part of Springer Nature 2018</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TwoSampleMR</span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/pQTLtools" class="external-link">pQTLtools</a></span><span class="op">)</span></span>
<span class="r-in"><span class="va">outcomes</span> <span class="op">&lt;-</span> <span class="st">"ebi-a-GCST007432"</span></span>
<span class="r-in"><span class="va">prot</span> <span class="op">&lt;-</span> <span class="st">"MMP.10"</span></span>
<span class="r-in"><span class="va">type</span> <span class="op">&lt;-</span> <span class="st">"cis"</span></span>
<span class="r-in"><span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prot</span>,<span class="st">"-"</span>,<span class="va">type</span>,<span class="st">".mrx"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">"pQTLtools"</span>,lib.loc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/libPaths.html" class="external-link">.libPaths</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,<span class="st">"tests"</span>,<span class="va">f</span><span class="op">)</span>,</span>
<span class="r-in">                header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">exposure</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/format_data.html" class="external-link">format_data</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">d</span>,<span class="op">{</span><span class="va">P</span><span class="op">=</span><span class="fl">10</span><span class="op">^</span><span class="va">logP</span><span class="op">}</span><span class="op">)</span>, phenotype_col<span class="op">=</span><span class="st">"prot"</span>, snp_col<span class="op">=</span><span class="st">"rsid"</span>,</span>
<span class="r-in">                        chr_col<span class="op">=</span><span class="st">"Chromosome"</span>, pos_col<span class="op">=</span><span class="st">"Posistion"</span>,</span>
<span class="r-in">                        effect_allele_col<span class="op">=</span><span class="st">"Allele1"</span>, other_allele_col<span class="op">=</span><span class="st">"Allele2"</span>,</span>
<span class="r-in">                        eaf_col<span class="op">=</span><span class="st">"Freq1"</span>, beta_col<span class="op">=</span><span class="st">"Effect"</span>, se_col<span class="op">=</span><span class="st">"StdErr"</span>,</span>
<span class="r-in">                        pval_col<span class="op">=</span><span class="st">"P"</span>, log_pval<span class="op">=</span><span class="cn">FALSE</span>,</span>
<span class="r-in">                        samplesize_col<span class="op">=</span><span class="st">"N"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">clump</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/clump_data.html" class="external-link">clump_data</a></span><span class="op">(</span><span class="va">exposure</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Please look at vignettes for options on running this locally if you need to run many instances of this command.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Clumping foelRu, 1106 variants, using EUR population reference</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Removing 1102 of 1106 variants due to LD with other variants or absence from LD reference panel</span>
<span class="r-in"><span class="va">outcome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html" class="external-link">extract_outcome_data</a></span><span class="op">(</span>snps<span class="op">=</span><span class="va">exposure</span><span class="op">$</span><span class="va">SNP</span>,outcomes<span class="op">=</span><span class="va">outcomes</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting data for 1106 SNP(s) from 1 GWAS(s)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finding proxies for 155 SNPs in outcome ebi-a-GCST007432</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting data for 155 SNP(s) from 1 GWAS(s)</span>
<span class="r-in"><span class="va">harmonise</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html" class="external-link">harmonise_data</a></span><span class="op">(</span><span class="va">clump</span>,<span class="va">outcome</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Harmonising MMP.10 (foelRu) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span>
<span class="r-in"><span class="va">prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">outcomes</span>,<span class="va">prot</span>,<span class="va">type</span>,sep<span class="op">=</span><span class="st">"-"</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">run_TwoSampleMR</span><span class="op">(</span><span class="va">harmonise</span>, mr_plot<span class="op">=</span><span class="st">"pQTLtools"</span>, prefix<span class="op">=</span><span class="va">prefix</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysing 'foelRu' on 'ebi-a-GCST007432'</span>
<span class="r-plt img"><img src="run_TwoSampleMR-1.png" alt="" width="700" height="433"></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Removed 1 rows containing missing values (geom_errorbarh).</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Removed 1 rows containing missing values (geom_point).</span>
<span class="r-plt img"><img src="run_TwoSampleMR-2.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="run_TwoSampleMR-3.png" alt="" width="700" height="433"></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Removed 1 rows containing missing values (geom_errorbarh).</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Removed 1 rows containing missing values (geom_point).</span>
<span class="r-plt img"><img src="run_TwoSampleMR-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span class="va">caption</span> <span class="op">&lt;-</span> <span class="st">"Table. MMP.10 variants and FEV1"</span></span>
<span class="r-in"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-result.txt"</span><span class="op">)</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">             caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption</span>, <span class="st">"(result)"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Table: Table. MMP.10 variants and FEV1 (result)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |id.exposure |id.outcome       |outcome                               |exposure |method                    | nsnp|         b|      se|  pval|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |:-----------|:----------------|:-------------------------------------|:--------|:-------------------------|----:|---------:|-------:|-----:|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |MR Egger                  |    4| -0.000577| 0.01001| 0.959|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Weighted median           |    4|  0.002002| 0.00622| 0.748|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Inverse variance weighted |    4|  0.002067| 0.00594| 0.728|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Simple mode               |    4|  0.004152| 0.00902| 0.677|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Weighted mode             |    4|  0.001134| 0.00685| 0.879|</span>
<span class="r-in"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-heterogeneity.txt"</span><span class="op">)</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">             caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption</span>,<span class="st">"(heterogeneity)"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Table: Table. MMP.10 variants and FEV1 (heterogeneity)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |id.exposure |id.outcome       |outcome                               |exposure |method                    |     Q| Q_df| Q_pval|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |:-----------|:----------------|:-------------------------------------|:--------|:-------------------------|-----:|----:|------:|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |MR Egger                  | 0.114|    2|  0.944|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Inverse variance weighted | 0.222|    3|  0.974|</span>
<span class="r-in"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-pleiotropy.txt"</span><span class="op">)</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">             caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption</span>,<span class="st">"(pleiotropy)"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Table: Table. MMP.10 variants and FEV1 (pleiotropy)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |id.exposure |id.outcome       |outcome                               |exposure | egger_intercept|      se|  pval|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |:-----------|:----------------|:-------------------------------------|:--------|---------------:|-------:|-----:|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |foelRu      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |         0.00143| 0.00434| 0.774|</span>
<span class="r-in"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-single.txt"</span><span class="op">)</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">             caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption</span>,<span class="st">"(single)"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Table: Table. MMP.10 variants and FEV1 (single)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |exposure |outcome                               |id.exposure |id.outcome       | samplesize|SNP                             |         b|      se|     p|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |:--------|:-------------------------------------|:-----------|:----------------|----------:|:-------------------------------|---------:|-------:|-----:|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs11225354                      |  0.006425| 0.01217| 0.598|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs142915220                     | -0.003993| 0.02856| 0.889|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs17099622                      |  0.004360| 0.02229| 0.845|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs17860955                      |  0.000616| 0.00739| 0.934|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|All - Inverse variance weighted |  0.002067| 0.00594| 0.728|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|All - MR Egger                  | -0.000577| 0.01001| 0.959|</span>
<span class="r-in"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-loo.txt"</span><span class="op">)</span>,header<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">             caption<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">caption</span>,<span class="st">"(loo)"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Table: Table. MMP.10 variants and FEV1 (loo)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |exposure |outcome                               |id.exposure |id.outcome       | samplesize|SNP         |        b|      se|     p|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |:--------|:-------------------------------------|:-----------|:----------------|----------:|:-----------|--------:|-------:|-----:|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs11225354  | 0.000703| 0.00681| 0.918|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs142915220 | 0.002341| 0.00608| 0.700|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs17099622  | 0.001891| 0.00617| 0.759|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|rs17860955  | 0.004730| 0.01001| 0.636|</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |foelRu      |ebi-a-GCST007432 |     321047|All         | 0.002067| 0.00594| 0.728|</span>
<span class="r-in"><span class="kw">for</span> <span class="op">(</span><span class="va">x</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"result"</span>,<span class="st">"heterogeneity"</span>,<span class="st">"pleiotropy"</span>,<span class="st">"single"</span>,<span class="st">"loo"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">prefix</span>,<span class="st">"-"</span>,<span class="va">x</span>,<span class="st">".txt"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jing hua Zhao.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

    </footer></div>

  

  

  </body></html>

