<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Basic TwoSampleMR analysis — run_TwoSampleMR • pQTLtools</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Basic TwoSampleMR analysis — run_TwoSampleMR" />
<meta property="og:description" content="Given harmonised data, this function conducts a two-sample MR analysis." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pQTLtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jinghuazhao/pQTLtools">Source</a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Basic TwoSampleMR analysis</h1>
    
    <div class="hidden name"><code>run_TwoSampleMR.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given harmonised data, this function conducts a two-sample MR analysis.</p>
    </div>

    <pre class="usage"><span class='fu'>run_TwoSampleMR</span><span class='op'>(</span><span class='va'>TwoSampleMRinput</span>, mr_plot<span class='op'>=</span><span class='st'>"None"</span>, <span class='va'>prefix</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>TwoSampleMRinput</th>
      <td><p>Harmonised data</p></td>
    </tr>
    <tr>
      <th>mr_plot</th>
      <td><p>one of "None", "TwoSampleMR", "New" for no, the original and (some) new plots, respectively</p></td>
    </tr>
    <tr>
      <th>prefix</th>
      <td><p>a prefix for output files</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>As TwoSampleMR faces seemingly perplexing options, this function intends to simplify various steps in a two-sample MR. It is
particularly useful when a large numbher of MRs are necessary, e.g., multiple proteins and their cis/trans regions need to be examined,
in which case prefix could direct the output to various directories.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>No value is returned but several files.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Dimou NL, Tsilidis KK. A Primer in Mendelian Randomization Methodology with a Focus on Utilizing Published Summary Association Data. In
Evangelos Evangelou (ed.), Genetic Epidemiology: Methods and Protocols, Methods in Molecular Biology, vol. 1793,
https://doi.org/10.1007/978-1-4939-7868-7_13, Springer Science+Business Media, LLC, part of Springer Nature 2018</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Adapted from script by Dimou NL, Tsilidis KK</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>TwoSampleMR</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>TwoSampleMR version 0.5.6 </span>
#&gt; <span class='message'>[&gt;] New: Option to use non-European LD reference panels for clumping etc</span>
#&gt; <span class='message'>[&gt;] Some studies temporarily quarantined to verify effect allele</span>
#&gt; <span class='message'>[&gt;] See news(package='TwoSampleMR') and https://gwas.mrcieu.ac.uk for further details</span></div><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/jinghuazhao/pQTLtools'>pQTLtools</a></span><span class='op'>)</span>
<span class='va'>outcomes</span> <span class='op'>&lt;-</span> <span class='st'>"ebi-a-GCST007432"</span>
<span class='va'>prot</span> <span class='op'>&lt;-</span> <span class='st'>"MMP.10"</span>
<span class='va'>type</span> <span class='op'>&lt;-</span> <span class='st'>"cis"</span>
<span class='va'>f</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prot</span>,<span class='st'>"-"</span>,<span class='va'>type</span>,<span class='st'>".mrx"</span><span class='op'>)</span>
<span class='va'>d</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.table</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/find.package.html'>find.package</a></span><span class='op'>(</span><span class='st'>"pQTLtools"</span>,lib.loc<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/libPaths.html'>.libPaths</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>,<span class='st'>"tests"</span>,<span class='va'>f</span><span class='op'>)</span>,
                header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>exposure</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/format_data.html'>format_data</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/with.html'>within</a></span><span class='op'>(</span><span class='va'>d</span>,<span class='op'>{</span><span class='va'>P</span><span class='op'>=</span><span class='fl'>10</span><span class='op'>^</span><span class='va'>logP</span><span class='op'>}</span><span class='op'>)</span>, phenotype_col<span class='op'>=</span><span class='st'>"prot"</span>, snp_col<span class='op'>=</span><span class='st'>"rsid"</span>,
                        chr_col<span class='op'>=</span><span class='st'>"Chromosome"</span>, pos_col<span class='op'>=</span><span class='st'>"Posistion"</span>,
                        effect_allele_col<span class='op'>=</span><span class='st'>"Allele1"</span>, other_allele_col<span class='op'>=</span><span class='st'>"Allele2"</span>,
                        eaf_col<span class='op'>=</span><span class='st'>"Freq1"</span>, beta_col<span class='op'>=</span><span class='st'>"Effect"</span>, se_col<span class='op'>=</span><span class='st'>"StdErr"</span>,
                        pval_col<span class='op'>=</span><span class='st'>"P"</span>, log_pval<span class='op'>=</span><span class='cn'>FALSE</span>,
                        samplesize_col<span class='op'>=</span><span class='st'>"N"</span><span class='op'>)</span>
<span class='va'>clump</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/clump_data.html'>clump_data</a></span><span class='op'>(</span><span class='va'>exposure</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>API: public: http://gwas-api.mrcieu.ac.uk/</span></div><div class='output co'>#&gt; <span class='message'>Please look at vignettes for options on running this locally if you need to run many instances of this command.</span></div><div class='output co'>#&gt; <span class='message'>Clumping ImbABK, 1106 variants, using EUR population reference</span></div><div class='output co'>#&gt; <span class='message'>Removing 1102 of 1106 variants due to LD with other variants or absence from LD reference panel</span></div><div class='input'><span class='va'>outcome</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/extract_outcome_data.html'>extract_outcome_data</a></span><span class='op'>(</span>snps<span class='op'>=</span><span class='va'>exposure</span><span class='op'>$</span><span class='va'>SNP</span>,outcomes<span class='op'>=</span><span class='va'>outcomes</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Extracting data for 1106 SNP(s) from 1 GWAS(s)</span></div><div class='output co'>#&gt; <span class='message'>Finding proxies for 155 SNPs in outcome ebi-a-GCST007432</span></div><div class='output co'>#&gt; <span class='message'>Extracting data for 155 SNP(s) from 1 GWAS(s)</span></div><div class='input'><span class='va'>harmonise</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/TwoSampleMR/man/harmonise_data.html'>harmonise_data</a></span><span class='op'>(</span><span class='va'>clump</span>,<span class='va'>outcome</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Harmonising MMP.10 (ImbABK) and FEV1 || id:ebi-a-GCST007432 (ebi-a-GCST007432)</span></div><div class='input'><span class='va'>prefix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>outcomes</span>,<span class='va'>prot</span>,<span class='va'>type</span>,sep<span class='op'>=</span><span class='st'>"-"</span><span class='op'>)</span>
<span class='fu'>run_TwoSampleMR</span><span class='op'>(</span><span class='va'>harmonise</span>, mr_plot<span class='op'>=</span><span class='st'>"New"</span>, prefix<span class='op'>=</span><span class='va'>prefix</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Analysing 'ImbABK' on 'ebi-a-GCST007432'</span></div><div class='img'><img src='run_TwoSampleMR-1.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1 rows containing missing values (geom_errorbarh).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1 rows containing missing values (geom_point).</span></div><div class='img'><img src='run_TwoSampleMR-2.png' alt='' width='700' height='433' /></div><div class='img'><img src='run_TwoSampleMR-3.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1 rows containing missing values (geom_errorbarh).</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Removed 1 rows containing missing values (geom_point).</span></div><div class='img'><img src='run_TwoSampleMR-4.png' alt='' width='700' height='433' /></div><div class='input'><span class='va'>caption</span> <span class='op'>&lt;-</span> <span class='st'>"Table. MMP.10 variants and FEV1"</span>
<span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.delim</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prefix</span>,<span class='st'>"-result.txt"</span><span class='op'>)</span>,header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>,caption<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>caption</span>, <span class='st'>"(result)"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; 
#&gt; Table: Table. MMP.10 variants and FEV1 (result)
#&gt; 
#&gt; |id.exposure |id.outcome       |outcome                               |exposure |method                    | nsnp|         b|      se|  pval|
#&gt; |:-----------|:----------------|:-------------------------------------|:--------|:-------------------------|----:|---------:|-------:|-----:|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |MR Egger                  |    4| -0.000577| 0.01001| 0.959|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Weighted median           |    4|  0.002002| 0.00615| 0.745|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Inverse variance weighted |    4|  0.002067| 0.00594| 0.728|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Simple mode               |    4|  0.004152| 0.00892| 0.673|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Weighted mode             |    4|  0.001134| 0.00719| 0.885|</div><div class='input'><span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.delim</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prefix</span>,<span class='st'>"-heterogeneity.txt"</span><span class='op'>)</span>,header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>,caption<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>caption</span>,<span class='st'>"(heterogeneity)"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; 
#&gt; Table: Table. MMP.10 variants and FEV1 (heterogeneity)
#&gt; 
#&gt; |id.exposure |id.outcome       |outcome                               |exposure |method                    |     Q| Q_df| Q_pval|
#&gt; |:-----------|:----------------|:-------------------------------------|:--------|:-------------------------|-----:|----:|------:|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |MR Egger                  | 0.114|    2|  0.944|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |Inverse variance weighted | 0.222|    3|  0.974|</div><div class='input'><span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.delim</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prefix</span>,<span class='st'>"-pleiotropy.txt"</span><span class='op'>)</span>,header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>,caption<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>caption</span>,<span class='st'>"(pleiotropy)"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; 
#&gt; Table: Table. MMP.10 variants and FEV1 (pleiotropy)
#&gt; 
#&gt; |id.exposure |id.outcome       |outcome                               |exposure | egger_intercept|      se|  pval|
#&gt; |:-----------|:----------------|:-------------------------------------|:--------|---------------:|-------:|-----:|
#&gt; |ImbABK      |ebi-a-GCST007432 |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |MMP.10   |         0.00143| 0.00434| 0.774|</div><div class='input'><span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.delim</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prefix</span>,<span class='st'>"-single.txt"</span><span class='op'>)</span>,header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>,caption<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>caption</span>,<span class='st'>"(single)"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; 
#&gt; Table: Table. MMP.10 variants and FEV1 (single)
#&gt; 
#&gt; |exposure |outcome                               |id.exposure |id.outcome       | samplesize|SNP                             |         b|      se|     p|
#&gt; |:--------|:-------------------------------------|:-----------|:----------------|----------:|:-------------------------------|---------:|-------:|-----:|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs11225354                      |  0.006425| 0.01217| 0.598|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs142915220                     | -0.003993| 0.02856| 0.889|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs17099622                      |  0.004360| 0.02229| 0.845|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs17860955                      |  0.000616| 0.00739| 0.934|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|All - Inverse variance weighted |  0.002067| 0.00594| 0.728|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|All - MR Egger                  | -0.000577| 0.01001| 0.959|</div><div class='input'><span class='fu'>knitr</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/knitr/man/kable.html'>kable</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.delim</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prefix</span>,<span class='st'>"-loo.txt"</span><span class='op'>)</span>,header<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span>,caption<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>caption</span>,<span class='st'>"(loo)"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; 
#&gt; Table: Table. MMP.10 variants and FEV1 (loo)
#&gt; 
#&gt; |exposure |outcome                               |id.exposure |id.outcome       | samplesize|SNP         |        b|      se|     p|
#&gt; |:--------|:-------------------------------------|:-----------|:----------------|----------:|:-----------|--------:|-------:|-----:|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs11225354  | 0.000703| 0.00681| 0.918|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs142915220 | 0.002341| 0.00608| 0.700|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs17099622  | 0.001891| 0.00617| 0.759|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|rs17860955  | 0.004730| 0.01001| 0.636|
#&gt; |MMP.10   |FEV1 &amp;#124;&amp;#124; id:ebi-a-GCST007432 |ImbABK      |ebi-a-GCST007432 |     321047|All         | 0.002067| 0.00594| 0.728|</div><div class='input'><span class='kw'>for</span> <span class='op'>(</span><span class='va'>x</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"result"</span>,<span class='st'>"heterogeneity"</span>,<span class='st'>"pleiotropy"</span>,<span class='st'>"single"</span>,<span class='st'>"loo"</span><span class='op'>)</span><span class='op'>)</span> <span class='fu'><a href='https://rdrr.io/r/base/unlink.html'>unlink</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prefix</span>,<span class='st'>"-"</span>,<span class='va'>x</span>,<span class='st'>".txt"</span><span class='op'>)</span><span class='op'>)</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jing hua Zhao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


