<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>A generic wrapper for TwoSampleMR analysis — run_TwoSampleMR • pQTLtools</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/united/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="A generic wrapper for TwoSampleMR analysis — run_TwoSampleMR" />
<meta property="og:description" content="Given a compressed or uncompressed file containing exposure data, this function conducts a two-sample MR analysis using MR-Base." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pQTLtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jinghuazhao/pQTLtools">Source</a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>A generic wrapper for TwoSampleMR analysis</h1>
    
    <div class="hidden name"><code>run_TwoSampleMR.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Given a compressed or uncompressed file containing exposure data, this function conducts a two-sample MR analysis using MR-Base.</p>
    </div>

    <pre class="usage"><span class='fu'>run_TwoSampleMR</span><span class='op'>(</span>exposure.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>format_file.args</a></span><span class='op'>(</span><span class='op'>)</span>,outcome.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>extract_outcome_data.args</a></span><span class='op'>(</span><span class='op'>)</span>,
                       clump.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>clump_data.args</a></span><span class='op'>(</span><span class='op'>)</span>,harmonise.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>harmonise_data.args</a></span><span class='op'>(</span><span class='op'>)</span>,<span class='va'>prefix</span>,<span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>exposure.args</th>
      <td><p>a list of options appropriate for TwoSampleMR's format_data() in a file,</p><ol>
<li><p>file. NULL (file)</p></li>
<li><p>type. exposure (or outcome)</p></li>
<li><p>snps. NULL</p></li>
<li><p>header. TRUE (or FALSE)</p></li>
<li><p>phenotype_col. "Phenotype" (protein column)</p></li>
<li><p>snp_col. "SNP"</p></li>
<li><p>beta_col. "beta"</p></li>
<li><p>se_col. "se"</p></li>
<li><p>eaf_col. "eaf"</p></li>
<li><p>effect_allele_col. "effect_allele"</p></li>
<li><p>other_allele_col. "other_allele"</p></li>
<li><p>pval_col. "pval"</p></li>
<li><p>units_col. "units"</p></li>
<li><p>ncase_col. "ncase"</p></li>
<li><p>ncontrol_col. "ncontrol"</p></li>
<li><p>samplesize_col. "samplesize"</p></li>
<li><p>gene_col. "gene"</p></li>
<li><p>id_col. "id"</p></li>
<li><p>min_pval. 1e-200</p></li>
<li><p>z_col. "z"</p></li>
<li><p>info_col. "info"</p></li>
<li><p>chr_col. "chr"</p></li>
<li><p>pos_col. "pos"</p></li>
<li><p>log_pval. FALSE (or TRUE)</p></li>
</ol>
</td>
    </tr>
    <tr>
      <th>outcome.args</th>
      <td><p>a list of options appropriate for TwoSampleMR's extract_outcome_data(),</p><ol>
<li><p>snp. NULL (to derive from exposure.args)</p></li>
<li><p>outcomes. NULL</p></li>
<li><p>proxies. TRUE or FALSE</p></li>
<li><p>rsq. r^2</p></li>
<li><p>align_alleles. 1</p></li>
<li><p>palindromes. 1</p></li>
<li><p>maf_threshold. 0.3</p></li>
<li><p>access_token. token</p></li>
<li><p>splitsize. 10000</p></li>
<li><p>proxy_splitsize. 500</p></li>
</ol>
</td>
    </tr>
    <tr>
      <th>clump.args</th>
      <td><p>a list of options appropriate for TwoSampleMR's clump_data(),</p><ol>
<li><p>dat. NULL (format_data() output)</p></li>
<li><p>clump_kb. 10000</p></li>
<li><p>clump_r2. 0.001</p></li>
<li><p>clump_p1. 1</p></li>
<li><p>clump_p2. 1</p></li>
<li><p>pop. "EUR" ("EUR", "SAS", "EAS", "AFR", "AMR")</p></li>
</ol>
</td>
    </tr>
    <tr>
      <th>harmonise.args</th>
      <td><p>a list of options appropriate for harmonise_data(),</p><ol>
<li><p>exposure_data. NULL (exposure data from format_data())</p></li>
<li><p>outcome_data. NULL (outcome data from format_data())</p></li>
<li><p>action. 2 (1, 2, 3)</p></li>
</ol>
</td>
    </tr>
    <tr>
      <th>prefix</th>
      <td><p>a prefix for output files</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>other options to be added</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>As TwoSampleMR includes seemingly daunting tasks, this function intends to simplify various steps in a two-sample MR. It is
particularly useful when a large numbher of MRs are necessary, e.g., multiple proteins and their cis/trans regions need to be examined,
in which case prefix could direct the output to various directories.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>No value is returned but several files.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Dimou NL, Tsilidis KK. A Primer in Mendelian Randomization Methodology with a Focus on Utilizing Published Summary Association Data. In
Evangelos Evangelou (ed.), Genetic Epidemiology: Methods and Protocols, Methods in Molecular Biology, vol. 1793,
https://doi.org/10.1007/978-1-4939-7868-7_13, Springer Science+Business Media, LLC, part of Springer Nature 2018</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Adapted from script by Dimou NL, Tsilidis KK</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># SCALLOP-INF</span>
<span class='va'>INF</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/Sys.getenv.html'>Sys.getenv</a></span><span class='op'>(</span><span class='st'>"INF"</span><span class='op'>)</span>
<span class='va'>outcomes</span> <span class='op'>&lt;-</span> <span class='st'>"ukb-b-19732"</span>
<span class='va'>prot</span> <span class='op'>&lt;-</span> <span class='st'>"MMP.10"</span>
<span class='va'>type</span> <span class='op'>&lt;-</span> <span class='st'>"cis"</span>
<span class='va'>wd</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>INF</span>,<span class='st'>"mr"</span>,<span class='va'>type</span><span class='op'>)</span>
<span class='va'>f</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='va'>prot</span>,<span class='st'>"-"</span>,<span class='va'>type</span>,<span class='st'>".mrx"</span><span class='op'>)</span>
<span class='va'>d</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>wd</span>,<span class='va'>f</span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'>TwoSampleMR</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>TwoSampleMR version 0.5.5 </span>
#&gt; <span class='message'>[&gt;] New: Option to use non-European LD reference panels for clumping etc</span>
#&gt; <span class='message'>[&gt;] Some studies temporarily quarantined to verify effect allele</span>
#&gt; <span class='message'>[&gt;] See news(package='TwoSampleMR') and https://gwas.mrcieu.ac.uk for further details</span></div><div class='input'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span>
<span class='fu'>run_TwoSampleMR</span><span class='op'>(</span>exposure.args<span class='op'>=</span><span class='fu'>foramt_file.args</span><span class='op'>(</span><span class='va'>d</span><span class='op'>)</span>,outcome.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>extract_outcome_data.args</a></span><span class='op'>(</span>outcomes<span class='op'>=</span><span class='va'>outcomes</span><span class='op'>)</span>,
                clump.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>clump_data.args</a></span><span class='op'>(</span><span class='op'>)</span>,harmonise.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>harmonise_data.args</a></span><span class='op'>(</span><span class='op'>)</span>,
                prefix<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>wd</span>,<span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>outcomes</span>,<span class='va'>prot</span>,<span class='va'>type</span>,sep<span class='op'>=</span><span class='st'>"-"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>}</span>
<span class='co'># pQTLtools</span>
<span class='va'>d</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/find.package.html'>find.package</a></span><span class='op'>(</span><span class='st'>"pQTLtools"</span><span class='op'>)</span>,<span class='st'>"tests"</span>,<span class='va'>f</span><span class='op'>)</span>
<span class='fu'>run_TwoSampleMR</span><span class='op'>(</span>exposure.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>format_file.args</a></span><span class='op'>(</span><span class='va'>d</span><span class='op'>)</span>,outcome.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>extract_outcome_data.args</a></span><span class='op'>(</span>outcomes<span class='op'>=</span><span class='va'>outcomes</span><span class='op'>)</span>,
                clump.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>clump_data.args</a></span><span class='op'>(</span><span class='op'>)</span>,harmonise.args<span class='op'>=</span><span class='fu'><a href='pQTLtools-internal.html'>harmonise_data.args</a></span><span class='op'>(</span><span class='op'>)</span>,
                prefix<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste</a></span><span class='op'>(</span><span class='va'>outcomes</span>,<span class='va'>prot</span>,<span class='va'>type</span>,sep<span class='op'>=</span><span class='st'>"-"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in format_file.args(d): object 'dat' not found</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Jing hua Zhao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


