<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Manhattan plots — turboman • pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><script src="../deps/MathJax-3.2.2/tex-chtml.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Manhattan plots — turboman"><meta name="description" content="Manhattan plots"><meta property="og:description" content="Manhattan plots"><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html" aria-label="Home"><span class="fa fa-home"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a></li>
    <li><a class="dropdown-item" href="../articles/esse.html">ExpressionSet/SummarizedExperiment usage</a></li>
    <li><a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a></li>
    <li><a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a></li>
    <li><a class="dropdown-item" href="../articles/spectrum.html">Spectrum data analysis</a></li>
    <li><a class="dropdown-item" href="../articles/SCALLOP-Caprion.html">SCALLOP-Caprion scripts</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../reference/pQTLtools.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools" aria-label="GitHub"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/" aria-label="R packages"><span class="fa fab fa-r-project"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/" aria-label="AlphaFold"><span class="fa fas fa-fast-backward"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-urls" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">URLs</button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-urls"><li><a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a></li>
    <li><a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a></li>
    <li><a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a></li>
    <li><a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a></li>
    <li><a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a></li>
    <li><a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a></li>
    <li><a class="external-link dropdown-item" href="https://www.histo.fyi/">histo.fyi</a></li>
    <li><a class="external-link dropdown-item" href="https://metascape.org/">metascape</a></li>
    <li><a class="external-link dropdown-item" href="https://ligand-discovery.ai/">Ligand discovery</a></li>
    <li><a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a></li>
    <li><a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a></li>
    <li><a class="external-link dropdown-item" href="https://string-db.org/">STRING</a></li>
    <li><a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a></li>
    <li><a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a></li>
  </ul></li>
<li class="nav-item"><a class="external-link nav-link" href="https://www.uniprot.org/" aria-label="UniProt"><span class="fa fas fa-fast-forward"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Manhattan plots</h1>
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/R/turboman.r" class="external-link"><code>R/turboman.r</code></a></small>
      <div class="d-none name"><code>turboman.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Manhattan plots</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">turboman</span><span class="op">(</span></span>
<span>  <span class="va">input_data_path</span>,</span>
<span>  <span class="va">custom_peak_annotation_file_path</span>,</span>
<span>  <span class="va">reference_file_path</span>,</span>
<span>  <span class="va">pvalue_sign</span>,</span>
<span>  <span class="va">plot_title</span>,</span>
<span>  vertical_resolution <span class="op">=</span> <span class="fl">1800</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-input-data-path">input_data_path<a class="anchor" aria-label="anchor" href="#arg-input-data-path"></a></dt>
<dd><p>Path of the input association data.</p></dd>


<dt id="arg-custom-peak-annotation-file-path">custom_peak_annotation_file_path<a class="anchor" aria-label="anchor" href="#arg-custom-peak-annotation-file-path"></a></dt>
<dd><p>Path of the custom annotation of variants.</p></dd>


<dt id="arg-reference-file-path">reference_file_path<a class="anchor" aria-label="anchor" href="#arg-reference-file-path"></a></dt>
<dd><p>Path to the 'turboman_hg19_reference_data.rda' / 'turboman_hg19_reference_data.rda' reference file.</p></dd>


<dt id="arg-pvalue-sign">pvalue_sign<a class="anchor" aria-label="anchor" href="#arg-pvalue-sign"></a></dt>
<dd><p>Significance threshold p-value.</p></dd>


<dt id="arg-plot-title">plot_title<a class="anchor" aria-label="anchor" href="#arg-plot-title"></a></dt>
<dd><p>Plot title which will be displayed on top of the plot.</p></dd>


<dt id="arg-vertical-resolution">vertical_resolution<a class="anchor" aria-label="anchor" href="#arg-vertical-resolution"></a></dt>
<dd><p>A fixed number of points (pixels) to be plotted vertically.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><strong>Input association data file / input_data_path</strong></p>
<p>Define the path of the input association data.
The input data needs to be a file that has:</p><ol><li><p>Spaces as field separators.</p></li>
<li><p>One header line.</p></li>
<li><p>Option I. (no extreme p-values present): 3 columns, being chromosome, position, pvalue in order, column names are not important.
Option II. (extreme p-values present): 5 columns, being chromosome, position, pvalue, beta, se in order, column names are not important.</p></li>
</ol><p><strong>Custom annotation file / custom_peak_annotation_file_path</strong></p>
<p>Define the path of the custom annotation of variants.
The input data needs to be a file that has:</p><ol><li><p>Spaces / tabs as field separators.</p></li>
<li><p>One header line with exact column names (order not important).</p></li>
<li><p>Columns: chromosome, position, label (e.g., gene name) / nearest_gene_name, cis/trans flag (optional).
NB!: If no label is given, variants will be automatically annotated</p></li>
</ol><p><strong>Reference file / reference_file_path</strong></p>
<p>Define the path to the 'turboman_hg19_reference_data.rda' / 'turboman_hg38_reference_data.rda' reference file that
contains the LD block breaks as in berisa16;textualpQTLtools
and gene coordinates used to construct and annotate the Manhattan plot. Both
are available from the <code>turboman</code> directory of the installed package, e.g.,
<code>file.path(find.package('pQTLtools'),'turboman','turboman_hg38_reference_data.rda')</code>.</p>
<p><strong>Significance threshold p-value / pvalue_sign</strong></p>
<p>Define the significance threshold.
This will be used to</p><ol><li><p>Highlight signal peaks that come above this significance threshold.</p></li>
<li><p>Annotate the nearest gene to the top signal in the peak.</p></li>
<li><p>Draw a horizontal reference line equal to this threshold.</p></li>
</ol><p><strong>Title of the plot / plot_title</strong></p>
<p>Define title on top of the plot.</p>
<p><strong>Number of pixels on vertical axis / vertical_resolution</strong></p>
<p>Define a fixed number of points (pixels) on vertical axis.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p></p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Arthur Gilly, Chris Finan, Bram Prins, see <a href="https://github.com/bpprins/turboman" class="external-link">https://github.com/bpprins/turboman</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Screen output</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://jinghuazhao.github.io/R/" class="external-link">gap.datasets</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">test</span> <span class="op">&lt;-</span> <span class="va">mhtdata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'chr'</span>,<span class="st">'pos'</span>,<span class="st">'p'</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">test</span>,file<span class="op">=</span><span class="st">'test.txt'</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,quote<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">annotate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">mhtdata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'chr'</span>,<span class="st">'start'</span>,<span class="st">'gene'</span>,<span class="st">'p'</span><span class="op">)</span><span class="op">]</span>,<span class="va">p</span><span class="op">&lt;</span><span class="fl">5e-8</span> <span class="op">&amp;</span> <span class="va">gene</span><span class="op">!=</span><span class="st">''</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'chromosome'</span>,<span class="st">'position'</span>,<span class="st">'nearest_gene_name'</span>,<span class="st">'p'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">[</span>,<span class="op">-</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span>,file<span class="op">=</span><span class="st">'annotate.txt'</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,quote<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">input_data_path</span> <span class="op">&lt;-</span> <span class="st">'test.txt'</span></span></span>
<span class="r-in"><span><span class="va">custom_peak_annotation_file_path</span> <span class="op">&lt;-</span> <span class="st">'annotate.txt'</span></span></span>
<span class="r-in"><span><span class="va">reference_file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">'pQTLtools'</span>,lib.loc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/libPaths.html" class="external-link">.libPaths</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  <span class="st">'turboman'</span>,<span class="st">'turboman_hg19_reference_data.rda'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pvalue_sign</span> <span class="op">&lt;-</span> <span class="fl">5e-8</span></span></span>
<span class="r-in"><span><span class="va">plot_title</span> <span class="op">&lt;-</span> <span class="st">'gap.datasets example'</span></span></span>
<span class="r-in"><span><span class="fu">turboman</span><span class="op">(</span><span class="va">input_data_path</span>, <span class="va">custom_peak_annotation_file_path</span>,</span></span>
<span class="r-in"><span>         <span class="va">reference_file_path</span>, <span class="va">pvalue_sign</span>, <span class="va">plot_title</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Figure shown on https://github.com/jinghuazhao/tests/tree/main/turboman</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/png.html" class="external-link">png</a></span><span class="op">(</span><span class="st">'IL12B.png'</span>,width<span class="op">=</span><span class="fl">3600</span>, height<span class="op">=</span><span class="fl">3600</span>, pointsize <span class="op">=</span> <span class="fl">12</span>, res<span class="op">=</span><span class="fl">450</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">input_data_path</span> <span class="op">&lt;-</span> <span class="st">'IL.12B.txt.gz'</span></span></span>
<span class="r-in"><span><span class="va">custom_peak_annotation_file_path</span> <span class="op">&lt;-</span> <span class="st">'IL.12B.annotate'</span></span></span>
<span class="r-in"><span><span class="va">reference_file_path</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/find.package.html" class="external-link">find.package</a></span><span class="op">(</span><span class="st">'pQTLtools'</span><span class="op">)</span>,<span class="st">'turboman'</span>,<span class="st">'turboman_hg19_reference_data.rda'</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pvalue_sign</span> <span class="op">&lt;-</span> <span class="fl">5e-8</span></span></span>
<span class="r-in"><span><span class="va">plot_title</span> <span class="op">&lt;-</span> <span class="st">'IL12B'</span></span></span>
<span class="r-in"><span><span class="fu">turboman</span><span class="op">(</span><span class="va">input_data_path</span>, <span class="va">custom_peak_annotation_file_path</span>,</span></span>
<span class="r-in"><span>         <span class="va">reference_file_path</span>, <span class="va">pvalue_sign</span>, <span class="va">plot_title</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html" class="external-link">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer></div>



  <script type="module">import mermaid from "https://unpkg.com/mermaid@10.4.0/dist/mermaid.esm.min.mjs";mermaid.initialize({});</script></body></html>

