<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Manhattan plots"><title>Manhattan plots — turboman • pQTLtools</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Manhattan plots — turboman"><meta property="og:description" content="Manhattan plots"><meta property="og:image" content="https://jinghuazhao.github.io/pQTLtools/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">pQTLtools</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <h6 class="dropdown-header" data-toc-skip>Vignettes:</h6>
    <a class="dropdown-item" href="../articles/pQTLtools.html">An overview of pQTLtools</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Articles:</h6>
    <a class="dropdown-item" href="../articles/bioconductor.html">Bioconductor notes</a>
    <a class="dropdown-item" href="../articles/es.html">ExpressionSet usage</a>
    <a class="dropdown-item" href="../articles/LocusZoom.js.html">LocusZoom.js</a>
    <a class="dropdown-item" href="../articles/snakemake.html">snakemake showcases</a>
    <a class="dropdown-item" href="../articles/SCALLOP-INF.html">SCALLOP-INF scripts</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/pQTLtools.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jinghuazhao/pQTLtools/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://jinghuazhao.github.io/R/">
    <span class="fab fa fab fa-r-project"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://alphafold.ebi.ac.uk/">
    <span class="fas fa fas fa-fast-backward"></span>
     
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-urls">URLs</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-urls">
    <a class="external-link dropdown-item" href="https://alphafold.ebi.ac.uk/">AlphaFold</a>
    <a class="external-link dropdown-item" href="https://github.com/Bioconductor">Bioconductor</a>
    <a class="external-link dropdown-item" href="https://biorender.com/">BioRender</a>
    <a class="external-link dropdown-item" href="https://cran.r-project.org/">CRAN</a>
    <a class="external-link dropdown-item" href="https://cytoscape.org/">Cytoscape</a>
    <a class="external-link dropdown-item" href="https://www.ensembl.org/">Ensembl</a>
    <a class="external-link dropdown-item" href="https://esmatlas.com/">ESMfold</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/eqtl/">eQTL catalog</a>
    <a class="external-link dropdown-item" href="https://www.ebi.ac.uk/gwas/">GWAS catalog</a>
    <a class="external-link dropdown-item" href="https://metascape.org/">metascape</a>
    <a class="external-link dropdown-item" href="https://www.mrbase.org/">MRBase</a>
    <a class="external-link dropdown-item" href="https://www.proteomicsdb.org/">ProteomicsDB</a>
    <a class="external-link dropdown-item" href="https://pubmed.ncbi.nlm.nih.gov/">PubMed</a>
    <a class="external-link dropdown-item" href="https://string-db.org/">STRING</a>
    <a class="external-link dropdown-item" href="https://www.proteinatlas.org/">The Human Protein Atlas</a>
    <a class="external-link dropdown-item" href="https://www.uniprot.org/">UniProt</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.uniprot.org/">
    <span class="fas fa fas fa-fast-forward"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Manhattan plots</h1>
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/pQTLtools/blob/HEAD/R/turboman.r" class="external-link"><code>R/turboman.r</code></a></small>
      <div class="d-none name"><code>turboman.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Manhattan plots</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">turboman</span><span class="op">(</span></span>
<span>  <span class="va">input_data_path</span>,</span>
<span>  <span class="va">output_data_rootname</span>,</span>
<span>  <span class="va">custom_peak_annotation_file_path</span>,</span>
<span>  <span class="va">reference_file_path</span>,</span>
<span>  <span class="va">pvalue_sign</span>,</span>
<span>  <span class="va">plot_title</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>input_data_path</dt>
<dd><p>Path of the input association data</p></dd>


<dt>output_data_rootname</dt>
<dd><p>Root name of the plot output file</p></dd>


<dt>custom_peak_annotation_file_path</dt>
<dd><p>Path of the custom annotation of variants</p></dd>


<dt>reference_file_path</dt>
<dd><p>Path to the "turboman_hg19_reference_data.rda" reference file</p></dd>


<dt>pvalue_sign</dt>
<dd><p>Significance threshold p-value</p></dd>


<dt>plot_title</dt>
<dd><p>Plot title which will be displayed on top of the plot</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><strong>Input association data path / input_data_path</strong></p>
<p>Define the path of the input association data...
The input data needs to be a file that has:</p><ol><li><p>Spaces as field separators</p></li>
<li><p>One header line</p></li>
<li><p>Two option:</p><ul><li><p>I (no extreme p-values present): 3 columns, being chromosome, position, pvalue in order, column names are not important</p></li>
<li><p>II (extreme p-values present): 5 columns, being chromosome, position, pvalue, beta, se in order, column names are not important</p></li>
</ul></li>
</ol><p><strong>Output data root name / output_data_rootname</strong></p>
<p>Define the root name of the plot output file...
Ex.: "my_man_plot" will result in an output file named "my_man_plot.png"</p>
<p><strong>OPTIONAL: Custom annotation file / custom_peak_annotation_file_path</strong></p>
<p>Define the path of the custom annotation of variants...
The input data needs to be a file that has:</p><ol><li><p>Spaces / tabs as field separators</p></li>
<li><p>One header line with exact column names (order not important)</p></li>
<li><p>3 columns: chromosome, position, label (e.g., gene name)
Column names chromosome, position, nearest_gene_name
NB!: If no label is given, variants will be automatically annotated</p></li>
</ol><p><strong>Reference file path / reference_file_path</strong></p>
<p>Define the path to the "turboman_hg19_reference_data.rda" reference file that
contains the LD block breaks as in Berisa and Pickrell (2016)

and gene coordinates used to construct and annotate the Manhattan plot</p>
<p><strong>Significance threshold p-value / pvalue_sign</strong></p>
<p>Define the significance threshold...
This will be used to...</p><ol><li><p>Highlight signal peaks that come above this significance threshold</p></li>
<li><p>Annotate the nearest gene to the top signal in the peak</p></li>
<li><p>Draw a horizontal reference line equal to this threshold</p></li>
</ol></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Berisa T, Pickrell JK (2016).
“Approximately independent linkage disequilibrium blocks in human populations.”
<em>Bioinformatics</em>, <b>32</b>(2), 283-285.
<a href="https://doi.org/10.1093/bioinformatics/btv546" class="external-link">doi:10.1093/bioinformatics/btv546</a>
.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Arthur Gilly, Chris Finan, Bram Prins</p>
<p><strong>Changelog</strong></p><ul><li><p>2017/12/04. Added Varying window size in which gene names are plotted based on the length of gene names so that gene names will not be truncated in the plot</p></li>
<li><p>2017/12/04. Added an option allowing the user to supply a table with top SNPs to highlight, which can, for example, be selected by conditional analyses / LD pruning</p></li>
<li><p>2017/12/04. Fixed a bug (found by Tao) that leads to erroneous binning when plotting extremely sparse data</p></li>
<li><p>2017/12/05. Added optional log-pval calculations from beta/SE for extreme p-values that run below <code>.Machine$double.xmin</code></p></li>
<li><p>2017/12/05. Added an automated option for plotting already log-transformed p-values (user can supply either)</p></li>
<li><p>2017/12/05. Increased contrast between chromosomes on the plot (Eric H)</p></li>
<li><p>2017/12/05. Set '10' in the y-axis label (-log10 pvalue) to subscript (Eric H)</p></li>
</ul><p><strong>To do list</strong></p><ol><li><p>Display gene names bold / non-bold when genic / intergenic</p></li>
<li><p>Create LD blocks for hg38</p></li>
<li><p>Add many more sanity checks (are you giving pvalues, the file does not exist, etc.)</p></li>
<li><p>Create a function version of the script (Eric H)</p></li>
<li><p>...</p></li>
</ol></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://jinghuazhao.github.io/R/" class="external-link">gap.datasets</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">test</span> <span class="op">&lt;-</span> <span class="va">mhtdata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">"pos"</span>,<span class="st">"p"</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">test</span>,file<span class="op">=</span><span class="st">"test.txt"</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,quote<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">annotate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">mhtdata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="st">"start"</span>,<span class="st">"gene"</span>,<span class="st">"p"</span><span class="op">)</span><span class="op">]</span>,<span class="va">p</span><span class="op">&lt;</span><span class="fl">5e-8</span> <span class="op">&amp;</span> <span class="va">gene</span><span class="op">!=</span><span class="st">""</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chromosome"</span>,<span class="st">"position"</span>,<span class="st">"nearest_gene_name"</span>,<span class="st">"p"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">annotate</span><span class="op">[</span>,<span class="op">-</span><span class="fl">4</span><span class="op">]</span>,file<span class="op">=</span><span class="st">"annotate.txt"</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,quote<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">input_data_path</span> <span class="op">&lt;-</span> <span class="st">"test.txt"</span></span></span>
<span class="r-in"><span><span class="va">output_data_rootname</span> <span class="op">&lt;-</span> <span class="st">"test_man"</span></span></span>
<span class="r-in"><span><span class="va">custom_peak_annotation_file_path</span> <span class="op">&lt;-</span> <span class="st">"annotate.txt"</span></span></span>
<span class="r-in"><span><span class="va">reference_file_path</span> <span class="op">&lt;-</span> <span class="st">"turboman_hg19_reference_data.rda"</span></span></span>
<span class="r-in"><span><span class="va">pvalue_sign</span> <span class="op">&lt;-</span> <span class="fl">5e-8</span></span></span>
<span class="r-in"><span><span class="va">plot_title</span> <span class="op">&lt;-</span> <span class="st">"gap.datasets example"</span></span></span>
<span class="r-in"><span><span class="fu">turboman</span><span class="op">(</span><span class="va">input_data_path</span>, <span class="va">output_data_rootname</span>, <span class="va">custom_peak_annotation_file_path</span>, <span class="va">reference_file_path</span>, <span class="va">pvalue_sign</span>, <span class="va">plot_title</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">input_data_path</span> <span class="op">&lt;-</span> <span class="st">"FCN3.txt.gz"</span></span></span>
<span class="r-in"><span><span class="va">output_data_rootname</span> <span class="op">&lt;-</span> <span class="st">"FCN3"</span></span></span>
<span class="r-in"><span><span class="va">custom_peak_annotation_file_path</span> <span class="op">&lt;-</span> <span class="st">"FCN3.txt"</span></span></span>
<span class="r-in"><span><span class="va">reference_file_path</span> <span class="op">&lt;-</span> <span class="st">"turboman_hg19_reference_data.rda"</span></span></span>
<span class="r-in"><span><span class="va">pvalue_sign</span> <span class="op">&lt;-</span> <span class="fl">5e-8</span></span></span>
<span class="r-in"><span><span class="va">plot_title</span> <span class="op">&lt;-</span> <span class="st">"FCN3"</span></span></span>
<span class="r-in"><span><span class="fu">turboman</span><span class="op">(</span><span class="va">input_data_path</span>, <span class="va">output_data_rootname</span>, <span class="va">custom_peak_annotation_file_path</span>, <span class="va">reference_file_path</span>, <span class="va">pvalue_sign</span>, <span class="va">plot_title</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Crafted by <a href="https://jinghuazhao.github.io/" class="external-link"><img src="https://jinghuazhao.github.io/bees.svg" width="24"> Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer></div>

  

  

  </body></html>

